---
title: "Global_variation"
author: "Lauren Blake"
date: "November 26, 2016"
output: html_document
---

The goal of this script is to assess the extent of regulatory variation in our cell types. 

## Check data

```{r}
# Load libraries

library("ggplot2")
library("qvalue")
source("~/Desktop/Endoderm_TC/ashlar-trial/analysis/chunk-options.R")
library("RColorBrewer")

# Load colors
pal <- c(brewer.pal(9, "Set1"), brewer.pal(8, "Set2"), brewer.pal(12, "Set3"))


# Load data
cpm_in_cutoff <- read.delim("~/Desktop/Endoderm_TC/ashlar-trial/data/cpm_norm_data.txt")
```


```{r}
# Take the mean of the technical replicates when available

  # Day 0 technical replicates
D0_28815 <- as.data.frame(apply(cpm_in_cutoff[,5:6], 1, mean))
D0_3647 <- as.data.frame(apply(cpm_in_cutoff[,8:9], 1, mean))
D0_3649 <- as.data.frame(apply(cpm_in_cutoff[,10:11], 1, mean))
D0_40300 <- as.data.frame(apply(cpm_in_cutoff[,12:13], 1, mean))
D0_4955 <- as.data.frame(apply(cpm_in_cutoff[,14:15], 1, mean))
  # Day 1 technical replicates
D1_20157 <- as.data.frame(apply(cpm_in_cutoff[,16:17], 1, mean))
D1_28815 <- as.data.frame(apply(cpm_in_cutoff[,21:22], 1, mean))
D1_3647 <- as.data.frame(apply(cpm_in_cutoff[,24:25], 1, mean))
D1_3649 <- as.data.frame(apply(cpm_in_cutoff[,26:27], 1, mean))
D1_40300 <- as.data.frame(apply(cpm_in_cutoff[,28:29], 1, mean))
D1_4955 <- as.data.frame(apply(cpm_in_cutoff[,30:31], 1, mean))
  # Day 2 technical replicates
D2_20157 <- as.data.frame(apply(cpm_in_cutoff[,32:33], 1, mean))
D2_28815 <- as.data.frame(apply(cpm_in_cutoff[,37:38], 1, mean))
D2_3647 <- as.data.frame(apply(cpm_in_cutoff[,40:41], 1, mean))
D2_3649 <- as.data.frame(apply(cpm_in_cutoff[,42:43], 1, mean))
D2_40300 <- as.data.frame(apply(cpm_in_cutoff[,44:45], 1, mean))
D2_4955 <- as.data.frame(apply(cpm_in_cutoff[,46:47], 1, mean))
  # Day 3 technical replicates
D3_20157 <- as.data.frame(apply(cpm_in_cutoff[,48:49], 1, mean))
D3_28815 <- as.data.frame(apply(cpm_in_cutoff[,53:54], 1, mean))
D3_3647 <- as.data.frame(apply(cpm_in_cutoff[,56:57], 1, mean))
D3_3649 <- as.data.frame(apply(cpm_in_cutoff[,58:59], 1, mean))
D3_40300 <- as.data.frame(apply(cpm_in_cutoff[,60:61], 1, mean))
D3_4955 <- as.data.frame(apply(cpm_in_cutoff[,62:63], 1, mean))

# Create a new data frame with all of the combined technical replicates
mean_tech_reps <- cbind(cpm_in_cutoff[,1:4], D0_28815, cpm_in_cutoff[,7], D0_3647, D0_3649, D0_40300, D0_4955, D1_20157, cpm_in_cutoff[,18:20], D1_28815, cpm_in_cutoff[,23], D1_3647, D1_3649, D1_40300, D1_4955, D2_20157, cpm_in_cutoff[,34:36], D2_28815, cpm_in_cutoff[,39], D2_3647, D2_3649, D2_40300, D2_4955, D3_20157, cpm_in_cutoff[,50:52], D3_28815, cpm_in_cutoff[,55], D3_3647, D3_3649, D3_40300, D3_4955)


colnames(mean_tech_reps) <- c("D0_20157", "D0_20961", "D0_21792", "D0_28162", "D0_28815", "D0_29089", "D0_3647", "D0_3649", "D0_40300", "D0_4955", "D1_20157", "D1_20961", "D1_21792", "D1_28162", "D1_28815", "D1_29089", "D1_3647", "D1_3649", "D1_40300", "D1_4955", "D2_20157", "D2_20961", "D2_21792", "D2_28162", "D2_28815", "D2_29089", "D2_3647", "D2_3649", "D2_40300", "D2_4955", "D3_20157", "D3_20961", "D3_21792", "D3_28162", "D3_28815", "D3_29089", "D3_3647", "D3_3649", "D3_40300", "D3_4955")

dim(mean_tech_reps)

# Make a column for which are averaged or not

averaged_status <- c(1,1,1,1,2,1,2,2,2,2,2,1,1,1,2,1,2,2,2,2,2,1,1,1,2,1,2,2,2,2,2,1,1,1,2,1,2,2,2,2)

# Find the technical factors for the biological replicates (no technical replicates)

bio_rep_samplefactors <- read.delim("~/Downloads/samplefactors-filtered.txt", stringsAsFactors=FALSE)
day <- bio_rep_samplefactors$Day
species <- bio_rep_samplefactors$Species

# Make PCA plots with the factors colored by day

pca_genes <- prcomp(t(mean_tech_reps), scale = T, retx = TRUE, center = TRUE)

matrixpca <- pca_genes$x
pc1 <- matrixpca[,1]
pc2 <- matrixpca[,2]
pc3 <- matrixpca[,3]
pc4 <- matrixpca[,4]
pc5 <- matrixpca[,5]

pcs <- data.frame(pc1, pc2, pc3, pc4, pc5)

summary <- summary(pca_genes)

#dev.off()

ggplot(data=pcs, aes(x=pc1, y=pc2, color=as.factor(day+1), shape=as.factor(averaged_status), size=2)) + geom_point() + xlab(paste("PC1 (",(summary$importance[2,1]*100), "% of variance)")) + ylab(paste("PC2 (",(summary$importance[2,2]*100), "% of variance)")) + theme_minimal() + guides(color = guide_legend(order=1), size = FALSE, shape = guide_legend(order=3)) + scale_color_discrete(name  ="Day", labels = c("0", "1", "2", "3")) +  scale_shape_manual(values = c(5, 7)) + scale_shape_discrete(name  ="Replicate status", labels = c("Single" ,"Averaged")) + labs(title = "PCs 1 and 2 from global normalized expression")

ggplot(data=pcs, aes(x=pc1, y=pc2, color=as.factor(day+1), shape=as.factor(averaged_status), size=2)) + geom_point() + xlab(paste("PC1 (",(summary$importance[2,1]*100), "% of variance)")) + ylab(paste("PC2 (",(summary$importance[2,2]*100), "% of variance)")) + scale_shape_manual(values = c(5, 8)) + scale_color_discrete(name  ="Day", labels = c("0", "1", "2", "3"))  + labs(title = "PCs 1 and 2 from global normalized gene expression (Averaged bet. technical replicates)")

#ggplotly()


```

## Evaluate the variances between day-species pairs

```{r}
# Calculate the mean for each species-time pair

humans_day0_var <- as.data.frame(apply(as.data.frame(mean_tech_reps[,1:6]),1, var) )
colnames(humans_day0_var) <- c("Variance") 
chimps_day0_var <- as.data.frame(apply(as.data.frame(mean_tech_reps[,7:10]),1, var))
colnames(chimps_day0_var) <- c("Variance") 
humans_day1_var <- as.data.frame(apply(as.data.frame(mean_tech_reps[,11:16]),1, var))
colnames(humans_day1_var) <- c("Variance") 
chimps_day1_var <- as.data.frame(apply(as.data.frame(mean_tech_reps[,17:20]),1, var))
colnames(chimps_day1_var) <- c("Variance")
humans_day2_var <- as.data.frame(apply(as.data.frame(mean_tech_reps[,21:26]),1, var))
colnames(humans_day2_var) <- c("Variance") 
chimps_day2_var <- as.data.frame(apply(as.data.frame(mean_tech_reps[,27:30]),1, var))
colnames(chimps_day2_var) <- c("Variance")
humans_day3_var <- as.data.frame(apply(as.data.frame(mean_tech_reps[,31:36]),1, var))
colnames(humans_day3_var) <- c("Variance") 
chimps_day3_var <- as.data.frame(apply(as.data.frame(mean_tech_reps[,37:40]),1, var))
colnames(chimps_day3_var) <- c("Variance")

 
humans_day0_mean <- apply(as.data.frame(mean_tech_reps[,1:6]),1, mean)
chimps_day0_mean <- apply(as.data.frame(mean_tech_reps[,7:10]),1, mean)
humans_day1_mean <- apply(as.data.frame(mean_tech_reps[,11:16]),1, mean)
chimps_day1_mean <- apply(as.data.frame(mean_tech_reps[,17:20]),1, mean)
humans_day2_mean <- apply(as.data.frame(mean_tech_reps[,21:26]),1, mean)
chimps_day2_mean <- apply(as.data.frame(mean_tech_reps[,27:30]),1, mean)
humans_day3_mean <- apply(as.data.frame(mean_tech_reps[,31:36]),1, mean)
chimps_day3_mean <- apply(as.data.frame(mean_tech_reps[,37:40]),1, mean)

# Make arrays with all the means

labels1 <- array("Chimp Day 0", dim = c(10304, 1)) 
labels2 <- array("Chimp Day 1", dim = c(10304, 1)) 
labels3 <- array("Chimp Day 2", dim = c(10304, 1)) 
labels4 <- array("Chimp Day 3", dim = c(10304, 1)) 
labels5 <- array("Human Day 0", dim = c(10304, 1)) 
labels6 <- array("Human Day 1", dim = c(10304, 1)) 
labels7 <- array("Human Day 2", dim = c(10304, 1)) 
labels8 <- array("Human Day 3", dim = c(10304, 1)) 

# Make species-day labels
labels9 <- rbind(labels1, labels2, labels3, labels4, labels5, labels6, labels7, labels8)
labels <- as.numeric(as.factor(labels9))
# Make labels so same days from different species are the same color
labels10 <- rbind(labels1, labels2, labels3, labels4, labels1, labels2, labels3, labels4)
labels10 <- as.numeric(as.factor(labels10))

#1 Overall trend across genes

# Boxplot of variances gives general trend
HC_var <- rbind(as.data.frame(chimps_day0_var), as.data.frame(chimps_day1_var), as.data.frame(chimps_day2_var), as.data.frame(chimps_day3_var), as.data.frame(humans_day0_var), as.data.frame(humans_day1_var), as.data.frame(humans_day2_var), as.data.frame(humans_day3_var))

HC_var_labels <- cbind(HC_var, labels, labels10)

p <- ggplot(HC_var_labels, aes(x = factor(labels), y = HC_var))
p <- p + geom_violin(aes(fill = factor(labels10)), show.legend = FALSE) + geom_boxplot(aes(fill = factor(labels10)), show.legend = FALSE, outlier.shape = NA,width=0.2) + scale_y_continuous(trans = "log2") + ggtitle("Variance for each gene by species and day") + xlab("Species-Day Pair") + ylab("Variance for each gene") 
p <- p + scale_x_discrete(labels=c("1" = "C Day 0", "2" = "C Day 1", "3" = "C Day 2", "4" = "C Day 3", "5" = "H Day 0", "6" = "H Day 1", "7" = "H Day 2", "8" = "H Day 3"))
p
       
```

### Check to see if differences between means (of the variances of all of the genes) within humans and chimps

```{r}

########## Within chimps ##########

# Test at least one of the means is different
group <- rbind(labels1, labels2, labels3, labels4)

var_day_species <- rbind(chimps_day0_var,chimps_day1_var, chimps_day2_var, chimps_day3_var)

aov_groups <- cbind(var_day_species, group)

dim(aov_groups)

aov_bet <- oneway.test(aov_groups$Variance ~ aov_groups$group)
aov_bet

# F = 16.803, num df = 3, denom df = 22848, p-value = 6.715e-11

# Test groups individually

# Day 0 and Day 1
t.test(chimps_day0_var, chimps_day1_var) # 0.4586
t.test(chimps_day0_var, chimps_day1_var, alternative = "greater")
# Day 1 and Day 2
t.test(chimps_day1_var, chimps_day2_var) # 0.03103
# Day 2 and Day 3
t.test(chimps_day2_var, chimps_day3_var) # 2.992e-06
# Day 1 and Day 3
t.test(chimps_day1_var, chimps_day3_var) # 4e-11


########## Within humans ##########

# Test at least one of the means is different
group <- rbind(labels5, labels6, labels7, labels8)

var_day_species <- rbind(humans_day0_var,humans_day1_var, humans_day2_var, humans_day3_var)

aov_groups <- cbind(var_day_species, group)

dim(aov_groups)

aov_bet <- oneway.test(aov_groups$Variance ~ aov_groups$group)
aov_bet

# F = 80.035, num df = 3, denom df = 22652, p-value < 2.2e-16

# Test groups individually

# Day 0 and Day 1
t.test(humans_day0_var, humans_day1_var) # < 2.2e-16
# Day 1 and Day 2
t.test(humans_day1_var, humans_day2_var) # < 2.2e-16
# Day 2 and Day 3
t.test(humans_day2_var, humans_day3_var) # 7.301e-07

```

Conclusion: For both species, there is at least 1 mean (of the variances) that is different; however, when testing between the chimps day 0 and 1, there is not enough statistical evidence to suggest a difference in the means of the variances (p = 0.4586). Even if you make the argument that biologically we expect a reduction in variance between day 0 an day 1 (due to cannalization, for example) and therefore try a one-sided t-test, the p-value is still not statistically significant (p = 0.2293). 

## Characterizing variance trends between days

We are interested in the genes that experience either an increase or reduction in variance between days in both species. 

### Find genes for which variance in day 0 is higher than the variance for day 1 in both species

```{r}
#2 Gene-specific trend

humans_day0_var <- apply(as.data.frame(mean_tech_reps[,1:6]),1, var) 
chimps_day0_var <-apply(as.data.frame(mean_tech_reps[,7:10]),1, var)
humans_day1_var <- apply(as.data.frame(mean_tech_reps[,11:16]),1, var)

chimps_day1_var <- apply(as.data.frame(mean_tech_reps[,17:20]),1, var)

# Chimps: some genes have a higher variance at day 0 than day 1

plot(log10(chimps_day1_var), log10(chimps_day0_var), ylab = "Log10 variance for each gene- Chimps Day 0", xlab = "Log 10 variance for each gene- Chimps Day 1", main = "Per gene variance for Chimps")
abline(0,1, col = "red")
cor.test(log10(chimps_day1_var), log10(chimps_day0_var), method = "spearman")

# Humans: more genes have a higher variance at day 0 than day 1
plot(log10(humans_day1_var), log10(humans_day0_var), ylab = "Log10 variance for each gene- Humans Day 0", xlab = "Log 10 variance for each gene- Humans Day 1")
abline(0,1, col = "red")
cor.test(log10(humans_day1_var), log10(humans_day0_var), method = "spearman")

```

```{r}

# We are going to first use a parametric test

# Variance in day 0 > var in day 1

############# Chimps #####################

# Make an array to store the p-values

chimp_var_pval <- array(NA, dim = c(10304, 1))

for(i in 1:10304){
  x <- t(mean_tech_reps[i,7:10])
  y <- t(mean_tech_reps[i,17:20])
  htest <- var.test(x, y, alternative = c("greater"))
  chimp_var_pval[i,1] <- htest$p.value
}

hist(chimp_var_pval)

############## For humans

human_var_pval <- array(NA, dim = c(10304, 1))

for(i in 1:10304){
  x <- t(mean_tech_reps[i,1:6])
  y <- t(mean_tech_reps[i,11:16])
  htest <- var.test(x, y, alternative = c("greater"))
  human_var_pval[i,1] <- htest$p.value
}

hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
rownames(var_pval) <- rownames(mean_tech_reps)
colnames(var_pval) <- c("Chimpanzee", "Human")
summary(var_pval)

dim(var_pval)

# Set p-value

p_val <- 0.05

# P-val < 0.05 for chimps only
num_var_pval_chimps <- var_pval[ which(var_pval[,1] < p_val), ]
dim(num_var_pval_chimps)
                    
# P-val < 0.05 for humans only
num_var_pval_humans <- var_pval[ which(var_pval[,2] < p_val), ]
dim(num_var_pval_humans)

# Note: there are 612 with p-val < 0.05 in the chimps and 2,289 with p-val < 0.05 in the humans. 

# P-val < 0.05 for both humans and chimps (tested separately)

# num_var_pval <- var_pval[ which(var_pval[,1] < 0.05 & var_pval[,2] < 0.05), ]

# dim(num_var_pval)

# See the gene names of the 159 that overlap 

# rownames(num_var_pval)

```

Conclusion: 612 genes undergo a reduction in variance in the chimps between day 0 and day 1 (cutoff of p-value < 0.05) and 2,289 genes undergo a reduction in variance in the humans between day 0 and day 1 (cutoff of p-value < 0.05). 

We will have incomplete power if we take the overlap of the p-values of the F statistic. Therefore, we will use a method which looks at the p-values of the F statistics for chimps, then the p-values of the human F statistics given that it was significant in the chimps, and then calculate the estimate of sharing that accounts for incomplete power of the replication tests (Storey's pi0, Storey and Tibshirani 2003 as implemented in Nick Banovich et al's paper http://biorxiv.org/content/early/2016/12/05/091660). Storey and Tibshirani's method estimates the true proportion of null statistics from a given p-value distribution, pi0. We can then calculate the proportion of significant tests from a p-value distribution by calculating pi1 = 1 - pi0.  



```{r}
# Get the p-values for the chimps

num_var_pval_chimps_only <- as.data.frame(var_pval[ which(var_pval[,1] < 0.05), 1])
hist(num_var_pval_chimps_only[,1])

# Find the p-values of the human F-statistic given that it was significant in the chimps. 

subset_var <- as.data.frame(var_pval[ which(var_pval[,1] < 0.05), 1:2])
num_var_pval_chimp_given_chimp <- as.data.frame(subset_var[,1])
num_var_pval_human_given_chimp <- as.data.frame(subset_var[,2])
plot(subset_var)

#subset_var[ which(subset_var[,1] == subset_var[,2], 1:2])
              
hist(num_var_pval_human_given_chimp[,1])

```

### Make the qqplots of the p-vlaues from the F tests of the variances

Original code kindly provided by Bryce van de Geijn. 

```{r}
#### Function List #####                                                                                                                 

#pickpoints - used to subsample points for plotting                                                                                       
#addqqplot - adds a set of points to an existing plot                                                                                     
#newqqplot - adds a set of points to a new plot                                                                                           

####### Functions #######                                                                                                                 

#pickpoints can be used to subsample from a number of points, preferentially keeping                                                      
#more points from the end of the array.  This is useful for reducing the number of                                                        
#points in a QQ plot without changing what it looks like.                                                                                 
#Input 3 parameters:                                                                                                                      
#num.points - number of points to subsample                                                                                               
#always.plot - number of points to always keep from the end of the array (optional)                                                       
#density - the density of points to be kept (optional)                                                                                    
#Output:                                                                                                                                  
#boolean array (T/F) giving which points are kept                                                                                         

pickpoints.helper = function(x,env=new.env(),num.points=1000,always.plot=500,density=35){
  a=get("count",envir=env)
  if(num.points-x<=always.plot){
    return(T)
  }else{
    a=a+density/(num.points-x-always.plot)
    if(a >= 1){
      assign("count",a-1,envir=env)
      return(T)
    }
  }
  assign("count",a,envir=env)
  return(F)
}

pickpoints=function(num.points,always.plot=500,density=35){
  env=new.env()
  assign("count",1,envir=env)
  sapply(1:num.points,pickpoints.helper,env=env,num.points=num.points,always.plot=always.plot,density=density)
}

#addqqplot adds an additional set of qq points to an existing plot                                                                        
#Input 3 parameters:                                                                                                                      
#Required:                                                                                                                                
#pvals - log10 pvalues to be plotted                                                                                                      
#Optional for subsampling points (no subsampling if always.plot is -1):                                                                   
#always.plot - number of points to number of points to always keep from the end of the array                                              
#density - the density of points to be kept (optional)                                                                                    
#Output:                                                                                                                                  
#adds a set of qq points to an existing graph                                                                                             

addqqplot=function(pvals, always.plot,density, col_designated){
  len = length(pvals)
  res=qqplot(-log10((1:len)/(1+len)),pvals,plot.it=F)
  if(always.plot>=0){
    pp=pickpoints(len,always.plot,density)
    points(res$x[pp],res$y[pp], pch = 16, col = col_designated)
    return(res)
  }else{
    points(res$x,res$y, pch = 16, col = col_designated)
    return(res)
  }
}

#newqqplot creates a new plot with a set of qq points                                                                                     
#Input 3 parameters:                                                                                                                      
#Required:                                                                                                                                
#pvals - -log10 pvalues to be plotted                                                                                                     
#Optional for subsampling points (no subsampling if always.plot is -1):                                                                   
#always.plot - number of points to number of points to always keep from the end of the array                                              
#density - the density of points to be kept (optional)                                                                                    
#Output:                                                                                                                                  
#creates a new qq plot                                                                                                                    

newqqplot=function(pvals, always.plot,density){
  len = length(pvals)
  res=qqplot(-log10((1:len)/(1+len)),pvals,plot.it=F)
  if(always.plot>=0){
    pp=pickpoints(len,always.plot,density)
    plot(res$x[pp],res$y[pp], pch = 16, ylim = c(0, 10))
      return(res)
  }else{
    # Added axes labels
    plot(res$x,res$y, pch = 16, ylim = c(0, 6))
    return(res)
  }
}

# Run for the chimps
num_var_pval_chimps_neg_log <- -log10(var_pval[,1])

#qq_values <- -log10((1:10304)/(1+10304))
#plot(as.matrix(qq_values), as.matrix(num_var_pval_chimps_neg_log ))

res_chimp <- newqqplot(num_var_pval_chimps_neg_log, -1, 100)
# res_attempt <- newqqplot(num_var_pval_shared_neg_log[,1], -1, 100)

# Run for the shared
num_var_pval_shared_neg_log <- -log10(num_var_pval_human_given_chimp)

res_attempt <- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])
res_shared <- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])


legend("topleft", c("Chimpanzee", "Shared"),  col=c("black", "#4DAF4A"), pch = c(16,16, 16))
abline(0,1)

# Plot hist of p-values on top of each other (to make sure that we are in fact pulling from two different distributions)

hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


# Plot everything

max(res_chimp$x)
max(res_chimp$y)
max(res_shared$x)
max(res_shared$y)

plot(res_chimp$x,res_chimp$y, pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved")
abline(0,1, col = "black")


# Add to the plot
points(res_shared$x,res_shared$y, pch = 16, col = pal[3])

# Add a legend
legend("topleft", c("Chimpanzee", "Shared"),  col=pal[2:3], pch = c(16,16))


# QC 

# Make sure the values aren't the same
inshared_lists = res_attempt$x %in% res_shared$x
which(inshared_lists == TRUE)

inshared_lists = num_var_pval_chimps_neg_log %in% num_var_pval_shared_neg_log[,1]
which(inshared_lists == TRUE)

inshared_lists = human_var_pval %in% chimp_var_pval
which(inshared_lists == TRUE)

inshared_lists = chimp_var_pval %in% human_var_pval
which(inshared_lists == TRUE)

# Exploratory analysis
cor(chimp_var_pval, human_var_pval)
cor(subset_var[,1], subset_var[,2])
cor(-log10(subset_var[,1]), -log10(subset_var[,2]))
plot(subset_var[,1], subset_var[,2])
plot(subset_var[,1], subset_var[,2], ylim = c(0,0.05))
plot(-log10(subset_var[,1]), -log10(subset_var[,2]), xlim = c(0,5))
```

### Make plots of the -log10(variance) for all consecutive species-day combinations 

```{r}
# Find the variances for each species-day 
humans_day0_var <- apply(as.data.frame(mean_tech_reps[,1:6]),1, var) 
chimps_day0_var <-apply(as.data.frame(mean_tech_reps[,7:10]),1, var)
humans_day1_var <- apply(as.data.frame(mean_tech_reps[,11:16]),1, var)
chimps_day1_var <- apply(as.data.frame(mean_tech_reps[,17:20]),1, var)
humans_day2_var <- apply(as.data.frame(mean_tech_reps[,21:26]),1, var) 
chimps_day2_var <-apply(as.data.frame(mean_tech_reps[,27:30]),1, var)
humans_day3_var <- apply(as.data.frame(mean_tech_reps[,31:36]),1, var)
chimps_day3_var <- apply(as.data.frame(mean_tech_reps[,37:40]),1, var)

# Exploratory analysis of variances

#par(mfrow=c(2, 3))
# Day 0 versus 1
  # Chimps: some genes have a higher variance at day 0 than day 1

plot(-log10(chimps_day1_var), -log10(chimps_day0_var), ylab = "-log10 variance for each gene (Chimps at day 0)", xlab = "-log10 variance for each gene (Chimps at day 1)", main = "Per gene variance for Chimps (day 0 versus 1)")
abline(0,1, col = "red")
cor.test(-log10(chimps_day1_var), -log10(chimps_day0_var), method = "spearman")

# Day 1 versus day 2

  # Chimps
plot(-log10(chimps_day2_var), -log10(chimps_day1_var), ylab = "-log10 variance for each gene (Chimps at day 1)", xlab = "-log10 variance for each gene (Chimps at day 2)", main = "Per gene variance for Chimps (day 1 versus 2)")
abline(0,1, col = "red")
cor.test(-log10(chimps_day2_var), -log10(chimps_day1_var), method = "spearman")

# Day 2 versus day 3

  # Chimps
plot(-log10(chimps_day3_var), -log10(chimps_day2_var), ylab = "-log10 variance for each gene (Chimps at day 2)", xlab = "-log10 variance for each gene (Chimps at day 3)", main = "Per gene variance for Chimps (day 2 versus 3)")
abline(0,1, col = "red")
cor.test(-log10(chimps_day3_var), -log10(chimps_day2_var), method = "spearman")

# Day 0 versus 1

  # Humans: more genes have a higher variance at day 0 than day 1
plot(-log10(humans_day1_var), -log10(humans_day0_var), ylab = "-log10 variance for each gene (Humans at day 0)", xlab = "-log10 variance for each gene (Humans at day 1)",  main = "Per gene variance for Humans (day 0 versus 1)")
abline(0,1, col = "red")
cor.test(log10(humans_day1_var), log10(humans_day0_var), method = "spearman")

# Day 1 versus day 2


  # Humans
plot(-log10(humans_day2_var), -log10(humans_day1_var), ylab = "-log10 variance for each gene (Humans at day 1)", xlab = "-log10 variance for each gene (Humans at day 2)",  main = "Per gene variance for Humans (day 1 versus 2)")
abline(0,1, col = "red")
cor.test(log10(humans_day2_var), log10(humans_day1_var), method = "spearman")

# Day 2 versus day 3

  # Humans
plot(-log10(humans_day3_var), -log10(humans_day2_var), ylab = "-log10 variance for each gene (Humans at day 2)", xlab = "-log10 variance for each gene (Humans at day 3)",  main = "Per gene variance for Humans (day 2 versus 3)")
abline(0,1, col = "red")
cor.test(log10(humans_day3_var), log10(humans_day2_var), method = "spearman")

dev.off()

```

### General functions for reduction/increase in variance (with Human F statistics conditioned on Chimp samples)

```{r}
#Output:                                                                                                                                  
#adds a set of qq points to an existing graph                                                                                             

addqqplot=function(pvals, always.plot,density, col_designated){
  len = length(pvals)
  res=qqplot(-log10((1:len)/(1+len)),pvals,plot.it=F)
  return(res)
}

#newqqplot creates a new plot with a set of qq points                                                                                     
#Output:                                                                                                                                  
#creates a new qq plot                                                                                                                    

newqqplot=function(pvals, always.plot,density){
  len = length(pvals)
  res=qqplot(-log10((1:len)/(1+len)),pvals,plot.it=F)
}

# We need 4 values for each species: day t-1 start and end values and day t start and end values.
#par(mfrow=c(1, 2))
find_qqplot_red <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval[ which(var_pval[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval[ which(var_pval[,2] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
  num_var_pval_chimps_neg_log <- -log10(var_pval[,1])

  res_chimp <- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the human F-statistic given that it was significant in the chimps.   

  subset_var <- as.data.frame(var_pval[ which(var_pval[,1] < 0.05), 1:2])
  num_var_pval_human_given_chimp_no_df <- subset_var[,2]
  num_var_pval_chimp_given_chimp <- as.data.frame(subset_var[,1])
  num_var_pval_human_given_chimp <- as.data.frame(subset_var[,2])

  num_var_pval_shared_neg_log <- -log10(num_var_pval_human_given_chimp)

  res_shared <- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(chimp_var_pval, num_var_pval_human_given_chimp_no_df, sig_p_val, res_chimp$x, res_chimp$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}


find_qqplot_inc <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval[ which(var_pval[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval[ which(var_pval[,2] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
  num_var_pval_chimps_neg_log <- -log10(var_pval[,1])

  res_chimp <- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the human F-statistic given that it was significant in the chimps.   

  subset_var <- as.data.frame(var_pval[ which(var_pval[,1] < 0.05), 1:2])
  num_var_pval_chimp_given_chimp <- as.data.frame(subset_var[,1])
  num_var_pval_human_given_chimp_no_df <- subset_var[,2]
  num_var_pval_human_given_chimp <- as.data.frame(subset_var[,2])

  num_var_pval_shared_neg_log <- -log10(num_var_pval_human_given_chimp)

  res_shared <- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(chimp_var_pval, num_var_pval_human_given_chimp_no_df, sig_p_val, res_chimp$x, res_chimp$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}

par(mfrow=c(3, 2))
# Reduction in variance from day 0 to day 1

qqplot_red_day01 <- find_qqplot_red(7,10,17,20,1,6,11,16)

# Reduction in variance from day 1 to day 2

qqplot_red_day12 <- find_qqplot_red(17,20,27,30,11,16,21,26)

# Reduction in variance from day 2 to day 3

qqplot_red_day23 <- find_qqplot_red(27,30,37,40,21,26,31,36)

# Reduction in variance from day 0 to day 2

qqplot_red_day02 <- find_qqplot_red(7,10,27,30,1,6,21,26)

# Inc. in variance from day 0 to day 1

qqplot_inc_day01 <- find_qqplot_inc(7,10,17,20,1,6,11,16)

# Inc. in variance from day 1 to day 2

qqplot_inc_day12 <- find_qqplot_inc(17,20,27,30,11,16,21,26)

# Inc. in variance from day 2 to day 3

qqplot_inc_day23 <- find_qqplot_inc(27,30,37,40,21,26,31,36)

# Inc. in variance from day 0 to day 3

qqplot_inc_day02 <- find_qqplot_inc(7,10,27,30,1,6,21,26)


dev.off()

#par(mfrow=c(2, 3))

# Reduction of variance qqplots

par(mfrow=c(2,4),oma = c(2, 2, 2, 2))

# Reduction of variance to day 0 to 1
plot(qqplot_red_day01[[4]],qqplot_red_day01[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 0 to 1")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day01[[6]],qqplot_red_day01[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 1 to 2
plot(qqplot_red_day12[[4]],qqplot_red_day12[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 1 to 2")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day12[[6]],qqplot_red_day12[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 2 to 3
plot(qqplot_red_day23[[4]],qqplot_red_day23[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 2 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day23[[6]],qqplot_red_day23[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 0 to 2
plot(qqplot_red_day02[[4]],qqplot_red_day02[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 0 to 2")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day02[[6]],qqplot_red_day02[[7]], pch = 16, col = pal[3])

# Add bottom title

mtext("Reduction in Variation", font = 2, side = 3, line = -1.5, outer = TRUE)

### Increase in variance

# Inc variance to day 0 to 1
plot(qqplot_inc_day01[[4]],qqplot_inc_day01[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 0 to 1")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_inc_day01[[6]],qqplot_inc_day01[[7]], pch = 16, col = pal[3])

# Increase of variance to day 1 to 2
plot(qqplot_inc_day12[[4]],qqplot_inc_day12[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 1 to 2")
abline(0,1, col = "black")



# Add to the plot
points(qqplot_inc_day12[[6]],qqplot_inc_day12[[7]], pch = 16, col = pal[3])


# Increase of variance to day 2 to 3
plot(qqplot_inc_day23[[4]],qqplot_inc_day23[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 2 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_inc_day23[[6]],qqplot_inc_day23[[7]], pch = 16, col = pal[3])

# Increase of variance to day 0 to 2
plot(qqplot_inc_day02[[4]],qqplot_inc_day02[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 0 to 2")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_inc_day02[[6]],qqplot_inc_day02[[7]], pch = 16, col = pal[3])


mtext("Increase in Variation", font = 2, side = 3, line = -29, outer = TRUE)

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")
legend("bottomright", c("Chimpanzee", "Shared"), xpd = TRUE, horiz = TRUE, inset = c(0, 
    0), bty = "n", pch = c(16, 16), col = pal[2:3], cex = 1.5)

dev.off()

```

### Repeat test of reduction/increase in variance with Chimp F statistics conditioned on Human samples

```{r}
### A function for testing the reduction in variance with chimp F statistics being conditioned on human samples (above the human F statistics are conditioned on the chimp samples)

find_qqplot_red_shared_chimp <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval[ which(var_pval[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval[ which(var_pval[,2] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the humans
  num_var_pval_humans_neg_log <- -log10(var_pval[,2])

  res_human <- newqqplot(num_var_pval_humans_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the human F-statistic given that it was significant in the chimps.   

  subset_var <- as.data.frame(var_pval[ which(var_pval[,2] < 0.05), 1:2])
  num_var_pval_chimp_given_human_no_df <- subset_var[,1]
  num_var_pval_chimp_given_human <- as.data.frame(subset_var[,1])

  num_var_pval_shared_neg_log <- -log10(num_var_pval_chimp_given_human)

  res_shared <- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(human_var_pval, num_var_pval_chimp_given_human_no_df, sig_p_val, res_human$x, res_human$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}

# Now, we will write a function for determining an increase in variation

find_qqplot_inc_shared_chimp <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval[ which(var_pval[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval[ which(var_pval[,2] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the humans
  num_var_pval_humans_neg_log <- -log10(var_pval[,2])

  res_human <- newqqplot(num_var_pval_humans_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the human F-statistic given that it was significant in the chimps.   

  subset_var <- as.data.frame(var_pval[ which(var_pval[,2] < 0.05), 1:2])
  num_var_pval_chimp_given_human <- as.data.frame(subset_var[,1])
  num_var_pval_chimp_given_human_no_df <- subset_var[,1]
  num_var_pval_chimp_given_human <- as.data.frame(subset_var[,1])

  num_var_pval_shared_neg_log <- -log10(num_var_pval_chimp_given_human)

  res_shared <- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(human_var_pval, num_var_pval_chimp_given_human_no_df, sig_p_val, res_human$x, res_human$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}


# Reduction in variance from day 0 to day 1

qqplot_red_day01_shared_chimp <- find_qqplot_red_shared_chimp(7,10,17,20,1,6,11,16)

# Reduction in variance from day 1 to day 2

qqplot_red_day12_shared_chimp <- find_qqplot_red_shared_chimp(17,20,27,30,11,16,21,26)

# Reduction in variance from day 2 to day 3

qqplot_red_day23_shared_chimp <- find_qqplot_red_shared_chimp(27,30,37,40,21,26,31,36)

# Reduction in variance from day 0 to day 2

qqplot_red_day02_shared_chimp <- find_qqplot_red_shared_chimp(7,10,27,30,1,6,21,26)

# Inc. in variance from day 0 to day 1

qqplot_inc_day01_shared_chimp <- find_qqplot_inc_shared_chimp(7,10,17,20,1,6,11,16)

# Inc. in variance from day 1 to day 2

qqplot_inc_day12_shared_chimp <- find_qqplot_inc_shared_chimp(17,20,27,30,11,16,21,26)

# Inc. in variance from day 2 to day 3

qqplot_inc_day23_shared_chimp <- find_qqplot_inc_shared_chimp(27,30,37,40,21,26,31,36)

# Inc. in variance from day 0 to day 2

qqplot_inc_day02_shared_chimp <- find_qqplot_inc_shared_chimp(7,10,27,30,1,6,21,26)


dev.off()

#par(mfrow=c(2, 3))

# Reduction of variance qqplots

par(mfrow=c(2,4),oma = c(2, 2, 2, 2))

# Reduction of variance to day 0 to 1
plot(qqplot_red_day01_shared_chimp[[4]],qqplot_red_day01_shared_chimp[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 0 to 1")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day01_shared_chimp[[6]],qqplot_red_day01_shared_chimp[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 1 to 2
plot(qqplot_red_day12_shared_chimp[[4]],qqplot_red_day12_shared_chimp[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 1 to 2")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day12_shared_chimp[[6]],qqplot_red_day12_shared_chimp[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 2 to 3
plot(qqplot_red_day23_shared_chimp[[4]],qqplot_red_day23_shared_chimp[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 2 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day23_shared_chimp[[6]],qqplot_red_day23_shared_chimp[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 0 to 2
plot(qqplot_red_day02_shared_chimp[[4]],qqplot_red_day02_shared_chimp[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 0 to 2")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day02_shared_chimp[[6]],qqplot_red_day02_shared_chimp[[7]], pch = 16, col = pal[3])

# Add bottom title

mtext("Reduction in Variation", font = 2, side = 3, line = -1.5, outer = TRUE)

### Increase in variance

# Inc variance to day 0 to 1
plot(qqplot_inc_day01_shared_chimp[[4]],qqplot_inc_day01_shared_chimp[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 0 to 1")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_inc_day01_shared_chimp[[6]],qqplot_inc_day01_shared_chimp[[7]], pch = 16, col = pal[3])

# Increase of variance to day 1 to 2
plot(qqplot_inc_day12_shared_chimp[[4]],qqplot_inc_day12_shared_chimp[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 1 to 2")
abline(0,1, col = "black")



# Add to the plot
points(qqplot_inc_day12_shared_chimp[[6]],qqplot_inc_day12_shared_chimp[[7]], pch = 16, col = pal[3])


# Increase of variance to day 2 to 3
plot(qqplot_inc_day23_shared_chimp[[4]],qqplot_inc_day23_shared_chimp[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 2 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_inc_day23_shared_chimp[[6]],qqplot_inc_day23_shared_chimp[[7]], pch = 16, col = pal[3])

# Increase of variance to day 0 to 3
plot(qqplot_inc_day02_shared_chimp[[4]],qqplot_inc_day02_shared_chimp[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 0 to 2")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_inc_day02_shared_chimp[[6]],qqplot_inc_day02_shared_chimp[[7]], pch = 16, col = pal[3])


mtext("Increase in Variation", font = 2, side = 3, line = -29, outer = TRUE)

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")
legend("bottomright", c("Human", "Shared"), xpd = TRUE, horiz = TRUE, inset = c(0, 
    0), bty = "n", pch = c(16, 16), col = pal[2:3], cex = 1.5)

dev.off()
```





### Repeat chimpanzees conditioned on humans with a non-parametric test (Kruskal-Wallis Test)

Might be important to use a non-parametric test because of sample size. The downside is that we can only assess different than 0.


```{r}
dev.off()
# Normal distribution?
boxplot(t(mean_tech_reps[1:50,1:6]),  xaxt="n", ylab = "Log2(CPM) Expression", xlab = "OrthoExon Genes 1-50 (Day 0 Humans)", main = "Gene Expression Values for 50 Genes (Day 0 Humans)")

boxplot(t(mean_tech_reps[1:50,7:10]), xaxt="n")

#htest <- kruskal.test(list(x, y))

find_qqplot_kruskal <- function(chimp_var_begin_day, chimp_var_end_day, human_var_begin_day, human_var_end_day){

  # Make variances data frames
  chimp_var_begin_day <- as.data.frame(chimp_var_begin_day)
  chimp_var_end_day <- as.data.frame(chimp_var_end_day)
  human_var_begin_day <- as.data.frame(chimp_var_begin_day)
  human_var_end_day <- as.data.frame(chimp_var_end_day)
  
  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- chimp_var_begin_day[i,1]
    y <- chimp_var_end_day[i,1]
    htest <- kruskal.test(list(x, y))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- kruskal.test(list(x, y))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval[ which(var_pval[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval[ which(var_pval[,2] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
  num_var_pval_chimps_neg_log <- -log10(var_pval[,1])

  res_chimp <- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the human F-statistic given that it was significant in the chimps.   

  subset_var <- as.data.frame(var_pval[ which(var_pval[,1] < 0.05), 1:2])
  num_var_pval_human_given_chimp_no_df <- subset_var[,2]
  num_var_pval_chimp_given_chimp <- as.data.frame(subset_var[,1])
  num_var_pval_human_given_chimp <- as.data.frame(subset_var[,2])

  num_var_pval_shared_neg_log <- -log10(num_var_pval_human_given_chimp)

  res_shared <- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(chimp_var_pval, num_var_pval_human_given_chimp_no_df, sig_p_val, res_chimp$x, res_chimp$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}


# Reduction in variance from day 0 to day 1

qqplot_kruskal_day01 <- find_qqplot_kruskal(7,10,17,20,1,6,11,16)

# Reduction in variance from day 1 to day 2

qqplot_kruskal_day12 <- find_qqplot_kruskal(17,20,27,30,11,16,21,26)

# Reduction in variance from day 2 to day 3

qqplot_kruskal_day23 <- find_qqplot_kruskal(27,30,37,40,21,26,31,36)


dev.off()

par(mfrow=c(2,4),oma = c(2, 2, 2, 2))

# Reduction of variance to day 0 to 1
plot(qqplot_red_day01[[4]],qqplot_red_day01[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 0 to 1")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day01[[6]],qqplot_red_day01[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 1 to 2
plot(qqplot_red_day12[[4]],qqplot_red_day12[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 1 to 2")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day12[[6]],qqplot_red_day12[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 2 to 3
plot(qqplot_red_day23[[4]],qqplot_red_day23[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 2 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day23[[6]],qqplot_red_day23[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 0 to 3
plot(qqplot_red_day03[[4]],qqplot_red_day03[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 0 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day03[[6]],qqplot_red_day03[[7]], pch = 16, col = pal[3])

# Add bottom title

mtext("Difference in Distributions", font = 2, side = 3, line = -1.5, outer = TRUE)


```

### Repeat with a non-parametric test (Mood)

```{r}
find_qqplot_red_mood <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- ansari.test(x, y, alternative = c("greater"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- ansari.test(x, y, alternative = c("greater"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval[ which(var_pval[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval[ which(var_pval[,2] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
  num_var_pval_chimps_neg_log <- -log10(var_pval[,1])

  res_chimp <- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the human F-statistic given that it was significant in the chimps.   

  subset_var <- as.data.frame(var_pval[ which(var_pval[,1] < 0.05), 1:2])
  num_var_pval_human_given_chimp_no_df <- subset_var[,2]
  num_var_pval_chimp_given_chimp <- as.data.frame(subset_var[,1])
  num_var_pval_human_given_chimp <- as.data.frame(subset_var[,2])

  num_var_pval_shared_neg_log <- -log10(num_var_pval_human_given_chimp)

  res_shared <- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(chimp_var_pval, num_var_pval_human_given_chimp_no_df, sig_p_val, res_chimp$x, res_chimp$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}


find_qqplot_inc_mood <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- ansari.test(x, y, alternative = c("less"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- ansari.test(x, y, alternative = c("less"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval[ which(var_pval[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval[ which(var_pval[,2] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
  num_var_pval_chimps_neg_log <- -log10(var_pval[,1])

  res_chimp <- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the human F-statistic given that it was significant in the chimps.   

  subset_var <- as.data.frame(var_pval[ which(var_pval[,1] < 0.05), 1:2])
  num_var_pval_chimp_given_chimp <- as.data.frame(subset_var[,1])
  num_var_pval_human_given_chimp_no_df <- subset_var[,2]
  num_var_pval_human_given_chimp <- as.data.frame(subset_var[,2])

  num_var_pval_shared_neg_log <- -log10(num_var_pval_human_given_chimp)

  res_shared <- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(chimp_var_pval, num_var_pval_human_given_chimp_no_df, sig_p_val, res_chimp$x, res_chimp$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}

dev.off()
# Reduction in variance from day 0 to day 1

qqplot_red_day01 <- find_qqplot_red_mood(7,10,17,20,1,6,11,16)

# Reduction in variance from day 1 to day 2

qqplot_red_day12 <- find_qqplot_red_mood(17,20,27,30,11,16,21,26)

# Reduction in variance from day 2 to day 3

qqplot_red_day23 <- find_qqplot_red_mood(27,30,37,40,21,26,31,36)

# Reduction in variance from day 0 to day 3

qqplot_red_day03 <- find_qqplot_red_mood(7,10,37,40,1,6,31,36)

# Inc. in variance from day 0 to day 1

qqplot_inc_day01 <- find_qqplot_inc_mood(7,10,17,20,1,6,11,16)

# Inc. in variance from day 1 to day 2

qqplot_inc_day12 <- find_qqplot_inc_mood(17,20,27,30,11,16,21,26)

# Inc. in variance from day 2 to day 3

qqplot_inc_day23 <- find_qqplot_inc_mood(27,30,37,40,21,26,31,36)

# Inc. in variance from day 0 to day 3

qqplot_inc_day03 <- find_qqplot_inc_mood(7,10,37,40,1,6,31,36)


dev.off()

#par(mfrow=c(2, 3))

# Reduction of variance qqplots

par(mfrow=c(2,4),oma = c(2, 2, 2, 2))

# Reduction of variance to day 0 to 1
plot(qqplot_red_day01[[4]],qqplot_red_day01[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 0 to 1")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day01[[6]],qqplot_red_day01[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 1 to 2
plot(qqplot_red_day12[[4]],qqplot_red_day12[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 1 to 2")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day12[[6]],qqplot_red_day12[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 2 to 3
plot(qqplot_red_day23[[4]],qqplot_red_day23[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 2 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day23[[6]],qqplot_red_day23[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 0 to 3
plot(qqplot_red_day03[[4]],qqplot_red_day03[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved", main = "Day 0 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_red_day03[[6]],qqplot_red_day03[[7]], pch = 16, col = pal[3])

# Add bottom title

mtext("Reduction in Variation", font = 2, side = 3, line = -1.5, outer = TRUE)

### Increase in variance

# Inc variance to day 0 to 1
plot(qqplot_inc_day01[[4]],qqplot_inc_day01[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 0 to 1")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_inc_day01[[6]],qqplot_inc_day01[[7]], pch = 16, col = pal[3])

# Increase of variance to day 1 to 2
plot(qqplot_inc_day12[[4]],qqplot_inc_day12[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 1 to 2")
abline(0,1, col = "black")



# Add to the plot
points(qqplot_inc_day12[[6]],qqplot_inc_day12[[7]], pch = 16, col = pal[3])


# Increase of variance to day 2 to 3
plot(qqplot_inc_day23[[4]],qqplot_inc_day23[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 2 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_inc_day23[[6]],qqplot_inc_day23[[7]], pch = 16, col = pal[3])

# Increase of variance to day 0 to 3
plot(qqplot_inc_day03[[4]],qqplot_inc_day03[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = "-log10(p) Expected", ylab = "-log10(p) Obeserved",  main = "Day 0 to 3")
abline(0,1, col = "black")


# Add to the plot
points(qqplot_inc_day03[[6]],qqplot_inc_day03[[7]], pch = 16, col = pal[3])


mtext("Increase in Variation", font = 2, side = 3, line = -29, outer = TRUE)

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")
legend("bottomright", c("Chimpanzee", "Shared"), xpd = TRUE, horiz = TRUE, inset = c(0, 
    0), bty = "n", pch = c(16, 16), col = pal[2:3], cex = 1.5)

dev.off()

```


### Estimate pi0 and pi1; make the density plots with p0 hat (humans conditioned on chimps)
```{r}
# Pi_0 estimate for the chimps
pi0est(var_pval[,1], lambda = seq(0.05, 0.95, 0.05))
pi0est_chimps <- pi0est(var_pval[,1], lambda = seq(0.05, 0.95, 0.05))$pi0

qobj_chimps <- qvalue(var_pval[,1])
hist(qobj_chimps)

# Pi_1 estimate for the chimps

pi1est_chimps <- 1 - pi0est_chimps
pi1est_chimps

# Pi_0 estimate for the humans given significant in the chimps
pi0est(num_var_pval_human_given_chimp, lambda = seq(0.05, 0.95, 0.05))

pi0est_humans_given_chimps <- pi0est(num_var_pval_human_given_chimp, lambda = seq(0.05, 0.95, 0.05))$pi0

qobj_humans_given_chimps <- qvalue(num_var_pval_human_given_chimp)
hist(qobj_humans_given_chimps)

# Pi_1 estimate for the humans

pi1est_humans_given_chimps <- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps
pi1est_humans_given_chimps*as.data.frame(dim(num_var_pval_human_given_chimp))[1,1]

# Let's overlap the two histograms

abline(1,0)
hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_c <- array("Chimpanzee", dim = c(as.data.frame(dim(var_pval))[1,1]),1)
labels_h <- array("Shared (Human|Chimpanzee)", dim = c(as.data.frame(dim(num_var_pval_human_given_chimp))[1,1]))

chimps <- cbind(labels_c, var_pval)
colnames(chimps) <- c("Species", "pval")
humans <- cbind(labels_h, num_var_pval_human_given_chimp)
colnames(humans) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(chimps[,1:2], humans[,1:2])


p <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = T, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() + 
  geom_hline(yintercept = pi0est_humans_given_chimps) + 
  geom_hline(yintercept = pi0est_chimps) +       
  annotate("text", label='italic(hat(pi)[0][", chimp"])=="0.74"', parse=TRUE, x=0.6, y=(pi0est_chimps+0.05)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.12"', parse=TRUE, x=0.6, y=(pi0est_humans_given_chimps+0.05))

p
  

# Why does density extend past 0 and 1? I think it's a feature of ggplot2 plotting because the histograms do that even though the p-values are clearly in the range of 0 to 1

summary(all_pval)
ggplot(all_pval, aes(pval, fill = Species)) +
  geom_histogram(show.legend = T, alpha = 0.3) + scale_x_continuous(limits = c(-1,1.5), breaks = seq(0,1,0.25))


```

### Make density plots with p0 hat for all species-day combinations (humans conditioned on chimps)

```{r}
################ REDUCTION IN VARIATION ################

# Day 0 to 1


# Obtain pvalues
chimp_p_values_gen <- as.data.frame(qqplot_red_day01[1])
shared_p_values_gen <- as.data.frame(qqplot_red_day01[2])


# Pi_0 estimate for the chimps

pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_chimps <- pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps

# Pi_1 estimate for the chimps

pi1est_chimps <- 1 - pi0est_chimps
pi1est_chimps

# Pi_0 estimate for the humans given significant in the chimps
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_humans_given_chimps <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_humans_given_chimps

# Pi_1 estimate for the humans

pi1est_humans_given_chimps <- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps
pi1est_humans_given_chimps*as.data.frame(qqplot_red_day01[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_c <- array("Chimpanzee", dim = c(10304,1))
labels_h <- array("Shared (Human|Chimpanzee)", dim = c(as.data.frame(qqplot_red_day01[[3]][1]),1))

chimps <- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) <- c("Species", "pval")
humans <- cbind(labels_h, shared_p_values_gen)
colnames(humans) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(chimps[,1:2], humans[,1:2])


p1 <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 0 to 1") +
  geom_hline(yintercept = pi0est_humans_given_chimps) + 
  geom_hline(yintercept = pi0est_chimps) +       
  annotate("text", label='italic(hat(pi)[0][", chimp"])=="0.74"', parse=TRUE, x=0.6, y=(pi0est_chimps+0.07)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.12"', parse=TRUE, x=0.6, y=(pi0est_humans_given_chimps+0.07))

p1


# Day 1 to 2

# Obtain pvalues
chimp_p_values_gen <- as.data.frame(qqplot_red_day12[1])
shared_p_values_gen <- as.data.frame(qqplot_red_day12[2])


# Pi_0 estimate for the chimps

pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_chimps <- pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps

# Pi_1 estimate for the chimps

pi1est_chimps <- 1 - pi0est_chimps
pi1est_chimps

# Pi_0 estimate for the humans given significant in the chimps
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_humans_given_chimps <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_humans_given_chimps

# Pi_1 estimate for the humans

pi1est_humans_given_chimps <- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps
pi1est_humans_given_chimps*as.data.frame(qqplot_red_day12[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_c <- array("Chimpanzee", dim = c(10304,1))
labels_h <- array("Shared (Human|Chimpanzee)", dim = c(as.data.frame(qqplot_red_day12[[3]][1]),1))

chimps <- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) <- c("Species", "pval")
humans <- cbind(labels_h, shared_p_values_gen)
colnames(humans) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(chimps[,1:2], humans[,1:2])


p2 <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() + labs(title = "Day 1 to 2") +
  geom_hline(yintercept = pi0est_humans_given_chimps) + 
  geom_hline(yintercept = pi0est_chimps) +       
  annotate("text", label='italic(hat(pi)[0][", chimp"])=="1"', parse=TRUE, x=0.8, y=(pi0est_chimps+0.05)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="1"', parse=TRUE, x=0.8, y=(pi0est_humans_given_chimps-0.05))

p2
  
# Day 2 to 3

# Obtain pvalues
chimp_p_values_gen <- as.data.frame(qqplot_red_day23[1])
shared_p_values_gen <- as.data.frame(qqplot_red_day23[2])


# Pi_0 estimate for the chimps

pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_chimps <- pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps

# Pi_1 estimate for the chimps

pi1est_chimps <- 1 - pi0est_chimps
pi1est_chimps

# Pi_0 estimate for the humans given significant in the chimps
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_humans_given_chimps <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_humans_given_chimps

# Pi_1 estimate for the humans

pi1est_humans_given_chimps <- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps
pi1est_humans_given_chimps*as.data.frame(qqplot_red_day23[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_c <- array("Chimpanzee", dim = c(10304,1))
labels_h <- array("Shared (Human|Chimpanzee)", dim = c(as.data.frame(qqplot_red_day23[[3]][1]),1))

chimps <- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) <- c("Species", "pval")
humans <- cbind(labels_h, shared_p_values_gen)
colnames(humans) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(chimps[,1:2], humans[,1:2])


p3 <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() + labs(title = "Day 2 to 3") +
  geom_hline(yintercept = pi0est_humans_given_chimps) + 
  geom_hline(yintercept = pi0est_chimps) +       
  annotate("text", label='italic(hat(pi)[0][", chimp"])=="0.91"', parse=TRUE, x=0.65, y=(pi0est_chimps+0.05)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.71"', parse=TRUE, x=0.65, y=(pi0est_humans_given_chimps+0.05))

p3
  

# Day 0 to 2

# Obtain pvalues
chimp_p_values_gen <- as.data.frame(qqplot_red_day02[1])
shared_p_values_gen <- as.data.frame(qqplot_red_day02[2])


# Pi_0 estimate for the chimps

pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_chimps <- pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps

# Pi_1 estimate for the chimps

pi1est_chimps <- 1 - pi0est_chimps
pi1est_chimps

# Pi_0 estimate for the humans given significant in the chimps
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_humans_given_chimps <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_humans_given_chimps

# Pi_1 estimate for the humans

pi1est_humans_given_chimps <- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps
pi1est_humans_given_chimps*as.data.frame(qqplot_red_day02[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_c <- array("Chimpanzee", dim = c(10304,1))
labels_h <- array("Shared (Human|Chimpanzee)", dim = c(as.data.frame(qqplot_red_day02[[3]][1]),1))

chimps <- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) <- c("Species", "pval")
humans <- cbind(labels_h, shared_p_values_gen)
colnames(humans) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(chimps[,1:2], humans[,1:2])


p4 <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() + labs(title = "Day 0 to 2") +
  geom_hline(yintercept = pi0est_humans_given_chimps) + 
  geom_hline(yintercept = pi0est_chimps) +       
  annotate("text", label='italic(hat(pi)[0][", chimp"])=="1"', parse=TRUE, x=0.85, y=(pi0est_chimps+0.05)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.72"', parse=TRUE, x=0.85, y=(pi0est_humans_given_chimps+0.05))

p4



################ INCREASE IN VARIATION ################

# Day 0 to 1


# Obtain pvalues
chimp_p_values_gen <- as.data.frame(qqplot_inc_day01[1])
shared_p_values_gen <- as.data.frame(qqplot_inc_day01[2])


# Pi_0 estimate for the chimps

pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_chimps <- pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps

# Pi_1 estimate for the chimps

pi1est_chimps <- 1 - pi0est_chimps
pi1est_chimps

# Pi_0 estimate for the humans given significant in the chimps
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_humans_given_chimps <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_humans_given_chimps

# Pi_1 estimate for the humans

pi1est_humans_given_chimps <- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps
pi1est_humans_given_chimps*as.data.frame(qqplot_red_day01[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_c <- array("Chimpanzee", dim = c(10304,1))
labels_h <- array("Shared (Human|Chimpanzee)", dim = c(as.data.frame(qqplot_inc_day01[[3]][1]),1))

chimps <- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) <- c("Species", "pval")
humans <- cbind(labels_h, shared_p_values_gen)
colnames(humans) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(chimps[,1:2], humans[,1:2])


p5 <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 0 to 1") +
  geom_hline(yintercept = pi0est_humans_given_chimps) + 
  geom_hline(yintercept = pi0est_chimps) +       
  annotate("text", label='italic(hat(pi)[0][", chimp"])=="1"', parse=TRUE, x=0.2, y=(pi0est_chimps+0.05)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="1"', parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps-0.05))

p5


# Day 1 to 2

# Obtain pvalues
chimp_p_values_gen <- as.data.frame(qqplot_inc_day12[1])
shared_p_values_gen <- as.data.frame(qqplot_inc_day12[2])


# Pi_0 estimate for the chimps

pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_chimps <- pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps

# Pi_1 estimate for the chimps

pi1est_chimps <- 1 - pi0est_chimps
pi1est_chimps

# Pi_0 estimate for the humans given significant in the chimps
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_humans_given_chimps <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_humans_given_chimps

# Pi_1 estimate for the humans

pi1est_humans_given_chimps <- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps
pi1est_humans_given_chimps*as.data.frame(qqplot_inc_day12[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_c <- array("Chimpanzee", dim = c(10304,1))
labels_h <- array("Shared (Human|Chimpanzee)", dim = c(as.data.frame(qqplot_inc_day12[[3]][1]),1))

chimps <- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) <- c("Species", "pval")
humans <- cbind(labels_h, shared_p_values_gen)
colnames(humans) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(chimps[,1:2], humans[,1:2])


p6 <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() + labs(title = "Day 1 to 2") +
  geom_hline(yintercept = pi0est_humans_given_chimps) + 
  geom_hline(yintercept = pi0est_chimps) +       
  annotate("text", label='italic(hat(pi)[0][", chimp"])=="0.52"', parse=TRUE, x=0.2, y=(pi0est_chimps+0.07)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.11"', parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.07))

p6
  
# Day 2 to 3

# Obtain pvalues
chimp_p_values_gen <- as.data.frame(qqplot_inc_day23[1])
shared_p_values_gen <- as.data.frame(qqplot_inc_day23[2])


# Pi_0 estimate for the chimps

pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_chimps <- pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps

# Pi_1 estimate for the chimps

pi1est_chimps <- 1 - pi0est_chimps
pi1est_chimps

# Pi_0 estimate for the humans given significant in the chimps
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_humans_given_chimps <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_humans_given_chimps

# Pi_1 estimate for the humans

pi1est_humans_given_chimps <- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps
pi1est_humans_given_chimps*as.data.frame(qqplot_inc_day23[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_c <- array("Chimpanzee", dim = c(10304,1))
labels_h <- array("Shared (Human|Chimpanzee)", dim = c(as.data.frame(qqplot_inc_day23[[3]][1]),1))

chimps <- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) <- c("Species", "pval")
humans <- cbind(labels_h, shared_p_values_gen)
colnames(humans) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(chimps[,1:2], humans[,1:2])


p7 <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() + labs(title = "Day 2 to 3") +
  geom_hline(yintercept = pi0est_chimps) + 
  geom_hline(yintercept = pi0est_humans_given_chimps) +       
  annotate("text", label='italic(hat(pi)[0][", chimp"])=="0.37"', parse=TRUE, x=0.3, y=(pi0est_chimps+0.06)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.75"', parse=TRUE, x=0.3, y=(pi0est_humans_given_chimps+0.06))

p7
  

# Day 0 to 2

# Obtain pvalues
chimp_p_values_gen <- as.data.frame(qqplot_inc_day02[1])
shared_p_values_gen <- as.data.frame(qqplot_inc_day02[2])


# Pi_0 estimate for the chimps

pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_chimps <- pi0est(chimp_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps

# Pi_1 estimate for the chimps

pi1est_chimps <- 1 - pi0est_chimps
pi1est_chimps

# Pi_0 estimate for the humans given significant in the chimps
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_humans_given_chimps <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_humans_given_chimps

# Pi_1 estimate for the humans

pi1est_humans_given_chimps <- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps
pi1est_humans_given_chimps*as.data.frame(qqplot_inc_day03[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_c <- array("Chimpanzee", dim = c(10304,1))
labels_h <- array("Shared (Human|Chimpanzee)", dim = c(as.data.frame(qqplot_inc_day02[[3]][1]),1))

chimps <- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) <- c("Species", "pval")
humans <- cbind(labels_h, shared_p_values_gen)
colnames(humans) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(chimps[,1:2], humans[,1:2])


p8 <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() + labs(title = "Day 0 to 2") +
  geom_hline(yintercept = pi0est_humans_given_chimps) + 
  geom_hline(yintercept = pi0est_chimps) +       
  annotate("text", label='italic(hat(pi)[0][", chimp"])=="0.99"', parse=TRUE, x=0.7, y=(pi0est_chimps+0.07)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.90"', parse=TRUE, x=0.7, y=(pi0est_humans_given_chimps-0.07))

p8

# Make multiple ggplots on the same page

# Multiplot function (from http://www.cookbook-r.com/Graphs/Multiple_graphs_on_one_page_%28ggplot2%29/)

multiplot <- function(..., plotlist=NULL, file, cols=1, layout=NULL) {
  library(grid)

  # Make a list from the ... arguments and plotlist
  plots <- c(list(...), plotlist)

  numPlots = length(plots)

  # If layout is NULL, then use 'cols' to determine layout
  if (is.null(layout)) {
    # Make the panel
    # ncol: Number of columns of plots
    # nrow: Number of rows needed, calculated from # of cols
    layout <- matrix(seq(1, cols * ceiling(numPlots/cols)),
                    ncol = cols, nrow = ceiling(numPlots/cols))
  }

 if (numPlots==1) {
    print(plots[[1]])

  } else {
    # Set up the page
    grid.newpage()
    pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))

    # Make each plot, in the correct location
    for (i in 1:numPlots) {
      # Get the i,j matrix positions of the regions that contain this subplot
      matchidx <- as.data.frame(which(layout == i, arr.ind = TRUE))

      print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,
                                      layout.pos.col = matchidx$col))
    }
  }
}


multiplot(p1,p5,p2,p6,p3,p7,p4,p8,cols=4)

```

### Make density plots with p0 hat for all species-day combinations (chimps conditioned on humans)

```{r}
################ REDUCTION IN VARIATION ################

# Day 0 to 1


# Obtain pvalues
human_p_values_gen <- as.data.frame(qqplot_red_day01_shared_chimp[1])
shared_p_values_gen <- as.data.frame(qqplot_red_day01_shared_chimp[2])


# Pi_0 estimate for the humans

pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_human <- pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_human

# Pi_1 estimate for the humans

pi1est_human <- 1 - pi0est_human
pi1est_human

# Pi_0 estimate for the chimps given significant in the humans
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_chimps_given_humans <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps_given_humans

# Pi_1 estimate for the chimps

pi1est_chimps_given_humans <- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans
pi1est_chimps_given_humans*as.data.frame(qqplot_red_day01_shared_chimp[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_h <- array("Human", dim = c(10304,1))
labels_c <- array("Shared (Chimpanzee|Human)", dim = c(as.data.frame(qqplot_red_day01[[3]][2]),1))

humans <- cbind(labels_h, human_p_values_gen)
colnames(humans) <- c("Species", "pval")
chimps <- cbind(labels_c, shared_p_values_gen)
colnames(chimps) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(humans[,1:2], chimps[,1:2])


p1c <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 0 to 1") +
  geom_hline(yintercept = pi0est_humans_given_chimps) + 
  geom_hline(yintercept = pi0est_human) +       
  annotate("text", label='italic(hat(pi)[0][", human"])=="0.17"', parse=TRUE, x=0.25, y=(pi0est_human+0.08)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.52"', parse=TRUE, x=0.25, y=(pi0est_chimps_given_humans+0.08))

p1c


# Day 1 to 2

# Obtain pvalues
human_p_values_gen <- as.data.frame(qqplot_red_day12_shared_chimp[1])
shared_p_values_gen <- as.data.frame(qqplot_red_day12_shared_chimp[2])


# Pi_0 estimate for the humans

pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_human <- pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_human

# Pi_1 estimate for the humans

pi1est_human <- 1 - pi0est_human
pi1est_human

# Pi_0 estimate for the chimps given significant in the humans
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_chimps_given_humans <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps_given_humans

# Pi_1 estimate for the chimps

pi1est_chimps_given_humans <- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans
pi1est_chimps_given_humans*as.data.frame(qqplot_red_day12_shared_chimp[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_h <- array("Human", dim = c(10304,1))
labels_c <- array("Shared (Chimpanzee|Human)", dim = c(as.data.frame(qqplot_red_day12[[3]][2]),1))

humans <- cbind(labels_h, human_p_values_gen)
colnames(humans) <- c("Species", "pval")
chimps <- cbind(labels_c, shared_p_values_gen)
colnames(chimps) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(humans[,1:2], chimps[,1:2])


p2c <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 1 to 2") +
  geom_hline(yintercept = pi0est_human) + 
  geom_hline(yintercept = pi0est_chimps_given_humans) +       
  annotate("text", label='italic(hat(pi)[0][", human"])=="1"', parse=TRUE, x=0.45, y=(pi0est_human+0.07)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.57"', parse=TRUE, x=0.45, y=(pi0est_chimps_given_humans+0.07))

p2c

  
# Day 2 to 3

# Obtain pvalues
human_p_values_gen <- as.data.frame(qqplot_red_day23_shared_chimp[1])
shared_p_values_gen <- as.data.frame(qqplot_red_day23_shared_chimp[2])


# Pi_0 estimate for the humans

pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_human <- pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_human

# Pi_1 estimate for the humans

pi1est_human <- 1 - pi0est_human
pi1est_human

# Pi_0 estimate for the chimps given significant in the humans
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_chimps_given_humans <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps_given_humans

# Pi_1 estimate for the chimps

pi1est_chimps_given_humans <- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans
pi1est_chimps_given_humans*as.data.frame(qqplot_red_day12_shared_chimp[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_h <- array("Human", dim = c(10304,1))
labels_c <- array("Shared (Chimpanzee|Human)", dim = c(as.data.frame(qqplot_red_day23[[3]][2]),1))

humans <- cbind(labels_h, human_p_values_gen)
colnames(humans) <- c("Species", "pval")
chimps <- cbind(labels_c, shared_p_values_gen)
colnames(chimps) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(humans[,1:2], chimps[,1:2])


p3c <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 2 to 3") +
  geom_hline(yintercept = pi0est_human) + 
  geom_hline(yintercept = pi0est_chimps_given_humans) +       
  annotate("text", label='italic(hat(pi)[0][", human"])=="0.87"', parse=TRUE, x=0.7, y=(pi0est_human+0.06)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.97"', parse=TRUE, x=0.7, y=(pi0est_chimps_given_humans+0.06))

p3c
  

# Day 0 to 2


# Obtain pvalues
human_p_values_gen <- as.data.frame(qqplot_red_day02_shared_chimp[1])
shared_p_values_gen <- as.data.frame(qqplot_red_day02_shared_chimp[2])


# Pi_0 estimate for the humans

pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_human <- pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_human

# Pi_1 estimate for the humans

pi1est_human <- 1 - pi0est_human
pi1est_human

# Pi_0 estimate for the chimps given significant in the humans
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_chimps_given_humans <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps_given_humans

# Pi_1 estimate for the chimps

pi1est_chimps_given_humans <- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans
pi1est_chimps_given_humans*as.data.frame(qqplot_red_day02_shared_chimp[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_h <- array("Human", dim = c(10304,1))
labels_c <- array("Shared (Chimpanzee|Human)", dim = c(as.data.frame(qqplot_red_day03_shared_chimp[[3]][2]),1))

humans <- cbind(labels_h, human_p_values_gen)
colnames(humans) <- c("Species", "pval")
chimps <- cbind(labels_c, shared_p_values_gen)
colnames(chimps) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(humans[,1:2], chimps[,1:2])


p4c <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 0 to 2") +
  geom_hline(yintercept = pi0est_human) + 
  geom_hline(yintercept = pi0est_chimps_given_humans) +       
  annotate("text", label='italic(hat(pi)[0][", human"])=="1"', parse=TRUE, x=0.35, y=(pi0est_human+0.12)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.81"', parse=TRUE, x=0.35, y=(pi0est_chimps_given_humans+0.07))

p4c


################ INCREASE IN VARIATION ################

# Day 0 to 1

# Obtain pvalues
human_p_values_gen <- as.data.frame(qqplot_inc_day01_shared_chimp[1])
shared_p_values_gen <- as.data.frame(qqplot_inc_day01_shared_chimp[2])


# Pi_0 estimate for the humans

pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_human <- pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_human

# Pi_1 estimate for the humans

pi1est_human <- 1 - pi0est_human
pi1est_human

# Pi_0 estimate for the chimps given significant in the humans
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_chimps_given_humans <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps_given_humans

# Pi_1 estimate for the chimps

pi1est_chimps_given_humans <- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans
pi1est_chimps_given_humans*as.data.frame(qqplot_red_day12_shared_chimp[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_h <- array("Human", dim = c(10304,1))
labels_c <- array("Shared (Chimpanzee|Human)", dim = c(as.data.frame(qqplot_inc_day01[[3]][2]),1))

humans <- cbind(labels_h, human_p_values_gen)
colnames(humans) <- c("Species", "pval")
chimps <- cbind(labels_c, shared_p_values_gen)
colnames(chimps) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(humans[,1:2], chimps[,1:2])


p5c <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 0 to 1") +
  geom_hline(yintercept = pi0est_human) + 
  geom_hline(yintercept = pi0est_chimps_given_humans) +       
  annotate("text", label='italic(hat(pi)[0][", human"])=="1"', parse=TRUE, x=0.25, y=(pi0est_human+0.12)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.86"', parse=TRUE, x=0.25, y=(pi0est_chimps_given_humans-0.07))

p5c



# Day 1 to 2
# Obtain pvalues
human_p_values_gen <- as.data.frame(qqplot_inc_day12_shared_chimp[1])
shared_p_values_gen <- as.data.frame(qqplot_inc_day12_shared_chimp[2])


# Pi_0 estimate for the humans

pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_human <- pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_human

# Pi_1 estimate for the humans

pi1est_human <- 1 - pi0est_human
pi1est_human

# Pi_0 estimate for the chimps given significant in the humans
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_chimps_given_humans <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps_given_humans

# Pi_1 estimate for the chimps

pi1est_chimps_given_humans <- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans
pi1est_chimps_given_humans*as.data.frame(qqplot_red_day12_shared_chimp[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_h <- array("Human", dim = c(10304,1))
labels_c <- array("Shared (Chimpanzee|Human)", dim = c(as.data.frame(qqplot_inc_day12[[3]][2]),1))

humans <- cbind(labels_h, human_p_values_gen)
colnames(humans) <- c("Species", "pval")
chimps <- cbind(labels_c, shared_p_values_gen)
colnames(chimps) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(humans[,1:2], chimps[,1:2])


p6c <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 1 to 2") +
  geom_hline(yintercept = pi0est_human) + 
  geom_hline(yintercept = pi0est_chimps_given_humans) +       
  annotate("text", label='italic(hat(pi)[0][", human"])=="0.22"', parse=TRUE, x=0.25, y=(pi0est_human-0.07)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.32"', parse=TRUE, x=0.25, y=(pi0est_chimps_given_humans+0.07))

p6c

  
# Day 2 to 3

# Obtain pvalues
human_p_values_gen <- as.data.frame(qqplot_inc_day23_shared_chimp[1])
shared_p_values_gen <- as.data.frame(qqplot_inc_day23_shared_chimp[2])


# Pi_0 estimate for the humans

pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_human <- pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_human

# Pi_1 estimate for the humans

pi1est_human <- 1 - pi0est_human
pi1est_human

# Pi_0 estimate for the chimps given significant in the humans
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_chimps_given_humans <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps_given_humans

# Pi_1 estimate for the chimps

pi1est_chimps_given_humans <- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans
pi1est_chimps_given_humans*as.data.frame(qqplot_red_day12_shared_chimp[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_h <- array("Human", dim = c(10304,1))
labels_c <- array("Shared (Chimpanzee|Human)", dim = c(as.data.frame(qqplot_inc_day23[[3]][2]),1))

humans <- cbind(labels_h, human_p_values_gen)
colnames(humans) <- c("Species", "pval")
chimps <- cbind(labels_c, shared_p_values_gen)
colnames(chimps) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(humans[,1:2], chimps[,1:2])


p7c <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 2 to 3") +
  geom_hline(yintercept = pi0est_human) + 
  geom_hline(yintercept = pi0est_chimps_given_humans) +       
  annotate("text", label='italic(hat(pi)[0][", human"])=="0.87"', parse=TRUE, x=0.3, y=(pi0est_human+0.07)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.23"', parse=TRUE, x=0.3, y=(pi0est_chimps_given_humans+0.07))

p7c


# Day 0 to 2


# Obtain pvalues
human_p_values_gen <- as.data.frame(qqplot_inc_day02_shared_chimp[1])
shared_p_values_gen <- as.data.frame(qqplot_inc_day02_shared_chimp[2])


# Pi_0 estimate for the humans

pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))
pi0est_human <- pi0est(human_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_human

# Pi_1 estimate for the humans

pi1est_human <- 1 - pi0est_human
pi1est_human

# Pi_0 estimate for the chimps given significant in the humans
pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))

pi0est_chimps_given_humans <- pi0est(shared_p_values_gen, lambda = seq(0.05, 0.95, 0.05))$pi0
pi0est_chimps_given_humans

# Pi_1 estimate for the chimps

pi1est_chimps_given_humans <- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans
pi1est_chimps_given_humans*as.data.frame(qqplot_red_day12_shared_chimp[[3]][1])

# Let's overlap the two histograms

#abline(1,0)
#hist(num_var_pval_human_given_chimp[,1], breaks = 20, col=rgb(0,0,1,0.5), freq = FALSE)
#hist(var_pval[,1], col=rgb(1,0,0,0.5),  breaks = 20, freq = FALSE, add = T)


labels_h <- array("Human", dim = c(10304,1))
labels_c <- array("Shared (Chimpanzee|Human)", dim = c(as.data.frame(qqplot_inc_day03[[3]][2]),1))

humans <- cbind(labels_h, human_p_values_gen)
colnames(humans) <- c("Species", "pval")
chimps <- cbind(labels_c, shared_p_values_gen)
colnames(chimps) <- c("Species", "pval")

# Combine the two

all_pval <- rbind(humans[,1:2], chimps[,1:2])


p8c <- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 0.5) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_minimal() +  labs(title = "Day 0 to 2") +
  geom_hline(yintercept = pi0est_human) + 
  geom_hline(yintercept = pi0est_chimps_given_humans) +       
  annotate("text", label='italic(hat(pi)[0][", human"])=="1"', parse=TRUE, x=0.3, y=(pi0est_human+0.07)) +  annotate("text", label='italic(hat(pi)[0][", shared"])=="0.62"', parse=TRUE, x=0.3, y=(pi0est_chimps_given_humans+0.07))

p8c

multiplot(p1c,p5c,p2c,p6c,p3c,p7c,p4c,p8c,cols=4)

```


### Calculate pi0 and pi1 given a different # of genes (500, 1000, 1500, 2000)

```{r}


find_qqplot_red_top <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Convert to data frame
  top_var_pval <- var_pval[order(var_pval[,1]), ]
  
# Instead of going by p < 0.05, take the top 500 p-values
  
  num_var_pval <- top_var_pval[1:500, 1:2]
  
# Get chimps
  num_var_pval_chimps_500 <- top_var_pval[1:500, 1]
  
# Get sharing
  num_var_pval_shared_500 <- top_var_pval[1:500, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_500 <- pi0est(num_var_pval_shared_500, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_500 <- 1 - pi0est_humans_given_chimps_500

# Instead of going by p < 0.05, take the top 1000 p-values
  
  num_var_pval <- top_var_pval[1:1000, 1:2]
  
# Get chimps
  num_var_pval_chimps_1000 <- top_var_pval[1:1000, 1]
  
# Get sharing
  num_var_pval_shared_1000 <- top_var_pval[1:1000, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_1000 <- pi0est(num_var_pval_shared_1000, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_1000 <- 1 - pi0est_humans_given_chimps_1000
  
    # Instead of going by p < 0.05, take the top 1500 p-values
  
  num_var_pval <- top_var_pval[1:1500, 1:2]
  
# Get chimps
  num_var_pval_chimps_1500 <- top_var_pval[1:1500, 1]
  
# Get sharing
  num_var_pval_shared_1500 <- top_var_pval[1:1500, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_1500 <- pi0est(num_var_pval_shared_1500, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_1500 <- 1 - pi0est_humans_given_chimps_1500


  # Instead of going by p < 0.05, take the top 2000 p-values
  
  num_var_pval <- top_var_pval[1:2000, 1:2]
  
# Get chimps
  num_var_pval_chimps_2000 <- top_var_pval[1:2000, 1]
  
# Get sharing
  num_var_pval_shared_2000 <- top_var_pval[1:2000, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_2000 <- pi0est(num_var_pval_shared_2000, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_2000 <- 1 - pi0est_humans_given_chimps_2000

  # Instead of going by p < 0.05, take the top 1000 p-values
  
  num_var_pval <- top_var_pval[1:2000, 1:2]
  
# Get chimps
  num_var_pval_chimps_2000 <- top_var_pval[1:2000, 1]
  
# Get sharing
  num_var_pval_shared_2000 <- top_var_pval[1:2000, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_2000 <- pi0est(num_var_pval_shared_2000, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_2000 <- 1 - pi0est_humans_given_chimps_2000

# Instead of going by p < 0.05, take the top 3000 p-values
  
  num_var_pval <- top_var_pval[1:3000, 1:2]
  
# Get chimps
  num_var_pval_chimps_3000 <- top_var_pval[1:3000, 1]
  
# Get sharing
  num_var_pval_shared_3000 <- top_var_pval[1:3000, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_3000 <- pi0est(num_var_pval_shared_3000, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_3000 <- 1 - pi0est_humans_given_chimps_2000

  # Instead of going by p < 0.05, take the top 1000 p-values
  
  num_var_pval <- top_var_pval[1:3000, 1:2]
  
# Get chimps
  num_var_pval_chimps_3000 <- top_var_pval[1:3000, 1]
  
# Get sharing
  num_var_pval_shared_3000 <- top_var_pval[1:3000, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_3000 <- pi0est(num_var_pval_shared_3000, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_3000 <- 1 - pi0est_humans_given_chimps_3000

# Instead of going by p < 0.05, take the top 4000 p-values
  
  num_var_pval <- top_var_pval[1:4000, 1:2]
  
# Get chimps
  num_var_pval_chimps_4000 <- top_var_pval[1:4000, 1]
  
# Get sharing
  num_var_pval_shared_4000 <- top_var_pval[1:4000, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_4000 <- pi0est(num_var_pval_shared_4000, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_4000 <- 1 - pi0est_humans_given_chimps_4000

  # Instead of going by p < 0.05, take the top 4000 p-values
  
  num_var_pval <- top_var_pval[1:4000, 1:2]
  
# Get chimps
  num_var_pval_chimps_4000 <- top_var_pval[1:4000, 1]
  
# Get sharing
  num_var_pval_shared_4000 <- top_var_pval[1:4000, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_4000 <- pi0est(num_var_pval_shared_4000, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_4000 <- 1 - pi0est_humans_given_chimps_4000

  # Instead of going by p < 0.05, take the top 5000 p-values
  
  num_var_pval <- top_var_pval[1:5000, 1:2]
  
# Get chimps
  num_var_pval_chimps_5000 <- top_var_pval[1:5000, 1]
  
# Get sharing
  num_var_pval_shared_5000 <- top_var_pval[1:5000, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_5000 <- pi0est(num_var_pval_shared_5000, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_5000 <- 1 - pi0est_humans_given_chimps_5000

  # Instead of going by p < 0.05, take the top 5000 p-values
  
  num_var_pval <- top_var_pval[1:5000, 1:2]
  
# Get chimps
  num_var_pval_chimps_5000 <- top_var_pval[1:5000, 1]
  
# Get sharing
  num_var_pval_shared_5000 <- top_var_pval[1:5000, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
 pi0est_humans_given_chimps_5000 <- pi0est(num_var_pval_shared_5000, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans

  pi1est_humans_given_chimps_5000 <- 1 - pi0est_humans_given_chimps_5000

  
  
  list_values = list(pi1est_humans_given_chimps_500, pi1est_humans_given_chimps_1000, pi1est_humans_given_chimps_1500, pi1est_humans_given_chimps_2000, pi1est_humans_given_chimps_3000, pi1est_humans_given_chimps_4000, pi1est_humans_given_chimps_5000)
  
  
  return(list_values)

}

# Reduction in variance from day 0 to day 1

qqplot_red_day01 <- find_qqplot_red_top(7,10,17,20,1,6,11,16)

# Reduction in variance from day 1 to day 2

qqplot_red_day12 <- find_qqplot_red_top(17,20,27,30,11,16,21,26)

# Reduction in variance from day 2 to day 3

qqplot_red_day23 <- find_qqplot_red_top(27,30,37,40,21,26,31,36)

# Reduction in variance from day 0 to day 3

qqplot_red_day03 <- find_qqplot_red_mood(7,10,37,40,1,6,31,36)

# Inc. in variance from day 0 to day 1

qqplot_inc_day01 <- find_qqplot_inc_mood(7,10,17,20,1,6,11,16)

# Inc. in variance from day 1 to day 2

qqplot_inc_day12 <- find_qqplot_inc_mood(17,20,27,30,11,16,21,26)

# Inc. in variance from day 2 to day 3

qqplot_inc_day23 <- find_qqplot_inc_mood(27,30,37,40,21,26,31,36)

# Inc. in variance from day 0 to day 3

qqplot_inc_day03 <- find_qqplot_inc_mood(7,10,37,40,1,6,31,36)



```

### Sharing estimates (humans conditioned on chimps)
```{r}
find_qqplot_red_top_all <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Convert to data frame
  top_var_pval <- var_pval[order(var_pval[,1]), ]
  
# Instead of going by p < 0.05, take the top x number of p-values
  for_pi1_var_pval <- array(NA, dim = c(10205, 2))
  
  for(i in 100:10304){
    num_var_pval <- top_var_pval[1:i, 1:2]
  
# Get chimps
    num_var_pval_chimps_i <- top_var_pval[1:i, 1]
  
# Get sharing
    num_var_pval_shared_i <- top_var_pval[1:i, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
  pi0est_humans_given_chimps_i <- pi0est(num_var_pval_shared_i, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans
  for_pi1_var_pval[i-99,1] <- i
  for_pi1_var_pval[i-99,2] <- 1 - pi0est_humans_given_chimps_i
  }
  return(for_pi1_var_pval)
}

find_qqplot_inc_top_all <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Chimpanzee", "Human")
  summary(var_pval)

  dim(var_pval)

# Convert to data frame
  top_var_pval <- var_pval[order(var_pval[,1]), ]
  
# Instead of going by p < 0.05, take the top x number of p-values
  for_pi1_var_pval <- array(NA, dim = c(10205, 2))
  
  for(i in 100:10304){
    num_var_pval <- top_var_pval[1:i, 1:2]
  
# Get chimps
    num_var_pval_chimps_i <- top_var_pval[1:i, 1]
  
# Get sharing
    num_var_pval_shared_i <- top_var_pval[1:i, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
  pi0est_humans_given_chimps_i <- pi0est(num_var_pval_shared_i, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans
  for_pi1_var_pval[i-99,1] <- i
  for_pi1_var_pval[i-99,2] <- 1 - pi0est_humans_given_chimps_i
  }
  return(for_pi1_var_pval)
}

# Reduction in variance from day 0 to day 1

qqplot_red_top_all_day01 <- find_qqplot_red_top_all(7,10,17,20,1,6,11,16)

# Reduction in variance from day 1 to day 2

qqplot_red_top_all_day12 <- find_qqplot_red_top_all(17,20,27,30,11,16,21,26)

# Reduction in variance from day 2 to day 3

qqplot_red_top_all_day23 <- find_qqplot_red_top_all(27,30,37,40,21,26,31,36)

# Reduction in variance from day 0 to day 3

qqplot_red_top_all_day02 <- find_qqplot_red_top_all(7,10,27,30,1,6,21,26)


# Inc. in variance from day 0 to day 1

qqplot_inc_top_all_day01 <- find_qqplot_inc_top_all(7,10,17,20,1,6,11,16)

# Inc. in variance from day 1 to day 2

qqplot_inc_top_all_day12 <- find_qqplot_inc_top_all(17,20,27,30,11,16,21,26)

# Inc. in variance from day 2 to day 3

qqplot_inc_top_all_day23 <- find_qqplot_inc_top_all(27,30,37,40,21,26,31,36)

# Inc. in variance from day 0 to day 2

qqplot_inc_top_all_day02 <- find_qqplot_inc_top_all(7,10,27,30,1,6,21,26)

# Make everything a data frame
qqplot_red_top_all_day01 <- as.data.frame(qqplot_red_top_all_day01)
qqplot_red_top_all_day12 <- as.data.frame(qqplot_red_top_all_day12)
qqplot_red_top_all_day23 <- as.data.frame(qqplot_red_top_all_day23)
qqplot_red_top_all_day02 <- as.data.frame(qqplot_red_top_all_day02)
qqplot_inc_top_all_day01 <- as.data.frame(qqplot_inc_top_all_day01)
qqplot_inc_top_all_day12 <- as.data.frame(qqplot_inc_top_all_day12)
qqplot_inc_top_all_day23 <- as.data.frame(qqplot_inc_top_all_day23)
qqplot_inc_top_all_day02 <- as.data.frame(qqplot_inc_top_all_day02)

dev.off()
# Plot reduction in variation
ggplot(qqplot_red_top_all_day01, aes(x = qqplot_red_top_all_day01[,1], y = qqplot_red_top_all_day01[,2])) + geom_line(colour="#E69F00") + theme_bw() + geom_line(aes(x = qqplot_red_top_all_day12[,1], y = qqplot_red_top_all_day12[,2]), colour = "#000000") + geom_line(aes(x = qqplot_red_top_all_day23[,1], y = qqplot_red_top_all_day23[,2]), colour =  "#009E73") + geom_line(aes(x = qqplot_red_top_all_day02[,1], y = qqplot_red_top_all_day02[,2]), colour =  "#0072B2") + ggtitle("Proportion of Shared Genes with a Reduction in Variance between Days") + xlab("N Most Significant Genes") + ylab("Sharing Estimate") 

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")
legend("right", c("Day 0 to 1", "Day 1 to 2", "Day 2 to 3", "Day 0 to 2"), xpd = TRUE, horiz = FALSE, inset = c(0, 
    0), bty = "n", pch = c(15, 15, 15, 15), col = c("#E69F00", "#000000",  "#009E73","#0072B2" ), cex = 1)

dev.off()

# Plot inc in variation
ggplot(qqplot_inc_top_all_day01, aes(x = qqplot_inc_top_all_day01[,1], y = qqplot_inc_top_all_day01[,2])) + geom_line(colour="#E69F00") + theme_bw() + geom_line(aes(x = qqplot_inc_top_all_day12[,1], y = qqplot_inc_top_all_day12[,2]), colour = "#000000") + geom_line(aes(x = qqplot_inc_top_all_day23[,1], y = qqplot_inc_top_all_day23[,2]), colour =  "#009E73") + geom_line(aes(x = qqplot_inc_top_all_day02[,1], y = qqplot_inc_top_all_day02[,2]), colour =  "#0072B2") + ggtitle("Proportion of Shared Genes with an Increase in Variance between Days") + xlab("N Most Significant Genes") + ylab("Sharing Estimate") 

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")
legend("right", c("Day 0 to 1", "Day 1 to 2", "Day 2 to 3", "Day 0 to 2"), xpd = TRUE, horiz = FALSE, inset = c(0, 
    0), bty = "n", pch = c(15, 15, 15, 15), col = c("#E69F00", "#000000",  "#009E73","#0072B2" ), cex = 1)

```




### Sharing estimates (chimps conditioned on humans)
```{r}
find_qqplot_red_top_all_sharing_chimps <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind(human_var_pval, chimp_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c("Human", "Chimpanzee")
  summary(var_pval)

  dim(var_pval)

# Convert to data frame
  top_var_pval <- var_pval[order(var_pval[,1]), ]
  
# Instead of going by p < 0.05, take the top x number of p-values
  for_pi1_var_pval <- array(NA, dim = c(10205, 2))
  
  for(i in 100:10304){
    num_var_pval <- top_var_pval[1:i, 1:2]
  
# Get humans
    num_var_pval_chimps_i <- top_var_pval[1:i, 1]
  
# Get sharing
    num_var_pval_shared_i <- top_var_pval[1:i, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
  pi0est_humans_given_chimps_i <- pi0est(num_var_pval_shared_i, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans
  for_pi1_var_pval[i-99,1] <- i
  for_pi1_var_pval[i-99,2] <- 1 - pi0est_humans_given_chimps_i
  }
  return(for_pi1_var_pval)
}

find_qqplot_inc_top_all_sharing_chimps <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    chimp_var_pval[i,1] <- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    human_var_pval[i,1] <- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval <- as.data.frame(cbind( human_var_pval, chimp_var_pval))
  rownames(var_pval) <- rownames(mean_tech_reps)
  colnames(var_pval) <- c( "Human", "Chimpanzee")
  summary(var_pval)

  dim(var_pval)

# Convert to data frame
  top_var_pval <- var_pval[order(var_pval[,1]), ]
  
# Instead of going by p < 0.05, take the top x number of p-values
  for_pi1_var_pval <- array(NA, dim = c(10205, 2))
  
  for(i in 100:10304){
    num_var_pval <- top_var_pval[1:i, 1:2]
  
# Get humans
    num_var_pval_chimps_i <- top_var_pval[1:i, 1]
  
# Get sharing
    num_var_pval_shared_i <- top_var_pval[1:i, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
  pi0est_humans_given_chimps_i <- pi0est(num_var_pval_shared_i, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans
  for_pi1_var_pval[i-99,1] <- i
  for_pi1_var_pval[i-99,2] <- 1 - pi0est_humans_given_chimps_i
  }
  return(for_pi1_var_pval)
}

# Reduction in variance from day 0 to day 1

qqplot_red_top_all_day01 <- find_qqplot_red_top_all_sharing_chimps(7,10,17,20,1,6,11,16)

# Reduction in variance from day 1 to day 2

qqplot_red_top_all_day12 <- find_qqplot_red_top_all_sharing_chimps(17,20,27,30,11,16,21,26)

# Reduction in variance from day 2 to day 3

qqplot_red_top_all_day23 <- find_qqplot_red_top_all_sharing_chimps(27,30,37,40,21,26,31,36)

# Reduction in variance from day 0 to day 2

qqplot_red_top_all_day02 <- find_qqplot_red_top_all_sharing_chimps(7,10,27,30,1,6,21,26)


# Inc. in variance from day 0 to day 1

qqplot_inc_top_all_day01 <- find_qqplot_inc_top_all_sharing_chimps(7,10,17,20,1,6,11,16)

# Inc. in variance from day 1 to day 2

qqplot_inc_top_all_day12 <- find_qqplot_inc_top_all_sharing_chimps(17,20,27,30,11,16,21,26)

# Inc. in variance from day 2 to day 3

qqplot_inc_top_all_day23 <- find_qqplot_inc_top_all_sharing_chimps(27,30,37,40,21,26,31,36)

# Inc. in variance from day 0 to day 2

qqplot_inc_top_all_day02 <- find_qqplot_inc_top_all_sharing_chimps(7,10,27,30,1,6,21,26)

# Make everything a data frame
qqplot_red_top_all_day01 <- as.data.frame(qqplot_red_top_all_day01)
qqplot_red_top_all_day12 <- as.data.frame(qqplot_red_top_all_day12)
qqplot_red_top_all_day23 <- as.data.frame(qqplot_red_top_all_day23)
qqplot_red_top_all_day02 <- as.data.frame(qqplot_red_top_all_day02)
qqplot_inc_top_all_day01 <- as.data.frame(qqplot_inc_top_all_day01)
qqplot_inc_top_all_day12 <- as.data.frame(qqplot_inc_top_all_day12)
qqplot_inc_top_all_day23 <- as.data.frame(qqplot_inc_top_all_day23)
qqplot_inc_top_all_day02 <- as.data.frame(qqplot_inc_top_all_day02)

dev.off()
# Plot reduction in variation
ggplot(qqplot_red_top_all_day01, aes(x = qqplot_red_top_all_day01[,1], y = qqplot_red_top_all_day01[,2])) + geom_line(colour="#E69F00") + theme_bw() + geom_line(aes(x = qqplot_red_top_all_day12[,1], y = qqplot_red_top_all_day12[,2]), colour = "#000000") + geom_line(aes(x = qqplot_red_top_all_day23[,1], y = qqplot_red_top_all_day23[,2]), colour =  "#009E73") + geom_line(aes(x = qqplot_red_top_all_day02[,1], y = qqplot_red_top_all_day02[,2]), colour =  "#0072B2") + ylim(0,1) + ggtitle("Proportion of Shared Genes with a Reduction in Variance between Days") + xlab("N Most Significant Genes") + ylab("Sharing Estimate") 

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")
legend("right", c("Day 0 to 1", "Day 1 to 2", "Day 2 to 3", "Day 0 to 2"), xpd = TRUE, horiz = FALSE, inset = c(0, 
    0), bty = "n", pch = c(15, 15, 15, 15), col = c("#E69F00", "#000000",  "#009E73","#0072B2" ), cex = 1)

dev.off()

# Plot inc in variation
ggplot(qqplot_inc_top_all_day01, aes(x = qqplot_inc_top_all_day01[,1], y = qqplot_inc_top_all_day01[,2])) + geom_line(colour="#E69F00") + theme_bw() + geom_line(aes(x = qqplot_inc_top_all_day12[,1], y = qqplot_inc_top_all_day12[,2]), colour = "#000000") + geom_line(aes(x = qqplot_inc_top_all_day23[,1], y = qqplot_inc_top_all_day23[,2]), colour =  "#009E73") + geom_line(aes(x = qqplot_inc_top_all_day02[,1], y = qqplot_inc_top_all_day02[,2]), colour =  "#0072B2") + ylim(0,1) + ggtitle("Proportion of Shared Genes with an Increase in Variance between Days") + xlab("N Most Significant Genes") + ylab("Sharing Estimate") 

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = "n", bty = "n", xaxt = "n", yaxt = "n")
legend("right", c("Day 0 to 1", "Day 1 to 2", "Day 2 to 3", "Day 0 to 2"), xpd = TRUE, horiz = FALSE, inset = c(0, 
    0), bty = "n", pch = c(15, 15, 15, 15), col = c("#E69F00", "#000000",  "#009E73","#0072B2" ), cex = 1)

```

### Comparison of eta (chimpanzees conditioned on humans)

```{r}

find_eta_red <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values and F statistics

  chimp_var_pval <- array(NA, dim = c(10304, 1))
  chimp_var_Fstat <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    chimp_var_pval[i,1] <- htest$p.value
    chimp_var_Fstat[i,1] <- htest$statistic
  }

  hist(chimp_var_pval)
  hist(chimp_var_Fstat)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))
  human_var_Fstat <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    human_var_pval[i,1] <- htest$p.value
    human_var_Fstat[i,1] <- htest$statistic
  }

  hist(human_var_pval)
  hist(human_var_Fstat)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval_Fstat <- as.data.frame(cbind(chimp_var_pval, chimp_var_Fstat,  human_var_pval, human_var_Fstat))
  rownames(var_pval_Fstat) <- rownames(mean_tech_reps)
  colnames(var_pval_Fstat) <- c("Chimpanzee p", "Chimpanzee F", "Human p", "Human F")

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval_Fstat[ which(var_pval_Fstat[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval_Fstat[ which(var_pval_Fstat[,3] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
#  num_var_pval_chimps_neg_log <- -log10(var_pval[,1])

#  res_chimp <- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the chimp p-values and F-statistics given that it was significant in the humans.   

  subset_var <- as.data.frame(var_pval_Fstat[ which(var_pval_Fstat[,3] < 0.05), 1:2])
  num_var_pval_chimp_given_human_no_df <- subset_var[,1:2]
  
  # Transform the Fstatistics
  
  # Chimp (conditioned on chimp) ones
  eta_sq_chimps_given_humans <- (num_var_pval_chimp_given_human_no_df[,2]/(num_var_pval_chimp_given_human_no_df[,2]+2))
  
  # Human ones
  eta_sq_humans <- (human_var_Fstat/(human_var_Fstat+4))
                                 

  # Bind the humans and sort based on pvalue
  human_log10_p <- -log10(human_var_pval)
  humans_sort <- cbind(human_log10_p, human_var_Fstat, eta_sq_humans)
  humans_sort <- humans_sort[ order(humans_sort[,1]), ]
  
    # Bind the chimps and sort based on pvalue
  chimp_log10_p <- -log10(subset_var[,1])
  chimps_sort <- cbind(chimp_log10_p, subset_var[,2], eta_sq_chimps_given_humans)
  chimps_sort <- chimps_sort[ order(chimps_sort[,1]), ]
  
  # Grab the expected -log10 pvalues

    # For the humans
  
   res_human <- newqqplot(human_log10_p, -1, 100)
   
   res_chimp_given_human <- newqqplot(chimp_log10_p, -1, 100)
  
    # For the chimps given humans
  
  # Information for plotting

  list_values = list(human_var_pval, human_var_Fstat, eta_sq_humans, humans_sort[,1], humans_sort[,2], humans_sort[,3], subset_var[,1], subset_var[,2], eta_sq_chimps_given_humans, chimps_sort[,1], chimps_sort[,2], chimps_sort[,3], res_human$x, res_chimp_given_human$x)
                     
  return(list_values)

}

find_eta_red <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values and F statistics

  chimp_var_pval <- array(NA, dim = c(10304, 1))
  chimp_var_Fstat <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    chimp_var_pval[i,1] <- htest$p.value
    chimp_var_Fstat[i,1] <- htest$statistic
  }

  hist(chimp_var_pval)
  hist(chimp_var_Fstat)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))
  human_var_Fstat <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("greater"))
    human_var_pval[i,1] <- htest$p.value
    human_var_Fstat[i,1] <- htest$statistic
  }

  hist(human_var_pval)
  hist(human_var_Fstat)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval_Fstat <- as.data.frame(cbind(chimp_var_pval, chimp_var_Fstat,  human_var_pval, human_var_Fstat))
  rownames(var_pval_Fstat) <- rownames(mean_tech_reps)
  colnames(var_pval_Fstat) <- c("Chimpanzee p", "Chimpanzee F", "Human p", "Human F")

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval_Fstat[ which(var_pval_Fstat[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval_Fstat[ which(var_pval_Fstat[,3] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
#  num_var_pval_chimps_neg_log <- -log10(var_pval[,1])

#  res_chimp <- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the chimp p-values and F-statistics given that it was significant in the humans.   

  subset_var <- as.data.frame(var_pval_Fstat[ which(var_pval_Fstat[,3] < 0.05), 1:2])
  num_var_pval_chimp_given_human_no_df <- subset_var[,1:2]
  
  # Transform the Fstatistics
  
  # Chimp (conditioned on chimp) ones
  eta_sq_chimps_given_humans <- (num_var_pval_chimp_given_human_no_df[,2]/(num_var_pval_chimp_given_human_no_df[,2]+2))
  
  # Human ones
  eta_sq_humans <- (human_var_Fstat/(human_var_Fstat+4))
                                 

  # Bind the humans and sort based on pvalue
  human_log10_p <- -log10(human_var_pval)
  humans_sort <- cbind(human_log10_p, human_var_Fstat, eta_sq_humans)
  humans_sort <- humans_sort[ order(humans_sort[,1]), ]
  
    # Bind the chimps and sort based on pvalue
  chimp_log10_p <- -log10(subset_var[,1])
  chimps_sort <- cbind(chimp_log10_p, subset_var[,2], eta_sq_chimps_given_humans)
  chimps_sort <- chimps_sort[ order(chimps_sort[,1]), ]
  
  # Grab the expected -log10 pvalues

    # For the humans
  
   res_human <- newqqplot(human_log10_p, -1, 100)
   
   res_chimp_given_human <- newqqplot(chimp_log10_p, -1, 100)
  
    # For the chimps given humans
  
  # Information for plotting

  list_values = list(human_var_pval, human_var_Fstat, eta_sq_humans, humans_sort[,1], humans_sort[,2], humans_sort[,3], subset_var[,1], subset_var[,2], eta_sq_chimps_given_humans, chimps_sort[,1], chimps_sort[,2], chimps_sort[,3], res_human$x, res_chimp_given_human$x)
                     
  return(list_values)

}

find_eta_inc <- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values and F statistics

  chimp_var_pval <- array(NA, dim = c(10304, 1))
  chimp_var_Fstat <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y <- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    chimp_var_pval[i,1] <- htest$p.value
    chimp_var_Fstat[i,1] <- htest$statistic
  }

  hist(chimp_var_pval)
  hist(chimp_var_Fstat)

############## For humans

  human_var_pval <- array(NA, dim = c(10304, 1))
  human_var_Fstat <- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x <- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y <- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest <- var.test(x, y, alternative = c("less"))
    human_var_pval[i,1] <- htest$p.value
    human_var_Fstat[i,1] <- htest$statistic
  }

  hist(human_var_pval)
  hist(human_var_Fstat)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval_Fstat <- as.data.frame(cbind(chimp_var_pval, chimp_var_Fstat,  human_var_pval, human_var_Fstat))
  rownames(var_pval_Fstat) <- rownames(mean_tech_reps)
  colnames(var_pval_Fstat) <- c("Chimpanzee p", "Chimpanzee F", "Human p", "Human F")

# Set p-value

  p_val <- 0.05

# P-val < 0.05 for chimps only
  num_var_pval_chimps <- var_pval_Fstat[ which(var_pval_Fstat[,1] < p_val), ]

                    
# P-val < 0.05 for humans only
  num_var_pval_humans <- var_pval_Fstat[ which(var_pval_Fstat[,3] < p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val <- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
#  num_var_pval_chimps_neg_log <- -log10(var_pval[,1])

#  res_chimp <- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the chimp p-values and F-statistics given that it was significant in the humans.   

  subset_var <- as.data.frame(var_pval_Fstat[ which(var_pval_Fstat[,3] < 0.05), 1:2])
  num_var_pval_chimp_given_human_no_df <- subset_var[,1:2]
  
  # Transform the Fstatistics
  
  # Chimp (conditioned on chimp) ones
  eta_sq_chimps_given_humans <- (num_var_pval_chimp_given_human_no_df[,2]/(num_var_pval_chimp_given_human_no_df[,2]+2))
  
  # Human ones
  eta_sq_humans <- (human_var_Fstat/(human_var_Fstat+4))
                                 

  # Bind the humans and sort based on pvalue
  human_log10_p <- -log10(human_var_pval)
  humans_sort <- cbind(human_log10_p, human_var_Fstat, eta_sq_humans)
  humans_sort <- humans_sort[ order(humans_sort[,1]), ]
  
    # Bind the chimps and sort based on pvalue
  chimp_log10_p <- -log10(subset_var[,1])
  chimps_sort <- cbind(chimp_log10_p, subset_var[,2], eta_sq_chimps_given_humans)
  chimps_sort <- chimps_sort[ order(chimps_sort[,1]), ]
  
  # Grab the expected -log10 pvalues

    # For the humans
  
   res_human <- newqqplot(human_log10_p, -1, 100)
   
   res_chimp_given_human <- newqqplot(chimp_log10_p, -1, 100)
  
    # For the chimps given humans
  
  # Information for plotting

  list_values = list(human_var_pval, human_var_Fstat, eta_sq_humans, humans_sort[,1], humans_sort[,2], humans_sort[,3], subset_var[,1], subset_var[,2], eta_sq_chimps_given_humans, chimps_sort[,1], chimps_sort[,2], chimps_sort[,3], res_human$x, res_chimp_given_human$x)
                     
  return(list_values)

}

# 6 is eta_squared values for the human
# 12 is eta_squared values for the chimp given human
# 13 is expected p values for the human
# 14 is expected p values for the chimp given human

### Reduction
# Day 0 to 1
find_eta_red_01 <- find_eta_red(7,10,17,20,1,6,11,16)

# Day 1 to 2 
find_eta_red_12 <- find_eta_red(17,20,27,30,11,16,21,26)

# Day 2 to 3

find_eta_red_23 <- find_eta_red(27,30,37,40,21,26,31,36)
                                
# Day 0 to 2

find_eta_red_02 <- find_eta_red(7,10,27,30,1,6,21,26)


### Increase
# Day 0 to 1
find_eta_inc_01 <- find_eta_inc(7,10,17,20,1,6,11,16)

# Day 1 to 2 
find_eta_inc_12 <- find_eta_inc(17,20,27,30,11,16,21,26)

# Day 2 to 3

find_eta_inc_23 <- find_eta_inc(27,30,37,40,21,26,31,36)
                                
# Day 0 to 2

find_eta_inc_02 <- find_eta_inc(7,10,27,30,1,6,21,26)


dev.off()

#par(mfrow=c(2, 3))

# Reduction of variance qqplots

par(mfrow=c(2,4),oma = c(2, 2, 2, 2))

# Reduction of variance to day 0 to 1
plot(find_eta_red_01[[13]],find_eta_red_01[[6]], pch = 16, xlim = c(0,6), ylim = c(0,1), col = pal[2], ylab = "Eta^2", xlab = "-log10(p) Expected", main = "Day 0 to 1")

# Add to the plot
points(find_eta_red_01[[14]],find_eta_red_01[[12]], pch = 16, col = pal[3])

# Reduction of variance to day 1 to 2
plot(find_eta_red_12[[13]],find_eta_red_12[[6]], pch = 16, xlim = c(0,6), ylim = c(0,1), col = pal[2], ylab = "Eta^2", xlab = "-log10(p) Expected", main = "Day 1 to 2")

# Add to the plot
points(find_eta_red_12[[14]],find_eta_red_12[[12]], pch = 16, col = pal[3])

# Reduction of variance to day 2 to 3
plot(find_eta_red_23[[13]],find_eta_red_23[[6]], pch = 16, xlim = c(0,6), ylim = c(0,1), col = pal[2], ylab = "Eta^2", xlab = "-log10(p) Expected", main = "Day 2 to 3")

# Add to the plot
points(find_eta_red_23[[14]],find_eta_red_23[[12]], pch = 16, col = pal[3])

# Reduction of variance to day 0 to 2
plot(find_eta_red_02[[13]],find_eta_red_02[[6]], pch = 16, xlim = c(0,6), ylim = c(0,1), col = pal[2], ylab = "Eta^2", xlab = "-log10(p) Expected", main = "Day 0 to 2")

# Add to the plot
points(find_eta_red_02[[14]],find_eta_red_02[[12]], pch = 16, col = pal[3])



# Increase of variance to day 0 to 1
plot(find_eta_inc_01[[13]],rev(find_eta_inc_01[[6]]), pch = 16, xlim = c(0,6), ylim = c(0,1), col = pal[2], ylab = "Eta^2", xlab = "-log10(p) Expected", main = "Day 0 to 1")

# Add to the plot
points(find_eta_inc_01[[14]],rev(find_eta_inc_01[[12]]), pch = 16, col = pal[3])

# Increase of variance to day 1 to 2
plot(find_eta_inc_12[[13]],rev(find_eta_inc_12[[6]]), pch = 16, xlim = c(0,6), ylim = c(0,1), col = pal[2], ylab = "Eta^2", xlab = "-log10(p) Expected", main = "Day 1 to 2")

# Add to the plot
points(find_eta_inc_12[[14]],rev(find_eta_inc_12[[12]]), pch = 16, col = pal[3])

# Increase of variance to day 2 to 3
plot(find_eta_inc_23[[13]],rev(find_eta_inc_23[[6]]), pch = 16, xlim = c(0,6), ylim = c(0,1), col = pal[2], ylab = "Eta^2", xlab = "-log10(p) Expected", main = "Day 2 to 3")

# Add to the plot
points(find_eta_inc_23[[14]],rev(find_eta_inc_23[[12]]), pch = 16, col = pal[3])

# Increase of variance to day 0 to 2
plot(find_eta_inc_02[[13]],rev(find_eta_inc_02[[6]]), pch = 16, xlim = c(0,6), ylim = c(0,1), col = pal[2], ylab = "Eta^2", xlab = "-log10(p) Expected", main = "Day 0 to 2")

# Add to the plot
points(find_eta_inc_02[[14]],rev(find_eta_inc_02[[12]]), pch = 16, col = pal[3])

```





### Is the difference in variance between day 0 and day 1 being driven by differences in means? 

This is an important check because prior observation shows that in RNA-seq data, higher expression tends to have higher variance. 

```{r}

# Variance-mean dependency genome wide

############### Chimps
# day0 and day1
fit_day0 <- lm(chimps_day0_var ~ chimps_day0_mean)
fit_day1 <- lm(chimps_day1_var ~ chimps_day1_mean)
fit_day0; fit_day1
par(mfrow = c(1,2))
plot(x = chimps_day0_mean, y = sqrt(chimps_day0_var))
l1 <- lowess(chimps_day0_mean, chimps_day0_var)
lines(l1, col = "red")
plot(x = chimps_day1_mean, y = sqrt(chimps_day1_var))
l0 <- lowess(chimps_day1_mean, chimps_day1_var)
lines(l0, col = "red")


############### Humans
# day0 and day1
fit_day0 <- lm(humans_day0_var ~ humans_day0_mean)
fit_day1 <- lm(humans_day1_var ~ humans_day1_mean)
fit_day0; fit_day1
par(mfrow = c(1,2))
plot(x = humans_day0_mean, y = sqrt(humans_day0_var))
l1 <- lowess(humans_day0_mean, humans_day0_var)
lines(l1, col = "red")
plot(x = humans_day1_mean, y = sqrt(humans_day1_var))
l0 <- lowess(humans_day1_mean, humans_day1_var)
lines(l0, col = "red")

### What about in the genes that show a reduction in variance between days 0 and 1?

# Find the genes that show a reduction in variance in chimps

inshared_lists = row.names(mean_tech_reps) %in% rownames(num_var_pval_chimps)
inshared_lists_data <- as.data.frame(inshared_lists)
counts_genes_in <- cbind(mean_tech_reps, inshared_lists_data)
counts_genes_in_cutoff <- subset(counts_genes_in, inshared_lists_data == "TRUE")
norm_exp_chimp_612 <- counts_genes_in_cutoff[,1:40]

# Find the genes that show a reduction in variance in humans

inshared_lists = row.names(mean_tech_reps) %in% rownames(num_var_pval_humans)
inshared_lists_data <- as.data.frame(inshared_lists)
counts_genes_in <- cbind(mean_tech_reps, inshared_lists_data)
counts_genes_in_cutoff <- subset(counts_genes_in, inshared_lists_data == "TRUE")
norm_exp_human_2289 <- counts_genes_in_cutoff[,1:40]

# Define chimp and human means and variances
  # Means

norm_exp_human_2289_day0_mean <- apply(as.data.frame(norm_exp_human_2289[,1:6]),1, mean) 

norm_exp_chimp_612_day0_mean <-apply(as.data.frame(norm_exp_chimp_612[,7:10]),1, mean)

norm_exp_human_2289_day1_mean <- apply(as.data.frame(norm_exp_human_2289[,11:16]),1, mean)

norm_exp_chimp_612_day1_mean <- apply(as.data.frame(norm_exp_chimp_612[,17:20]),1, mean)


  # Variances
norm_exp_human_2289_day0_var <- apply(as.data.frame(norm_exp_human_2289[,1:6]),1, var) 

norm_exp_chimp_612_day0_var <-apply(as.data.frame(norm_exp_chimp_612[,7:10]),1, var)

norm_exp_human_2289_day1_var <- apply(as.data.frame(norm_exp_human_2289[,11:16]),1, var)

norm_exp_chimp_612_day1_var <- apply(as.data.frame(norm_exp_chimp_612[,17:20]),1, var)


############### Chimps

# day0 and day1
fit_day0 <- lm(norm_exp_chimp_612_day0_var ~ norm_exp_chimp_612_day0_mean)
fit_day1 <- lm(norm_exp_chimp_612_day1_var ~ norm_exp_chimp_612_day1_mean)
fit_day0; fit_day1
par(mfrow = c(1,2))
plot(x = norm_exp_chimp_612_day0_mean, y = sqrt(norm_exp_chimp_612_day0_var))
l1 <- lowess(norm_exp_chimp_612_day0_mean, norm_exp_chimp_612_day0_var)
lines(l1, col = "red")
plot(x = norm_exp_chimp_612_day1_mean, y = sqrt(norm_exp_chimp_612_day1_var))
l0 <- lowess(norm_exp_chimp_612_day1_mean, norm_exp_chimp_612_day1_var)
lines(l0, col = "red")


############### Humans
# day0 and day1
fit_day0 <- lm(norm_exp_human_2289_day0_var ~ norm_exp_human_2289_day0_mean)
fit_day1 <- lm(norm_exp_human_2289_day1_var ~ norm_exp_human_2289_day1_mean)
fit_day0; fit_day1
par(mfrow = c(1,2))
plot(x = norm_exp_human_2289_day0_mean, y = sqrt(norm_exp_human_2289_day0_var))
l1 <- lowess(norm_exp_human_2289_day0_mean, norm_exp_human_2289_day0_var)
lines(l1, col = "red")
plot(x = norm_exp_human_2289_day1_mean, y = sqrt(norm_exp_human_2289_day1_var))
l0 <- lowess(norm_exp_human_2289_day0_mean, norm_exp_human_2289_day0_var)
lines(l0, col = "red")


dev.off()
# Look at this genomewide

HC_var <- cbind((chimps_day1_mean - chimps_day0_mean), chimps_day1_var-chimps_day0_var)
plot(HC_var, xlab = "Differences in Means (Day 1 - Day 0)", ylab = "Differences in Variances (Day 1 - Day 0)", main = "Variances and Means in Day 1 compared to Day 0 in Chimps")


HC_var <- cbind((humans_day1_mean - humans_day0_mean), humans_day1_var-humans_day0_var)
plot(HC_var, xlab = "Differences in Means (Day 1 - Day 0)", ylab = "Differences in Variances (Day 1 - Day 0)", main = "Variances and Means in Day 1 compared to Day 0 in Humans")


# An example gene 
#iimean_day1GTday0 <- humans_day1_mean - humans_day0_mean 
#iivar_day1GTday0 <- humans_day1_var - humans_day0_var
#ii_gene <- iimean_day1GTday0 > 0 & iivar_day1GTday0 < 0 

#ii_gene_df <- as.data.frame(ii_gene)
#high_mean_low_var <- ii_gene_df[which(ii_gene_df == "TRUE") , ]

#head(which(ii_gene))
#iimean_day1GTday0[2]; iivar_day1GTday0[2]
#ii_expr <- unlist(mean_tech_reps[2, ])
#boxplot(c(ii_expr[1:6]), c(ii_expr[11:16]), ylab = "log2 normalized expression for ENSG00000000419", xlab = "Day", "Gene")

```

We can see that there are a number of genes where the mean is higher in day 1 than day 0 but the variance is lower in day 1 than day 0.

```{r}
### Let's look at these within the genes that have been previously identified as having a reduction in variance between day 0 and day 1 

# In chimps
HC_var <- cbind((norm_exp_chimp_612_day1_mean - norm_exp_chimp_612_day0_mean), norm_exp_chimp_612_day1_var-norm_exp_chimp_612_day0_var)
plot(HC_var, xlab = "Differences in Means (Day 1 - Day 0)", ylab = "Differences in Variances (Day 1 - Day 0)", main = "Variances and Means in Day 1 compared to Day 0 in Chimps (Subset)")

iimean_day1GTday0_chimps <- norm_exp_chimp_612_day1_mean - norm_exp_chimp_612_day0_mean
iivar_day1GTday0_chimps <- norm_exp_chimp_612_day1_var - norm_exp_chimp_612_day0_var
ii_gene <- iimean_day1GTday0_chimps > 0 & iivar_day1GTday0_chimps < 0 

ii_gene_df <- as.data.frame(ii_gene)

high_mean_low_var_chimps <- as.data.frame(ii_gene_df[which(ii_gene_df == "TRUE") , ])
high_mean_low_var_chimps <- as.data.frame(high_mean_low_var_chimps)

# 341 genes

head(which(ii_gene))
iimean_day1GTday0_chimps[2]; iivar_day1GTday0_chimps[2]

exp <- as.data.frame(mean_tech_reps[grepl("ENSG00000005810", rownames(mean_tech_reps)), ])

ii_expr <- unlist(exp)
boxplot(c(ii_expr[7:10]), c(ii_expr[17:20]), ylab = "log2 normalized expression", main = "Gene expression for ENSG00000005810 (Chimps)", xaxt="n", xlab = "Day")
axis(1, at=1:2, labels=c("Day 0", "Day 1"))

# In humans

HC_var <- cbind((norm_exp_human_2289_day1_mean - norm_exp_human_2289_day0_mean), norm_exp_human_2289_day1_var - norm_exp_human_2289_day0_var)
plot(HC_var, xlab = "Differences in Means (Day 1 - Day 0)", ylab = "Differences in Variances (Day 1 - Day 0)", main = "Variances and Means in Day 1 compared to Day 0 in Humans (Subset)")


iimean_day1GTday0_humans <- norm_exp_human_2289_day1_mean - norm_exp_human_2289_day0_mean
iivar_day1GTday0_humans <- norm_exp_human_2289_day1_var - norm_exp_human_2289_day0_var
ii_gene <- iimean_day1GTday0_humans > 0 & iivar_day1GTday0_humans < 0 

ii_gene_df <- as.data.frame(ii_gene)
high_mean_low_var <- ii_gene_df[which(ii_gene_df == "TRUE") , ]

# 1,283 genes

head(which(ii_gene))
iimean_day1GTday0_humans[2]; iivar_day1GTday0_humans[2]
ii_expr <- unlist(mean_tech_reps[2, ])
boxplot(c(ii_expr[1:6]), c(ii_expr[11:16]), xaxt="n", ylab = "log2 normalized expression", main = "Gene expression for ENSG00000000419 (Humans)", xlab = "Day")
axis(1, at=1:2, labels=c("Day 0", "Day 1"))

# In humans and chimps

inshared_lists = row.names(mean_tech_reps) %in% rownames(num_var_pval)
inshared_lists_data <- as.data.frame(inshared_lists)
counts_genes_in <- cbind(mean_tech_reps, inshared_lists_data)
counts_genes_in_cutoff <- subset(counts_genes_in, inshared_lists_data == "TRUE")
over_gene_red_var <- counts_genes_in_cutoff[,1:40]

# Define chimp and human means and variances
  # Means

over_human_day0_mean <- apply(as.data.frame(over_gene_red_var[,1:6]),1, mean) 

over_chimp_day0_mean <-apply(as.data.frame(over_gene_red_var[,7:10]),1, mean)

over_human_day1_mean <- apply(as.data.frame(over_gene_red_var[,11:16]),1, mean)

over_chimp_day1_mean <- apply(as.data.frame(over_gene_red_var[,17:20]),1, mean)


  # Variances
over_human_day0_var <- apply(as.data.frame(over_gene_red_var[,1:6]),1, var) 

over_chimp_day0_var <-apply(as.data.frame(over_gene_red_var[,7:10]),1, var)

over_human_day1_var <- apply(as.data.frame(over_gene_red_var[,11:16]),1, var)

over_chimp_day1_var <- apply(as.data.frame(over_gene_red_var[,17:20]),1, var)


iimean_day1GTday0_chimps <- over_chimp_day1_mean - over_chimp_day0_mean
iivar_day1GTday0_chimps <- over_chimp_day1_var - over_chimp_day0_var
iimean_day1GTday0_humans <- over_human_day1_mean - over_chimp_day0_mean
iivar_day1GTday0_humans <- over_human_day1_var - over_chimp_day0_var


ii_gene <- iimean_day1GTday0_chimps > 0 & iivar_day1GTday0_chimps < 0 & iimean_day1GTday0_humans > 0 & iivar_day1GTday0_humans < 0

ii_gene_df <- as.data.frame(ii_gene)
high_mean_low_var <- ii_gene_df[which(ii_gene_df == "TRUE") , ]

# 78 genes

head(which(ii_gene))
#iimean_day1GTday0[4]; iivar_day1GTday0[4]
ii_expr <- unlist(over_gene_red_var[4, ])
boxplot(c(ii_expr[7:10]), c(ii_expr[17:20]), c(ii_expr[1:6]), c(ii_expr[11:16]),  ylab = "log2 normalized expression",  xaxt="n", main = "Gene expression for ENSG00000011478", xlab = "Species-Day Pair")
axis(1, at=1:4, labels=c("C Day 0", "C Day 1", "H Day 0", "H Day 1"))


```

```{r}

#3.2 Driven by expression background
# for lowly expressed genes, more difference in variance

# Genomewide chimps
HC_var <- cbind((chimps_day1_mean + chimps_day0_mean)/2, chimps_day1_var-chimps_day0_var)
plot(HC_var, xlab = "(log2 expression mean day 0 + log2 expression mean day 1) / 2", ylab = "Difference in variance (Day 1-Day 0)", main = "Relationship between background and variance differences (chimps, all genes)")

# Genomewide humans
HC_var <- cbind((humans_day1_mean + humans_day0_mean)/2, humans_day1_var-humans_day0_var)
plot(HC_var, xlab = "(log2 expression mean day 0 + log2 expression mean day 1) / 2", ylab = "Difference in variance (Day 1-Day 0)", main = "Relationship bet. background and variance differences (humans, all genes)")

# Significantly reduced variance in chimps

HC_var <- cbind((norm_exp_chimp_612_day1_mean + norm_exp_chimp_612_day0_mean)/2, norm_exp_chimp_612_day1_var-norm_exp_chimp_612_day0_var)
plot(HC_var, xlab = "(log2 expression mean day 0 + log2 expression mean day 1) / 2", ylab = "Difference in variance (Day 1-Day 0)", main = "Relationship bet. background and variance differences (chimps, subset)")



# Significantly reduced variance in humans

HC_var <- cbind((norm_exp_human_2289_day1_mean + norm_exp_human_2289_day0_mean)/2, norm_exp_human_2289_day1_var-norm_exp_human_2289_day0_var)
plot(HC_var, xlab = "(log2 expression mean day 0 + log2 expression mean day 1) / 2", ylab = "Difference in variance (Day 1-Day 0)", main = "Relationship bet. background and variance differences (humans, subset)")


#HC_var <- cbind((chimps_day1_mean - chimps_day0_mean), chimps_day1_var-chimps_day0_var)
#plot(HC_var)

```


