<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />


<title>DE_exp_limma</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Endoderm_TimeCourse</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Endoderm_TC">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">DE_exp_limma</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>November 8, 2016</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#here-we-want-to-identify-de-genes-across-the-species.">Here, we want to identify DE genes across the species.</a></li>
<li><a href="#find-non-significant-genes-between-species-for-each-day">Find non-significant genes between species for each day</a></li>
<li><a href="#significant-interaction-term">Significant interaction term</a></li>
<li><a href="#repeat-the-analysis-but-only-using-a-subset-of-samples">Repeat the analysis but only using a subset of samples</a><ul>
<li><a href="#data-visualization">Data visualization</a></li>
</ul></li>
<li><a href="#subset-of-human-samples-with-the-interaction-term">Subset of human samples with the interaction term</a></li>
</ul>
</div>

<p>The goal of this script to to assess differential gene expression with pairwise comparisons using Limma.</p>
<pre class="r"><code># Load libraries

library(&quot;gplots&quot;)</code></pre>
<pre><code>## Warning: package &#39;gplots&#39; was built under R version 3.2.4</code></pre>
<pre><code>## 
## Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     lowess</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)</code></pre>
<pre><code>## Warning: package &#39;ggplot2&#39; was built under R version 3.2.4</code></pre>
<pre class="r"><code>library(&quot;RColorBrewer&quot;)
library(&quot;scales&quot;)</code></pre>
<pre><code>## Warning: package &#39;scales&#39; was built under R version 3.2.3</code></pre>
<pre class="r"><code>library(&quot;edgeR&quot;)</code></pre>
<pre><code>## Warning: package &#39;edgeR&#39; was built under R version 3.2.4</code></pre>
<pre><code>## Loading required package: limma</code></pre>
<pre><code>## Warning: package &#39;limma&#39; was built under R version 3.2.4</code></pre>
<pre class="r"><code>theme_set(theme_bw(base_size = 16))
library(&quot;biomaRt&quot;)
library(&quot;colorfulVennPlot&quot;)</code></pre>
<pre><code>## Loading required package: grid</code></pre>
<pre class="r"><code>library(&quot;VennDiagram&quot;)</code></pre>
<pre><code>## Warning: package &#39;VennDiagram&#39; was built under R version 3.2.5</code></pre>
<pre><code>## Loading required package: futile.logger</code></pre>
<pre><code>## Warning: package &#39;futile.logger&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>library(&quot;gridExtra&quot;)</code></pre>
<pre><code>## Warning: package &#39;gridExtra&#39; was built under R version 3.2.4</code></pre>
<pre class="r"><code>library(&quot;R.utils&quot;)</code></pre>
<pre><code>## Warning: package &#39;R.utils&#39; was built under R version 3.2.5</code></pre>
<pre><code>## Loading required package: R.oo</code></pre>
<pre><code>## Warning: package &#39;R.oo&#39; was built under R version 3.2.5</code></pre>
<pre><code>## Loading required package: R.methodsS3</code></pre>
<pre><code>## Warning: package &#39;R.methodsS3&#39; was built under R version 3.2.3</code></pre>
<pre><code>## R.methodsS3 v1.7.1 (2016-02-15) successfully loaded. See ?R.methodsS3 for help.</code></pre>
<pre><code>## R.oo v1.21.0 (2016-10-30) successfully loaded. See ?R.oo for help.</code></pre>
<pre><code>## 
## Attaching package: &#39;R.oo&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:methods&#39;:
## 
##     getClasses, getMethods</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     attach, detach, gc, load, save</code></pre>
<pre><code>## R.utils v2.4.0 (2016-09-13) successfully loaded. See ?R.utils for help.</code></pre>
<pre><code>## 
## Attaching package: &#39;R.utils&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:utils&#39;:
## 
##     timestamp</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     cat, commandArgs, getOption, inherits, isOpen, parse, warnings</code></pre>
<pre class="r"><code>source(&quot;~/Desktop/Endoderm_TC/ashlar-trial/analysis/chunk-options.R&quot;)</code></pre>
<pre><code>## Warning: package &#39;knitr&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code># Load colors 

colors &lt;- colorRampPalette(c(brewer.pal(9, &quot;Blues&quot;)[1],brewer.pal(9, &quot;Blues&quot;)[9]))(100)
pal &lt;- c(brewer.pal(9, &quot;Set1&quot;), brewer.pal(8, &quot;Set2&quot;), brewer.pal(12, &quot;Set3&quot;))

# Load normalized data

### Note: We want the object dge_in_cutoff for voom, which is why we import the counts data and do the normalization again. ###

gene_counts_combined_raw_data &lt;- read.delim(&quot;~/Desktop/Endoderm_TC/gene_counts_combined.txt&quot;)
counts_genes &lt;- gene_counts_combined_raw_data[1:30030,2:65]
rownames(counts_genes) &lt;- gene_counts_combined_raw_data[1:30030,1]

# Get data and sample info

counts_genes63 &lt;- counts_genes[,-2]
dim(counts_genes63)</code></pre>
<pre><code>## [1] 30030    63</code></pre>
<pre class="r"><code>After_removal_sample_info &lt;- read.csv(&quot;~/Desktop/Endoderm_TC/After_removal_sample_info.csv&quot;)

Species &lt;- After_removal_sample_info$Species
species &lt;- After_removal_sample_info$Species
day &lt;- After_removal_sample_info$Day
day &lt;- as.factor(day)
individual &lt;- After_removal_sample_info$Individual
Sample_ID &lt;- After_removal_sample_info$Sample_ID
labels &lt;- paste(Sample_ID, day, sep=&quot; &quot;)


# Log2(CPM)

cpm &lt;- cpm(counts_genes63, log=TRUE)

# Filter lowly expressed genes

humans &lt;- c(1:7, 16:23, 32:39, 48:55)
chimps &lt;- c(8:15, 24:31, 40:47, 56:63)

cpm_filtered &lt;- (rowSums(cpm[,humans] &gt; 1.5) &gt; 15 &amp; rowSums(cpm[,chimps] &gt; 1.5) &gt; 16)

genes_in_cutoff &lt;- cpm[cpm_filtered==TRUE,]
dim(genes_in_cutoff)</code></pre>
<pre><code>## [1] 10304    63</code></pre>
<pre class="r"><code># Find the original counts of all of the genes that fit the criteria 

counts_genes_in_cutoff &lt;- counts_genes63[cpm_filtered==TRUE,]
dim(counts_genes_in_cutoff)</code></pre>
<pre><code>## [1] 10304    63</code></pre>
<pre class="r"><code># Take the TMM of the counts only for the genes that remain after filtering

dge_in_cutoff &lt;- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff &lt;- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff &lt;- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)</code></pre>
<div id="here-we-want-to-identify-de-genes-across-the-species." class="section level2">
<h2>Here, we want to identify DE genes across the species.</h2>
<pre class="r"><code>species &lt;- c(&quot;H&quot;, &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;, &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;, &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;, &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;)

day &lt;- c(&quot;0&quot;, &quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;, &quot;0&quot;, &quot;0&quot;, &quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;, &quot;0&quot;, &quot;0&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;, &quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,  &quot;2&quot;, &quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;, &quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,  &quot;3&quot;, &quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,  &quot;3&quot;, &quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;)

condition &lt;- factor(paste(species,day,sep=&quot;.&quot;))
design &lt;- model.matrix(~ 0 + condition)
colnames(design) &lt;- gsub(&quot;condition&quot;, &quot;&quot;, dput(colnames(design)))</code></pre>
<pre><code>c(&quot;conditionC.0&quot;, &quot;conditionC.1&quot;, &quot;conditionC.2&quot;, &quot;conditionC.3&quot;, 
&quot;conditionH.0&quot;, &quot;conditionH.1&quot;, &quot;conditionH.2&quot;, &quot;conditionH.3&quot;
)</code></pre>
<pre class="r"><code># We want a random effect term for individual. As a result, we want to run voom twice. See https://support.bioconductor.org/p/59700/

cpm.voom &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;none&quot;)

# We want a random effect term for individual
#corfit &lt;- duplicateCorrelation(cpm.voom, design,block=individual)
corfit.correlation = 0.1947084

cpm.voom.corfit &lt;- voom(dge_in_cutoff, design, plot = TRUE, normalize.method=&quot;none&quot;, block=individual, correlation = corfit.correlation )</code></pre>
<p><img src="figure/DE_exp_limma_only.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot the density to see shape of the distribution

plotDensities(cpm.voom.corfit, col=pal[as.numeric(day)], legend = F)</code></pre>
<p><img src="figure/DE_exp_limma_only.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run lmFit and eBayes in limma

fit &lt;- lmFit(cpm.voom.corfit , design, block=individual, correlation=corfit.correlation)


# In the contrast matrix, we have the species DE at each day

cm2 &lt;- makeContrasts(HCday0 = c(1,0,0,0,-1,0,0,0), HCday1 = c(0,1,0,0,0,-1,0,0), HCday2 = c(0,0,1,0,0,0,-1,0), HCday3 = c(0,0,0,1,0,0,0,-1), levels = design)

# Fit the new model
diff_species &lt;- contrasts.fit(fit, cm2)
fit2 &lt;- eBayes(diff_species)


top3 &lt;- list(HvCday0 =topTable(fit2, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HvCday1 =topTable(fit2, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  HvCday2 =topTable(fit2, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  HvCday3 =topTable(fit2, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))


# Set FDR level at 5% 

FDR_level &lt;- 0.05
par(mfrow=c(1,1))

mylist &lt;- list()
mylist[[&quot;DE Day 0&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;DE Day 1&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;DE Day 2&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;DE Day 3&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]

# For Bryan- make a list of genes DE between the species

# Step 1- designate which in design matrix you are interested in (genes, logFC, adj.P.value)

important_columns &lt;- c(1,2,6)

# Find the genes that are DE at Day 0
HvCday0 =topTable(fit2, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
HvCday0_adj_pval &lt;- HvCday0[which(HvCday0$adj.P.Val &lt; 0.05), important_columns]                  

# Find the genes that are DE at Day 1
HvCday1 =topTable(fit2, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
HvCday1_adj_pval &lt;- HvCday1[which(HvCday1$adj.P.Val &lt; 0.05), important_columns] 

# Find the genes that are DE at Day 2
HvCday2 =topTable(fit2, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
HvCday2_adj_pval &lt;- HvCday2[which(HvCday2$adj.P.Val &lt; 0.05), important_columns] 

# Find the genes that are DE at Day 3
HvCday3 =topTable(fit2, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
HvCday3_adj_pval &lt;- HvCday3[which(HvCday3$adj.P.Val &lt; 0.05), important_columns] 

dim(HvCday0_adj_pval)</code></pre>
<pre><code>[1] 4376    3</code></pre>
<pre class="r"><code>dim(HvCday1_adj_pval)</code></pre>
<pre><code>[1] 4113    3</code></pre>
<pre class="r"><code>dim(HvCday2_adj_pval)</code></pre>
<pre><code>[1] 4422    3</code></pre>
<pre class="r"><code>dim(HvCday3_adj_pval)</code></pre>
<pre><code>[1] 4630    3</code></pre>
<pre class="r"><code># Make a table of the 4 sections

#write.table(HvCday0_adj_pval, &quot;~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day0.txt&quot;, sep=&quot;\t&quot;)
#write.table(HvCday1_adj_pval, &quot;~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day1.txt&quot;, sep=&quot;\t&quot;)
#write.table(HvCday2_adj_pval, &quot;~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day2.txt&quot;, sep=&quot;\t&quot;)
#write.table(HvCday3_adj_pval, &quot;~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day3.txt&quot;, sep=&quot;\t&quot;)

                  
# End section for Bryan

# Make as pdf 
Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between species per day (FDR&lt; 5%)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
pdf(file = &quot;Figure 7.D.pdf&quot;)
  grid.draw(Four_comp)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code># Significant between days 0 and 1 in each species

#FDR_level &lt;- 0.05
#par(mfrow=c(1,1))

#mylist &lt;- list()

#mylist[[&quot;H&quot;]] &lt;-  row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val &lt; FDR_level]
#mylist[[&quot;C&quot;]] &lt;- row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val &lt; FDR_level]

#orange_blue &lt;- c(2,5)
# Make as pdf 
#Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between days in each species (FDR&lt; 5%)&quot;, cex=4 , fill = pal[orange_blue], lty=1, height=2000, width=3000)
#pdf(file = &quot;Presentation_day12.pdf&quot;)
#  grid.draw(Four_comp)
#dev.off()

# Significant every day

sig_all &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 0&quot;]], mylist[[&quot;DE Day 1&quot;]]), mylist[[&quot;DE Day 2&quot;]]), mylist[[&quot;DE Day 3&quot;]]))

nonlist &lt;- list()
nonlist[[&quot;non DE Day 0&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &gt; FDR_level]
nonlist[[&quot;non DE Day 1&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &gt; FDR_level]
nonlist[[&quot;non DE Day 2&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &gt; FDR_level]
nonlist[[&quot;non DE Day 3&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &gt; FDR_level]

# Non-sign all
sig_none &lt;- length(intersect(intersect(intersect(nonlist[[&quot;non DE Day 0&quot;]], nonlist[[&quot;non DE Day 1&quot;]]), nonlist[[&quot;non DE Day 2&quot;]]), nonlist[[&quot;non DE Day 3&quot;]]))

# Find unique genes for each day 

sig_day_0_only &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 0&quot;]], nonlist[[&quot;non DE Day 1&quot;]]), nonlist[[&quot;non DE Day 2&quot;]]), nonlist[[&quot;non DE Day 3&quot;]]))

sig_day_1_only &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 1&quot;]], nonlist[[&quot;non DE Day 0&quot;]]), nonlist[[&quot;non DE Day 2&quot;]]), nonlist[[&quot;non DE Day 3&quot;]]))

sig_day_2_only &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 2&quot;]], nonlist[[&quot;non DE Day 0&quot;]]), nonlist[[&quot;non DE Day 1&quot;]]), nonlist[[&quot;non DE Day 3&quot;]]))

sig_day_3_only &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 3&quot;]], nonlist[[&quot;non DE Day 0&quot;]]), nonlist[[&quot;non DE Day 1&quot;]]), nonlist[[&quot;non DE Day 2&quot;]]))

# Find the rest of the values for the graph (DE in 2 or 3 days)

black_day_0 &lt;- length(mylist[[&quot;DE Day 0&quot;]]) - sig_day_0_only

black_day_1 &lt;- length(mylist[[&quot;DE Day 1&quot;]]) - sig_day_1_only

black_day_2 &lt;- length(mylist[[&quot;DE Day 2&quot;]]) - sig_day_2_only

black_day_3 &lt;- length(mylist[[&quot;DE Day 3&quot;]]) - sig_day_3_only

#### Plot the number of genes DE between species at each day

dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code># Numbers for non-unique DE genes
collect_total_barplot &lt;- c(black_day_0, black_day_1, black_day_2, black_day_3, sig_none, sig_all)

# Numbers for unique DE genes
collect_only &lt;- c(sig_day_0_only, sig_day_1_only, sig_day_2_only, sig_day_3_only, 0, 0)

# Combine the unique and non-unique DE genes
msig &lt;- rbind(collect_total_barplot, collect_only)
colnames(msig) &lt;- c(&quot;Day 0&quot;, &quot;Day 1&quot;, &quot;Day 2&quot;, &quot;Day 3&quot;, &quot;Never DE&quot;, &quot;Always DE&quot;)

# Make a stacked barplot

barplot(msig, ylab=&quot;Number of DE genes&quot;, ylim = c(0, 6000), main = &quot;Differentially Expressed Genes between Humans and Chimps Across Days&quot;)</code></pre>
</div>
<div id="find-non-significant-genes-between-species-for-each-day" class="section level2">
<h2>Find non-significant genes between species for each day</h2>
<pre class="r"><code># Not significant at each day
mylist &lt;- list()
mylist[[&quot;non DE Day 0&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &gt; FDR_level]
mylist[[&quot;non DE Day 1&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &gt; FDR_level]
mylist[[&quot;non DE Day 2&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &gt; FDR_level]
mylist[[&quot;non DE Day 3&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &gt; FDR_level]

# Not significant at any days
notsig &lt;- length(intersect(intersect(intersect(mylist[[&quot;non DE Day 0&quot;]], mylist[[&quot;non DE Day 1&quot;]]), mylist[[&quot;non DE Day 2&quot;]]), mylist[[&quot;non DE Day 3&quot;]]))</code></pre>
</div>
<div id="significant-interaction-term" class="section level2">
<h2>Significant interaction term</h2>
<p>Here, the model for expression has fixed effects for day, species, and a day-by-species interaction term.</p>
<pre class="r"><code># Make the design matrix with an interaction term between species and day

condition &lt;- factor(paste(species,day,sep=&quot;.&quot;))
design &lt;- model.matrix(~species+day+species*day)
colnames(design) &lt;- gsub(&quot;:&quot;, &quot;&quot;, dput(colnames(design)))</code></pre>
<pre><code>c(&quot;(Intercept)&quot;, &quot;speciesH&quot;, &quot;day1&quot;, &quot;day2&quot;, &quot;day3&quot;, &quot;speciesH:day1&quot;, 
&quot;speciesH:day2&quot;, &quot;speciesH:day3&quot;)</code></pre>
<pre class="r"><code>#colnames(design) &lt;- gsub(&quot;:&quot;, &quot;&quot;, dput(colnames(design)))

# We want a random effect term for individual. As a result, we want to run voom twice. See https://support.bioconductor.org/p/59700/

cpm.voom &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;none&quot;)

# We want a random effect term for individual
corfit &lt;- duplicateCorrelation(cpm.voom, design,block=individual)
corfit.correlation = 0.1947084

cpm.voom.corfit &lt;- voom(dge_in_cutoff, design, plot = TRUE, normalize.method=&quot;none&quot;, block=individual, correlation = corfit.correlation )</code></pre>
<p><img src="figure/DE_exp_limma_only.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot the density to see shape of the distribution

plotDensities(cpm.voom.corfit, col=pal[as.numeric(day)], legend = F)</code></pre>
<p><img src="figure/DE_exp_limma_only.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run lmFit and eBayes in limma

fit &lt;- lmFit(cpm.voom.corfit , design, block=individual, correlation=corfit.correlation)

# Fit the new model

fit2 &lt;- eBayes(fit)

top3 &lt;- list(day1inter =topTable(fit2, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), day2inter =topTable(fit2, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  day3inter =topTable(fit2, coef=8, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

mylist &lt;- list()
mylist[[&quot;Human x Day 1&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]

mylist[[&quot;Human x Day 2&quot;]] &lt;- row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]

mylist[[&quot;Human x Day 3&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]


# Make as pdf 
Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;Genes with significant species by day interactions (FDR&lt; 5%)&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
pdf(file = &quot;Figure 6.A.pdf&quot;)
  grid.draw(Four_comp)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div id="repeat-the-analysis-but-only-using-a-subset-of-samples" class="section level2">
<h2>Repeat the analysis but only using a subset of samples</h2>
<p>Now, we are going to subset to only include the 20157 (Batch 2), 28815 (Batch 1), and 28126 (Batch 2) in days 0-2 and only 20157 and 28126 in day 3.</p>
<pre class="r"><code># Subset the samples
samples_to_keep &lt;- c(1,4,5, 8:15, 17, 20, 21, 24:31, 33,36, 37,40:47, 49,  52, 56:63 )

species &lt;- c(&quot;H&quot;, &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;, &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;, &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;, &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;)

day &lt;- c(&quot;0&quot;, &quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;, &quot;0&quot;, &quot;0&quot;, &quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;,&quot;0&quot;, &quot;0&quot;, &quot;0&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;, &quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,&quot;1&quot;,  &quot;2&quot;, &quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;, &quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,&quot;2&quot;,  &quot;3&quot;, &quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,  &quot;3&quot;, &quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;,&quot;3&quot;)

# Subset the species and day
species1 &lt;- as.data.frame(species)
species1 &lt;- species1[samples_to_keep,]
day1 &lt;- as.data.frame(day)
day1 &lt;- day1[samples_to_keep,] 

Sample_ID_43 &lt;- Sample_ID[samples_to_keep]
#Sample_ID_43[1] &lt;- c(&quot;H1B&quot;)
labels &lt;- paste(Sample_ID_43, day1, sep=&quot; &quot;)



# Take the TMM of the counts only for the genes that remain after filtering

counts_genes_in_cutoff_43 &lt;- counts_genes_in_cutoff[samples_to_keep,]

dge_in_cutoff &lt;- DGEList(counts=as.matrix(counts_genes_in_cutoff[, samples_to_keep]), genes=rownames(counts_genes_in_cutoff[, samples_to_keep]), group = as.character(t(labels)))
dge_in_cutoff &lt;- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff &lt;- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)
head(cpm_in_cutoff)</code></pre>
<pre><code>                D0_20157 D0_28162 D0_28815  D0_3647 D0_36470116  D0_3649
ENSG00000000003 7.666887 7.552523 7.667892 7.999150    7.964802 7.759551
ENSG00000000419 6.602240 6.694251 6.397524 5.750014    6.659866 5.545887
ENSG00000000457 3.505002 3.767750 3.757056 4.514018    4.839825 4.860630
ENSG00000000460 5.586966 5.335995 5.508917 5.727276    5.630599 5.454519
ENSG00000001036 6.565851 6.478730 6.658377 6.662235    6.508938 6.438993
ENSG00000001084 6.785202 6.884728 6.823589 6.924570    6.866096 6.443599
                D0_3649_0116 D0_40300 D0_40300_0116  D0_4955 D0_4955_0116
ENSG00000000003     7.662517 7.749555      7.266127 8.069458     7.440003
ENSG00000000419     6.139484 5.518641      5.638588 5.920671     5.579494
ENSG00000000457     4.543224 4.786110      4.715185 5.050432     4.701635
ENSG00000000460     5.193904 5.409706      5.746273 5.501778     5.692062
ENSG00000001036     6.462931 6.516642      5.843244 6.657266     5.958483
ENSG00000001084     6.131580 6.629634      6.604443 6.587136     6.551615
                D1_20157_0116 D1_28162 D1_28815  D1_3647 D1_3647_0116
ENSG00000000003      7.574895 7.541381 7.693971 7.447703     7.314916
ENSG00000000419      6.873788 6.700503 7.006780 5.790362     6.175832
ENSG00000000457      3.515437 3.428026 3.931575 4.703579     4.742074
ENSG00000000460      5.649504 5.535184 5.730176 5.465811     5.457987
ENSG00000001036      6.267133 5.604307 6.043472 5.888861     6.148031
ENSG00000001084      6.938436 6.851612 7.022147 6.739406     6.447268
                 D1_3649 D1_3649_0116 D1_40300 D1_40300_0116  D1_4955
ENSG00000000003 7.727045     7.765187 7.882489      7.209700 7.858718
ENSG00000000419 5.975113     6.322324 5.999738      5.624330 6.290980
ENSG00000000457 5.043823     4.837075 4.811592      4.638943 5.035682
ENSG00000000460 5.661325     5.424358 5.626171      5.695962 5.489585
ENSG00000001036 6.421169     6.319581 6.146620      5.762352 6.420594
ENSG00000001084 6.394555     6.143154 6.572766      6.542276 6.604201
                D1_4955_0116 D2_20157_0116 D2_28162 D2_28815  D2_3647
ENSG00000000003     7.531477      7.117897 6.889788 5.976559 7.239082
ENSG00000000419     6.258963      6.888781 6.457160 6.954237 5.660610
ENSG00000000457     4.702041      3.629095 3.723929 5.129176 4.686855
ENSG00000000460     5.765495      4.976032 4.923510 5.535972 5.198996
ENSG00000001036     6.004825      6.924118 6.045860 5.871393 5.871721
ENSG00000001084     6.548956      7.265390 7.222978 7.513238 7.240329
                D2_3647_0116  D2_3649 D2_3649_0116 D2_40300 D2_40300_0116
ENSG00000000003     6.870626 7.682226     7.580646 7.390547      6.938549
ENSG00000000419     5.584074 5.939125     5.989422 5.821400      5.582401
ENSG00000000457     4.787525 5.085492     5.178652 5.022906      4.945512
ENSG00000000460     4.887299 5.226162     4.838616 5.051309      5.037235
ENSG00000001036     5.854454 6.144550     5.888692 6.026504      5.579631
ENSG00000001084     6.810469 6.660173     6.694783 6.610930      6.995792
                 D2_4955 D2_4955_0116 D3_20157_0116 D3_28162  D3_3647
ENSG00000000003 7.728951     7.126346      7.029282 6.782377 6.889327
ENSG00000000419 6.250441     5.920930      6.216045 6.375280 5.426692
ENSG00000000457 5.188839     5.054545      3.753318 3.728482 4.872069
ENSG00000000460 5.087000     5.373107      4.439093 4.253110 4.534587
ENSG00000001036 6.202545     5.691593      6.836537 5.982231 5.734257
ENSG00000001084 6.480148     6.965154      6.733239 7.186535 7.255290
                D3_3647_0116  D3_3649 D3_3649_0116 D3_40300 D3_40300_0116
ENSG00000000003     6.971864 7.196961     7.597596 6.945987      7.110069
ENSG00000000419     5.440665 5.511795     5.608552 5.457278      5.694422
ENSG00000000457     4.768674 5.224928     4.936371 5.289149      4.934258
ENSG00000000460     3.825519 3.970743     3.330032 4.429630      4.756462
ENSG00000001036     5.923268 5.682365     5.812589 5.877066      5.425665
ENSG00000001084     6.928603 6.697804     6.796982 6.627567      6.747722
                 D3_4955 D3_4955_0116
ENSG00000000003 6.967484     7.099858
ENSG00000000419 5.714709     5.634974
ENSG00000000457 5.358955     5.120214
ENSG00000000460 3.979420     4.495437
ENSG00000001036 5.767835     5.689313
ENSG00000001084 6.822175     7.126062</code></pre>
<pre class="r"><code>dim(cpm_in_cutoff)</code></pre>
<pre><code>[1] 10304    43</code></pre>
<div id="data-visualization" class="section level3">
<h3>Data visualization</h3>
<pre class="r"><code># Make PCA plots with the factors colored by day

pca_genes &lt;- prcomp(t(cpm_in_cutoff), scale = T, retx = TRUE, center = TRUE)
scores &lt;- pca_genes$x

matrixpca &lt;- pca_genes$x
pc1 &lt;- matrixpca[,1]
pc2 &lt;- matrixpca[,2]
pc3 &lt;- matrixpca[,3]
pc4 &lt;- matrixpca[,4]
pc5 &lt;- matrixpca[,5]

pcs &lt;- data.frame(pc1, pc2, pc3, pc4, pc5)

summary &lt;- summary(pca_genes)


ggplot(data=pcs, aes(x=pc1, y=pc2, color=as.factor(day1), shape=as.factor(species1), size=2)) + geom_point() + xlab(paste(&quot;PC1 (&quot;,(summary$importance[2,1]*100), &quot;% of variance)&quot;)) + ylab(paste(&quot;PC2 (&quot;,(summary$importance[2,2]*100), &quot;% of variance)&quot;))  + guides(color = guide_legend(order=1), size = FALSE, shape = guide_legend(order=2)) + scale_color_discrete(name  =&quot;Day&quot;) + labs(title = &quot;PCA of normalized endoderm data (Humans subset to match chimps)&quot;)</code></pre>
<p><img src="figure/DE_exp_limma_only.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Make the design matrix

condition &lt;- factor(paste(species1,day1,sep=&quot;.&quot;))
design &lt;- model.matrix(~ 0 + condition)
colnames(design) &lt;- gsub(&quot;condition&quot;, &quot;&quot;, dput(colnames(design)))</code></pre>
<pre><code>c(&quot;conditionC.0&quot;, &quot;conditionC.1&quot;, &quot;conditionC.2&quot;, &quot;conditionC.3&quot;, 
&quot;conditionH.0&quot;, &quot;conditionH.1&quot;, &quot;conditionH.2&quot;, &quot;conditionH.3&quot;
)</code></pre>
<pre class="r"><code># We want a random effect term for individual. As a result, we want to run voom twice. See https://support.bioconductor.org/p/59700/

cpm.voom &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;none&quot;)

# We want a random effect term for individual
#corfit &lt;- duplicateCorrelation(cpm.voom, design,block=Sample_ID_43)
corfit.correlation = 0.349764

cpm.voom.corfit &lt;- voom(dge_in_cutoff, design, plot = TRUE, normalize.method=&quot;none&quot;, block=Sample_ID_43, correlation = corfit.correlation )</code></pre>
<p><img src="figure/DE_exp_limma_only.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot the density to see shape of the distribution

plotDensities(cpm.voom.corfit, col=pal[as.numeric(day1)], legend = F)</code></pre>
<p><img src="figure/DE_exp_limma_only.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run lmFit and eBayes in limma

fit &lt;- lmFit(cpm.voom.corfit , design, block=Sample_ID_43, correlation=corfit.correlation)


# In the contrast matrix, we have the species DE at each day

cm2 &lt;- makeContrasts(HCday0 = c(1,0,0,0,-1,0,0,0), HCday1 = c(0,1,0,0,0,-1,0,0), HCday2 = c(0,0,1,0,0,0,-1,0), HCday3 = c(0,0,0,1,0,0,0,-1), levels = design)

# Fit the new model
diff_species &lt;- contrasts.fit(fit, cm2)
fit2 &lt;- eBayes(diff_species)


top3 &lt;- list(HvCday0 =topTable(fit2, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), HvCday1 =topTable(fit2, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  HvCday2 =topTable(fit2, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  HvCday3 =topTable(fit2, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))


# Set FDR level at 5% 

FDR_level &lt;- 0.05
par(mfrow=c(1,1))

mylist &lt;- list()
mylist[[&quot;DE Day 0&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;DE Day 1&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;DE Day 2&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]
mylist[[&quot;DE Day 3&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &lt; FDR_level]

# For Bryan- make a list of genes DE between the species

# Step 1- designate which in design matrix you are interested in (genes, logFC, adj.P.value)

important_columns &lt;- c(1,2,6)

# Find the genes that are DE at Day 0
HvCday0 =topTable(fit2, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
HvCday0_adj_pval &lt;- HvCday0[which(HvCday0$adj.P.Val &lt; 0.05), important_columns]                  

# Find the genes that are DE at Day 1
HvCday1 =topTable(fit2, coef=2, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
HvCday1_adj_pval &lt;- HvCday1[which(HvCday1$adj.P.Val &lt; 0.05), important_columns] 

# Find the genes that are DE at Day 2
HvCday2 =topTable(fit2, coef=3, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
HvCday2_adj_pval &lt;- HvCday2[which(HvCday2$adj.P.Val &lt; 0.05), important_columns] 

# Find the genes that are DE at Day 3
HvCday3 =topTable(fit2, coef=4, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;)
HvCday3_adj_pval &lt;- HvCday3[which(HvCday3$adj.P.Val &lt; 0.05), important_columns] 

dim(HvCday0_adj_pval)</code></pre>
<pre><code>[1] 3393    3</code></pre>
<pre class="r"><code>dim(HvCday1_adj_pval)</code></pre>
<pre><code>[1] 3138    3</code></pre>
<pre class="r"><code>dim(HvCday2_adj_pval)</code></pre>
<pre><code>[1] 3690    3</code></pre>
<pre class="r"><code>dim(HvCday3_adj_pval)</code></pre>
<pre><code>[1] 3062    3</code></pre>
<pre class="r"><code># Make a table of the 4 sections

#write.table(HvCday0_adj_pval, &quot;~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day0.txt&quot;, sep=&quot;\t&quot;)
#write.table(HvCday1_adj_pval, &quot;~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day1.txt&quot;, sep=&quot;\t&quot;)
#write.table(HvCday2_adj_pval, &quot;~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day2.txt&quot;, sep=&quot;\t&quot;)
#write.table(HvCday3_adj_pval, &quot;~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day3.txt&quot;, sep=&quot;\t&quot;)

                  
# End section for Bryan

# Make as pdf 
#Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between species per day (FDR&lt; 5%)&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
#pdf(file = &quot;Figure 7.D.pdf&quot;)
#  grid.draw(Four_comp)
#dev.off()

# Significant between days 0 and 1 in each species

#FDR_level &lt;- 0.05
#par(mfrow=c(1,1))

#mylist &lt;- list()

#mylist[[&quot;H&quot;]] &lt;-  row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val &lt; FDR_level]
#mylist[[&quot;C&quot;]] &lt;- row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val &lt; FDR_level]

#orange_blue &lt;- c(2,5)
# Make as pdf 
#Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between days in each species (FDR&lt; 5%)&quot;, cex=4 , fill = pal[orange_blue], lty=1, height=2000, width=3000)
#pdf(file = &quot;Presentation_day12.pdf&quot;)
#  grid.draw(Four_comp)
#dev.off()

# Significant every day

sig_all &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 0&quot;]], mylist[[&quot;DE Day 1&quot;]]), mylist[[&quot;DE Day 2&quot;]]), mylist[[&quot;DE Day 3&quot;]]))

nonlist &lt;- list()
nonlist[[&quot;non DE Day 0&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &gt; FDR_level]
nonlist[[&quot;non DE Day 1&quot;]] &lt;-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &gt; FDR_level]
nonlist[[&quot;non DE Day 2&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &gt; FDR_level]
nonlist[[&quot;non DE Day 3&quot;]] &lt;- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val &gt; FDR_level]

# Non-sign all
sig_none &lt;- length(intersect(intersect(intersect(nonlist[[&quot;non DE Day 0&quot;]], nonlist[[&quot;non DE Day 1&quot;]]), nonlist[[&quot;non DE Day 2&quot;]]), nonlist[[&quot;non DE Day 3&quot;]]))

# Find unique genes for each day 

sig_day_0_only &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 0&quot;]], nonlist[[&quot;non DE Day 1&quot;]]), nonlist[[&quot;non DE Day 2&quot;]]), nonlist[[&quot;non DE Day 3&quot;]]))

sig_day_1_only &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 1&quot;]], nonlist[[&quot;non DE Day 0&quot;]]), nonlist[[&quot;non DE Day 2&quot;]]), nonlist[[&quot;non DE Day 3&quot;]]))

sig_day_2_only &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 2&quot;]], nonlist[[&quot;non DE Day 0&quot;]]), nonlist[[&quot;non DE Day 1&quot;]]), nonlist[[&quot;non DE Day 3&quot;]]))

sig_day_3_only &lt;- length(intersect(intersect(intersect(mylist[[&quot;DE Day 3&quot;]], nonlist[[&quot;non DE Day 0&quot;]]), nonlist[[&quot;non DE Day 1&quot;]]), nonlist[[&quot;non DE Day 2&quot;]]))

sig_day_0_only</code></pre>
<pre><code>[1] 563</code></pre>
<pre class="r"><code>sig_day_1_only</code></pre>
<pre><code>[1] 265</code></pre>
<pre class="r"><code>sig_day_2_only</code></pre>
<pre><code>[1] 805</code></pre>
<pre class="r"><code>sig_day_3_only</code></pre>
<pre><code>[1] 587</code></pre>
<pre class="r"><code># Find the rest of the values for the graph (DE in 2 or 3 days)

black_day_0 &lt;- length(mylist[[&quot;DE Day 0&quot;]]) - sig_day_0_only

black_day_1 &lt;- length(mylist[[&quot;DE Day 1&quot;]]) - sig_day_1_only

black_day_2 &lt;- length(mylist[[&quot;DE Day 2&quot;]]) - sig_day_2_only

black_day_3 &lt;- length(mylist[[&quot;DE Day 3&quot;]]) - sig_day_3_only

#### Plot the number of genes DE between species at each day

dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code># Numbers for non-unique DE genes
collect_total_barplot &lt;- c(black_day_0, black_day_1, black_day_2, black_day_3, sig_none, sig_all)

# Numbers for unique DE genes
collect_only &lt;- c(sig_day_0_only, sig_day_1_only, sig_day_2_only, sig_day_3_only, 0, 0)

# Combine the unique and non-unique DE genes
msig &lt;- rbind(collect_total_barplot, collect_only)
colnames(msig) &lt;- c(&quot;Day 0&quot;, &quot;Day 1&quot;, &quot;Day 2&quot;, &quot;Day 3&quot;, &quot;Never DE&quot;, &quot;Always DE&quot;)

# Make a stacked barplot

barplot(msig, ylab=&quot;Number of DE genes&quot;, ylim = c(0, 6000), main = &quot;DE Genes between 2-3 Humans and Chimps Across Days&quot;)</code></pre>
</div>
</div>
<div id="subset-of-human-samples-with-the-interaction-term" class="section level2">
<h2>Subset of human samples with the interaction term</h2>
<pre class="r"><code># Make the design matrix with an interaction term between species and day

condition &lt;- factor(paste(species1,day1,sep=&quot;.&quot;))
design &lt;- model.matrix(~species1+day1+species1*day1)
colnames(design) &lt;- gsub(&quot;:&quot;, &quot;&quot;, dput(colnames(design)))</code></pre>
<pre><code>c(&quot;(Intercept)&quot;, &quot;species1H&quot;, &quot;day11&quot;, &quot;day12&quot;, &quot;day13&quot;, &quot;species1H:day11&quot;, 
&quot;species1H:day12&quot;, &quot;species1H:day13&quot;)</code></pre>
<pre class="r"><code>#colnames(design) &lt;- gsub(&quot;:&quot;, &quot;&quot;, dput(colnames(design)))

# We want a random effect term for individual. As a result, we want to run voom twice. See https://support.bioconductor.org/p/59700/

cpm.voom &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;none&quot;)

# We want a random effect term for individual
#corfit &lt;- duplicateCorrelation(cpm.voom, design,block=Sample_ID_43)
corfit.correlation = 0.1947084

cpm.voom.corfit &lt;- voom(dge_in_cutoff, design, plot = TRUE, normalize.method=&quot;none&quot;, block=Sample_ID_43, correlation = corfit.correlation )</code></pre>
<p><img src="figure/DE_exp_limma_only.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Plot the density to see shape of the distribution

plotDensities(cpm.voom.corfit, col=pal[as.numeric(day)], legend = F)</code></pre>
<p><img src="figure/DE_exp_limma_only.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Run lmFit and eBayes in limma

fit &lt;- lmFit(cpm.voom.corfit , design, block=Sample_ID_43, correlation=corfit.correlation)

# Fit the new model

fit2 &lt;- eBayes(fit)

top3 &lt;- list(day1inter =topTable(fit2, coef=6, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;), day2inter =topTable(fit2, coef=7, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;),  day3inter =topTable(fit2, coef=8, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;none&quot;))

# Find the name of the genes with a significant interaction term

mylist &lt;- list()
mylist[[&quot;Human x Day 1&quot;]] &lt;- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level]

mylist[[&quot;Human x Day 2&quot;]] &lt;- row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level]

mylist[[&quot;Human x Day 3&quot;]] &lt;- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level]

# Find the numbers of genes with a significant interaction term
sig_subset_interaction_day1 &lt;- length(row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val &lt; FDR_level])
sig_subset_interaction_day2 &lt;- length(row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val &lt; FDR_level])
sig_subset_interaction_day3 &lt;- length(row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val &lt; FDR_level])

sig_subset_interaction_day1</code></pre>
<pre><code>[1] 42</code></pre>
<pre class="r"><code>sig_subset_interaction_day2</code></pre>
<pre><code>[1] 717</code></pre>
<pre class="r"><code>sig_subset_interaction_day3</code></pre>
<pre><code>[1] 828</code></pre>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
