---
title: "DE_exp_limma"
author: "Lauren Blake"
date: "November 8, 2016"
output: html_document
---

The goal of this script to to assess differential gene expression with pairwise comparisons using Limma.

```{r}
# Load libraries

library("gplots")
library("ggplot2")
library("RColorBrewer")
library("scales")
library("edgeR")
theme_set(theme_bw(base_size = 16))
library("biomaRt")
library("colorfulVennPlot")
library("VennDiagram")
library("gridExtra")
library("R.utils")

source("~/Desktop/Endoderm_TC/ashlar-trial/analysis/chunk-options.R")

# Load colors 

colors <- colorRampPalette(c(brewer.pal(9, "Blues")[1],brewer.pal(9, "Blues")[9]))(100)
pal <- c(brewer.pal(9, "Set1"), brewer.pal(8, "Set2"), brewer.pal(12, "Set3"))

# Load normalized data

### Note: We want the object dge_in_cutoff for voom, which is why we import the counts data and do the normalization again. ###

gene_counts_combined_raw_data <- read.delim("~/Desktop/Endoderm_TC/gene_counts_combined.txt")
counts_genes <- gene_counts_combined_raw_data[1:30030,2:65]
rownames(counts_genes) <- gene_counts_combined_raw_data[1:30030,1]

# Get data and sample info

counts_genes63 <- counts_genes[,-2]
dim(counts_genes63)

After_removal_sample_info <- read.csv("~/Desktop/Endoderm_TC/After_removal_sample_info.csv")

Species <- After_removal_sample_info$Species
species <- After_removal_sample_info$Species
day <- After_removal_sample_info$Day
day <- as.factor(day)
individual <- After_removal_sample_info$Individual
Sample_ID <- After_removal_sample_info$Sample_ID
labels <- paste(Sample_ID, day, sep=" ")


# Log2(CPM)

cpm <- cpm(counts_genes63, log=TRUE)

# Filter lowly expressed genes

humans <- c(1:7, 16:23, 32:39, 48:55)
chimps <- c(8:15, 24:31, 40:47, 56:63)

cpm_filtered <- (rowSums(cpm[,humans] > 1.5) > 15 & rowSums(cpm[,chimps] > 1.5) > 16)

genes_in_cutoff <- cpm[cpm_filtered==TRUE,]
dim(genes_in_cutoff)

# Find the original counts of all of the genes that fit the criteria 

counts_genes_in_cutoff <- counts_genes63[cpm_filtered==TRUE,]
dim(counts_genes_in_cutoff)

# Take the TMM of the counts only for the genes that remain after filtering

dge_in_cutoff <- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff <- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff <- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)



```


## Here, we want to identify DE genes across the species.

```{r}


species <- c("H", "H","H","H","H","H","H", "C", "C","C","C","C","C","C","C","H","H","H","H","H","H","H","H",  "C", "C","C","C","C","C","C","C", "H","H","H","H","H","H","H","H",  "C", "C","C","C","C","C","C","C", "H","H","H","H","H","H","H","H",  "C", "C","C","C","C","C","C","C")

day <- c("0", "0","0","0","0","0","0", "0", "0", "0","0","0","0","0", "0", "0","1","1","1","1","1","1","1","1", "1","1","1","1","1","1","1","1",  "2", "2","2","2","2","2","2","2","2", "2","2","2","2","2","2","2",  "3", "3","3","3","3","3","3","3",  "3", "3","3","3","3","3","3")

condition <- factor(paste(species,day,sep="."))
design <- model.matrix(~ 0 + condition)
colnames(design) <- gsub("condition", "", dput(colnames(design)))

# We want a random effect term for individual. As a result, we want to run voom twice. See https://support.bioconductor.org/p/59700/

cpm.voom <- voom(dge_in_cutoff, design, normalize.method="none")

# We want a random effect term for individual
#corfit <- duplicateCorrelation(cpm.voom, design,block=individual)
corfit.correlation = 0.1947084

cpm.voom.corfit <- voom(dge_in_cutoff, design, plot = TRUE, normalize.method="none", block=individual, correlation = corfit.correlation )

# Plot the density to see shape of the distribution

plotDensities(cpm.voom.corfit, col=pal[as.numeric(day)], legend = F)

# Run lmFit and eBayes in limma

fit <- lmFit(cpm.voom.corfit , design, block=individual, correlation=corfit.correlation)


# In the contrast matrix, we have the species DE at each day

cm2 <- makeContrasts(HCday0 = c(1,0,0,0,-1,0,0,0), HCday1 = c(0,1,0,0,0,-1,0,0), HCday2 = c(0,0,1,0,0,0,-1,0), HCday3 = c(0,0,0,1,0,0,0,-1), levels = design)

# Fit the new model
diff_species <- contrasts.fit(fit, cm2)
fit2 <- eBayes(diff_species)


top3 <- list(HvCday0 =topTable(fit2, coef=1, adjust="BH", number=Inf, sort.by="none"), HvCday1 =topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none"),  HvCday2 =topTable(fit2, coef=3, adjust="BH", number=Inf, sort.by="none"),  HvCday3 =topTable(fit2, coef=4, adjust="BH", number=Inf, sort.by="none"))


# Set FDR level at 5% 

FDR_level <- 0.05
par(mfrow=c(1,1))

mylist <- list()
mylist[["DE Day 0"]] <- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val < FDR_level]
mylist[["DE Day 1"]] <-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val < FDR_level]
mylist[["DE Day 2"]] <- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val < FDR_level]
mylist[["DE Day 3"]] <- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val < FDR_level]

# For Bryan- make a list of genes DE between the species

# Step 1- designate which in design matrix you are interested in (genes, logFC, adj.P.value)

important_columns <- c(1,2,6)

# Find the genes that are DE at Day 0
HvCday0 =topTable(fit2, coef=1, adjust="BH", number=Inf, sort.by="none")
HvCday0_adj_pval <- HvCday0[which(HvCday0$adj.P.Val < 0.05), important_columns]                  

# Find the genes that are DE at Day 1
HvCday1 =topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none")
HvCday1_adj_pval <- HvCday1[which(HvCday1$adj.P.Val < 0.05), important_columns] 

# Find the genes that are DE at Day 2
HvCday2 =topTable(fit2, coef=3, adjust="BH", number=Inf, sort.by="none")
HvCday2_adj_pval <- HvCday2[which(HvCday2$adj.P.Val < 0.05), important_columns] 

# Find the genes that are DE at Day 3
HvCday3 =topTable(fit2, coef=4, adjust="BH", number=Inf, sort.by="none")
HvCday3_adj_pval <- HvCday3[which(HvCday3$adj.P.Val < 0.05), important_columns] 

dim(HvCday0_adj_pval)
dim(HvCday1_adj_pval)
dim(HvCday2_adj_pval)
dim(HvCday3_adj_pval)

# Make a table of the 4 sections

#write.table(HvCday0_adj_pval, "~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day0.txt", sep="\t")
#write.table(HvCday1_adj_pval, "~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day1.txt", sep="\t")
#write.table(HvCday2_adj_pval, "~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day2.txt", sep="\t")
#write.table(HvCday3_adj_pval, "~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day3.txt", sep="\t")

                  
# End section for Bryan

# Make as pdf 
Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between species per day (FDR< 5%)", cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
pdf(file = "Figure 7.D.pdf")
  grid.draw(Four_comp)
dev.off()

# Significant between days 0 and 1 in each species

#FDR_level <- 0.05
#par(mfrow=c(1,1))

#mylist <- list()

#mylist[["H"]] <-  row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val < FDR_level]
#mylist[["C"]] <- row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val < FDR_level]

#orange_blue <- c(2,5)
# Make as pdf 
#Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between days in each species (FDR< 5%)", cex=4 , fill = pal[orange_blue], lty=1, height=2000, width=3000)
#pdf(file = "Presentation_day12.pdf")
#  grid.draw(Four_comp)
#dev.off()

# Significant every day

sig_all <- length(intersect(intersect(intersect(mylist[["DE Day 0"]], mylist[["DE Day 1"]]), mylist[["DE Day 2"]]), mylist[["DE Day 3"]]))

nonlist <- list()
nonlist[["non DE Day 0"]] <- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val > FDR_level]
nonlist[["non DE Day 1"]] <-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val > FDR_level]
nonlist[["non DE Day 2"]] <- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val > FDR_level]
nonlist[["non DE Day 3"]] <- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val > FDR_level]

# Non-sign all
sig_none <- length(intersect(intersect(intersect(nonlist[["non DE Day 0"]], nonlist[["non DE Day 1"]]), nonlist[["non DE Day 2"]]), nonlist[["non DE Day 3"]]))

# Find unique genes for each day 

sig_day_0_only <- length(intersect(intersect(intersect(mylist[["DE Day 0"]], nonlist[["non DE Day 1"]]), nonlist[["non DE Day 2"]]), nonlist[["non DE Day 3"]]))

sig_day_1_only <- length(intersect(intersect(intersect(mylist[["DE Day 1"]], nonlist[["non DE Day 0"]]), nonlist[["non DE Day 2"]]), nonlist[["non DE Day 3"]]))

sig_day_2_only <- length(intersect(intersect(intersect(mylist[["DE Day 2"]], nonlist[["non DE Day 0"]]), nonlist[["non DE Day 1"]]), nonlist[["non DE Day 3"]]))

sig_day_3_only <- length(intersect(intersect(intersect(mylist[["DE Day 3"]], nonlist[["non DE Day 0"]]), nonlist[["non DE Day 1"]]), nonlist[["non DE Day 2"]]))

# Find the rest of the values for the graph (DE in 2 or 3 days)

black_day_0 <- length(mylist[["DE Day 0"]]) - sig_day_0_only

black_day_1 <- length(mylist[["DE Day 1"]]) - sig_day_1_only

black_day_2 <- length(mylist[["DE Day 2"]]) - sig_day_2_only

black_day_3 <- length(mylist[["DE Day 3"]]) - sig_day_3_only

#### Plot the number of genes DE between species at each day

dev.off()
# Numbers for non-unique DE genes
collect_total_barplot <- c(black_day_0, black_day_1, black_day_2, black_day_3, sig_none, sig_all)

# Numbers for unique DE genes
collect_only <- c(sig_day_0_only, sig_day_1_only, sig_day_2_only, sig_day_3_only, 0, 0)

# Combine the unique and non-unique DE genes
msig <- rbind(collect_total_barplot, collect_only)
colnames(msig) <- c("Day 0", "Day 1", "Day 2", "Day 3", "Never DE", "Always DE")

# Make a stacked barplot

barplot(msig, ylab="Number of DE genes", ylim = c(0, 6000), main = "Differentially Expressed Genes between Humans and Chimps Across Days")




```

## Find non-significant genes between species for each day

```{r}
# Not significant at each day
mylist <- list()
mylist[["non DE Day 0"]] <- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val > FDR_level]
mylist[["non DE Day 1"]] <-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val > FDR_level]
mylist[["non DE Day 2"]] <- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val > FDR_level]
mylist[["non DE Day 3"]] <- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val > FDR_level]

# Not significant at any days
notsig <- length(intersect(intersect(intersect(mylist[["non DE Day 0"]], mylist[["non DE Day 1"]]), mylist[["non DE Day 2"]]), mylist[["non DE Day 3"]]))


```


## Significant interaction term

Here, the model for expression has fixed effects for day, species, and a day-by-species interaction term. 

```{r}

# Make the design matrix with an interaction term between species and day

condition <- factor(paste(species,day,sep="."))
design <- model.matrix(~species+day+species*day)
colnames(design) <- gsub(":", "", dput(colnames(design)))
#colnames(design) <- gsub(":", "", dput(colnames(design)))

# We want a random effect term for individual. As a result, we want to run voom twice. See https://support.bioconductor.org/p/59700/

cpm.voom <- voom(dge_in_cutoff, design, normalize.method="none")

# We want a random effect term for individual
corfit <- duplicateCorrelation(cpm.voom, design,block=individual)
corfit.correlation = 0.1947084

cpm.voom.corfit <- voom(dge_in_cutoff, design, plot = TRUE, normalize.method="none", block=individual, correlation = corfit.correlation )

# Plot the density to see shape of the distribution

plotDensities(cpm.voom.corfit, col=pal[as.numeric(day)], legend = F)

# Run lmFit and eBayes in limma

fit <- lmFit(cpm.voom.corfit , design, block=individual, correlation=corfit.correlation)

# Fit the new model

fit2 <- eBayes(fit)

top3 <- list(day1inter =topTable(fit2, coef=6, adjust="BH", number=Inf, sort.by="none"), day2inter =topTable(fit2, coef=7, adjust="BH", number=Inf, sort.by="none"),  day3inter =topTable(fit2, coef=8, adjust="BH", number=Inf, sort.by="none"))

mylist <- list()
mylist[["Human x Day 1"]] <- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val < FDR_level]

mylist[["Human x Day 2"]] <- row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val < FDR_level]

mylist[["Human x Day 3"]] <- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val < FDR_level]


# Make as pdf 
Four_comp <- venn.diagram(mylist, filename= NULL, main="Genes with significant species by day interactions (FDR< 5%)", cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)
pdf(file = "Figure 6.A.pdf")
  grid.draw(Four_comp)
dev.off()



```

## Repeat the analysis but only using a subset of samples 

Now, we are going to subset to only include the 20157 (Batch 2), 28815 (Batch 1), and 28126 (Batch 2) in days 0-2 and only 20157 and 28126 in day 3. 

```{r}
# Subset the samples
samples_to_keep <- c(1,4,5, 8:15, 17, 20, 21, 24:31, 33,36, 37,40:47, 49,  52, 56:63 )

species <- c("H", "H","H","H","H","H","H", "C", "C","C","C","C","C","C","C","H","H","H","H","H","H","H","H",  "C", "C","C","C","C","C","C","C", "H","H","H","H","H","H","H","H",  "C", "C","C","C","C","C","C","C", "H","H","H","H","H","H","H","H",  "C", "C","C","C","C","C","C","C")

day <- c("0", "0","0","0","0","0","0", "0", "0", "0","0","0","0","0", "0", "0","1","1","1","1","1","1","1","1", "1","1","1","1","1","1","1","1",  "2", "2","2","2","2","2","2","2","2", "2","2","2","2","2","2","2",  "3", "3","3","3","3","3","3","3",  "3", "3","3","3","3","3","3")

# Subset the species and day
species1 <- as.data.frame(species)
species1 <- species1[samples_to_keep,]
day1 <- as.data.frame(day)
day1 <- day1[samples_to_keep,] 

Sample_ID_43 <- Sample_ID[samples_to_keep]
#Sample_ID_43[1] <- c("H1B")
labels <- paste(Sample_ID_43, day1, sep=" ")



# Take the TMM of the counts only for the genes that remain after filtering

counts_genes_in_cutoff_43 <- counts_genes_in_cutoff[samples_to_keep,]

dge_in_cutoff <- DGEList(counts=as.matrix(counts_genes_in_cutoff[, samples_to_keep]), genes=rownames(counts_genes_in_cutoff[, samples_to_keep]), group = as.character(t(labels)))
dge_in_cutoff <- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff <- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)
head(cpm_in_cutoff)
dim(cpm_in_cutoff)
```

### Data visualization

```{r}
# Make PCA plots with the factors colored by day

pca_genes <- prcomp(t(cpm_in_cutoff), scale = T, retx = TRUE, center = TRUE)
scores <- pca_genes$x

matrixpca <- pca_genes$x
pc1 <- matrixpca[,1]
pc2 <- matrixpca[,2]
pc3 <- matrixpca[,3]
pc4 <- matrixpca[,4]
pc5 <- matrixpca[,5]

pcs <- data.frame(pc1, pc2, pc3, pc4, pc5)

summary <- summary(pca_genes)


ggplot(data=pcs, aes(x=pc1, y=pc2, color=as.factor(day1), shape=as.factor(species1), size=2)) + geom_point() + xlab(paste("PC1 (",(summary$importance[2,1]*100), "% of variance)")) + ylab(paste("PC2 (",(summary$importance[2,2]*100), "% of variance)"))  + guides(color = guide_legend(order=1), size = FALSE, shape = guide_legend(order=2)) + scale_color_discrete(name  ="Day") + labs(title = "PCA of normalized endoderm data (Humans subset to match chimps)")

```


```{r}
# Make the design matrix

condition <- factor(paste(species1,day1,sep="."))
design <- model.matrix(~ 0 + condition)
colnames(design) <- gsub("condition", "", dput(colnames(design)))

# We want a random effect term for individual. As a result, we want to run voom twice. See https://support.bioconductor.org/p/59700/

cpm.voom <- voom(dge_in_cutoff, design, normalize.method="none")

# We want a random effect term for individual
#corfit <- duplicateCorrelation(cpm.voom, design,block=Sample_ID_43)
corfit.correlation = 0.349764

cpm.voom.corfit <- voom(dge_in_cutoff, design, plot = TRUE, normalize.method="none", block=Sample_ID_43, correlation = corfit.correlation )

# Plot the density to see shape of the distribution

plotDensities(cpm.voom.corfit, col=pal[as.numeric(day)], legend = F)

# Run lmFit and eBayes in limma

fit <- lmFit(cpm.voom.corfit , design, block=Sample_ID_43, correlation=corfit.correlation)


# In the contrast matrix, we have the species DE at each day

cm2 <- makeContrasts(HCday0 = c(1,0,0,0,-1,0,0,0), HCday1 = c(0,1,0,0,0,-1,0,0), HCday2 = c(0,0,1,0,0,0,-1,0), HCday3 = c(0,0,0,1,0,0,0,-1), levels = design)

# Fit the new model
diff_species <- contrasts.fit(fit, cm2)
fit2 <- eBayes(diff_species)


top3 <- list(HvCday0 =topTable(fit2, coef=1, adjust="BH", number=Inf, sort.by="none"), HvCday1 =topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none"),  HvCday2 =topTable(fit2, coef=3, adjust="BH", number=Inf, sort.by="none"),  HvCday3 =topTable(fit2, coef=4, adjust="BH", number=Inf, sort.by="none"))


# Set FDR level at 5% 

FDR_level <- 0.05
par(mfrow=c(1,1))

mylist <- list()
mylist[["DE Day 0"]] <- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val < FDR_level]
mylist[["DE Day 1"]] <-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val < FDR_level]
mylist[["DE Day 2"]] <- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val < FDR_level]
mylist[["DE Day 3"]] <- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val < FDR_level]

# For Bryan- make a list of genes DE between the species

# Step 1- designate which in design matrix you are interested in (genes, logFC, adj.P.value)

important_columns <- c(1,2,6)

# Find the genes that are DE at Day 0
HvCday0 =topTable(fit2, coef=1, adjust="BH", number=Inf, sort.by="none")
HvCday0_adj_pval <- HvCday0[which(HvCday0$adj.P.Val < 0.05), important_columns]                  

# Find the genes that are DE at Day 1
HvCday1 =topTable(fit2, coef=2, adjust="BH", number=Inf, sort.by="none")
HvCday1_adj_pval <- HvCday1[which(HvCday1$adj.P.Val < 0.05), important_columns] 

# Find the genes that are DE at Day 2
HvCday2 =topTable(fit2, coef=3, adjust="BH", number=Inf, sort.by="none")
HvCday2_adj_pval <- HvCday2[which(HvCday2$adj.P.Val < 0.05), important_columns] 

# Find the genes that are DE at Day 3
HvCday3 =topTable(fit2, coef=4, adjust="BH", number=Inf, sort.by="none")
HvCday3_adj_pval <- HvCday3[which(HvCday3$adj.P.Val < 0.05), important_columns] 

dim(HvCday0_adj_pval)
dim(HvCday1_adj_pval)
dim(HvCday2_adj_pval)
dim(HvCday3_adj_pval)

# Make a table of the 4 sections

#write.table(HvCday0_adj_pval, "~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day0.txt", sep="\t")
#write.table(HvCday1_adj_pval, "~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day1.txt", sep="\t")
#write.table(HvCday2_adj_pval, "~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day2.txt", sep="\t")
#write.table(HvCday3_adj_pval, "~/Desktop/Endoderm_TC/ashlar-trial/data/HvC_day3.txt", sep="\t")

                  
# End section for Bryan

# Make as pdf 
#Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between species per day (FDR< 5%)", cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
#pdf(file = "Figure 7.D.pdf")
#  grid.draw(Four_comp)
#dev.off()

# Significant between days 0 and 1 in each species

#FDR_level <- 0.05
#par(mfrow=c(1,1))

#mylist <- list()

#mylist[["H"]] <-  row.names(top3[[names(top3)[6]]])[top3[[names(top3)[6]]]$adj.P.Val < FDR_level]
#mylist[["C"]] <- row.names(top3[[names(top3)[5]]])[top3[[names(top3)[5]]]$adj.P.Val < FDR_level]

#orange_blue <- c(2,5)
# Make as pdf 
#Four_comp <- venn.diagram(mylist, filename= NULL, main="DE genes between days in each species (FDR< 5%)", cex=4 , fill = pal[orange_blue], lty=1, height=2000, width=3000)
#pdf(file = "Presentation_day12.pdf")
#  grid.draw(Four_comp)
#dev.off()

# Significant every day

sig_all <- length(intersect(intersect(intersect(mylist[["DE Day 0"]], mylist[["DE Day 1"]]), mylist[["DE Day 2"]]), mylist[["DE Day 3"]]))

nonlist <- list()
nonlist[["non DE Day 0"]] <- row.names(top3[[names(top3)[1]]])[top3[[names(top3)[1]]]$adj.P.Val > FDR_level]
nonlist[["non DE Day 1"]] <-  row.names(top3[[names(top3)[2]]])[top3[[names(top3)[2]]]$adj.P.Val > FDR_level]
nonlist[["non DE Day 2"]] <- row.names(top3[[names(top3)[3]]])[top3[[names(top3)[3]]]$adj.P.Val > FDR_level]
nonlist[["non DE Day 3"]] <- row.names(top3[[names(top3)[4]]])[top3[[names(top3)[4]]]$adj.P.Val > FDR_level]

# Non-sign all
sig_none <- length(intersect(intersect(intersect(nonlist[["non DE Day 0"]], nonlist[["non DE Day 1"]]), nonlist[["non DE Day 2"]]), nonlist[["non DE Day 3"]]))

# Find unique genes for each day 

sig_day_0_only <- length(intersect(intersect(intersect(mylist[["DE Day 0"]], nonlist[["non DE Day 1"]]), nonlist[["non DE Day 2"]]), nonlist[["non DE Day 3"]]))

sig_day_1_only <- length(intersect(intersect(intersect(mylist[["DE Day 1"]], nonlist[["non DE Day 0"]]), nonlist[["non DE Day 2"]]), nonlist[["non DE Day 3"]]))

sig_day_2_only <- length(intersect(intersect(intersect(mylist[["DE Day 2"]], nonlist[["non DE Day 0"]]), nonlist[["non DE Day 1"]]), nonlist[["non DE Day 3"]]))

sig_day_3_only <- length(intersect(intersect(intersect(mylist[["DE Day 3"]], nonlist[["non DE Day 0"]]), nonlist[["non DE Day 1"]]), nonlist[["non DE Day 2"]]))

sig_day_0_only
sig_day_1_only
sig_day_2_only
sig_day_3_only

# Find the rest of the values for the graph (DE in 2 or 3 days)

black_day_0 <- length(mylist[["DE Day 0"]]) - sig_day_0_only

black_day_1 <- length(mylist[["DE Day 1"]]) - sig_day_1_only

black_day_2 <- length(mylist[["DE Day 2"]]) - sig_day_2_only

black_day_3 <- length(mylist[["DE Day 3"]]) - sig_day_3_only

#### Plot the number of genes DE between species at each day

dev.off()
# Numbers for non-unique DE genes
collect_total_barplot <- c(black_day_0, black_day_1, black_day_2, black_day_3, sig_none, sig_all)

# Numbers for unique DE genes
collect_only <- c(sig_day_0_only, sig_day_1_only, sig_day_2_only, sig_day_3_only, 0, 0)

# Combine the unique and non-unique DE genes
msig <- rbind(collect_total_barplot, collect_only)
colnames(msig) <- c("Day 0", "Day 1", "Day 2", "Day 3", "Never DE", "Always DE")

# Make a stacked barplot

barplot(msig, ylab="Number of DE genes", ylim = c(0, 6000), main = "DE Genes between 2-3 Humans and Chimps Across Days")

```

