---
title: "Effect_size_updated_2lms"
author: "Lauren Blake"
date: "June 18, 2018"
output: html_document
---

This is a script to evaluate the effect sizes of the 2 linear models. 

## Load in data 


```{r chunk-options, include=FALSE}
source("chunk-options.R")
```


```{r}
# import sample labels
samples <- read.delim("../../../Reg_Evo_Primates/data/Sample_info_RNAseq_RIN.txt")
# expression
exprs <- read.table("../../../Reg_Evo_Primates/data/human_chimp_orth_exp_methyl_7725_hum.txt", sep="")
# methylation data
methyl <- read.csv("../../../Reg_Evo_Primates/data/chimp_human_orth_7725_avg_methyl_per_ts_gene.txt", sep="")

# Normalized gene expression data
cpm.voom.cyclic <- readRDS("../../../Reg_Evo_Primates/data/human_chimp_orth_cpm_voom_cyclic.rds")

# Counts

counts_12184 <- read.delim("~/Desktop/Reg_Evo_Primates/data/counts_12184.txt")

# Load libraries
library("edgeR")
library("limma")
library("plyr")
library("ashr")
library("cowplot")
library("devtools")
devtools::install_github("jhsiao999/mediation/pkg")
```

```{r}
#mediate.test.regressing <- function(Y, X, M) {

  library(limma)
  library(assertthat)

  # evaluate argument conditions
  assert_that(all.equal(dim(Y), dim(M)))
  assert_that(all.equal(length(unique(X)), 2))
  assert_that(all.equal(length(X), dim(Y)[2]))

  G <- nrow(Y)

  # model 1: Y_g ~ \tau_g X
  # specify tissue coding
  design_1 <- model.matrix(~X)

  Y_voom <- voom(Y, design=design_1, normalize.method = "none")
  model_1 <- lmFit(Y_voom, design_1)
  model_1 <- eBayes(model_1)

  # model 2: Y_g ~ \kappa_g M_g
  # to regress out the effect of mediating variable from Y
  Y_resid <- array(0, dim = dim(Y))
  for (g in 1:G){
    Y_resid[g,] <- lm(Y_voom$E[g, ] ~ as.numeric(M[g, ]))$resid
  }
  rownames(Y_resid) <- rownames(Y)
  Y_resid_voom <- vooma(Y_resid, design=design_1, plot=FALSE)

  # model 3: Y_resid_g ~ \tau^{\prime}_g X
  model_3 <- lmFit(Y_resid_voom, design_1)

  # get effect sizes
  beta1 <- coef(model_1[,2])
  beta3 <- coef(model_3[,2])

  se_beta1 <- se_beta2 <- se_beta3 <- cov_beta13 <- vector("numeric", G)

  # <---- get variances of \tau_g (condition effect)
  se_beta1 <- model_1$sigma*sqrt(model_1$cov.coefficients[2,2])

  # <---- get variances of \kappa_g mediator variable effect
  for (g in 1:length(se_beta2)) {
    design_2g <- model.matrix(~ as.numeric(M[g,]))
    sigma_g <- model_1$sigma[g]
    se_beta2[g] <- sigma_g*sqrt((solve(t(design_2g)%*%design_2g))[2,2])
  }

  # <---- get variances of \tau^{\prime}_g condition effect on expression after
  # regressing out mediating variable
  A <- solve(t(design_1)%*%design_1)%*%t(design_1)
  contr.vector <- array(c(0,1), dim=c(2,1))

  # compute beta3 by hand
  for (g in 1:length(se_beta3)) {
    M_g <- t(M[g,])
    design_2g <- model.matrix(~ as.numeric(M_g))
    A_2g <- solve(t(design_2g)%*%design_2g)%*%t(design_2g)
    sigma_g <- model_1$sigma[g]
    var_beta2g <- (se_beta2^2)[g]
    var_part1 <- (se_beta1[g])^2
    var_part2 <- ( A%*%M_g%*%var_beta2g%*%t(M_g)%*%t(A) )[2,2]
    var_part3 <- ( 2*(sigma_g^2)*A%*%t(A_2g)%*%contr.vector%*%t(M_g)%*%t(A) )[2,2]
    se_beta3[g] <- sqrt(var_part1 + var_part2 + var_part3)
  }

  # cov(beta1,beta3)
  for (g in 1:length(cov_beta13)) {
    M_g <- t(M[g,])
    design_2g <- model.matrix(~ as.numeric(M_g))
    A_2g <- solve(t(design_2g)%*%design_2g)%*%t(design_2g)
    sigma_g <- model_1$sigma[g]
    var_part1 <- (se_beta1[g])^2
    cov_beta13[g] <- var_part1 - (sigma_g^2)*((A %*% t(A_2g) %*%contr.vector%*%t(M_g)%*%t(A))[2,2])
  }

  # cov(beta1-beta3)
  d_se <- sqrt(se_beta1^2 + se_beta3^2 - 2*cov_beta13)
  d <- beta1-beta3

  df <- data.frame(d=as.numeric(d), d_se=d_se, tau = as.numeric(beta1),
                   tau_prime = as.numeric(beta3))
  rownames(df) <- rownames(Y)
  fit_ash <- ash(as.vector(df$d), d_se, mode=0)
#  return(df)
#}

# Check parameters

human_chimp_heart <- c(1, 5, 9, 13, 20, 24, 28)

# Methylation

exprs_methyl <- exprs[,49:79]

# Counts

counts_human_chimp_heart <- counts_12184[,human_chimp_heart]

human_chimp_heart1 <-  rownames(counts_human_chimp_heart) %in% rownames(exprs_methyl)
human_chimp_heart1 <- as.data.frame(human_chimp_heart1)
counts_genes_in <- cbind(counts_human_chimp_heart, human_chimp_heart1)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart1 == "TRUE")
counts_human_chimp_heart_subset <- counts_genes_in_cutoff[,1:7]

# Try running 

#mediate.test.regressing(counts_human_chimp_heart_subset, human_chimp_heart, exprs_methyl)

Y <- counts_human_chimp_heart_subset
two_species <- samples$Species[human_chimp_heart]
X <- droplevels.factor(two_species)
M <- exprs_methyl[,human_chimp_heart]

# Try just the DE genes

make_heart <- results[["HvC_Heart"]]

make_hearts_DE <- make_heart[which(make_heart$adj.P.Val < 0.05),]

human_chimp_heart1 <-  rownames(M) %in% rownames(make_hearts_DE)
human_chimp_heart1 <- as.data.frame(human_chimp_heart1)
counts_genes_in <- cbind(M, human_chimp_heart1)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart1 == "TRUE")
counts_human_chimp_heart_subset <- counts_genes_in_cutoff[,1:7]
M_DE <- counts_human_chimp_heart_subset

human_chimp_heart1 <-  rownames(Y) %in% rownames(make_hearts_DE)
human_chimp_heart1 <- as.data.frame(human_chimp_heart1)
counts_genes_in <- cbind(Y, human_chimp_heart1)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart1 == "TRUE")
counts_human_chimp_heart_subset <- counts_genes_in_cutoff[,1:7]
Y_DE <- counts_human_chimp_heart_subset

M <- M_DE
Y <- Y_DE

# DE based on 

# Run the linear model in limma
  design <- model.matrix(~ as.factor(tissue_no_extra))
  fit_all <- lmFit(cpm.voom.cyclic[,human_chimp_heart], design_1)
  fit_all <- eBayes(fit_all)

# Get results
  HvC_Heart_fit_all = topTable(fit_all, coef=2, adjust="BH", number=Inf, sort.by="none")
  HvC_Heart_fit_all_5perc <-
HvC_Heart_fit_all[which(HvC_Heart_fit_all$adj.P.Val < FDR_level), ]
  
 human_chimp_heart1 <-  rownames(M) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 <- as.data.frame(human_chimp_heart1)
counts_genes_in <- cbind(M, human_chimp_heart1)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart1 == "TRUE")
counts_human_chimp_heart_subset <- counts_genes_in_cutoff[,1:7]
M_DE <- counts_human_chimp_heart_subset

human_chimp_heart1 <-  rownames(Y) %in% rownames( HvC_Heart_fit_all_5perc)
human_chimp_heart1 <- as.data.frame(human_chimp_heart1)
counts_genes_in <- cbind(Y, human_chimp_heart1)
counts_genes_in_cutoff <- subset(counts_genes_in, human_chimp_heart1 == "TRUE")
counts_human_chimp_heart_subset <- counts_genes_in_cutoff[,1:7]
Y_DE <- counts_human_chimp_heart_subset

M <- M_DE
Y <- Y_DE 
```

