<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />


<title>Endoderm_original_analysis</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Endoderm_TimeCourse</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Endoderm_TC">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Endoderm_original_analysis</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>January 8, 2017</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#plots-of-unmapped-and-mapped-reads">PLOTS OF UNMAPPED AND MAPPED READS</a></li>
<li><a href="#visualization-of-the-raw-data">VISUALIZATION OF THE RAW DATA</a></li>
<li><a href="#initialize-normalization-64-samples">INITIALIZE NORMALIZATION (64 samples)</a><ul>
<li><a href="#filtering-lowly-expressed-genes">Filtering lowly expressed genes</a></li>
</ul></li>
<li><a href="#about-gc-content-normalization">About GC Content Normalization</a></li>
<li><a href="#correction-for-library-size">Correction for library size</a><ul>
<li><a href="#visualizing-the-filtered-data-64-samples-10304-genes">Visualizing the filtered data (64 samples, 10,304 genes)</a></li>
</ul></li>
<li><a href="#normalization-after-removal-of-sample-d0201570116">Normalization after removal of Sample D0201570116</a></li>
<li><a href="#final-visualization-of-the-data-all-technical-replicates-included-63-samples">Final Visualization of the data (all technical replicates included, 63 samples)</a></li>
<li><a href="#figures-for-the-paper">Figures for the paper</a></li>
<li><a href="#cormotif-analysis">Cormotif analysis</a><ul>
<li><a href="#set-up">Set up</a></li>
<li><a href="#identifying-genes-with-interspecies-differences-by-day">Identifying genes with interspecies differences by day</a></li>
<li><a href="#identifying-genes-showing-differences-during-differentiation">Identifying genes showing differences during differentiation</a></li>
<li><a href="#combined-interspecies-and-diffentiation-differences">Combined interspecies and diffentiation differences</a></li>
<li><a href="#differences-during-differentiation-in-each-species-separately">Differences during differentiation in each species separately</a></li>
</ul></li>
<li><a href="#evaluation-of-variance">Evaluation of variance</a></li>
<li><a href="#pairwise-correlations">Pairwise correlations</a><ul>
<li><a href="#using-mostly-batch-1-samples">Using mostly batch 1 samples</a></li>
<li><a href="#using-mostly-batch-2-samples">Using mostly batch 2 samples</a></li>
<li><a href="#using-average-of-log2cpm-expression-to-combine-technical-replicates">Using average of log2(CPM) expression to combine technical replicates</a></li>
<li><a href="#plot-the-correlations-genewise">Plot the correlations genewise</a></li>
</ul></li>
<li><a href="#assessing-differential-gene-expression-using-limma">Assessing differential gene expression using Limma</a><ul>
<li><a href="#make-a-venn-diagrams">Make a Venn Diagrams</a></li>
</ul></li>
<li><a href="#differentially-expressed-genes-using-a-model-with-no-interaction-term">Differentially expressed genes using a model with no interaction term</a></li>
<li><a href="#differences-between-species-at-each-time-points">Differences between species at each time points</a><ul>
<li><a href="#find-non-significant-genes-between-species-for-each-day">Find non-significant genes between species for each day</a></li>
<li><a href="#find-genes-between-species-for-each-day">Find genes between species for each day</a></li>
<li><a href="#find-genes-that-are-significant-between-species-only-for-one-day">Find genes that are significant between species only for one day</a></li>
<li><a href="#plot-the-number-of-genes-de-between-species-at-each-day">Plot the number of genes DE between species at each day</a></li>
<li><a href="#find-significant-genes-arising-at-a-given-day">Find significant genes arising at a given day</a><ul>
<li><a href="#plot-distributions-of-signficant-de-genes-between-species-at-each-day-line-946">Plot distributions of signficant DE genes between species at each day (line 946)</a></li>
</ul></li>
<li><a href="#plot-distributions-of-non-signficant-de-genes-between-species-at-each-day">Plot distributions of non-signficant DE genes between species at each day</a></li>
</ul></li>
<li><a href="#expression-values-for-individual-genes">Expression values for individual genes</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The goal of this script is to process the data from the 2 core runs of our endoderm time course project. The data was sequenced on an Illumina 4000 at the University of Chicago Genomics Core Facility. There are 2 species (humans and chimps) with 6 human iPSC lines with 2 replicates and 4 chimp iPSC lines with 4 replicates. RNA was extracted from the iPSCs for each day of the differentation into endoderm for a total of 4 days.</p>
</div>
<div id="plots-of-unmapped-and-mapped-reads" class="section level1">
<h1>PLOTS OF UNMAPPED AND MAPPED READS</h1>
<pre class="r"><code># Load necessary libraries
library(ggplot2)</code></pre>
<pre><code>## Warning: package &#39;ggplot2&#39; was built under R version 3.2.4</code></pre>
<pre class="r"><code>## Warning: package &#39;ggplot2&#39; was built under R version 3.2.3
library(scales)</code></pre>
<pre><code>## Warning: package &#39;scales&#39; was built under R version 3.2.3</code></pre>
<pre class="r"><code># Get data for unmapped and mapped reads
Endoderm_mapping_2_core_runs &lt;- read.csv(&quot;~/Desktop/Endoderm_TC/Endoderm_mapping_2_core_runs.csv&quot;)



# Plot mapped reads per sample with a line at 10 million reads

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Total_mapped, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (2 core runs)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample AFTER subsampling D0_3649_0116

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Total_mapped_post_subsample, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (2 core runs, post-subsampling)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-1-2.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample for core run 1

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_mapped_run1, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (core run 1)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-1-3.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample for core run 1 AFTER subsampling D0_3649_0116

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_mapped_post_subsample_run1, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (core run 1, post-subsampling)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-1-4.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample for core run 2

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_mapped_run2, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (core run 2)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-1-5.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample for core run 2

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_mapped_post_subsample_run2, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (core run 2, post-subsampling)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-1-6.png" width="672" /></p>
<pre class="r"><code># Plot for unmapped reads/sample for both core runs

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Total_unmapped, fill = Species)) + ylab(&quot;Number of unmapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Unmapped reads for all samples (2 core runs)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre class="r"><code># Plot for unmapped reads/sample

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_unmapped_run1, fill = Species)) + ylab(&quot;Number of unmapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Unmapped reads for all samples  (core run 1)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-2-2.png" width="672" /></p>
<pre class="r"><code># Plot for unmapped reads/lane in core run 1

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Lane_run1), y = Reads_unmapped_run1, fill = Species)) + ylab(&quot;Number of unmapped reads&quot;) + xlab(&quot;Lane number&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;)  + ggtitle(&quot;Unmapped reads for all samples  (core run 1)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-2-3.png" width="672" /></p>
<pre class="r"><code># Plot for unmapped reads/sample

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_unmapped_run2, fill = Species)) + ylab(&quot;Number of unmapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Unmapped reads for all samples  (core run 2)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-2-4.png" width="672" /></p>
<pre class="r"><code># Plot for unmapped reads/lane in core run 1

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Lane_run2), y = Reads_unmapped_run2, fill = Species)) + ylab(&quot;Number of unmapped reads&quot;) + xlab(&quot;Lane number&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + ggtitle(&quot;Unmapped reads for all samples  (core run 2)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-2-5.png" width="672" /></p>
<p>Originally, we had considered subsampling to the median.</p>
<pre class="r"><code># Originally, we had considered subsampling to the median. Find the median of the mapped reads (not including sample D0_3647_0116)

median_test &lt;- as.data.frame(Endoderm_mapping_2_core_runs$Total_mapped)
median_run &lt;- as.data.frame(median_test[-10,])

median(as.numeric(t(median_run)))</code></pre>
<pre><code>## [1] 35953173</code></pre>
<pre class="r"><code>#We will subsample Sample D036470116 to the median number of reads, 35953173 reads.
#Actual number of reads in  SampleD036470116 post subsampling: 35644778 reads.</code></pre>
<p>Given the distribution of the mapped reads, we will subsample to the second highest value.</p>
<pre class="r"><code># Plot mapped reads per sample with a line at 10 million reads

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Total_mapped, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (2 core runs)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample AFTER subsampling D0_3649_0116

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Total_mapped_post_subsample, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (2 core runs, post-subsampling)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-4-2.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample for core run 1

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_mapped_run1, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (core run 1)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-4-3.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample for core run 1 AFTER subsampling D0_3649_0116

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_mapped_post_subsample_run1, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (core run 1, post-subsampling)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-4-4.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample for core run 2

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_mapped_run2, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (core run 2)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-4-5.png" width="672" /></p>
<pre class="r"><code># Plot mapped reads per sample for core run 2

ggplot(Endoderm_mapping_2_core_runs, aes(x = factor(Sample), y = Reads_mapped_post_subsample_run2, fill = Species)) + ylab(&quot;Number of mapped reads&quot;) + xlab(&quot;Sample name&quot;) + geom_bar(stat = &quot;identity&quot;, colour = &quot;black&quot;) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + xlab(&quot;Sample name&quot;) + ggtitle(&quot;Mapped reads for all samples  (core run 2, post-subsampling)&quot;) + geom_hline(yintercept = 10000000) + scale_y_continuous(labels = comma)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-4-6.png" width="672" /></p>
</div>
<div id="visualization-of-the-raw-data" class="section level1">
<h1>VISUALIZATION OF THE RAW DATA</h1>
<pre class="r"><code># Load libraries

library(&quot;gplots&quot;)</code></pre>
<pre><code>## Warning: package &#39;gplots&#39; was built under R version 3.2.4</code></pre>
<pre><code>## 
## Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     lowess</code></pre>
<pre class="r"><code>library(&quot;RColorBrewer&quot;)
library(&quot;scales&quot;)
library(&quot;edgeR&quot;)</code></pre>
<pre><code>## Warning: package &#39;edgeR&#39; was built under R version 3.2.4</code></pre>
<pre><code>## Loading required package: limma</code></pre>
<pre><code>## Warning: package &#39;limma&#39; was built under R version 3.2.4</code></pre>
<pre class="r"><code># Load colors 

colors &lt;- colorRampPalette(c(brewer.pal(9, &quot;Blues&quot;)[1],brewer.pal(9, &quot;Blues&quot;)[9]))(100)
pal &lt;- c(brewer.pal(9, &quot;Set1&quot;), brewer.pal(8, &quot;Set2&quot;), brewer.pal(12, &quot;Set3&quot;))

# Set expression cutoff

expr_cutoff &lt;- 1.5

# Load count data

gene_counts_combined_raw_data &lt;- read.delim(&quot;~/Desktop/Endoderm_TC/gene_counts_combined.txt&quot;)
#gene_counts_combined_raw_data &lt;- read.delim(&quot;~/Dropbox/Endoderm TC/gene_counts_combined_raw_data.txt&quot;, header=FALSE, stringsAsFactors=FALSE)
counts_genes &lt;- gene_counts_combined_raw_data[1:30030,2:65]
rownames(counts_genes) &lt;- gene_counts_combined_raw_data[1:30030,1]
#counts_genes &lt;- gene_counts

# Load sample info

Endoderm_mapping_core_1 &lt;- read.csv(&quot;~/Desktop/Endoderm_TC/Endoderm_mapping_core_1.csv&quot;)

# Make labels with species and day
species &lt;- Endoderm_mapping_core_1$Species
day &lt;- Endoderm_mapping_core_1$Day
individual &lt;- Endoderm_mapping_core_1$Individual
Sample_ID &lt;- Endoderm_mapping_core_1$Sample_ID

labels &lt;- paste(species, day, sep=&quot; &quot;)</code></pre>
<pre class="r"><code># Hierarchical clustering on raw data

cors &lt;- cor(counts_genes, method=&quot;spearman&quot;, use=&quot;pairwise.complete.obs&quot;)

heatmap.2( cors, scale=&quot;column&quot;, col = colors, margins = c(12, 12), trace=&#39;none&#39;, denscol=&quot;white&quot;, labCol=labels, ColSideColors=pal[as.integer(as.factor(Endoderm_mapping_core_1$Species))], RowSideColors=pal[as.integer(as.factor(Endoderm_mapping_core_1$Day))+9], cexCol = 0.2 + 1/log10(15), cexRow = 0.2 + 1/log10(15))</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<pre class="r"><code>#PCA function (original code from Julien Roux)
#Load in the plot_scores function
plot_scores &lt;- function(pca, scores, n, m, cols, points=F, pchs =20, legend=F){
  xmin &lt;- min(scores[,n]) - (max(scores[,n]) - min(scores[,n]))*0.05
  if (legend == T){ ## let some room (35%) for a legend                                                                                                                                                 
    xmax &lt;- max(scores[,n]) + (max(scores[,n]) - min(scores[,n]))*0.50
  }
  else {
    xmax &lt;- max(scores[,n]) + (max(scores[,n]) - min(scores[,n]))*0.05
  }
  ymin &lt;- min(scores[,m]) - (max(scores[,m]) - min(scores[,m]))*0.05
  ymax &lt;- max(scores[,m]) + (max(scores[,m]) - min(scores[,m]))*0.05
  plot(scores[,n], scores[,m], xlab=paste(&quot;PC&quot;, n, &quot;: &quot;, round(summary(pca)$importance[2,n],3)*100, &quot;% variance explained&quot;, sep=&quot;&quot;), ylab=paste(&quot;PC&quot;, m, &quot;: &quot;, round(summary(pca)$importance[2,m],3)*100, &quot;% variance explained&quot;, sep=&quot;&quot;), xlim=c(xmin, xmax), ylim=c(ymin, ymax), type=&quot;n&quot;)
  if (points == F){
    text(scores[,n],scores[,m], rownames(scores), col=cols, cex=1)
  }
  else {
    points(scores[,n],scores[,m], col=cols, pch=pchs, cex=1.3)
  }
}

# Run the PCA 

# Check that there&#39;s no &quot;NAs&quot; in the data
select &lt;- counts_genes
summary(apply(select, 1, var) == 0)</code></pre>
<pre><code>##    Mode   FALSE    TRUE    NA&#39;s 
## logical   25466    4564       0</code></pre>
<pre class="r"><code>row_sub = apply(counts_genes, 1, function(row) all(row !=0 ))
counts_genes_no0 &lt;- counts_genes[row_sub,]

# Perform PCA

pca_genes &lt;- prcomp(t(counts_genes_no0), scale = T)
scores &lt;- pca_genes$x

#Make PCA plots with the factors colored by day

### PCs 1 and 2 Raw Data
for (n in 1:1){
  col.v &lt;- pal[as.integer(species)]
  plot_scores(pca_genes, scores, n, n+1, col.v)
}

for (n in 1:1){
  col.v &lt;- pal[as.integer(species)]
  plot_scores(pca_genes, scores, n, n+1, col.v)
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
<div id="initialize-normalization-64-samples" class="section level1">
<h1>INITIALIZE NORMALIZATION (64 samples)</h1>
<pre class="r"><code># Let&#39;s see what happens when we take the log2 of the raw counts

log_counts_genes &lt;- as.data.frame(log2(counts_genes))
head(log_counts_genes)</code></pre>
<pre><code>##                  D0_20157 D0_20157_0116  D0_20961  D0_21792  D0_28162
## ENSG00000000003 11.521110     10.709945 12.048487 12.100662 12.897278
## ENSG00000000005  5.321928      2.807355  5.169925  4.807355  7.228819
## ENSG00000000419 10.456354     10.198445 10.961450 11.321364 12.038919
## ENSG00000000457  7.357552      5.643856  8.588715  8.076816  9.111136
## ENSG00000000460  9.440869      7.189825 10.786270 10.148477 10.680360
## ENSG00000000938  1.584963      0.000000  4.584963  3.321928  4.954196
##                  D0_28815 D0_28815_0116  D0_29089   D0_3647 D0_36470116
## ENSG00000000003 11.989040     12.155451 11.224002 12.154818   13.254881
## ENSG00000000005  5.857981      7.087463  2.584963  3.169925    4.169925
## ENSG00000000419 10.718533     11.145932 10.459432  9.905387   11.949827
## ENSG00000000457  8.076816      9.002815  7.988685  8.668885   10.129283
## ENSG00000000460  9.829723     10.291171  9.479780  9.882643   10.920353
## ENSG00000000938  2.807355      3.906891  1.000000      -Inf    1.000000
##                   D0_3649 D0_3649_0116  D0_40300 D0_40300_0116  D0_4955
## ENSG00000000003 11.194141    10.038919 11.894818     11.224002 13.39460
## ENSG00000000005  2.321928     0.000000  3.807355      2.321928  4.70044
## ENSG00000000419  8.980140     8.515700  9.663558      9.596190 11.24555
## ENSG00000000457  8.294621     6.918863  8.930737      8.672425 10.37504
## ENSG00000000460  8.888743     7.569856  9.554589      9.703904 10.82655
## ENSG00000000938      -Inf     0.000000  1.000000          -Inf  1.00000
##                 D0_4955_0116  D1_20157 D1_20157_0116  D1_20961  D1_21792
## ENSG00000000003    11.038233 12.735133     11.849014 11.160502 11.843529
## ENSG00000000005     2.584963  7.700440      7.169925  6.794416  7.658211
## ENSG00000000419     9.177420 11.967226     11.147841 10.714246 11.277287
## ENSG00000000457     8.299208  8.778077      7.787903  7.357552  7.748193
## ENSG00000000460     9.290019 10.556506      9.923327  9.503826 10.049849
## ENSG00000000938     0.000000  1.584963      0.000000  0.000000  0.000000
##                  D1_28162  D1_28815 D1_28815_0116  D1_29089   D1_3647
## ENSG00000000003 12.377211 12.658435     12.395534 11.656872 11.034799
## ENSG00000000005  6.832890  8.294621      8.422065  6.357552  3.459432
## ENSG00000000419 11.536247 11.971184     11.871135 11.176173  9.377211
## ENSG00000000457  8.262095  8.894818      9.014020  7.832890  8.290019
## ENSG00000000460 10.370687 10.694358     11.091435 10.005625  9.052568
## ENSG00000000938  2.807355  1.000000      0.000000  1.000000      -Inf
##                 D1_3647_0116   D1_3649 D1_3649_0116  D1_40300
## ENSG00000000003    10.550747 11.401413    11.483312 12.233919
## ENSG00000000005     1.000000  2.000000     0.000000  2.000000
## ENSG00000000419     9.411511  9.649256    10.040290 10.350939
## ENSG00000000457     7.977280  8.717676     8.554589  9.162391
## ENSG00000000460     8.693487  9.335390     9.142107  9.977280
## ENSG00000000938     0.000000  0.000000         -Inf      -Inf
##                 D1_40300_0116   D1_4955 D1_4955_0116  D2_20157
## ENSG00000000003     11.051209 11.438272    12.670878 11.876133
## ENSG00000000005      2.321928  3.584963     2.321928  5.209453
## ENSG00000000419      9.465566  9.870365    11.398209 11.547377
## ENSG00000000457      8.479780  8.614710     9.840778  8.573647
## ENSG00000000460      9.537218  9.068778    10.904635  9.583083
## ENSG00000000938          -Inf      -Inf         -Inf      -Inf
##                 D2_20157_0116  D2_20961  D2_21792  D2_28162  D2_28815
## ENSG00000000003     10.650154 10.291171 11.325868 12.052908 10.650154
## ENSG00000000005      3.321928  4.459432  4.857981  2.807355  6.375039
## ENSG00000000419     10.421013 10.130571 11.186114 11.620220 11.627990
## ENSG00000000457      7.159871  7.321928  8.049849  8.885696  9.802516
## ENSG00000000460      8.507795  8.599913  9.719389 10.086136 10.209453
## ENSG00000000938      0.000000      -Inf  0.000000  2.584963  2.000000
##                 D2_28815_0116  D2_29089   D2_3647 D2_3647_0116   D2_3649
## ENSG00000000003     11.740202 11.353698 11.174926    10.471675 11.527966
## ENSG00000000005      6.845490  4.954196  0.000000         -Inf      -Inf
## ENSG00000000419     11.273213 11.032046  9.596190     9.184875  9.784635
## ENSG00000000457      9.296916  8.285402  8.622052     8.388017  8.930737
## ENSG00000000460     10.491853  9.831307  9.134426     8.487840  9.071462
## ENSG00000000938      1.584963  1.000000      -Inf         -Inf  0.000000
##                 D2_3649_0116  D2_40300 D2_40300_0116   D2_4955
## ENSG00000000003    10.803324 11.479275     10.381543 10.807355
## ENSG00000000005     0.000000  0.000000      0.000000  2.807355
## ENSG00000000419     9.211888  9.909893      9.025140  9.328675
## ENSG00000000457     8.400879  9.111136      8.388017  8.266787
## ENSG00000000460     8.060696  9.139551      8.479780  8.164907
## ENSG00000000938     0.000000      -Inf          -Inf      -Inf
##                 D2_4955_0116  D3_20157 D3_20157_0116  D3_20961  D3_21792
## ENSG00000000003    11.355351 11.591522     10.720244 10.552669 11.376668
## ENSG00000000005         -Inf  3.700440      2.321928  4.169925  4.700440
## ENSG00000000419    10.149747 11.239599      9.906891 10.210671 11.446566
## ENSG00000000457     9.283088  8.643856      7.442943  7.636625  8.945444
## ENSG00000000460     9.601771  9.714246      8.129283  8.744834 10.103288
## ENSG00000000938         -Inf      -Inf          -Inf      -Inf  0.000000
##                  D3_28162  D3_28815 D3_28815_0116  D3_29089   D3_3647
## ENSG00000000003 11.734710 11.029287     12.028597 11.053926 10.936638
## ENSG00000000005  2.321928  4.000000      8.794416  4.169925  3.000000
## ENSG00000000419 11.327553 10.564149     11.224002 10.551708  9.473706
## ENSG00000000457  8.679480  8.164907      9.142107  7.845490  8.918863
## ENSG00000000460  9.204571  8.361944     10.221587  9.011227  8.581201
## ENSG00000000938  1.000000      -Inf      0.000000  0.000000      -Inf
##                 D3_3647_0116   D3_3649 D3_3649_0116  D3_40300
## ENSG00000000003    11.250891 11.122181     9.661778 10.505812
## ENSG00000000005     1.000000  0.000000         -Inf  2.000000
## ENSG00000000419     9.719389  9.436712     7.672425  9.016808
## ENSG00000000457     9.047124  9.149747     7.000000  8.848623
## ENSG00000000460     8.103288  7.894818     5.392317  7.988685
## ENSG00000000938     0.000000  0.000000     2.000000      -Inf
##                 D3_40300_0116   D3_4955 D3_4955_0116
## ENSG00000000003     11.142107 10.512740    10.627534
## ENSG00000000005          -Inf  1.000000         -Inf
## ENSG00000000419      9.726218  9.259743     9.162391
## ENSG00000000457      8.965784  8.903882     8.647458
## ENSG00000000460      8.787903  7.523562     8.022368
## ENSG00000000938          -Inf      -Inf         -Inf</code></pre>
<pre class="r"><code># Plot density (a) by species and (b) by day
plotDensities(log_counts_genes, col=pal[as.numeric(Endoderm_mapping_core_1$Species)], legend=&quot;topright&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre class="r"><code>plotDensities(log_counts_genes, col=pal[as.numeric(Endoderm_mapping_core_1$Day)], legend=&quot;topright&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-8-2.png" width="672" /></p>
<pre class="r"><code># Log2(CPM)

cpm &lt;- cpm(counts_genes, log=TRUE)

# Make plot 
hist(cpm, main = &quot;log2(CPM) values in unfiltered data (n = 64 samples)&quot;, breaks = 100, ylim = c(0, 50000), xlab = &quot;log2(CPM) values&quot;)
abline(v = expr_cutoff, col = &quot;red&quot;, lwd = 3)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-8-3.png" width="672" /></p>
<pre class="r"><code># Plot density (a) by species and (b) by day
plotDensities(cpm, col=pal[as.numeric(Endoderm_mapping_core_1$Species)], legend=&quot;topright&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-8-4.png" width="672" /></p>
<pre class="r"><code>plotDensities(cpm, col=pal[as.numeric(Endoderm_mapping_core_1$Day)], legend=&quot;topright&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-8-5.png" width="672" /></p>
<pre class="r"><code># Plot library size

#boxplot_library_size &lt;- ggplot(dge_original$samples, aes(x = as.factor(Endoderm_mapping_core_1$Day), y = dge_original$samples$lib.size, fill = Endoderm_mapping_core_1$Species)) + geom_boxplot()

#boxplot_library_size + labs(title = &quot;Library size by day&quot;) + labs(y = &quot;Library size&quot;) + labs(x = &quot;Day&quot;) + guides(fill=guide_legend(title=&quot;Species&quot;))</code></pre>
<div id="filtering-lowly-expressed-genes" class="section level2">
<h2>Filtering lowly expressed genes</h2>
<p>We are beginning with 30030 genes and 64 samples (8 samples/species x 4 timepoints/species x 2 species) Based on what we have learned from Roux and Blake (<a href="http://lauren-blake.github.io/Reg_Evo_Primates/analysis/Correlation_bet_tech_factors_in_best_set_and_expression_stringent_filtering.html" class="uri">http://lauren-blake.github.io/Reg_Evo_Primates/analysis/Correlation_bet_tech_factors_in_best_set_and_expression_stringent_filtering.html</a>), we will use a cutoff of log2(CPM) &gt; 1.5 in at least 16 of the human samples and 16 of the chimp samples.</p>
<pre class="r"><code>expr_cutoff &lt;- 1.5

# Make plot 
hist(cpm, main = &quot;log2(CPM) values in unfiltered data (n = 64 samples)&quot;, breaks = 100, ylim = c(0, 50000), xlab = &quot;log2(CPM) values&quot;)
abline(v = expr_cutoff, col = &quot;red&quot;, lwd = 3)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<pre class="r"><code># Filter data

humans &lt;- c(1:8, 17:24, 33:40, 49:56)
chimps &lt;- c(9:16, 25:32, 41:48, 57:64 )


cpm_filtered &lt;- (rowSums(cpm[,humans] &gt; 1.5) &gt; 16 &amp; rowSums(cpm[,chimps] &gt; 1.5) &gt; 16)

genes_in_cutoff &lt;- cpm[cpm_filtered==TRUE,]
dim(genes_in_cutoff)</code></pre>
<pre><code>## [1] 10270    64</code></pre>
<pre class="r"><code># Make a histogram of the filtered data 

hist(as.numeric(unlist(genes_in_cutoff)), main = &quot;log2(CPM) values in filtered data (n = 64 samples, 10,270 genes)&quot;, breaks = 100, ylim = c(0, 50000), xlab = &quot;log2(CPM) values&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-9-2.png" width="672" /></p>
</div>
</div>
<div id="about-gc-content-normalization" class="section level1">
<h1>About GC Content Normalization</h1>
<p>Roux and Blakeâ€™s comparative analysis (<a href="http://lauren-blake.github.io/Reg_Evo_Primates/analysis/GC_content_normalization_CHT.html" class="uri">http://lauren-blake.github.io/Reg_Evo_Primates/analysis/GC_content_normalization_CHT.html</a>) showed that GC content normalization does not substantially impact gene counts. Therefore, we will not perform it in this analysis.</p>
</div>
<div id="correction-for-library-size" class="section level1">
<h1>Correction for library size</h1>
<pre class="r"><code># Find the original counts of all of the genes that fit the criteria 

counts_genes_in_cutoff &lt;- counts_genes[cpm_filtered==TRUE,]
dim(counts_genes_in_cutoff)</code></pre>
<pre><code>## [1] 10270    64</code></pre>
<pre class="r"><code># Take the TMM of the counts only for the genes that remain after filtering

dge_in_cutoff &lt;- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff &lt;- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff &lt;- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)

head(summary(cpm_in_cutoff))</code></pre>
<pre><code>##     D0_20157         D0_20157_0116         D0_20961       
##  &quot;Min.   :-0.9894  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-3.625  &quot;
##  &quot;1st Qu.: 3.9680  &quot; &quot;1st Qu.: 3.066  &quot; &quot;1st Qu.: 4.048  &quot;
##  &quot;Median : 5.3500  &quot; &quot;Median : 5.165  &quot; &quot;Median : 5.354  &quot;
##  &quot;Mean   : 5.2974  &quot; &quot;Mean   : 4.839  &quot; &quot;Mean   : 5.308  &quot;
##  &quot;3rd Qu.: 6.5794  &quot; &quot;3rd Qu.: 6.874  &quot; &quot;3rd Qu.: 6.575  &quot;
##  &quot;Max.   :12.7214  &quot; &quot;Max.   :14.024  &quot; &quot;Max.   :12.545  &quot;
##     D0_21792           D0_28162           D0_28815       
##  &quot;Min.   :-3.906  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-6.277  &quot;
##  &quot;1st Qu.: 3.840  &quot; &quot;1st Qu.: 4.030  &quot; &quot;1st Qu.: 3.981  &quot;
##  &quot;Median : 5.292  &quot; &quot;Median : 5.396  &quot; &quot;Median : 5.345  &quot;
##  &quot;Mean   : 5.206  &quot; &quot;Mean   : 5.311  &quot; &quot;Mean   : 5.277  &quot;
##  &quot;3rd Qu.: 6.600  &quot; &quot;3rd Qu.: 6.591  &quot; &quot;3rd Qu.: 6.574  &quot;
##  &quot;Max.   :12.780  &quot; &quot;Max.   :12.369  &quot; &quot;Max.   :12.697  &quot;
##  D0_28815_0116         D0_29089           D0_3647        
##  &quot;Min.   :-6.277  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-3.847  &quot;
##  &quot;1st Qu.: 3.983  &quot; &quot;1st Qu.: 4.058  &quot; &quot;1st Qu.: 3.919  &quot;
##  &quot;Median : 5.350  &quot; &quot;Median : 5.357  &quot; &quot;Median : 5.308  &quot;
##  &quot;Mean   : 5.286  &quot; &quot;Mean   : 5.313  &quot; &quot;Mean   : 5.251  &quot;
##  &quot;3rd Qu.: 6.564  &quot; &quot;3rd Qu.: 6.584  &quot; &quot;3rd Qu.: 6.596  &quot;
##  &quot;Max.   :12.859  &quot; &quot;Max.   :12.393  &quot; &quot;Max.   :12.686  &quot;
##   D0_36470116          D0_3649          D0_3649_0116     
##  &quot;Min.   :-6.277  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-6.277  &quot;
##  &quot;1st Qu.: 3.708  &quot; &quot;1st Qu.: 3.895  &quot; &quot;1st Qu.: 3.757  &quot;
##  &quot;Median : 5.246  &quot; &quot;Median : 5.299  &quot; &quot;Median : 5.267  &quot;
##  &quot;Mean   : 5.164  &quot; &quot;Mean   : 5.222  &quot; &quot;Mean   : 5.168  &quot;
##  &quot;3rd Qu.: 6.608  &quot; &quot;3rd Qu.: 6.578  &quot; &quot;3rd Qu.: 6.610  &quot;
##  &quot;Max.   :12.476  &quot; &quot;Max.   :12.764  &quot; &quot;Max.   :12.925  &quot;
##     D0_40300        D0_40300_0116         D0_4955        
##  &quot;Min.   :-6.277  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-4.746  &quot;
##  &quot;1st Qu.: 3.952  &quot; &quot;1st Qu.: 3.897  &quot; &quot;1st Qu.: 3.850  &quot;
##  &quot;Median : 5.327  &quot; &quot;Median : 5.325  &quot; &quot;Median : 5.286  &quot;
##  &quot;Mean   : 5.246  &quot; &quot;Mean   : 5.255  &quot; &quot;Mean   : 5.205  &quot;
##  &quot;3rd Qu.: 6.590  &quot; &quot;3rd Qu.: 6.603  &quot; &quot;3rd Qu.: 6.582  &quot;
##  &quot;Max.   :12.897  &quot; &quot;Max.   :12.788  &quot; &quot;Max.   :12.819  &quot;
##   D0_4955_0116         D1_20157        D1_20157_0116     
##  &quot;Min.   :-6.277  &quot; &quot;Min.   :-2.359  &quot; &quot;Min.   :-2.563  &quot;
##  &quot;1st Qu.: 3.870  &quot; &quot;1st Qu.: 3.950  &quot; &quot;1st Qu.: 3.958  &quot;
##  &quot;Median : 5.314  &quot; &quot;Median : 5.350  &quot; &quot;Median : 5.348  &quot;
##  &quot;Mean   : 5.234  &quot; &quot;Mean   : 5.303  &quot; &quot;Mean   : 5.301  &quot;
##  &quot;3rd Qu.: 6.604  &quot; &quot;3rd Qu.: 6.608  &quot; &quot;3rd Qu.: 6.593  &quot;
##  &quot;Max.   :12.687  &quot; &quot;Max.   :12.742  &quot; &quot;Max.   :12.874  &quot;
##     D1_20961           D1_21792           D1_28162       
##  &quot;Min.   :-1.452  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-3.585  &quot;
##  &quot;1st Qu.: 3.995  &quot; &quot;1st Qu.: 3.940  &quot; &quot;1st Qu.: 4.023  &quot;
##  &quot;Median : 5.357  &quot; &quot;Median : 5.351  &quot; &quot;Median : 5.380  &quot;
##  &quot;Mean   : 5.312  &quot; &quot;Mean   : 5.293  &quot; &quot;Mean   : 5.315  &quot;
##  &quot;3rd Qu.: 6.587  &quot; &quot;3rd Qu.: 6.603  &quot; &quot;3rd Qu.: 6.590  &quot;
##  &quot;Max.   :12.358  &quot; &quot;Max.   :12.642  &quot; &quot;Max.   :12.838  &quot;
##     D1_28815        D1_28815_0116         D1_29089       
##  &quot;Min.   :-6.277  &quot; &quot;Min.   :-4.527  &quot; &quot;Min.   :-3.943  &quot;
##  &quot;1st Qu.: 3.913  &quot; &quot;1st Qu.: 3.965  &quot; &quot;1st Qu.: 4.016  &quot;
##  &quot;Median : 5.344  &quot; &quot;Median : 5.353  &quot; &quot;Median : 5.365  &quot;
##  &quot;Mean   : 5.285  &quot; &quot;Mean   : 5.307  &quot; &quot;Mean   : 5.317  &quot;
##  &quot;3rd Qu.: 6.601  &quot; &quot;3rd Qu.: 6.596  &quot; &quot;3rd Qu.: 6.600  &quot;
##  &quot;Max.   :12.705  &quot; &quot;Max.   :12.724  &quot; &quot;Max.   :12.346  &quot;
##     D1_3647          D1_3647_0116         D1_3649        
##  &quot;Min.   :-2.492  &quot; &quot;Min.   :-2.163  &quot; &quot;Min.   :-6.277  &quot;
##  &quot;1st Qu.: 3.932  &quot; &quot;1st Qu.: 3.912  &quot; &quot;1st Qu.: 3.841  &quot;
##  &quot;Median : 5.324  &quot; &quot;Median : 5.329  &quot; &quot;Median : 5.306  &quot;
##  &quot;Mean   : 5.285  &quot; &quot;Mean   : 5.293  &quot; &quot;Mean   : 5.240  &quot;
##  &quot;3rd Qu.: 6.617  &quot; &quot;3rd Qu.: 6.643  &quot; &quot;3rd Qu.: 6.618  &quot;
##  &quot;Max.   :12.896  &quot; &quot;Max.   :12.942  &quot; &quot;Max.   :12.758  &quot;
##   D1_3649_0116         D1_40300        D1_40300_0116     
##  &quot;Min.   :-3.535  &quot; &quot;Min.   :-4.028  &quot; &quot;Min.   :-1.474  &quot;
##  &quot;1st Qu.: 3.749  &quot; &quot;1st Qu.: 3.813  &quot; &quot;1st Qu.: 3.947  &quot;
##  &quot;Median : 5.260  &quot; &quot;Median : 5.290  &quot; &quot;Median : 5.356  &quot;
##  &quot;Mean   : 5.195  &quot; &quot;Mean   : 5.206  &quot; &quot;Mean   : 5.306  &quot;
##  &quot;3rd Qu.: 6.613  &quot; &quot;3rd Qu.: 6.580  &quot; &quot;3rd Qu.: 6.615  &quot;
##  &quot;Max.   :12.607  &quot; &quot;Max.   :13.001  &quot; &quot;Max.   :12.780  &quot;
##     D1_4955          D1_4955_0116         D2_20157       
##  &quot;Min.   :-6.277  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-1.643  &quot;
##  &quot;1st Qu.: 3.759  &quot; &quot;1st Qu.: 3.813  &quot; &quot;1st Qu.: 4.034  &quot;
##  &quot;Median : 5.281  &quot; &quot;Median : 5.311  &quot; &quot;Median : 5.383  &quot;
##  &quot;Mean   : 5.189  &quot; &quot;Mean   : 5.243  &quot; &quot;Mean   : 5.348  &quot;
##  &quot;3rd Qu.: 6.612  &quot; &quot;3rd Qu.: 6.643  &quot; &quot;3rd Qu.: 6.572  &quot;
##  &quot;Max.   :12.896  &quot; &quot;Max.   :12.712  &quot; &quot;Max.   :13.725  &quot;
##  D2_20157_0116         D2_20961            D2_21792       
##  &quot;Min.   :-1.862  &quot; &quot;Min.   :-0.2512  &quot; &quot;Min.   :-1.138  &quot;
##  &quot;1st Qu.: 3.977  &quot; &quot;1st Qu.: 4.0231  &quot; &quot;1st Qu.: 4.019  &quot;
##  &quot;Median : 5.370  &quot; &quot;Median : 5.3629  &quot; &quot;Median : 5.366  &quot;
##  &quot;Mean   : 5.316  &quot; &quot;Mean   : 5.3503  &quot; &quot;Mean   : 5.349  &quot;
##  &quot;3rd Qu.: 6.594  &quot; &quot;3rd Qu.: 6.5847  &quot; &quot;3rd Qu.: 6.580  &quot;
##  &quot;Max.   :13.646  &quot; &quot;Max.   :13.2244  &quot; &quot;Max.   :12.932  &quot;
##     D2_28162           D2_28815        D2_28815_0116     
##  &quot;Min.   :-2.997  &quot; &quot;Min.   :-2.890  &quot; &quot;Min.   :-3.607  &quot;
##  &quot;1st Qu.: 4.030  &quot; &quot;1st Qu.: 4.096  &quot; &quot;1st Qu.: 4.021  &quot;
##  &quot;Median : 5.386  &quot; &quot;Median : 5.415  &quot; &quot;Median : 5.383  &quot;
##  &quot;Mean   : 5.334  &quot; &quot;Mean   : 5.381  &quot; &quot;Mean   : 5.339  &quot;
##  &quot;3rd Qu.: 6.583  &quot; &quot;3rd Qu.: 6.669  &quot; &quot;3rd Qu.: 6.569  &quot;
##  &quot;Max.   :13.897  &quot; &quot;Max.   :12.443  &quot; &quot;Max.   :13.251  &quot;
##     D2_29089           D2_3647           D2_3647_0116     
##  &quot;Min.   :-1.372  &quot; &quot;Min.   :-0.9076  &quot; &quot;Min.   :-1.522  &quot;
##  &quot;1st Qu.: 4.092  &quot; &quot;1st Qu.: 3.9531  &quot; &quot;1st Qu.: 4.039  &quot;
##  &quot;Median : 5.370  &quot; &quot;Median : 5.3625  &quot; &quot;Median : 5.390  &quot;
##  &quot;Mean   : 5.357  &quot; &quot;Mean   : 5.3251  &quot; &quot;Mean   : 5.342  &quot;
##  &quot;3rd Qu.: 6.560  &quot; &quot;3rd Qu.: 6.6357  &quot; &quot;3rd Qu.: 6.619  &quot;
##  &quot;Max.   :12.522  &quot; &quot;Max.   :13.1232  &quot; &quot;Max.   :13.607  &quot;
##     D2_3649          D2_3649_0116         D2_40300        
##  &quot;Min.   :-2.161  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-0.7417  &quot;
##  &quot;1st Qu.: 4.003  &quot; &quot;1st Qu.: 3.881  &quot; &quot;1st Qu.: 3.9560  &quot;
##  &quot;Median : 5.366  &quot; &quot;Median : 5.340  &quot; &quot;Median : 5.3484  &quot;
##  &quot;Mean   : 5.317  &quot; &quot;Mean   : 5.234  &quot; &quot;Mean   : 5.3038  &quot;
##  &quot;3rd Qu.: 6.601  &quot; &quot;3rd Qu.: 6.610  &quot; &quot;3rd Qu.: 6.6030  &quot;
##  &quot;Max.   :13.594  &quot; &quot;Max.   :12.862  &quot; &quot;Max.   :13.2013  &quot;
##  D2_40300_0116         D2_4955          D2_4955_0116     
##  &quot;Min.   :-1.791  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-6.277  &quot;
##  &quot;1st Qu.: 3.960  &quot; &quot;1st Qu.: 3.940  &quot; &quot;1st Qu.: 3.931  &quot;
##  &quot;Median : 5.359  &quot; &quot;Median : 5.329  &quot; &quot;Median : 5.364  &quot;
##  &quot;Mean   : 5.314  &quot; &quot;Mean   : 5.271  &quot; &quot;Mean   : 5.290  &quot;
##  &quot;3rd Qu.: 6.648  &quot; &quot;3rd Qu.: 6.577  &quot; &quot;3rd Qu.: 6.655  &quot;
##  &quot;Max.   :13.587  &quot; &quot;Max.   :13.100  &quot; &quot;Max.   :13.514  &quot;
##     D3_20157        D3_20157_0116         D3_20961        
##  &quot;Min.   :-2.060  &quot; &quot;Min.   :-2.545  &quot; &quot;Min.   :-0.5907  &quot;
##  &quot;1st Qu.: 4.114  &quot; &quot;1st Qu.: 4.091  &quot; &quot;1st Qu.: 4.1101  &quot;
##  &quot;Median : 5.409  &quot; &quot;Median : 5.414  &quot; &quot;Median : 5.3860  &quot;
##  &quot;Mean   : 5.379  &quot; &quot;Mean   : 5.361  &quot; &quot;Mean   : 5.3744  &quot;
##  &quot;3rd Qu.: 6.606  &quot; &quot;3rd Qu.: 6.606  &quot; &quot;3rd Qu.: 6.5787  &quot;
##  &quot;Max.   :13.317  &quot; &quot;Max.   :14.284  &quot; &quot;Max.   :13.1152  &quot;
##     D3_21792           D3_28162           D3_28815       
##  &quot;Min.   :-0.958  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-2.907  &quot;
##  &quot;1st Qu.: 4.106  &quot; &quot;1st Qu.: 4.031  &quot; &quot;1st Qu.: 3.981  &quot;
##  &quot;Median : 5.395  &quot; &quot;Median : 5.431  &quot; &quot;Median : 5.388  &quot;
##  &quot;Mean   : 5.379  &quot; &quot;Mean   : 5.346  &quot; &quot;Mean   : 5.316  &quot;
##  &quot;3rd Qu.: 6.580  &quot; &quot;3rd Qu.: 6.613  &quot; &quot;3rd Qu.: 6.611  &quot;
##  &quot;Max.   :12.253  &quot; &quot;Max.   :14.448  &quot; &quot;Max.   :13.826  &quot;
##  D3_28815_0116         D3_29089           D3_3647         
##  &quot;Min.   :-3.671  &quot; &quot;Min.   :-3.900  &quot; &quot;Min.   :-0.9919  &quot;
##  &quot;1st Qu.: 4.034  &quot; &quot;1st Qu.: 4.173  &quot; &quot;1st Qu.: 4.0378  &quot;
##  &quot;Median : 5.402  &quot; &quot;Median : 5.405  &quot; &quot;Median : 5.3931  &quot;
##  &quot;Mean   : 5.354  &quot; &quot;Mean   : 5.388  &quot; &quot;Mean   : 5.3686  &quot;
##  &quot;3rd Qu.: 6.590  &quot; &quot;3rd Qu.: 6.541  &quot; &quot;3rd Qu.: 6.6459  &quot;
##  &quot;Max.   :13.294  &quot; &quot;Max.   :12.500  &quot; &quot;Max.   :13.5213  &quot;
##   D3_3647_0116         D3_3649          D3_3649_0116     
##  &quot;Min.   :-6.277  &quot; &quot;Min.   :-6.277  &quot; &quot;Min.   :-6.277  &quot;
##  &quot;1st Qu.: 4.021  &quot; &quot;1st Qu.: 4.083  &quot; &quot;1st Qu.: 3.917  &quot;
##  &quot;Median : 5.409  &quot; &quot;Median : 5.412  &quot; &quot;Median : 5.384  &quot;
##  &quot;Mean   : 5.312  &quot; &quot;Mean   : 5.323  &quot; &quot;Mean   : 5.222  &quot;
##  &quot;3rd Qu.: 6.646  &quot; &quot;3rd Qu.: 6.595  &quot; &quot;3rd Qu.: 6.641  &quot;
##  &quot;Max.   :14.144  &quot; &quot;Max.   :14.109  &quot; &quot;Max.   :13.168  &quot;
##     D3_40300        D3_40300_0116         D3_4955        
##  &quot;Min.   :-6.277  &quot; &quot;Min.   :-3.749  &quot; &quot;Min.   :-6.277  &quot;
##  &quot;1st Qu.: 4.070  &quot; &quot;1st Qu.: 4.016  &quot; &quot;1st Qu.: 4.038  &quot;
##  &quot;Median : 5.410  &quot; &quot;Median : 5.403  &quot; &quot;Median : 5.429  &quot;
##  &quot;Mean   : 5.353  &quot; &quot;Mean   : 5.314  &quot; &quot;Mean   : 5.336  &quot;
##  &quot;3rd Qu.: 6.606  &quot; &quot;3rd Qu.: 6.629  &quot; &quot;3rd Qu.: 6.631  &quot;
##  &quot;Max.   :13.482  &quot; &quot;Max.   :14.081  &quot; &quot;Max.   :13.279  &quot;
##   D3_4955_0116     
##  &quot;Min.   :-6.277  &quot;
##  &quot;1st Qu.: 3.977  &quot;
##  &quot;Median : 5.398  &quot;
##  &quot;Mean   : 5.285  &quot;
##  &quot;3rd Qu.: 6.646  &quot;
##  &quot;Max.   :14.189  &quot;</code></pre>
<pre class="r"><code># Make density plots of the filtered data

plotDensities(cpm_in_cutoff, col=pal[as.numeric(species)], legend=&quot;topright&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<pre class="r"><code>plotDensities(cpm_in_cutoff, col=pal[as.numeric(day)], legend=&quot;topright&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-10-2.png" width="672" /></p>
<pre class="r"><code>#gplots::heatmap.2(x=as.matrix(t(cpm_in_cutoff)),
#                  , distfun = dist(x, method = &quot;euclidean&quot;), 
#                  hclustfun = function(x) hclust(dist(x), method = &quot;average&quot;), tracecol=NA, col=colors, denscol=&quot;white&quot;, ColSideColors=pal[as.integer(as.factor(Day))+9])

cors &lt;- cor(cpm_in_cutoff, method=&quot;spearman&quot;, use=&quot;pairwise.complete.obs&quot;)

heatmap.2( cors, scale=&quot;column&quot;, col = colors, margins = c(12, 12), trace=&#39;none&#39;, denscol=&quot;white&quot;, labCol=labels, ColSideColors=pal[as.integer(as.factor(Endoderm_mapping_core_1$Species))], RowSideColors=pal[as.integer(as.factor(Endoderm_mapping_core_1$Day))+9], cexCol = 0.2 + 1/log10(15), cexRow = 0.2 + 1/log10(15))</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-10-3.png" width="672" /></p>
<div id="visualizing-the-filtered-data-64-samples-10304-genes" class="section level2">
<h2>Visualizing the filtered data (64 samples, 10,304 genes)</h2>
<pre class="r"><code># Make PCA plots with the factors colored by day

pca_genes &lt;- prcomp(t(cpm_in_cutoff), scale = T)
scores &lt;- pca_genes$x

### PCs 1 and 2 
for (n in 1:1){
  col.v &lt;- pal[as.integer(species)]
  plot_scores(pca_genes, scores, n, n+1, col.v)
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<pre class="r"><code>for (n in 2:2){
  col.v &lt;- pal[as.integer(species)]
  plot_scores(pca_genes, scores, n, n+1, col.v)
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-11-2.png" width="672" /></p>
<p>From the PCA plot, we see that Sample D0201570116 is a clear outlier and doesnâ€™t cluster with anything, including its technical replicate. We will remove this sample and then go through the same process as above to visualize the data.</p>
</div>
</div>
<div id="normalization-after-removal-of-sample-d0201570116" class="section level1">
<h1>Normalization after removal of Sample D0201570116</h1>
<pre class="r"><code># Get data and sample info

counts_genes63 &lt;- counts_genes[,-2]
dim(counts_genes63)</code></pre>
<pre><code>## [1] 30030    63</code></pre>
<pre class="r"><code>After_removal_sample_info &lt;- read.csv(&quot;~/Desktop/Endoderm_TC/After_removal_sample_info.csv&quot;)

Species &lt;- After_removal_sample_info$Species
species &lt;- After_removal_sample_info$Species
day &lt;- After_removal_sample_info$Day
individual &lt;- After_removal_sample_info$Individual
Sample_ID &lt;- After_removal_sample_info$Sample_ID
labels &lt;- paste(Sample_ID, day, sep=&quot; &quot;)
rep_batch &lt;- After_removal_sample_info$Reprogramming_batch

# Log2(CPM)

cpm &lt;- cpm(counts_genes63, log=TRUE)

# Make plot 
hist(cpm, main = &quot;log2(CPM) values in unfiltered data (n = 64 samples)&quot;, breaks = 100, ylim = c(0, 50000), xlab = &quot;log2(CPM) values&quot;)
abline(v = 1.5, col = &quot;red&quot;, lwd = 3)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre class="r"><code># Filter lowly expressed genes

humans &lt;- c(1:7, 16:23, 32:39, 48:55)
chimps &lt;- c(8:15, 24:31, 40:47, 56:63)

cpm_filtered &lt;- (rowSums(cpm[,humans] &gt; 1.5) &gt; 15 &amp; rowSums(cpm[,chimps] &gt; 1.5) &gt; 16)

genes_in_cutoff &lt;- cpm[cpm_filtered==TRUE,]
dim(genes_in_cutoff)</code></pre>
<pre><code>## [1] 10304    63</code></pre>
<pre class="r"><code># Find the original counts of all of the genes that fit the criteria 

counts_genes_in_cutoff &lt;- counts_genes63[cpm_filtered==TRUE,]
dim(counts_genes_in_cutoff)</code></pre>
<pre><code>## [1] 10304    63</code></pre>
<pre class="r"><code># Take the TMM of the counts only for the genes that remain after filtering

dge_in_cutoff &lt;- DGEList(counts=as.matrix(counts_genes_in_cutoff), genes=rownames(counts_genes_in_cutoff), group = as.character(t(labels)))
dge_in_cutoff &lt;- calcNormFactors(dge_in_cutoff)

cpm_in_cutoff &lt;- cpm(dge_in_cutoff, normalized.lib.sizes=TRUE, log=TRUE)

head(summary(cpm_in_cutoff))</code></pre>
<pre><code>##     D0_20157           D0_20961           D0_21792       
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-4.394  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 3.969  &quot; &quot;1st Qu.: 4.063  &quot; &quot;1st Qu.: 3.831  &quot;
##  &quot;Median : 5.358  &quot; &quot;Median : 5.375  &quot; &quot;Median : 5.291  &quot;
##  &quot;Mean   : 5.294  &quot; &quot;Mean   : 5.320  &quot; &quot;Mean   : 5.196  &quot;
##  &quot;3rd Qu.: 6.592  &quot; &quot;3rd Qu.: 6.597  &quot; &quot;3rd Qu.: 6.600  &quot;
##  &quot;Max.   :12.735  &quot; &quot;Max.   :12.572  &quot; &quot;Max.   :12.787  &quot;
##     D0_28162           D0_28815        D0_28815_0116     
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-6.284  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 4.027  &quot; &quot;1st Qu.: 3.976  &quot; &quot;1st Qu.: 3.979  &quot;
##  &quot;Median : 5.397  &quot; &quot;Median : 5.355  &quot; &quot;Median : 5.354  &quot;
##  &quot;Mean   : 5.305  &quot; &quot;Mean   : 5.278  &quot; &quot;Mean   : 5.286  &quot;
##  &quot;3rd Qu.: 6.597  &quot; &quot;3rd Qu.: 6.584  &quot; &quot;3rd Qu.: 6.576  &quot;
##  &quot;Max.   :12.377  &quot; &quot;Max.   :12.712  &quot; &quot;Max.   :12.873  &quot;
##     D0_29089           D0_3647          D0_36470116      
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-3.860  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 4.068  &quot; &quot;1st Qu.: 3.888  &quot; &quot;1st Qu.: 3.759  &quot;
##  &quot;Median : 5.368  &quot; &quot;Median : 5.284  &quot; &quot;Median : 5.305  &quot;
##  &quot;Mean   : 5.318  &quot; &quot;Mean   : 5.225  &quot; &quot;Mean   : 5.220  &quot;
##  &quot;3rd Qu.: 6.599  &quot; &quot;3rd Qu.: 6.577  &quot; &quot;3rd Qu.: 6.671  &quot;
##  &quot;Max.   :12.413  &quot; &quot;Max.   :12.671  &quot; &quot;Max.   :12.543  &quot;
##     D0_3649          D0_3649_0116         D0_40300       
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-6.284  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 3.889  &quot; &quot;1st Qu.: 3.790  &quot; &quot;1st Qu.: 3.935  &quot;
##  &quot;Median : 5.297  &quot; &quot;Median : 5.298  &quot; &quot;Median : 5.322  &quot;
##  &quot;Mean   : 5.215  &quot; &quot;Mean   : 5.194  &quot; &quot;Mean   : 5.238  &quot;
##  &quot;3rd Qu.: 6.578  &quot; &quot;3rd Qu.: 6.646  &quot; &quot;3rd Qu.: 6.585  &quot;
##  &quot;Max.   :12.769  &quot; &quot;Max.   :12.963  &quot; &quot;Max.   :12.901  &quot;
##  D0_40300_0116         D0_4955          D0_4955_0116     
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-4.729  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 3.876  &quot; &quot;1st Qu.: 3.869  &quot; &quot;1st Qu.: 3.847  &quot;
##  &quot;Median : 5.311  &quot; &quot;Median : 5.309  &quot; &quot;Median : 5.302  &quot;
##  &quot;Mean   : 5.236  &quot; &quot;Mean   : 5.222  &quot; &quot;Mean   : 5.216  &quot;
##  &quot;3rd Qu.: 6.593  &quot; &quot;3rd Qu.: 6.608  &quot; &quot;3rd Qu.: 6.595  &quot;
##  &quot;Max.   :12.781  &quot; &quot;Max.   :12.848  &quot; &quot;Max.   :12.680  &quot;
##     D1_20157        D1_20157_0116         D1_20961       
##  &quot;Min.   :-4.542  &quot; &quot;Min.   :-6.284  &quot; &quot;Min.   :-1.443  &quot;
##  &quot;1st Qu.: 3.918  &quot; &quot;1st Qu.: 3.939  &quot; &quot;1st Qu.: 3.986  &quot;
##  &quot;Median : 5.330  &quot; &quot;Median : 5.347  &quot; &quot;Median : 5.359  &quot;
##  &quot;Mean   : 5.276  &quot; &quot;Mean   : 5.291  &quot; &quot;Mean   : 5.308  &quot;
##  &quot;3rd Qu.: 6.591  &quot; &quot;3rd Qu.: 6.595  &quot; &quot;3rd Qu.: 6.591  &quot;
##  &quot;Max.   :12.731  &quot; &quot;Max.   :12.879  &quot; &quot;Max.   :12.367  &quot;
##     D1_21792           D1_28162           D1_28815       
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-3.571  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 3.910  &quot; &quot;1st Qu.: 4.022  &quot; &quot;1st Qu.: 3.883  &quot;
##  &quot;Median : 5.330  &quot; &quot;Median : 5.391  &quot; &quot;Median : 5.332  &quot;
##  &quot;Mean   : 5.269  &quot; &quot;Mean   : 5.320  &quot; &quot;Mean   : 5.266  &quot;
##  &quot;3rd Qu.: 6.591  &quot; &quot;3rd Qu.: 6.604  &quot; &quot;3rd Qu.: 6.591  &quot;
##  &quot;Max.   :12.632  &quot; &quot;Max.   :12.857  &quot; &quot;Max.   :12.700  &quot;
##  D1_28815_0116         D1_29089           D1_3647        
##  &quot;Min.   :-4.525  &quot; &quot;Min.   :-3.941  &quot; &quot;Min.   :-2.519  &quot;
##  &quot;1st Qu.: 3.951  &quot; &quot;1st Qu.: 4.010  &quot; &quot;1st Qu.: 3.880  &quot;
##  &quot;Median : 5.353  &quot; &quot;Median : 5.361  &quot; &quot;Median : 5.290  &quot;
##  &quot;Mean   : 5.300  &quot; &quot;Mean   : 5.306  &quot; &quot;Mean   : 5.247  &quot;
##  &quot;3rd Qu.: 6.596  &quot; &quot;3rd Qu.: 6.595  &quot; &quot;3rd Qu.: 6.586  &quot;
##  &quot;Max.   :12.730  &quot; &quot;Max.   :12.350  &quot; &quot;Max.   :12.868  &quot;
##   D1_3647_0116         D1_3649          D1_3649_0116     
##  &quot;Min.   :-2.185  &quot; &quot;Min.   :-6.284  &quot; &quot;Min.   :-3.497  &quot;
##  &quot;1st Qu.: 3.869  &quot; &quot;1st Qu.: 3.828  &quot; &quot;1st Qu.: 3.779  &quot;
##  &quot;Median : 5.300  &quot; &quot;Median : 5.295  &quot; &quot;Median : 5.299  &quot;
##  &quot;Mean   : 5.261  &quot; &quot;Mean   : 5.225  &quot; &quot;Mean   : 5.231  &quot;
##  &quot;3rd Qu.: 6.615  &quot; &quot;3rd Qu.: 6.608  &quot; &quot;3rd Qu.: 6.655  &quot;
##  &quot;Max.   :12.919  &quot; &quot;Max.   :12.753  &quot; &quot;Max.   :12.652  &quot;
##     D1_40300        D1_40300_0116         D1_4955        
##  &quot;Min.   :-4.004  &quot; &quot;Min.   :-1.499  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 3.830  &quot; &quot;1st Qu.: 3.908  &quot; &quot;1st Qu.: 3.774  &quot;
##  &quot;Median : 5.317  &quot; &quot;Median : 5.320  &quot; &quot;Median : 5.309  &quot;
##  &quot;Mean   : 5.228  &quot; &quot;Mean   : 5.270  &quot; &quot;Mean   : 5.211  &quot;
##  &quot;3rd Qu.: 6.609  &quot; &quot;3rd Qu.: 6.585  &quot; &quot;3rd Qu.: 6.641  &quot;
##  &quot;Max.   :13.033  &quot; &quot;Max.   :12.755  &quot; &quot;Max.   :12.929  &quot;
##   D1_4955_0116         D2_20157        D2_20157_0116     
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-4.271  &quot; &quot;Min.   :-3.292  &quot;
##  &quot;1st Qu.: 3.792  &quot; &quot;1st Qu.: 4.037  &quot; &quot;1st Qu.: 3.999  &quot;
##  &quot;Median : 5.300  &quot; &quot;Median : 5.395  &quot; &quot;Median : 5.394  &quot;
##  &quot;Mean   : 5.227  &quot; &quot;Mean   : 5.357  &quot; &quot;Mean   : 5.335  &quot;
##  &quot;3rd Qu.: 6.633  &quot; &quot;3rd Qu.: 6.588  &quot; &quot;3rd Qu.: 6.619  &quot;
##  &quot;Max.   :12.707  &quot; &quot;Max.   :13.745  &quot; &quot;Max.   :13.676  &quot;
##     D2_20961            D2_21792           D2_28162       
##  &quot;Min.   :-0.8512  &quot; &quot;Min.   :-4.110  &quot; &quot;Min.   :-2.973  &quot;
##  &quot;1st Qu.: 4.0502  &quot; &quot;1st Qu.: 4.037  &quot; &quot;1st Qu.: 4.043  &quot;
##  &quot;Median : 5.3955  &quot; &quot;Median : 5.390  &quot; &quot;Median : 5.406  &quot;
##  &quot;Mean   : 5.3806  &quot; &quot;Mean   : 5.368  &quot; &quot;Mean   : 5.352  &quot;
##  &quot;3rd Qu.: 6.6221  &quot; &quot;3rd Qu.: 6.607  &quot; &quot;3rd Qu.: 6.605  &quot;
##  &quot;Max.   :13.2650  &quot; &quot;Max.   :12.962  &quot; &quot;Max.   :13.925  &quot;
##     D2_28815        D2_28815_0116         D2_29089       
##  &quot;Min.   :-2.881  &quot; &quot;Min.   :-3.593  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 4.100  &quot; &quot;1st Qu.: 4.024  &quot; &quot;1st Qu.: 4.096  &quot;
##  &quot;Median : 5.418  &quot; &quot;Median : 5.393  &quot; &quot;Median : 5.378  &quot;
##  &quot;Mean   : 5.384  &quot; &quot;Mean   : 5.347  &quot; &quot;Mean   : 5.361  &quot;
##  &quot;3rd Qu.: 6.676  &quot; &quot;3rd Qu.: 6.582  &quot; &quot;3rd Qu.: 6.572  &quot;
##  &quot;Max.   :12.454  &quot; &quot;Max.   :13.269  &quot; &quot;Max.   :12.539  &quot;
##     D2_3647           D2_3647_0116         D2_3649        
##  &quot;Min.   :-0.9384  &quot; &quot;Min.   :-1.542  &quot; &quot;Min.   :-2.164  &quot;
##  &quot;1st Qu.: 3.9096  &quot; &quot;1st Qu.: 4.018  &quot; &quot;1st Qu.: 3.998  &quot;
##  &quot;Median : 5.3231  &quot; &quot;Median : 5.363  &quot; &quot;Median : 5.358  &quot;
##  &quot;Mean   : 5.2865  &quot; &quot;Mean   : 5.316  &quot; &quot;Mean   : 5.307  &quot;
##  &quot;3rd Qu.: 6.6014  &quot; &quot;3rd Qu.: 6.596  &quot; &quot;3rd Qu.: 6.592  &quot;
##  &quot;Max.   :13.0918  &quot; &quot;Max.   :13.587  &quot; &quot;Max.   :13.590  &quot;
##   D2_3649_0116         D2_40300        D2_40300_0116     
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-0.721  &quot; &quot;Min.   :-1.810  &quot;
##  &quot;1st Qu.: 3.899  &quot; &quot;1st Qu.: 3.966  &quot; &quot;1st Qu.: 3.933  &quot;
##  &quot;Median : 5.365  &quot; &quot;Median : 5.361  &quot; &quot;Median : 5.334  &quot;
##  &quot;Mean   : 5.256  &quot; &quot;Mean   : 5.319  &quot; &quot;Mean   : 5.288  &quot;
##  &quot;3rd Qu.: 6.634  &quot; &quot;3rd Qu.: 6.622  &quot; &quot;3rd Qu.: 6.624  &quot;
##  &quot;Max.   :12.890  &quot; &quot;Max.   :13.223  &quot; &quot;Max.   :13.569  &quot;
##     D2_4955          D2_4955_0116         D3_20157       
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-6.284  &quot; &quot;Min.   :-4.450  &quot;
##  &quot;1st Qu.: 3.967  &quot; &quot;1st Qu.: 3.906  &quot; &quot;1st Qu.: 4.121  &quot;
##  &quot;Median : 5.359  &quot; &quot;Median : 5.344  &quot; &quot;Median : 5.425  &quot;
##  &quot;Mean   : 5.303  &quot; &quot;Mean   : 5.269  &quot; &quot;Mean   : 5.391  &quot;
##  &quot;3rd Qu.: 6.612  &quot; &quot;3rd Qu.: 6.639  &quot; &quot;3rd Qu.: 6.624  &quot;
##  &quot;Max.   :13.138  &quot; &quot;Max.   :13.500  &quot; &quot;Max.   :13.339  &quot;
##  D3_20157_0116         D3_20961           D3_21792       
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-2.778  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 4.096  &quot; &quot;1st Qu.: 4.129  &quot; &quot;1st Qu.: 4.113  &quot;
##  &quot;Median : 5.417  &quot; &quot;Median : 5.409  &quot; &quot;Median : 5.407  &quot;
##  &quot;Mean   : 5.362  &quot; &quot;Mean   : 5.395  &quot; &quot;Mean   : 5.387  &quot;
##  &quot;3rd Qu.: 6.614  &quot; &quot;3rd Qu.: 6.602  &quot; &quot;3rd Qu.: 6.595  &quot;
##  &quot;Max.   :14.296  &quot; &quot;Max.   :13.145  &quot; &quot;Max.   :12.272  &quot;
##     D3_28162           D3_28815        D3_28815_0116     
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-2.882  &quot; &quot;Min.   :-3.658  &quot;
##  &quot;1st Qu.: 4.055  &quot; &quot;1st Qu.: 3.998  &quot; &quot;1st Qu.: 4.037  &quot;
##  &quot;Median : 5.457  &quot; &quot;Median : 5.410  &quot; &quot;Median : 5.413  &quot;
##  &quot;Mean   : 5.371  &quot; &quot;Mean   : 5.335  &quot; &quot;Mean   : 5.362  &quot;
##  &quot;3rd Qu.: 6.645  &quot; &quot;3rd Qu.: 6.637  &quot; &quot;3rd Qu.: 6.603  &quot;
##  &quot;Max.   :14.483  &quot; &quot;Max.   :13.854  &quot; &quot;Max.   :13.312  &quot;
##     D3_29089           D3_3647          D3_3647_0116     
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-1.014  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 4.166  &quot; &quot;1st Qu.: 4.004  &quot; &quot;1st Qu.: 4.003  &quot;
##  &quot;Median : 5.402  &quot; &quot;Median : 5.365  &quot; &quot;Median : 5.393  &quot;
##  &quot;Mean   : 5.381  &quot; &quot;Mean   : 5.340  &quot; &quot;Mean   : 5.293  &quot;
##  &quot;3rd Qu.: 6.540  &quot; &quot;3rd Qu.: 6.619  &quot; &quot;3rd Qu.: 6.628  &quot;
##  &quot;Max.   :12.502  &quot; &quot;Max.   :13.499  &quot; &quot;Max.   :14.131  &quot;
##     D3_3649          D3_3649_0116         D3_40300       
##  &quot;Min.   :-6.284  &quot; &quot;Min.   :-6.284  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 4.062  &quot; &quot;1st Qu.: 3.934  &quot; &quot;1st Qu.: 4.042  &quot;
##  &quot;Median : 5.393  &quot; &quot;Median : 5.392  &quot; &quot;Median : 5.391  &quot;
##  &quot;Mean   : 5.304  &quot; &quot;Mean   : 5.233  &quot; &quot;Mean   : 5.334  &quot;
##  &quot;3rd Qu.: 6.580  &quot; &quot;3rd Qu.: 6.654  &quot; &quot;3rd Qu.: 6.591  &quot;
##  &quot;Max.   :14.095  &quot; &quot;Max.   :13.184  &quot; &quot;Max.   :13.469  &quot;
##  D3_40300_0116         D3_4955          D3_4955_0116     
##  &quot;Min.   :-3.739  &quot; &quot;Min.   :-6.284  &quot; &quot;Min.   :-6.284  &quot;
##  &quot;1st Qu.: 4.025  &quot; &quot;1st Qu.: 4.026  &quot; &quot;1st Qu.: 3.957  &quot;
##  &quot;Median : 5.413  &quot; &quot;Median : 5.418  &quot; &quot;Median : 5.380  &quot;
##  &quot;Mean   : 5.321  &quot; &quot;Mean   : 5.325  &quot; &quot;Mean   : 5.268  &quot;
##  &quot;3rd Qu.: 6.641  &quot; &quot;3rd Qu.: 6.623  &quot; &quot;3rd Qu.: 6.635  &quot;
##  &quot;Max.   :14.095  &quot; &quot;Max.   :13.274  &quot; &quot;Max.   :14.179  &quot;</code></pre>
<pre class="r"><code># Make plot 
hist(cpm_in_cutoff, main = &quot;log2(CPM) values in filtered data (10,304 genes from 63 samples)&quot;, breaks = 100, ylim = c(0, 50000), xlab = &quot;log2(CPM) values&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-12-2.png" width="672" /></p>
<pre class="r"><code># Make table of the normalized data

# write.table(cpm_in_cutoff,file=&quot;~/Dropbox/Endoderm TC/gene_counts_combined_norm_data.txt&quot;,sep=&quot;\t&quot;, col.names = T, row.names = T)</code></pre>
<p>There are 10,304 genes remaining.</p>
</div>
<div id="final-visualization-of-the-data-all-technical-replicates-included-63-samples" class="section level1">
<h1>Final Visualization of the data (all technical replicates included, 63 samples)</h1>
<pre class="r"><code>library(&quot;ggplot2&quot;)
dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
<pre class="r"><code>sample &lt;- Endoderm_mapping_2_core_runs$Sample[-2]

# Make PCA plots with the factors colored by day

pca_genes &lt;- prcomp(t(cpm_in_cutoff), scale = T, retx = TRUE, center = TRUE)
scores &lt;- pca_genes$x

### PCs Plots on filtered and normalized data
# PCs 1 and 2
for (n in 1:1){
  col.v &lt;- pal[as.integer(species)]
  plot_scores(pca_genes, scores, n, n+1, col.v)
}
for (n in 1:1){
  col.v &lt;- pal[as.integer(day)]
  plot_scores(pca_genes, scores, n, n+1, col.v)
}

matrixpca &lt;- pca_genes$x
pc1 &lt;- matrixpca[,1]
pc2 &lt;- matrixpca[,2]
pc3 &lt;- matrixpca[,3]
pc4 &lt;- matrixpca[,4]
pc5 &lt;- matrixpca[,5]
pc6 &lt;- matrixpca[,6]
pc7 &lt;- matrixpca[,7]
pc8 &lt;- matrixpca[,8]
pc9 &lt;- matrixpca[,9]
pc10 &lt;- matrixpca[,10]

pcs &lt;- data.frame(pc1, pc2, pc3, pc4, pc5, pc6, pc7, pc8, pc9, pc10)

summary &lt;- summary(pca_genes)


ggplot(data=pcs, aes(x=pc1, y=pc2, color=as.factor(day+1), shape=species, size=2)) +
geom_point(aes(text = sample)) + xlab(paste(&quot;PC1 (&quot;,(summary$importance[2,1]*100), &quot;% of variance)&quot;)) + ylab(paste(&quot;PC2 (&quot;,(summary$importance[2,2]*100), &quot;% of variance)&quot;)) + title(&quot;PCs 1 and 2 of Normalized Data&quot;) + guides(color = guide_legend(order=1), size = FALSE, shape = guide_legend(order=2))  + scale_color_discrete(name  =&quot;Day&quot;, labels=c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;)) + labs(title = &quot;PCA plot of normalized endoderm time course data&quot;)
#ggplotly()

# PCs 2 and 3
for (n in 2:2){
  col.v &lt;- pal[as.integer(species)]
  plot_scores(pca_genes, scores, n, n+1, col.v)
}
ggplot(data=pcs, aes(x=pc2, y=pc3, color=as.factor(day), shape=Species, size=2)) + geom_point(aes(text = sample)) + xlab(paste(&quot;PC2 (&quot;,(summary$importance[2,1]*100), &quot;% of variance)&quot;)) + ylab(paste(&quot;PC3 (&quot;,(summary$importance[2,2]*100), &quot;% of variance)&quot;)) + title(&quot;PCs 2 and 3 of Normalized Data&quot;) + guides(color = guide_legend(order=1), size = FALSE, shape = guide_legend(order=2)) + scale_color_discrete(name  =&quot;Day&quot;) + labs(title = &quot;PCA plot of normalized endoderm time course data&quot;)
#ggplotly()

# PCs 3 and 4
#for (n in 3:3){
#  col.v &lt;- pal[as.integer(species)]
#  plot_scores(pca_genes, scores, n, n+1, col.v)
#}

ggplot(data=pcs, aes(x=pc3, y=pc4, color=as.factor(day), shape=Species, size=2)) + geom_point(aes(text = sample)) + xlab(paste(&quot;PC3 (&quot;,(summary$importance[2,1]*100), &quot;% of variance)&quot;)) + ylab(paste(&quot;PC4 (&quot;,(summary$importance[2,2]*100), &quot;% of variance)&quot;)) + title(&quot;PCs 3 and 4 of Normalized Data&quot;) + guides(color = guide_legend(order=1), size = FALSE, shape = guide_legend(order=2)) + scale_color_discrete(name  =&quot;Day&quot;) + labs(title = &quot;PCA plot of normalized endoderm time course data&quot;)
#ggplotly()

# PCs 4 and 5
#for (n in 4:4){
#  col.v &lt;- pal[as.integer(species)]
#  plot_scores(pca_genes, scores, n, n+1, col.v)
#}

ggplot(data=pcs, aes(x=pc4, y=pc5, color=as.factor(day), shape=Species, size=2)) + geom_point(aes(text = sample)) + xlab(paste(&quot;PC4 (&quot;,(summary$importance[2,1]*100), &quot;% of variance)&quot;)) + ylab(paste(&quot;PC5 (&quot;,(summary$importance[2,2]*100), &quot;% of variance)&quot;)) + title(&quot;PCs 4 and 5 of Normalized Data&quot;) + guides(color = guide_legend(order=1), size = FALSE, shape = guide_legend(order=2)) + scale_color_discrete(name  =&quot;Day&quot;) + labs(title = &quot;PCA plot of normalized endoderm time course data&quot;)
#ggplotly()


# Hierarchical clustering on normalized data
dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
<pre class="r"><code>cors &lt;- cor(cpm_in_cutoff, method=&quot;pearson&quot;, use=&quot;pairwise.complete.obs&quot;)

heatmap.2( cors, scale=&quot;none&quot;, col = colors, margins = c(12, 12), trace=&#39;none&#39;, denscol=&quot;white&quot;, labCol= labels,labRow=&#39; &#39;, ColSideColors=pal[as.integer(as.factor(species))], RowSideColors=pal[as.integer(as.factor(day))+9], cexCol = 0.2 + 1/log10(15), cexRow = 0.2 + 1/log10(15))

#gplots::heatmap.2(x=as.matrix(t(cpm_in_cutoff)),
#                  , distfun = dist(x, method = &quot;euclidean&quot;), 
#                  hclustfun = function(x) hclust(dist(x), method = &quot;average&quot;), tracecol=NA, col=colors, denscol=&quot;white&quot;, ColSideColors=pal[as.integer(as.factor(Day))+9])</code></pre>
<p>PC 1 is correlated with day, PC 2 with species, PC 3 is somewhat by species, PC 4 with day, and PC 5 separates one sample (D228815) from the rest of the samples.</p>
</div>
<div id="figures-for-the-paper" class="section level1">
<h1>Figures for the paper</h1>
<pre class="r"><code>library(&quot;Biobase&quot;)</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following object is masked from &#39;package:limma&#39;:
## 
##     plotMA</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, as.vector, cbind,
##     colnames, do.call, duplicated, eval, evalq, Filter, Find, get,
##     grep, grepl, intersect, is.unsorted, lapply, lengths, Map,
##     mapply, match, mget, order, paste, pmax, pmax.int, pmin,
##     pmin.int, Position, rank, rbind, Reduce, rownames, sapply,
##     setdiff, sort, table, tapply, union, unique, unlist, unsplit</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre class="r"><code>library(&quot;GEOquery&quot;)</code></pre>
<pre><code>## Setting options(&#39;download.file.method.GEOquery&#39;=&#39;auto&#39;)</code></pre>
<pre><code>## Setting options(&#39;GEOquery.inmemory.gpl&#39;=FALSE)</code></pre>
<pre class="r"><code>library(&quot;plyr&quot;)</code></pre>
<pre><code>## Warning: package &#39;plyr&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>#theme_set(theme_bw(base_size = 16))
library(&quot;biomaRt&quot;)
library(&quot;colorfulVennPlot&quot;)</code></pre>
<pre><code>## Loading required package: grid</code></pre>
<pre class="r"><code>library(&quot;VennDiagram&quot;)</code></pre>
<pre><code>## Warning: package &#39;VennDiagram&#39; was built under R version 3.2.5</code></pre>
<pre><code>## Loading required package: futile.logger</code></pre>
<pre><code>## Warning: package &#39;futile.logger&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>library(&quot;gridExtra&quot;)</code></pre>
<pre><code>## Warning: package &#39;gridExtra&#39; was built under R version 3.2.4</code></pre>
<pre><code>## 
## Attaching package: &#39;gridExtra&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     combine</code></pre>
<pre><code>## The following object is masked from &#39;package:BiocGenerics&#39;:
## 
##     combine</code></pre>
<pre class="r"><code>library(&quot;car&quot;)</code></pre>
<pre><code>## Warning: package &#39;car&#39; was built under R version 3.2.5</code></pre>
<pre><code>## 
## Attaching package: &#39;car&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:VennDiagram&#39;:
## 
##     ellipse</code></pre>
<pre class="r"><code>library(&quot;topGO&quot;)</code></pre>
<pre><code>## Loading required package: graph</code></pre>
<pre><code>## 
## Attaching package: &#39;graph&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:plyr&#39;:
## 
##     join</code></pre>
<pre><code>## Loading required package: GO.db</code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Warning: package &#39;AnnotationDbi&#39; was built under R version 3.2.3</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Warning: package &#39;IRanges&#39; was built under R version 3.2.3</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Warning: package &#39;S4Vectors&#39; was built under R version 3.2.3</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:plyr&#39;:
## 
##     rename</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:plyr&#39;:
## 
##     desc</code></pre>
<pre><code>## The following object is masked from &#39;package:gplots&#39;:
## 
##     space</code></pre>
<pre><code>## Loading required package: DBI</code></pre>
<pre><code>## Warning: package &#39;DBI&#39; was built under R version 3.2.5</code></pre>
<pre><code>## </code></pre>
<pre><code>## Loading required package: SparseM</code></pre>
<pre><code>## Warning: package &#39;SparseM&#39; was built under R version 3.2.5</code></pre>
<pre><code>## 
## Attaching package: &#39;SparseM&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     backsolve</code></pre>
<pre><code>## Warning: replacing previous import by &#39;graph::.__C__dist&#39; when loading
## &#39;topGO&#39;</code></pre>
<pre><code>## 
## groupGOTerms:    GOBPTerm, GOMFTerm, GOCCTerm environments built.</code></pre>
<pre><code>## 
## Attaching package: &#39;topGO&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:IRanges&#39;:
## 
##     members</code></pre>
<pre><code>## The following object is masked from &#39;package:grid&#39;:
## 
##     depth</code></pre>
<pre class="r"><code>library(&quot;rowr&quot;)</code></pre>
<pre><code>## 
## Attaching package: &#39;rowr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:plyr&#39;:
## 
##     count</code></pre>
<pre class="r"><code>library(&quot;Cormotif&quot;)</code></pre>
<pre><code>## Loading required package: affy</code></pre>
<pre class="r"><code>library(&quot;ggfortify&quot;)</code></pre>
<pre><code>## Warning: package &#39;ggfortify&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>library(&quot;dplyr&quot;)</code></pre>
<pre><code>## Warning: package &#39;dplyr&#39; was built under R version 3.2.5</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:rowr&#39;:
## 
##     coalesce, count</code></pre>
<pre><code>## The following object is masked from &#39;package:AnnotationDbi&#39;:
## 
##     select</code></pre>
<pre><code>## The following objects are masked from &#39;package:IRanges&#39;:
## 
##     collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:S4Vectors&#39;:
## 
##     intersect, rename, setdiff, union</code></pre>
<pre><code>## The following object is masked from &#39;package:graph&#39;:
## 
##     union</code></pre>
<pre><code>## The following object is masked from &#39;package:car&#39;:
## 
##     recode</code></pre>
<pre><code>## The following object is masked from &#39;package:gridExtra&#39;:
## 
##     combine</code></pre>
<pre><code>## The following object is masked from &#39;package:biomaRt&#39;:
## 
##     select</code></pre>
<pre><code>## The following objects are masked from &#39;package:plyr&#39;:
## 
##     arrange, count, desc, failwith, id, mutate, rename, summarise,
##     summarize</code></pre>
<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from &#39;package:BiocGenerics&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(&quot;tidyr&quot;)</code></pre>
<pre><code>## Warning: package &#39;tidyr&#39; was built under R version 3.2.5</code></pre>
<pre><code>## 
## Attaching package: &#39;tidyr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:IRanges&#39;:
## 
##     expand</code></pre>
<pre class="r"><code>library(&quot;statmod&quot;)</code></pre>
<pre><code>## Warning: package &#39;statmod&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>library(&quot;qtlcharts&quot;)</code></pre>
<pre><code>## Warning: package &#39;qtlcharts&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>library(&quot;mygene&quot;)</code></pre>
<pre><code>## Loading required package: GenomicFeatures</code></pre>
<pre><code>## Warning: package &#39;GenomicFeatures&#39; was built under R version 3.2.3</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Warning: package &#39;GenomeInfoDb&#39; was built under R version 3.2.3</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Warning: package &#39;GenomicRanges&#39; was built under R version 3.2.3</code></pre>
<pre><code>## 
## Attaching package: &#39;GenomicFeatures&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:topGO&#39;:
## 
##     genes</code></pre>
<pre><code>## 
## Attaching package: &#39;mygene&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     query</code></pre>
<pre><code>## The following object is masked from &#39;package:biomaRt&#39;:
## 
##     getGene</code></pre>
</div>
<div id="cormotif-analysis" class="section level1">
<h1>Cormotif analysis</h1>
<div id="set-up" class="section level2">
<h2>Set up</h2>
<pre class="r"><code># Create a factor for the experiments, with the levels ordered such that as.numeric converts them to numbers in a defined manner

species &lt;- c(&quot;H&quot;, &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;, &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,
             &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,
             &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;)
group_fac &lt;- factor(paste(species, day, sep = &quot;.&quot;),
                    levels = c(&quot;C.0&quot;, &quot;C.1&quot;, &quot;C.2&quot;, &quot;C.3&quot;,
                               &quot;H.0&quot;, &quot;H.1&quot;, &quot;H.2&quot;, &quot;H.3&quot;))
groupid &lt;- as.numeric(group_fac)
# Conditions to compare:
# day0, day1, day2, day3, 
compid_day &lt;- data.frame(c1 = c(1, 2, 3, 4),
                         c2 = c(5, 6, 7, 8))
compid_tran &lt;- data.frame(c1 = c(1, 1, 1, 5, 5, 5),
                          c2 = c(2, 3, 4, 6, 7, 8))
compid_all &lt;- data.frame(c1 = c(1, 2, 3, 4, 1, 2, 3, 5, 6, 7),
                         c2 = c(5, 6, 7, 8, 2, 3, 4, 6, 7, 8))
                         
compid_chimp &lt;- data.frame(c1=c(1,2,3), c2=c(2,3,4))                         
compid_human &lt;- data.frame(c1=c(5,6,7), c2=c(6,7,8))</code></pre>
</div>
<div id="identifying-genes-with-interspecies-differences-by-day" class="section level2">
<h2>Identifying genes with interspecies differences by day</h2>
<pre class="r"><code>#interspecies differences at each day
set.seed(12345)
cormotif_day &lt;- cormotiffit(exprs = genes_in_cutoff , groupid = groupid, compid = compid_day, K = 3:5)</code></pre>
<pre><code>## [1] &quot;We have run the first 50 iterations for K=3&quot;
## [1] &quot;We have run the first 100 iterations for K=3&quot;
## [1] &quot;We have run the first 50 iterations for K=4&quot;
## [1] &quot;We have run the first 100 iterations for K=4&quot;
## [1] &quot;We have run the first 50 iterations for K=5&quot;
## [1] &quot;We have run the first 100 iterations for K=5&quot;</code></pre>
<pre class="r"><code>plotIC(cormotif_day)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<pre class="r"><code>plotMotif(cormotif_day)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-16-2.png" width="672" /></p>
<pre class="r"><code>gene_prob_day &lt;- cormotif_day$bestmotif$p.post
rownames(gene_prob_day) &lt;- rownames(cpm_in_cutoff)

# Note: Originally, the code had the &quot;sep2013 archive&quot; but it looks like that has been disabled online. 

ensembl &lt;- useMart(host = &quot;dec2013.archive.ensembl.org&quot;, biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)
gene_names &lt;- getBM(filters= &quot;ensembl_gene_id&quot;, attributes= c(&quot;ensembl_gene_id&quot;,&quot;hgnc_symbol&quot;) ,values=rownames(gene_prob_day),mart= ensembl)
gene_prob_day &lt;- gene_prob_day[(rownames(gene_prob_day) %in% gene_names[,1]),]
#rownames(gene_names) &lt;- gene_names[,1]
gene_names &lt;- gene_names[!duplicated(gene_names[,1]),]
all(rownames(gene_prob_day) == gene_names[,1])</code></pre>
<pre><code>## [1] TRUE</code></pre>
<pre class="r"><code>rownames(gene_prob_day) &lt;- gene_names[,2]

prob_sp_1 &lt;- rownames(gene_prob_day[(gene_prob_day[,1] &lt; 0.5 &amp; gene_prob_day[,2] &lt;0.5 &amp; gene_prob_day[,3] &lt;0.5 &amp; gene_prob_day[,4] &lt;0.5),])
length(prob_sp_1)</code></pre>
<pre><code>## [1] 3925</code></pre>
<pre class="r"><code>prob_sp_2 &lt;- rownames(gene_prob_day[(gene_prob_day[,1] &gt;0.5 &amp; gene_prob_day[,2] &gt;0.7 &amp; gene_prob_day[,3] &lt;0.5 &amp; gene_prob_day[,4] &lt;0.5),])
length(prob_sp_2)</code></pre>
<pre><code>## [1] 619</code></pre>
<pre class="r"><code>prob_sp_3 &lt;- rownames(gene_prob_day[(gene_prob_day[,1] &gt;0.9 &amp; gene_prob_day[,2] &gt;0.9 &amp; gene_prob_day[,3] &gt; 0.9 &amp; gene_prob_day[,4] &gt; 0.9),])
length(prob_sp_3)</code></pre>
<pre><code>## [1] 2727</code></pre>
<pre class="r"><code>prob_sp_4 &lt;- rownames(gene_prob_day[(gene_prob_day[,1] &lt;0.5&amp; gene_prob_day[,2] &lt;0.5 &amp; gene_prob_day[,3]&gt;0.7 &amp; gene_prob_day[,4] &gt;0.9),])
length(prob_sp_4)</code></pre>
<pre><code>## [1] 324</code></pre>
<pre class="r"><code>prob_sp_5 &lt;- rownames(gene_prob_day[(gene_prob_day[,1] &gt;0.9 &amp; gene_prob_day[,2] &gt;0.9 &amp; gene_prob_day[,3] &lt; 0.5 &amp; gene_prob_day[,4] &lt;0.5),])
length(prob_sp_5)</code></pre>
<pre><code>## [1] 322</code></pre>
<pre class="r"><code>3866+438+3207+346+294</code></pre>
<pre><code>## [1] 8151</code></pre>
<pre class="r"><code>cor_collect &lt;- cbind.fill(prob_sp_1, prob_sp_2, prob_sp_3, prob_sp_4, prob_sp_5, fill=NA)


write.table(cor_collect, file = &quot;corcollect.txt&quot;)
write.table(prob_sp_1, file=&quot;pr1.txt&quot;, row.names=FALSE)
write.table(prob_sp_2, file=&quot;pr2.txt&quot;, row.names=FALSE)
write.table(prob_sp_3, file=&quot;pr3.txt&quot;, row.names=FALSE)
write.table(prob_sp_4, file=&quot;pr4.txt&quot;, row.names=FALSE)
write.table(prob_sp_5, file=&quot;pr5.txt&quot;, row.names=FALSE)
all_genes &lt;- rownames(gene_prob_day)
write.csv(all_genes, file=&quot;all_genes.txt&quot;, row.names=FALSE)</code></pre>
</div>
<div id="identifying-genes-showing-differences-during-differentiation" class="section level2">
<h2>Identifying genes showing differences during differentiation</h2>
<pre class="r"><code>#differences during diffentiation
set.seed(12345)
cormotif_tran &lt;- cormotiffit(exprs = cpm_in_cutoff, groupid = groupid,
                        compid = compid_tran, K = 5:10)</code></pre>
<pre><code>## [1] &quot;We have run the first 50 iterations for K=5&quot;
## [1] &quot;We have run the first 100 iterations for K=5&quot;
## [1] &quot;We have run the first 50 iterations for K=6&quot;
## [1] &quot;We have run the first 100 iterations for K=6&quot;
## [1] &quot;We have run the first 50 iterations for K=7&quot;
## [1] &quot;We have run the first 100 iterations for K=7&quot;
## [1] &quot;We have run the first 50 iterations for K=8&quot;
## [1] &quot;We have run the first 100 iterations for K=8&quot;
## [1] &quot;We have run the first 50 iterations for K=9&quot;
## [1] &quot;We have run the first 100 iterations for K=9&quot;
## [1] &quot;We have run the first 50 iterations for K=10&quot;
## [1] &quot;We have run the first 100 iterations for K=10&quot;</code></pre>
<pre class="r"><code>gene_prob_tran &lt;- cormotif_tran$bestmotif$p.post
rownames(gene_prob_tran) &lt;- rownames( cpm_in_cutoff)
plotIC(cormotif_tran)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<pre class="r"><code>plotMotif(cormotif_tran)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-17-2.png" width="672" /></p>
<pre class="r"><code>dim(gene_prob_tran)</code></pre>
<pre><code>## [1] 10304     6</code></pre>
<pre class="r"><code># See note in previous chunk about the archive date

ensembl &lt;- useMart(host = &quot;dec2013.archive.ensembl.org&quot;,
                   biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                   dataset = &quot;hsapiens_gene_ensembl&quot;)
gene_names &lt;- getBM(filters= &quot;ensembl_gene_id&quot;, attributes= c(&quot;ensembl_gene_id&quot;,&quot;hgnc_symbol&quot;) ,values=rownames(gene_prob_tran),mart= ensembl)

gene_prob_tran &lt;- gene_prob_tran[rownames(gene_prob_tran) %in% gene_names[,1],]
for (i in 1:nrow(gene_prob_tran)) {
if(rownames(gene_prob_tran)[i] == gene_names[i,1]) {
rownames(gene_prob_tran)[i] &lt;- gene_names[i,2]
}
}

########

prob_day_1  &lt;- rownames(gene_prob_tran[(gene_prob_tran[,1] &lt;0.5 &amp; gene_prob_tran[,2] &lt;0.5 &amp; gene_prob_tran[,3] &lt;0.5 &amp; gene_prob_tran[,4] &lt;0.5 &amp; gene_prob_tran[,5] &lt; 0.5 &amp; gene_prob_tran[,6]&lt;0.5),])
length(prob_day_1)</code></pre>
<pre><code>## [1] 2852</code></pre>
<pre class="r"><code>write.table(prob_day_1, file=&quot;prob_cluster_1.txt&quot;)

prob_day_2  &lt;- rownames(gene_prob_tran[(gene_prob_tran[,1] &lt;0.5 &amp; gene_prob_tran[,2] &gt;0.7 &amp; gene_prob_tran[,3] &gt;0.9 &amp; gene_prob_tran[,4] &lt;0.5 &amp; gene_prob_tran[,5] &lt; 0.5 &amp; gene_prob_tran[,6]&lt;0.5),])
length(prob_day_2)</code></pre>
<pre><code>## [1] 169</code></pre>
<pre class="r"><code>write.table(prob_day_2, file=&quot;prob_cluster_2.txt&quot;)

prob_day_3  &lt;- rownames(gene_prob_tran[(gene_prob_tran[,1] &lt;0.5 &amp; gene_prob_tran[,2] &lt;0.5 &amp; gene_prob_tran[,3] &gt;0.9 &amp; gene_prob_tran[,4] &lt;0.5 &amp; gene_prob_tran[,5] &lt; 0.5 &amp; gene_prob_tran[,6]&gt;0.9),])
length(prob_day_3)</code></pre>
<pre><code>## [1] 399</code></pre>
<pre class="r"><code>write.table(prob_day_3, file=&quot;prob_cluster_3.txt&quot;)

prob_day_4  &lt;- rownames(gene_prob_tran[(gene_prob_tran[,1] &gt;0.9 &amp; gene_prob_tran[,2] &lt;0.5 &amp; gene_prob_tran[,3] &lt;0.5 &amp; gene_prob_tran[,4] &gt;0.9 &amp; gene_prob_tran[,5] &lt; 0.5 &amp; gene_prob_tran[,6]&lt;0.5),])
length(prob_day_4)</code></pre>
<pre><code>## [1] 255</code></pre>
<pre class="r"><code>write.table(prob_day_4, file=&quot;prob_cluster_4.txt&quot;)

prob_day_5  &lt;- rownames(gene_prob_tran[(gene_prob_tran[,1] &gt;0.9 &amp; gene_prob_tran[,2] &gt;0.9 &amp; gene_prob_tran[,3] &gt;0.9 &amp; gene_prob_tran[,4] &gt;0.5 &amp; gene_prob_tran[,5] &lt; 0.5 &amp; gene_prob_tran[,6]&lt;0.5),])
length(prob_day_5)</code></pre>
<pre><code>## [1] 66</code></pre>
<pre class="r"><code>write.table(prob_day_5, file=&quot;prob_cluster_5.txt&quot;)

prob_day_6  &lt;- rownames(gene_prob_tran[(gene_prob_tran[,1] &lt;0.5 &amp; gene_prob_tran[,2] &gt;0.9 &amp; gene_prob_tran[,3] &gt;0.9 &amp; gene_prob_tran[,4] &lt;0.5 &amp; gene_prob_tran[,5] &gt; 0.9 &amp; gene_prob_tran[,6]&gt;0.9),])
length(prob_day_6)</code></pre>
<pre><code>## [1] 1184</code></pre>
<pre class="r"><code>write.table(prob_day_6, file=&quot;prob_cluster_6.txt&quot;)

prob_day_7  &lt;- rownames(gene_prob_tran[(gene_prob_tran[,1] &gt;0.5 &amp; gene_prob_tran[,2] &lt;0.5 &amp; gene_prob_tran[,3] &lt;0.5 &amp; gene_prob_tran[,4] &gt;0.7 &amp; gene_prob_tran[,5] &lt; 0.9 &amp; gene_prob_tran[,6]&lt;0.9),])
length(prob_day_7)</code></pre>
<pre><code>## [1] 524</code></pre>
<pre class="r"><code>write.table(prob_day_7, file=&quot;prob_cluster_7.txt&quot;)

prob_day_8  &lt;- rownames(gene_prob_tran[(gene_prob_tran[,1] &gt;0.9 &amp; gene_prob_tran[,2] &gt;0.9 &amp; gene_prob_tran[,3] &gt;0.9 &amp; gene_prob_tran[,4] &gt;0.9 &amp; gene_prob_tran[,5] &gt; 0.9 &amp; gene_prob_tran[,6]&gt;0.9),])
length(prob_day_8)</code></pre>
<pre><code>## [1] 1477</code></pre>
<pre class="r"><code>write.table(prob_day_8, file=&quot;prob_cluster_8.txt&quot;)





####### SPLINES ##########

#prob_day_1  &lt;- rownames(ENSG_gene_prob_tran[(ENSG_gene_prob_tran[,1] &lt;0.5 &amp; ENSG_gene_prob_tran[,2] &lt;0.5 &amp; ENSG_gene_prob_tran[,3] &lt;0.5 &amp; ENSG_gene_prob_tran[,4] &lt;0.5 &amp; ENSG_gene_prob_tran[,5] &lt; 0.5 &amp; ENSG_gene_prob_tran[,6]&lt;0.5),])

#prob_day_1  &lt;- as.data.frame(prob_day_1)

#inshared_lists = row.names(cpm_in_cutoff) %in% prob_day_1[,1]
#inshared_lists_data &lt;- as.data.frame(inshared_lists)
#counts_in &lt;- cbind(cpm_in_cutoff, inshared_lists_data)
#counts_in_2_of_4 &lt;- as.data.frame(theme_bw()(counts_in, inshared_lists_data == &quot;TRUE&quot;))
#counts_in_2_of_4_col1 &lt;- as.data.frame(counts_in_2_of_4[,1:63])
#rownames(counts_in_2_of_4_col1) &lt;- rownames(counts_in_2_of_4)
#dim(counts_in_2_of_4_col1)
#write.table(counts_in_2_of_4_col1, file=&quot;ENSG_prob_cluster_1_cpm.txt&quot;)


#human_day0 &lt;- apply(counts_in_2_of_4_col1[,1:7], 1, median)
#chimp_day0 &lt;- apply(counts_in_2_of_4_col1[,8:15], 1, median)
#human_day1 &lt;- apply(counts_in_2_of_4_col1[,16:23], 1, median)
#chimp_day1 &lt;- apply(counts_in_2_of_4_col1[,24:31], 1, median)
#human_day2 &lt;- apply(counts_in_2_of_4_col1[,32:39], 1, median)
#chimp_day2 &lt;- apply(counts_in_2_of_4_col1[,40:47], 1, median)
#human_day3 &lt;- apply(counts_in_2_of_4_col1[,48:55], 1, median)
#chimp_day3 &lt;- apply(counts_in_2_of_4_col1[,56:63], 1, median)

#med &lt;- cbind(human_day0, chimp_day0, human_day1, chimp_day1,  human_day2,  chimp_day2,   human_day3, chimp_day3 )

#time &lt;- c(0,0,1,1,2,2,3,3)
#species &lt;- c(1,2,1,2,1,2,1,2)
#med2 &lt;- rbind(time, species, med)
#med3 &lt;- as.data.frame(t(med2))






#prob_day_2  &lt;- rownames(ENSG_gene_prob_tran[(ENSG_gene_prob_tran[,1] &lt;0.5 &amp; ENSG_gene_prob_tran[,2] &gt;0.7 &amp; ENSG_gene_prob_tran[,3] &gt;0.9 &amp; ENSG_gene_prob_tran[,4] &lt;0.5 &amp; ENSG_gene_prob_tran[,5] &lt; 0.5 &amp; ENSG_gene_prob_tran[,6]&lt;0.5),])
#length(prob_day_2)

#prob_day_2  &lt;- as.data.frame(prob_day_2)

#inshared_lists = row.names(cpm_in_cutoff) %in% prob_day_2[,1]
#inshared_lists_data &lt;- as.data.frame(inshared_lists)
#counts_in &lt;- cbind(cpm_in_cutoff, inshared_lists_data)
#counts_in_2_of_4 &lt;- as.data.frame(subset(counts_in, inshared_lists_data == &quot;TRUE&quot;))
#counts_in_2_of_4_col2 &lt;- as.data.frame(counts_in_2_of_4[,1:63])
#rownames(counts_in_2_of_4_col2) &lt;- rownames(counts_in_2_of_4)

#write.table(counts_in_2_of_4_col2, file=&quot;ENSG_prob_cluster_2.txt&quot;)

#prob_day_3  &lt;- rownames(ENSG_gene_prob_tran[(ENSG_gene_prob_tran[,1] &lt;0.5 &amp; ENSG_gene_prob_tran[,2] &lt;0.5 &amp; ENSG_gene_prob_tran[,3] &gt;0.9 &amp; ENSG_gene_prob_tran[,4] &lt;0.5 &amp; ENSG_gene_prob_tran[,5] &lt; 0.5 &amp; ENSG_gene_prob_tran[,6]&gt;0.9),])
#length(prob_day_3)

#prob_day_3  &lt;- as.data.frame(prob_day_3)
#dim(prob_day_3)

#inshared_lists = row.names(cpm_in_cutoff) %in% prob_day_3[,1]
#inshared_lists_data &lt;- as.data.frame(inshared_lists)
#counts_in &lt;- cbind(cpm_in_cutoff, inshared_lists_data)
#counts_in_2_of_4 &lt;- as.data.frame(subset(counts_in, inshared_lists_data == &quot;TRUE&quot;))
#counts_in_2_of_4_col3 &lt;- as.data.frame(counts_in_2_of_4[,1:63])
#rownames(counts_in_2_of_4_col3) &lt;- rownames(counts_in_2_of_4)
#dim(counts_in_2_of_4_col3)

#write.table(counts_in_2_of_4_col3, file=&quot;ENSG_prob_cluster_3.txt&quot;)

#prob_day_4  &lt;- rownames(ENSG_gene_prob_tran[(ENSG_gene_prob_tran[,1] &gt;0.9 &amp; ENSG_gene_prob_tran[,2] &lt;0.5 &amp; ENSG_gene_prob_tran[,3] &lt;0.5 &amp; ENSG_gene_prob_tran[,4] &gt;0.9 &amp; ENSG_gene_prob_tran[,5] &lt; 0.5 &amp; ENSG_gene_prob_tran[,6]&lt;0.5),])

#prob_day_4  &lt;- as.data.frame(prob_day_4)
#dim(prob_day_4)

#inshared_lists = row.names(cpm_in_cutoff) %in% prob_day_4[,1]
#inshared_lists_data &lt;- as.data.frame(inshared_lists)
#counts_in &lt;- cbind(cpm_in_cutoff, inshared_lists_data)
#counts_in_2_of_4 &lt;- as.data.frame(subset(counts_in, inshared_lists_data == &quot;TRUE&quot;))
#counts_in_2_of_4_col4 &lt;- as.data.frame(counts_in_2_of_4[,1:63])
#rownames(counts_in_2_of_4_col4) &lt;- rownames(counts_in_2_of_4)
#dim(counts_in_2_of_4_col4)

#write.table(counts_in_2_of_4_col4, file=&quot;ENSG_prob_cluster_4.txt&quot;)

#prob_day_5  &lt;- rownames(ENSG_gene_prob_tran[(ENSG_gene_prob_tran[,1] &gt;0.9 &amp; ENSG_gene_prob_tran[,2] &gt;0.9 &amp; ENSG_gene_prob_tran[,3] &gt;0.9 &amp; ENSG_gene_prob_tran[,4] &gt;0.5 &amp; ENSG_gene_prob_tran[,5] &lt; 0.5 &amp; ENSG_gene_prob_tran[,6]&lt;0.5),])

#prob_day_5  &lt;- as.data.frame(prob_day_5)
#dim(prob_day_5)

#inshared_lists = row.names(cpm_in_cutoff) %in% prob_day_5[,1]
#inshared_lists_data &lt;- as.data.frame(inshared_lists)
#counts_in &lt;- cbind(cpm_in_cutoff, inshared_lists_data)
#counts_in_2_of_4 &lt;- as.data.frame(subset(counts_in, inshared_lists_data == &quot;TRUE&quot;))
#counts_in_2_of_4_col5 &lt;- as.data.frame(counts_in_2_of_4[,1:63])
#rownames(counts_in_2_of_4_col5) &lt;- rownames(counts_in_2_of_4)
#dim(counts_in_2_of_4_col5)

#write.table(counts_in_2_of_4_col5, file=&quot;ENSG_prob_cluster_5.txt&quot;)

#prob_day_6  &lt;- rownames(ENSG_gene_prob_tran[(ENSG_gene_prob_tran[,1] &lt;0.5 &amp; ENSG_gene_prob_tran[,2] &gt;0.9 &amp; ENSG_gene_prob_tran[,3] &gt;0.9 &amp; ENSG_gene_prob_tran[,4] &lt;0.5 &amp; ENSG_gene_prob_tran[,5] &gt; 0.9 &amp; ENSG_gene_prob_tran[,6]&gt;0.9),])
#prob_day_6  &lt;- as.data.frame(prob_day_6)
#dim(prob_day_6)

#inshared_lists = row.names(cpm_in_cutoff) %in% prob_day_6[,1]
#inshared_lists_data &lt;- as.data.frame(inshared_lists)
#counts_in &lt;- cbind(cpm_in_cutoff, inshared_lists_data)
#counts_in_2_of_4 &lt;- as.data.frame(subset(counts_in, inshared_lists_data == &quot;TRUE&quot;))
#counts_in_2_of_4_col6 &lt;- as.data.frame(counts_in_2_of_4[,1:63])
#rownames(counts_in_2_of_4_col6) &lt;- rownames(counts_in_2_of_4)
#dim(counts_in_2_of_4_col6)

#write.table(counts_in_2_of_4_col6, file=&quot;ENSG_prob_cluster_6.txt&quot;)

#prob_day_7  &lt;- rownames(ENSG_gene_prob_tran[(ENSG_gene_prob_tran[,1] &gt;0.5 &amp; ENSG_gene_prob_tran[,2] &lt;0.5 &amp; ENSG_gene_prob_tran[,3] &lt;0.5 &amp; ENSG_gene_prob_tran[,4] &gt;0.7 &amp; ENSG_gene_prob_tran[,5] &gt; 0.9 &amp; ENSG_gene_prob_tran[,6]&gt;0.9),])
#length(prob_day_7)

#prob_day_7  &lt;- as.data.frame(prob_day_7)
#dim(prob_day_7)

#inshared_lists = row.names(cpm_in_cutoff) %in% prob_day_7[,1]
#inshared_lists_data &lt;- as.data.frame(inshared_lists)
#counts_in &lt;- cbind(cpm_in_cutoff, inshared_lists_data)
#counts_in_2_of_4 &lt;- as.data.frame(subset(counts_in, inshared_lists_data == &quot;TRUE&quot;))
#counts_in_2_of_4_col7 &lt;- as.data.frame(counts_in_2_of_4[,1:63])
#rownames(counts_in_2_of_4_col7) &lt;- rownames(counts_in_2_of_4)
#dim(counts_in_2_of_4_col7)

#write.table(counts_in_2_of_4_col7, file=&quot;ENSG_prob_cluster_7.txt&quot;)


#prob_day_8  &lt;- rownames(ENSG_gene_prob_tran[(ENSG_gene_prob_tran[,1] &gt;0.9 &amp; ENSG_gene_prob_tran[,2] &gt;0.9 &amp; ENSG_gene_prob_tran[,3] &gt;0.9 &amp; ENSG_gene_prob_tran[,4] &gt;0.9 &amp; ENSG_gene_prob_tran[,5] &gt; 0.9 &amp; ENSG_gene_prob_tran[,6]&gt;0.9),])

#prob_day_8  &lt;- as.data.frame(prob_day_8)

#inshared_lists = row.names(cpm_in_cutoff) %in% prob_day_8[,1]
#inshared_lists_data &lt;- as.data.frame(inshared_lists)
#counts_in &lt;- cbind(cpm_in_cutoff, inshared_lists_data)
#counts_in_2_of_4 &lt;- as.data.frame(subset(counts_in, inshared_lists_data == &quot;TRUE&quot;))
#counts_in_2_of_4_col8 &lt;- as.data.frame(counts_in_2_of_4[,1:63])
#rownames(counts_in_2_of_4_col8) &lt;- rownames(counts_in_2_of_4)
#dim(counts_in_2_of_4_col8)


#write.table(counts_in_2_of_4_col8, file=&quot;ENSG_prob_cluster_8.txt&quot;)

#plot(counts_in_2_of_4_col8[5,])</code></pre>
</div>
<div id="combined-interspecies-and-diffentiation-differences" class="section level2">
<h2>Combined interspecies and diffentiation differences</h2>
<pre class="r"><code>set.seed(12345)
cormotif_all &lt;- cormotiffit(exprs = cpm_in_cutoff, groupid = groupid,
                            compid = compid_all, K = 9:15)</code></pre>
<pre><code>## [1] &quot;We have run the first 50 iterations for K=9&quot;
## [1] &quot;We have run the first 100 iterations for K=9&quot;
## [1] &quot;We have run the first 50 iterations for K=10&quot;
## [1] &quot;We have run the first 100 iterations for K=10&quot;
## [1] &quot;We have run the first 50 iterations for K=11&quot;
## [1] &quot;We have run the first 100 iterations for K=11&quot;
## [1] &quot;We have run the first 50 iterations for K=12&quot;
## [1] &quot;We have run the first 100 iterations for K=12&quot;
## [1] &quot;We have run the first 50 iterations for K=13&quot;
## [1] &quot;We have run the first 100 iterations for K=13&quot;
## [1] &quot;We have run the first 50 iterations for K=14&quot;
## [1] &quot;We have run the first 100 iterations for K=14&quot;
## [1] &quot;We have run the first 50 iterations for K=15&quot;
## [1] &quot;We have run the first 100 iterations for K=15&quot;</code></pre>
<pre class="r"><code>gene_prob_all &lt;- cormotif_all$bestmotif$p.post
rownames(gene_prob_all) &lt;- rownames(cpm_in_cutoff)                            
plotIC(cormotif_all)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<pre class="r"><code>plotMotif(cormotif_all)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-18-2.png" width="672" /></p>
</div>
<div id="differences-during-differentiation-in-each-species-separately" class="section level2">
<h2>Differences during differentiation in each species separately</h2>
<pre class="r"><code>set.seed(12345)
cormotif_chimp &lt;- cormotiffit(exprs = cpm_in_cutoff, groupid = groupid,
                            compid = compid_chimp, K = 5:15)</code></pre>
<pre><code>## [1] &quot;We have run the first 50 iterations for K=5&quot;
## [1] &quot;We have run the first 100 iterations for K=5&quot;
## [1] &quot;We have run the first 50 iterations for K=6&quot;
## [1] &quot;We have run the first 100 iterations for K=6&quot;
## [1] &quot;We have run the first 50 iterations for K=7&quot;
## [1] &quot;We have run the first 100 iterations for K=7&quot;
## [1] &quot;We have run the first 50 iterations for K=8&quot;
## [1] &quot;We have run the first 100 iterations for K=8&quot;
## [1] &quot;We have run the first 50 iterations for K=9&quot;
## [1] &quot;We have run the first 100 iterations for K=9&quot;
## [1] &quot;We have run the first 50 iterations for K=10&quot;
## [1] &quot;We have run the first 100 iterations for K=10&quot;
## [1] &quot;We have run the first 50 iterations for K=11&quot;
## [1] &quot;We have run the first 100 iterations for K=11&quot;
## [1] &quot;We have run the first 50 iterations for K=12&quot;
## [1] &quot;We have run the first 100 iterations for K=12&quot;
## [1] &quot;We have run the first 50 iterations for K=13&quot;
## [1] &quot;We have run the first 100 iterations for K=13&quot;
## [1] &quot;We have run the first 50 iterations for K=14&quot;
## [1] &quot;We have run the first 100 iterations for K=14&quot;
## [1] &quot;We have run the first 50 iterations for K=15&quot;
## [1] &quot;We have run the first 100 iterations for K=15&quot;</code></pre>
<pre class="r"><code>gene_prob_chimp &lt;- cormotif_chimp$bestmotif$p.post
rownames(gene_prob_chimp) &lt;- rownames(cpm_in_cutoff)                            
plotIC(cormotif_chimp)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<pre class="r"><code>plotMotif(cormotif_chimp)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-19-2.png" width="672" /></p>
<pre class="r"><code>cormotif_human &lt;- cormotiffit(exprs = cpm_in_cutoff, groupid = groupid,
                            compid = compid_human, K = 5:15)</code></pre>
<pre><code>## [1] &quot;We have run the first 50 iterations for K=5&quot;
## [1] &quot;We have run the first 100 iterations for K=5&quot;
## [1] &quot;We have run the first 50 iterations for K=6&quot;
## [1] &quot;We have run the first 100 iterations for K=6&quot;
## [1] &quot;We have run the first 50 iterations for K=7&quot;
## [1] &quot;We have run the first 100 iterations for K=7&quot;
## [1] &quot;We have run the first 50 iterations for K=8&quot;
## [1] &quot;We have run the first 100 iterations for K=8&quot;
## [1] &quot;We have run the first 50 iterations for K=9&quot;
## [1] &quot;We have run the first 100 iterations for K=9&quot;
## [1] &quot;We have run the first 50 iterations for K=10&quot;
## [1] &quot;We have run the first 100 iterations for K=10&quot;
## [1] &quot;We have run the first 50 iterations for K=11&quot;
## [1] &quot;We have run the first 100 iterations for K=11&quot;
## [1] &quot;We have run the first 50 iterations for K=12&quot;
## [1] &quot;We have run the first 100 iterations for K=12&quot;
## [1] &quot;We have run the first 50 iterations for K=13&quot;
## [1] &quot;We have run the first 100 iterations for K=13&quot;
## [1] &quot;We have run the first 50 iterations for K=14&quot;
## [1] &quot;We have run the first 100 iterations for K=14&quot;
## [1] &quot;We have run the first 50 iterations for K=15&quot;
## [1] &quot;We have run the first 100 iterations for K=15&quot;</code></pre>
<pre class="r"><code>gene_prob_human &lt;- cormotif_human$bestmotif$p.post
rownames(gene_prob_human) &lt;- rownames(cpm_in_cutoff)
plotIC(cormotif_human)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-19-3.png" width="672" /></p>
<pre class="r"><code>plotMotif(cormotif_human)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-19-4.png" width="672" /></p>
</div>
</div>
<div id="evaluation-of-variance" class="section level1">
<h1>Evaluation of variance</h1>
<pre class="r"><code># Pull cpm for different combinations of species and day



d0b &lt;- cpm_in_cutoff[,day==&quot;0&quot;]
d1b &lt;- cpm_in_cutoff[,day==&quot;1&quot;]
d2b &lt;- cpm_in_cutoff[,day==&quot;2&quot;]
d3b &lt;- cpm_in_cutoff[,day==&quot;3&quot;]

d0H &lt;- cpm_in_cutoff[,day==&quot;0&quot; &amp; species==&quot;H&quot;]
dim(d0H)</code></pre>
<pre><code>## [1] 10304     7</code></pre>
<pre class="r"><code>d0C &lt;- cpm_in_cutoff[,day==&quot;0&quot; &amp; species==&quot;C&quot;]
dim(d0C)</code></pre>
<pre><code>## [1] 10304     8</code></pre>
<pre class="r"><code>d1H &lt;- cpm_in_cutoff[,day==&quot;1&quot; &amp; species==&quot;H&quot;]
dim(d1H)</code></pre>
<pre><code>## [1] 10304     8</code></pre>
<pre class="r"><code>d1C &lt;- cpm_in_cutoff[,day==&quot;1&quot; &amp; species==&quot;C&quot;]
dim(d1C)</code></pre>
<pre><code>## [1] 10304     8</code></pre>
<pre class="r"><code>d2H &lt;- cpm_in_cutoff[,day==&quot;2&quot; &amp; species==&quot;H&quot;]
dim(d2H)</code></pre>
<pre><code>## [1] 10304     8</code></pre>
<pre class="r"><code>d2C &lt;- cpm_in_cutoff[,day==&quot;2&quot; &amp; species==&quot;C&quot;]
dim(d2C)</code></pre>
<pre><code>## [1] 10304     8</code></pre>
<pre class="r"><code>d3H &lt;- cpm_in_cutoff[,day==&quot;3&quot; &amp; species==&quot;H&quot;]
dim(d3H)</code></pre>
<pre><code>## [1] 10304     8</code></pre>
<pre class="r"><code>d3C &lt;- cpm_in_cutoff[,day==&quot;3&quot; &amp; species==&quot;C&quot;]
dim(d3C)</code></pre>
<pre><code>## [1] 10304     8</code></pre>
<pre class="r"><code>var_d0b &lt;- apply(d0b, 1, var)
var_d1b &lt;- apply(d1b, 1, var)
var_d2b &lt;- apply(d2b, 1, var)
var_d3b &lt;- apply(d3b, 1, var)

m0b &lt;- mean(var_d0b)
m1b &lt;- mean(var_d1b)
m2b &lt;- mean(var_d2b)
m3b &lt;- mean(var_d3b)
meansb &lt;- c(m0b, m1b, m2b, m3b)
plot(meansb)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<pre class="r"><code>variance &lt;- data.frame(var=c(var_d0b, var_d1b, var_d2b, var_d3b), day = rep(c(&quot;D0&quot;, &quot;D1&quot;, &quot;D2&quot;, &quot;D3&quot;), times =c(length(var_d0b), length(var_d1b), length(var_d2b), length(var_d3b))))

# Make density plot 
ggplot(variance, aes(x=var, fill=day)) + geom_density(alpha=0.5)+xlab(&quot;Variance&quot;) + theme(legend.position=c(.75,.75), legend.title=element_blank(), axis.title=element_text(size=14), panel.background=element_rect(fill=&#39;white&#39;)) + theme(text = element_text(size=18)) +xlim(c(-0.1,1))+ theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())</code></pre>
<pre><code>## Warning: Removed 2391 rows containing non-finite values (stat_density).</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-20-2.png" width="672" /></p>
<pre class="r"><code>t.test(var_d1b, var_d3b)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  var_d1b and var_d3b
## t = -12.868, df = 18588, p-value &lt; 2.2e-16
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.1415122 -0.1041001
## sample estimates:
## mean of x mean of y 
## 0.2612758 0.3840820</code></pre>
<pre class="r"><code>var_d0H &lt;- apply(d0H, 1, var)
hist(var_d0H)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<pre class="r"><code>m0h &lt;- mean(var_d0H)
var_d1H &lt;- apply(d1H, 1, var)
hist(var_d1H)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-2.png" width="672" /></p>
<pre class="r"><code>m1h &lt;- mean(var_d1H)
var_d2H &lt;- apply(d2H, 1, var)
hist(var_d2H)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-3.png" width="672" /></p>
<pre class="r"><code>m2h &lt;- mean(var_d2H)
var_d3H &lt;- apply(d3H, 1, var)
hist(var_d3H)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-4.png" width="672" /></p>
<pre class="r"><code>m3h &lt;- mean(var_d3H)

var_d0C &lt;- apply(d0C, 1, var)
hist(var_d0C)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-5.png" width="672" /></p>
<pre class="r"><code>m0c &lt;- mean(var_d0C)
var_d1C &lt;- apply(d1C, 1, var)
hist(var_d1C)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-6.png" width="672" /></p>
<pre class="r"><code>m1c &lt;- mean(var_d1C)
var_d2C &lt;- apply(d2C, 1, var)
hist(var_d2C)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-7.png" width="672" /></p>
<pre class="r"><code>m2c &lt;- mean(var_d2C)
var_d3C &lt;- apply(d3C, 1, var)
hist(var_d3C)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-8.png" width="672" /></p>
<pre class="r"><code>m3c &lt;- mean(var_d3C)

means &lt;- c(m0h, m1h, m2h, m3h, m0c, m1c, m2c, m3c)
plot(means)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-9.png" width="672" /></p>
<pre class="r"><code>boxplot(var_d0H, var_d0C, var_d1H, var_d1C, var_d2H, var_d2C, var_d3H, var_d3C, ylim=c(0,4)) </code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-10.png" width="672" /></p>
<pre class="r"><code>variance_humans &lt;- data.frame(var=c(var_d0H, var_d1H, var_d2H, var_d3H), day = rep(c(&quot;D0&quot;,&quot;D1&quot;, &quot;D2&quot;, &quot;D3&quot;), times =c(length(var_d0H), length(var_d1H), length(var_d2H), length(var_d3H))))

ggplot(variance_humans, aes(x=var, fill=day)) + geom_density(alpha=0.5) +xlim(-.25,8.0)+xlab(&quot;Variance&quot;) + theme(legend.position=c(.75,.75), legend.title=element_blank(), axis.title=element_text(size=14), panel.background=element_rect(fill=&#39;white&#39;)) + theme(text = element_text(size=18)) +xlim(-.01, 2) + labs(title = &quot;Density plot of variance for humans&quot;)</code></pre>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 186 rows containing non-finite values (stat_density).</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-11.png" width="672" /></p>
<pre class="r"><code>variance_chimps &lt;- data.frame(var=c(var_d0C, var_d1C, var_d2C, var_d3C), day = rep(c(&quot;D0&quot;,&quot;D1&quot;, &quot;D2&quot;, &quot;D3&quot;), times =c(length(var_d0C), length(var_d1C), length(var_d2C), length(var_d3C))))
t.test(var_d0C, var_d1C)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  var_d0C and var_d1C
## t = 3.1294, df = 20605, p-value = 0.001754
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.009405086 0.040935338
## sample estimates:
## mean of x mean of y 
## 0.2287599 0.2035896</code></pre>
<pre class="r"><code>mean(var_d0H)</code></pre>
<pre><code>## [1] 0.1748098</code></pre>
<pre class="r"><code>ggplot(variance_chimps, aes(x=var, fill=day)) + geom_density(alpha=0.5) +xlim(-.25,8.0)+xlab(&quot;Variance&quot;) + theme(legend.position=c(.75,.75), legend.title=element_blank(), axis.title=element_text(size=14), panel.background=element_rect(fill=&#39;white&#39;)) + theme(text = element_text(size=18)) +xlim(-.01, 2) + theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) + labs(title = &quot;Density plot of variance for chimps&quot;)</code></pre>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<pre><code>## Warning: Removed 393 rows containing non-finite values (stat_density).</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-21-12.png" width="672" /></p>
<pre class="r"><code>t.test(var_d1H, var_d0H)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  var_d1H and var_d0H
## t = -12.193, df = 19256, p-value &lt; 2.2e-16
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.09096555 -0.06576993
## sample estimates:
##  mean of x  mean of y 
## 0.09644211 0.17480985</code></pre>
</div>
<div id="pairwise-correlations" class="section level1">
<h1>Pairwise correlations</h1>
<div id="using-mostly-batch-1-samples" class="section level2">
<h2>Using mostly batch 1 samples</h2>
<pre class="r"><code>library(&quot;tidyr&quot;)
library(&quot;dplyr&quot;)

# Run the same samples as Sammy
bio_rep &lt;- c(1:5, 7, 8, 10, 12, 14, 16, 18:21, 23, 24, 26, 28, 30, 32, 34:37, 39, 40, 42, 44, 46, 48, 50:53, 55, 56, 58, 60, 62)

bio_rep_cpm_in_cutoff &lt;- cpm_in_cutoff[,bio_rep]

# Find the technical factors for the biological replicates (no technical replicates)

bio_rep_samplefactors &lt;- read.delim(&quot;~/Downloads/samplefactors-filtered.txt&quot;, stringsAsFactors=FALSE) 

pair_cor &lt;- cor(bio_rep_cpm_in_cutoff, method = &quot;spearman&quot;)
pair_cor &lt;- as.data.frame(pair_cor)
pair_cor$sample1 &lt;- rownames(pair_cor)

pair_cor_long &lt;- gather(pair_cor, key = &quot;sample2&quot;, value = &quot;r&quot;, -sample1) %&gt;%
  mutate(sample2 = as.character(sample2)) %&gt;%
  filter(sample1 &lt; sample2) %&gt;% # ensures only one of the two entries is
  arrange(sample1, sample2)    # maintained
stopifnot(pair_cor_long$r &lt; 1,
          nrow(pair_cor_long) == (40*40 - 40) / 2)
head(pair_cor_long)</code></pre>
<pre><code>##    sample1  sample2         r
## 1 D0_20157 D0_20961 0.9566170
## 2 D0_20157 D0_21792 0.9624744
## 3 D0_20157 D0_28162 0.9595328
## 4 D0_20157 D0_28815 0.9806249
## 5 D0_20157 D0_29089 0.9542512
## 6 D0_20157  D0_3647 0.9089989</code></pre>
<pre class="r"><code>rownames(bio_rep_samplefactors) &lt;- colnames(bio_rep_cpm_in_cutoff)
pair_cor_long$day1 &lt;- &quot;&quot;
pair_cor_long$day2 &lt;- &quot;&quot;
pair_cor_long$species1 &lt;- &quot;&quot;
pair_cor_long$species2 &lt;- &quot;&quot;

for (i in 1:nrow(pair_cor_long)) {
  pair_cor_long$day1[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample1[i], &quot;Day&quot;]
  pair_cor_long$day2[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample2[i], &quot;Day&quot;]
  pair_cor_long$species1[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample1[i], &quot;Species&quot;]
  pair_cor_long$species2[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample2[i], &quot;Species&quot;]
}
head(pair_cor_long)</code></pre>
<pre><code>##    sample1  sample2         r day1 day2 species1 species2
## 1 D0_20157 D0_20961 0.9566170    0    0        H        H
## 2 D0_20157 D0_21792 0.9624744    0    0        H        H
## 3 D0_20157 D0_28162 0.9595328    0    0        H        H
## 4 D0_20157 D0_28815 0.9806249    0    0        H        H
## 5 D0_20157 D0_29089 0.9542512    0    0        H        H
## 6 D0_20157  D0_3647 0.9089989    0    0        H        C</code></pre>
<pre class="r"><code>pair_cor_long$human &lt;- NA
pair_cor_long$chimp &lt;- NA
pair_cor_long$cross &lt;- NA
pair_cor_long$all &lt;- NA
for (i in 1:nrow(pair_cor_long)) {
  # Do not include comparisons across days
  if (pair_cor_long$day1[i] != pair_cor_long$day2[i]) {
    next
  }
  pair_cor_long$all[i] &lt;- pair_cor_long$r[i]
  if (pair_cor_long$species1[i] == &quot;H&quot; &amp;
      pair_cor_long$species2[i] == &quot;H&quot;) {
    pair_cor_long$human[i] &lt;- pair_cor_long$r[i]
  } else if (pair_cor_long$species1[i] == &quot;C&quot; &amp;
             pair_cor_long$species2[i] == &quot;C&quot;) {
    pair_cor_long$chimp[i] &lt;- pair_cor_long$r[i]
  } else {
    pair_cor_long$cross[i] &lt;- pair_cor_long$r[i]
  }
}
head(pair_cor_long)</code></pre>
<pre><code>##    sample1  sample2         r day1 day2 species1 species2     human chimp
## 1 D0_20157 D0_20961 0.9566170    0    0        H        H 0.9566170    NA
## 2 D0_20157 D0_21792 0.9624744    0    0        H        H 0.9624744    NA
## 3 D0_20157 D0_28162 0.9595328    0    0        H        H 0.9595328    NA
## 4 D0_20157 D0_28815 0.9806249    0    0        H        H 0.9806249    NA
## 5 D0_20157 D0_29089 0.9542512    0    0        H        H 0.9542512    NA
## 6 D0_20157  D0_3647 0.9089989    0    0        H        C        NA    NA
##       cross       all
## 1        NA 0.9566170
## 2        NA 0.9624744
## 3        NA 0.9595328
## 4        NA 0.9806249
## 5        NA 0.9542512
## 6 0.9089989 0.9089989</code></pre>
<pre class="r"><code>df_plot &lt;- gather(pair_cor_long, key = &quot;comparison&quot;, value = &quot;correlation&quot;,
                  human, chimp, cross, all, na.rm = TRUE)
stopifnot(sum(df_plot$comparison == &quot;all&quot;) ==
            sum(df_plot$comparison == &quot;human&quot;) +
            sum(df_plot$comparison == &quot;chimp&quot;) +
            sum(df_plot$comparison == &quot;cross&quot;),
          df_plot$day1 == df_plot$day2)
head(df_plot)</code></pre>
<pre><code>##     sample1  sample2         r day1 day2 species1 species2 comparison
## 1  D0_20157 D0_20961 0.9566170    0    0        H        H      human
## 2  D0_20157 D0_21792 0.9624744    0    0        H        H      human
## 3  D0_20157 D0_28162 0.9595328    0    0        H        H      human
## 4  D0_20157 D0_28815 0.9806249    0    0        H        H      human
## 5  D0_20157 D0_29089 0.9542512    0    0        H        H      human
## 40 D0_20961 D0_21792 0.9285112    0    0        H        H      human
##    correlation
## 1    0.9566170
## 2    0.9624744
## 3    0.9595328
## 4    0.9806249
## 5    0.9542512
## 40   0.9285112</code></pre>
<pre class="r"><code>df_plot$comparison &lt;- factor(df_plot$comparison,
                             levels = c(&quot;chimp&quot;, &quot;human&quot;, &quot;cross&quot;, &quot;all&quot;),
                             labels = c(&quot;Within chimps&quot;, &quot;Within humans&quot;,
                                        &quot;Between species&quot;, &quot;All comparisons&quot;))
table(df_plot$comparison)</code></pre>
<pre><code>## 
##   Within chimps   Within humans Between species All comparisons 
##              24              60              96             180</code></pre>
<pre class="r"><code>ggplot(df_plot, aes(x = day1, y = correlation)) +
  geom_boxplot() +
  facet_wrap(~comparison) +
  labs(x = &quot;Day&quot;, y = &quot;Spearman&#39;s r&quot;, title = &quot;When original samples used (mostly from batch 1)&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<pre class="r"><code>  day1cor_cross &lt;- pair_cor_long[pair_cor_long$day1==1,]$cross
  day0cor_cross &lt;- pair_cor_long[pair_cor_long$day1==0,]$cross
  day2cor_cross &lt;- pair_cor_long[pair_cor_long$day1==2,]$cross
  day3cor_cross &lt;- pair_cor_long[pair_cor_long$day1==3,]$cross
    day1cor_H &lt;- pair_cor_long[pair_cor_long$day1==1,]$human
  day0cor_H &lt;- pair_cor_long[pair_cor_long$day1==0,]$human
  day2cor_H &lt;- pair_cor_long[pair_cor_long$day1==2,]$human
  day3cor_H &lt;- pair_cor_long[pair_cor_long$day1==3,]$human
     day1cor_C &lt;- pair_cor_long[pair_cor_long$day1==1,]$chimp
  day0cor_C &lt;- pair_cor_long[pair_cor_long$day1==0,]$chimp
  day2cor_C &lt;- pair_cor_long[pair_cor_long$day1==2,]$chimp
  day3cor_C &lt;- pair_cor_long[pair_cor_long$day1==3,]$chimp
  t.test(na.omit(day1cor_cross), na.omit(day3cor_cross))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  na.omit(day1cor_cross) and na.omit(day3cor_cross)
## t = 7.2917, df = 41.567, p-value = 5.919e-09
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.02726897 0.04814808
## sample estimates:
## mean of x mean of y 
## 0.9204149 0.8827064</code></pre>
<pre class="r"><code>  t.test(na.omit(day1cor_H), na.omit(day0cor_H))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  na.omit(day1cor_H) and na.omit(day0cor_H)
## t = 3.9624, df = 21.752, p-value = 0.0006727
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.00914519 0.02925811
## sample estimates:
## mean of x mean of y 
## 0.9776890 0.9584874</code></pre>
<pre class="r"><code>  t.test(na.omit(day1cor_C), na.omit(day0cor_C))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  na.omit(day1cor_C) and na.omit(day0cor_C)
## t = 0.49519, df = 6.6744, p-value = 0.6363
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.01474842  0.02246449
## sample estimates:
## mean of x mean of y 
## 0.9764115 0.9725535</code></pre>
<pre class="r"><code>mean(na.omit(day1cor_cross))</code></pre>
<pre><code>## [1] 0.9204149</code></pre>
<pre class="r"><code>mean(na.omit(day0cor_cross))</code></pre>
<pre><code>## [1] 0.903167</code></pre>
<pre class="r"><code># Real statistics with the correlations

# Within species
#Make all the correlations

cor_D0H &lt;- as.data.frame(na.omit(day0cor_H))
colnames(cor_D0H) &lt;- c(&quot;correlations&quot;)
cor_D0C &lt;- as.data.frame(na.omit(day0cor_C))
colnames(cor_D0C) &lt;- c(&quot;correlations&quot;)
cor_D1H &lt;- as.data.frame(na.omit(day1cor_H)) 
colnames(cor_D1H) &lt;- c(&quot;correlations&quot;)
cor_D1C &lt;- as.data.frame(na.omit(day1cor_C))
colnames(cor_D1C) &lt;- c(&quot;correlations&quot;)
cor_D2H &lt;- as.data.frame(na.omit(day2cor_H))
colnames(cor_D2H) &lt;- c(&quot;correlations&quot;)
cor_D2C &lt;- as.data.frame(na.omit(day2cor_C))
colnames(cor_D2C) &lt;- c(&quot;correlations&quot;)
cor_D3H &lt;- as.data.frame(na.omit(day3cor_H))
colnames(cor_D3H) &lt;- c(&quot;correlations&quot;)
cor_D3C &lt;- as.data.frame(na.omit(day3cor_C))
colnames(cor_D3C) &lt;- c(&quot;correlations&quot;)


corr_day_species &lt;- rbind(cor_D0H, cor_D0C, cor_D1H, cor_D1C, cor_D2H, cor_D2C, cor_D3H, cor_D3C)

D0H &lt;- as.data.frame(rep(&quot;D0H&quot;, each = 15))
colnames(D0H) &lt;- c(&quot;Sample&quot;)

D0C &lt;- as.data.frame(rep(&quot;D0C&quot;, each = 6))
colnames(D0C) &lt;- c(&quot;Sample&quot;)
D1H &lt;- as.data.frame(rep(&quot;D1H&quot;, each = 15))
colnames(D1H) &lt;- c(&quot;Sample&quot;)
D1C &lt;- as.data.frame(rep(&quot;D1C&quot;, each = 6))
colnames(D1C) &lt;- c(&quot;Sample&quot;)
D2H &lt;- as.data.frame(rep(&quot;D2H&quot;, each = 15))
colnames(D2H) &lt;- c(&quot;Sample&quot;)
D2C &lt;- as.data.frame(rep(&quot;D2C&quot;, each = 6))
colnames(D2C) &lt;- c(&quot;Sample&quot;)
D3H &lt;- as.data.frame(rep(&quot;D3H&quot;, each = 15))
colnames(D3H) &lt;- c(&quot;Sample&quot;)
D3C &lt;- as.data.frame(rep(&quot;D3C&quot;, each = 6))
colnames(D3C) &lt;- c(&quot;Sample&quot;)



#### For within species chimps

group &lt;- rbind(D0C,D1C, D2C,  D3C)

corr_day_species &lt;- rbind(cor_D0C,  cor_D1C,  cor_D2C, cor_D3C)

aov_groups &lt;- cbind(corr_day_species, group)

dim(aov_groups)</code></pre>
<pre><code>## [1] 24  2</code></pre>
<pre class="r"><code>oneway.test(aov_groups$correlations ~ aov_groups$Sample)</code></pre>
<pre><code>## 
##  One-way analysis of means (not assuming equal variances)
## 
## data:  aov_groups$correlations and aov_groups$Sample
## F = 2.4052, num df = 3.000, denom df = 9.903, p-value = 0.1288</code></pre>
<pre class="r"><code>t.test(cor_D0C, cor_D1C) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D0C and cor_D1C
## t = -0.49519, df = 6.6744, p-value = 0.6363
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.02246449  0.01474842
## sample estimates:
## mean of x mean of y 
## 0.9725535 0.9764115</code></pre>
<pre class="r"><code>t.test(cor_D2C, cor_D1C)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D2C and cor_D1C
## t = -1.9301, df = 6.7533, p-value = 0.09643
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.032904535  0.003450165
## sample estimates:
## mean of x mean of y 
## 0.9616844 0.9764115</code></pre>
<pre class="r"><code>t.test(cor_D3C, cor_D1C) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D3C and cor_D1C
## t = -2.2987, df = 5.7302, p-value = 0.06324
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.054538505  0.002015366
## sample estimates:
## mean of x mean of y 
## 0.9501500 0.9764115</code></pre>
<pre class="r"><code>#### For within species humans

group &lt;- rbind(D0H,D1H, D2H,  D3H)

corr_day_species &lt;- rbind(cor_D0H,  cor_D1H,  cor_D2H, cor_D3H)

aov_groups &lt;- cbind(corr_day_species, group)

dim(aov_groups)</code></pre>
<pre><code>## [1] 60  2</code></pre>
<pre class="r"><code>aov_bet &lt;- oneway.test(aov_groups$correlations ~ aov_groups$Sample)

t.test(cor_D0H, cor_D1H) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D0H and cor_D1H
## t = -3.9624, df = 21.752, p-value = 0.0006727
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.02925811 -0.00914519
## sample estimates:
## mean of x mean of y 
## 0.9584874 0.9776890</code></pre>
<pre class="r"><code>t.test(cor_D2H, cor_D1H)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D2H and cor_D1H
## t = -3.0478, df = 15, p-value = 0.008139
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.06505996 -0.01151131
## sample estimates:
## mean of x mean of y 
## 0.9394034 0.9776890</code></pre>
<pre class="r"><code>t.test(cor_D3H, cor_D1H) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D3H and cor_D1H
## t = -4.9037, df = 20.286, p-value = 8.256e-05
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.03725544 -0.01503300
## sample estimates:
## mean of x mean of y 
## 0.9515448 0.9776890</code></pre>
<pre class="r"><code># For between species 

#Make all the correlations

cor_D0H &lt;- as.data.frame(na.omit(day0cor_H))
colnames(cor_D0H) &lt;- c(&quot;correlations&quot;)
cor_D0C &lt;- as.data.frame(na.omit(day0cor_C))
colnames(cor_D0C) &lt;- c(&quot;correlations&quot;)

cor_D0 &lt;- rbind(cor_D0H, cor_D0C)

cor_D1H &lt;- as.data.frame(na.omit(day1cor_H)) 
colnames(cor_D1H) &lt;- c(&quot;correlations&quot;)
cor_D1C &lt;- as.data.frame(na.omit(day1cor_C))
colnames(cor_D1C) &lt;- c(&quot;correlations&quot;)
cor_D1 &lt;- rbind(cor_D1H, cor_D1C)

cor_D2H &lt;- as.data.frame(na.omit(day2cor_H))
colnames(cor_D2H) &lt;- c(&quot;correlations&quot;)
cor_D2C &lt;- as.data.frame(na.omit(day2cor_C))
colnames(cor_D2C) &lt;- c(&quot;correlations&quot;)
cor_D2 &lt;- rbind(cor_D2H, cor_D2C)
colnames(cor_D2) &lt;- c(&quot;correlations&quot;)

cor_D3H &lt;- as.data.frame(na.omit(day3cor_H))
colnames(cor_D3H) &lt;- c(&quot;correlations&quot;)
cor_D3C &lt;- as.data.frame(na.omit(day3cor_C))
colnames(cor_D3C) &lt;- c(&quot;correlations&quot;)
cor_D3 &lt;- rbind(cor_D3H, cor_D3C)


corr_day_species &lt;- rbind(cor_D0H, cor_D0C, cor_D1H, cor_D1C, cor_D2H, cor_D2C, cor_D3H, cor_D3C)

D0 &lt;- as.data.frame(rep(&quot;D0&quot;, each = 21))
colnames(D0) &lt;- c(&quot;Sample&quot;)

D1 &lt;- as.data.frame(rep(&quot;D1&quot;, each = 21))
colnames(D1) &lt;- c(&quot;Sample&quot;)

D2 &lt;- as.data.frame(rep(&quot;D2&quot;, each = 21))
colnames(D2) &lt;- c(&quot;Sample&quot;)

D3 &lt;- as.data.frame(rep(&quot;D3&quot;, each = 21))
colnames(D3) &lt;- c(&quot;Sample&quot;)


group &lt;- rbind(D0, D1, D2, D3)

aov_groups &lt;- cbind(corr_day_species, group)

# Perform the ANOVA

oneway.test(aov_groups$correlations ~ aov_groups$Sample)  </code></pre>
<pre><code>## 
##  One-way analysis of means (not assuming equal variances)
## 
## data:  aov_groups$correlations and aov_groups$Sample
## F = 14.113, num df = 3.000, denom df = 39.613, p-value = 2.097e-06</code></pre>
<pre class="r"><code>t.test(cor_D0, cor_D1)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D0 and cor_D1
## t = -3.4817, df = 28.714, p-value = 0.001614
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.023525818 -0.006109703
## sample estimates:
## mean of x mean of y 
## 0.9625063 0.9773240</code></pre>
<pre class="r"><code>t.test(cor_D2, cor_D1) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D2 and cor_D1
## t = -3.3611, df = 21.592, p-value = 0.002874
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.05104587 -0.01206344
## sample estimates:
## mean of x mean of y 
## 0.9457694 0.9773240</code></pre>
<pre class="r"><code>t.test(cor_D3, cor_D1) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D3 and cor_D1
## t = -5.3918, df = 26.51, p-value = 1.127e-05
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.03614814 -0.01620736
## sample estimates:
## mean of x mean of y 
## 0.9511463 0.9773240</code></pre>
</div>
<div id="using-mostly-batch-2-samples" class="section level2">
<h2>Using mostly batch 2 samples</h2>
<pre class="r"><code># Run samples from batch 2 when available
bio_rep &lt;- c(1:4, 6, 7, 9, 11, 13, 15, 17, 18:20, 22, 23, 25, 27, 29, 31, 33, 34:36, 38, 39, 41, 43, 45, 47, 49, 50:52,54, 55, 57, 59, 61, 63)

bio_rep_cpm_in_cutoff &lt;- cpm_in_cutoff[,bio_rep]

# Find the technical factors for the biological replicates (no technical replicates)

bio_rep_samplefactors &lt;- read.delim(&quot;~/Downloads/samplefactors-filtered.txt&quot;, stringsAsFactors=FALSE) 

pair_cor &lt;- cor(bio_rep_cpm_in_cutoff, method = &quot;spearman&quot;)
pair_cor &lt;- as.data.frame(pair_cor)
pair_cor$sample1 &lt;- rownames(pair_cor)

pair_cor_long &lt;- gather(pair_cor, key = &quot;sample2&quot;, value = &quot;r&quot;, -sample1) %&gt;%
  mutate(sample2 = as.character(sample2)) %&gt;%
  filter(sample1 &lt; sample2) %&gt;% # ensures only one of the two entries is
  arrange(sample1, sample2)    # maintained
stopifnot(pair_cor_long$r &lt; 1,
          nrow(pair_cor_long) == (40*40 - 40) / 2)
head(pair_cor_long)</code></pre>
<pre><code>##    sample1       sample2         r
## 1 D0_20157      D0_20961 0.9566170
## 2 D0_20157      D0_21792 0.9624744
## 3 D0_20157      D0_28162 0.9595328
## 4 D0_20157 D0_28815_0116 0.9687576
## 5 D0_20157      D0_29089 0.9542512
## 6 D0_20157   D0_36470116 0.9170885</code></pre>
<pre class="r"><code>rownames(bio_rep_samplefactors) &lt;- colnames(bio_rep_cpm_in_cutoff)
pair_cor_long$day1 &lt;- &quot;&quot;
pair_cor_long$day2 &lt;- &quot;&quot;
pair_cor_long$species1 &lt;- &quot;&quot;
pair_cor_long$species2 &lt;- &quot;&quot;

for (i in 1:nrow(pair_cor_long)) {
  pair_cor_long$day1[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample1[i], &quot;Day&quot;]
  pair_cor_long$day2[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample2[i], &quot;Day&quot;]
  pair_cor_long$species1[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample1[i], &quot;Species&quot;]
  pair_cor_long$species2[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample2[i], &quot;Species&quot;]
}
head(pair_cor_long)</code></pre>
<pre><code>##    sample1       sample2         r day1 day2 species1 species2
## 1 D0_20157      D0_20961 0.9566170    0    0        H        H
## 2 D0_20157      D0_21792 0.9624744    0    0        H        H
## 3 D0_20157      D0_28162 0.9595328    0    0        H        H
## 4 D0_20157 D0_28815_0116 0.9687576    0    0        H        H
## 5 D0_20157      D0_29089 0.9542512    0    0        H        H
## 6 D0_20157   D0_36470116 0.9170885    0    0        H        C</code></pre>
<pre class="r"><code>pair_cor_long$human &lt;- NA
pair_cor_long$chimp &lt;- NA
pair_cor_long$cross &lt;- NA
pair_cor_long$all &lt;- NA
for (i in 1:nrow(pair_cor_long)) {
  # Do not include comparisons across days
  if (pair_cor_long$day1[i] != pair_cor_long$day2[i]) {
    next
  }
  pair_cor_long$all[i] &lt;- pair_cor_long$r[i]
  if (pair_cor_long$species1[i] == &quot;H&quot; &amp;
      pair_cor_long$species2[i] == &quot;H&quot;) {
    pair_cor_long$human[i] &lt;- pair_cor_long$r[i]
  } else if (pair_cor_long$species1[i] == &quot;C&quot; &amp;
             pair_cor_long$species2[i] == &quot;C&quot;) {
    pair_cor_long$chimp[i] &lt;- pair_cor_long$r[i]
  } else {
    pair_cor_long$cross[i] &lt;- pair_cor_long$r[i]
  }
}
head(pair_cor_long)</code></pre>
<pre><code>##    sample1       sample2         r day1 day2 species1 species2     human
## 1 D0_20157      D0_20961 0.9566170    0    0        H        H 0.9566170
## 2 D0_20157      D0_21792 0.9624744    0    0        H        H 0.9624744
## 3 D0_20157      D0_28162 0.9595328    0    0        H        H 0.9595328
## 4 D0_20157 D0_28815_0116 0.9687576    0    0        H        H 0.9687576
## 5 D0_20157      D0_29089 0.9542512    0    0        H        H 0.9542512
## 6 D0_20157   D0_36470116 0.9170885    0    0        H        C        NA
##   chimp     cross       all
## 1    NA        NA 0.9566170
## 2    NA        NA 0.9624744
## 3    NA        NA 0.9595328
## 4    NA        NA 0.9687576
## 5    NA        NA 0.9542512
## 6    NA 0.9170885 0.9170885</code></pre>
<pre class="r"><code>df_plot &lt;- gather(pair_cor_long, key = &quot;comparison&quot;, value = &quot;correlation&quot;,
                  human, chimp, cross, all, na.rm = TRUE)
stopifnot(sum(df_plot$comparison == &quot;all&quot;) ==
            sum(df_plot$comparison == &quot;human&quot;) +
            sum(df_plot$comparison == &quot;chimp&quot;) +
            sum(df_plot$comparison == &quot;cross&quot;),
          df_plot$day1 == df_plot$day2)
head(df_plot)</code></pre>
<pre><code>##     sample1       sample2         r day1 day2 species1 species2 comparison
## 1  D0_20157      D0_20961 0.9566170    0    0        H        H      human
## 2  D0_20157      D0_21792 0.9624744    0    0        H        H      human
## 3  D0_20157      D0_28162 0.9595328    0    0        H        H      human
## 4  D0_20157 D0_28815_0116 0.9687576    0    0        H        H      human
## 5  D0_20157      D0_29089 0.9542512    0    0        H        H      human
## 40 D0_20961      D0_21792 0.9285112    0    0        H        H      human
##    correlation
## 1    0.9566170
## 2    0.9624744
## 3    0.9595328
## 4    0.9687576
## 5    0.9542512
## 40   0.9285112</code></pre>
<pre class="r"><code>df_plot$comparison &lt;- factor(df_plot$comparison,
                             levels = c(&quot;chimp&quot;, &quot;human&quot;, &quot;cross&quot;, &quot;all&quot;),
                             labels = c(&quot;Within chimps&quot;, &quot;Within humans&quot;,
                                        &quot;Between species&quot;, &quot;All comparisons&quot;))
table(df_plot$comparison)</code></pre>
<pre><code>## 
##   Within chimps   Within humans Between species All comparisons 
##              24              60              96             180</code></pre>
<pre class="r"><code>ggplot(df_plot, aes(x = day1, y = correlation)) +
  geom_boxplot() +
  facet_wrap(~comparison) +
  labs(x = &quot;Day&quot;, y = &quot;Spearman&#39;s r&quot;, title = &quot;When original samples used (mostly from batch 2)&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<pre class="r"><code>  day1cor_cross &lt;- pair_cor_long[pair_cor_long$day1==1,]$cross
  day0cor_cross &lt;- pair_cor_long[pair_cor_long$day1==0,]$cross
  day2cor_cross &lt;- pair_cor_long[pair_cor_long$day1==2,]$cross
  day3cor_cross &lt;- pair_cor_long[pair_cor_long$day1==3,]$cross
    day1cor_H &lt;- pair_cor_long[pair_cor_long$day1==1,]$human
  day0cor_H &lt;- pair_cor_long[pair_cor_long$day1==0,]$human
  day2cor_H &lt;- pair_cor_long[pair_cor_long$day1==2,]$human
  day3cor_H &lt;- pair_cor_long[pair_cor_long$day1==3,]$human
     day1cor_C &lt;- pair_cor_long[pair_cor_long$day1==1,]$chimp
  day0cor_C &lt;- pair_cor_long[pair_cor_long$day1==0,]$chimp
  day2cor_C &lt;- pair_cor_long[pair_cor_long$day1==2,]$chimp
  day3cor_C &lt;- pair_cor_long[pair_cor_long$day1==3,]$chimp
  t.test(na.omit(day1cor_cross), na.omit(day3cor_cross))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  na.omit(day1cor_cross) and na.omit(day3cor_cross)
## t = 6.7959, df = 28.744, p-value = 1.929e-07
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.03514817 0.06542831
## sample estimates:
## mean of x mean of y 
## 0.9175020 0.8672137</code></pre>
<pre class="r"><code>  t.test(na.omit(day1cor_H), na.omit(day0cor_H))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  na.omit(day1cor_H) and na.omit(day0cor_H)
## t = 4.5192, df = 17.413, p-value = 0.0002866
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.01133634 0.03112308
## sample estimates:
## mean of x mean of y 
## 0.9795512 0.9583215</code></pre>
<pre class="r"><code>  t.test(na.omit(day1cor_C), na.omit(day0cor_C))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  na.omit(day1cor_C) and na.omit(day0cor_C)
## t = -0.3782, df = 9.5065, p-value = 0.7136
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.03141787  0.02235482
## sample estimates:
## mean of x mean of y 
## 0.9524782 0.9570097</code></pre>
<pre class="r"><code>mean(na.omit(day1cor_cross))</code></pre>
<pre><code>## [1] 0.917502</code></pre>
<pre class="r"><code>mean(na.omit(day0cor_cross))</code></pre>
<pre><code>## [1] 0.9106549</code></pre>
<pre class="r"><code># Real statistics with the correlations

# Within species
#Make all the correlations

cor_D0H &lt;- as.data.frame(na.omit(day0cor_H))
colnames(cor_D0H) &lt;- c(&quot;correlations&quot;)
cor_D0C &lt;- as.data.frame(na.omit(day0cor_C))
colnames(cor_D0C) &lt;- c(&quot;correlations&quot;)
cor_D1H &lt;- as.data.frame(na.omit(day1cor_H)) 
colnames(cor_D1H) &lt;- c(&quot;correlations&quot;)
cor_D1C &lt;- as.data.frame(na.omit(day1cor_C))
colnames(cor_D1C) &lt;- c(&quot;correlations&quot;)
cor_D2H &lt;- as.data.frame(na.omit(day2cor_H))
colnames(cor_D2H) &lt;- c(&quot;correlations&quot;)
cor_D2C &lt;- as.data.frame(na.omit(day2cor_C))
colnames(cor_D2C) &lt;- c(&quot;correlations&quot;)
cor_D3H &lt;- as.data.frame(na.omit(day3cor_H))
colnames(cor_D3H) &lt;- c(&quot;correlations&quot;)
cor_D3C &lt;- as.data.frame(na.omit(day3cor_C))
colnames(cor_D3C) &lt;- c(&quot;correlations&quot;)


corr_day_species &lt;- rbind(cor_D0H, cor_D0C, cor_D1H, cor_D1C, cor_D2H, cor_D2C, cor_D3H, cor_D3C)

D0H &lt;- as.data.frame(rep(&quot;D0H&quot;, each = 15))
colnames(D0H) &lt;- c(&quot;Sample&quot;)

D0C &lt;- as.data.frame(rep(&quot;D0C&quot;, each = 6))
colnames(D0C) &lt;- c(&quot;Sample&quot;)
D1H &lt;- as.data.frame(rep(&quot;D1H&quot;, each = 15))
colnames(D1H) &lt;- c(&quot;Sample&quot;)
D1C &lt;- as.data.frame(rep(&quot;D1C&quot;, each = 6))
colnames(D1C) &lt;- c(&quot;Sample&quot;)
D2H &lt;- as.data.frame(rep(&quot;D2H&quot;, each = 15))
colnames(D2H) &lt;- c(&quot;Sample&quot;)
D2C &lt;- as.data.frame(rep(&quot;D2C&quot;, each = 6))
colnames(D2C) &lt;- c(&quot;Sample&quot;)
D3H &lt;- as.data.frame(rep(&quot;D3H&quot;, each = 15))
colnames(D3H) &lt;- c(&quot;Sample&quot;)
D3C &lt;- as.data.frame(rep(&quot;D3C&quot;, each = 6))
colnames(D3C) &lt;- c(&quot;Sample&quot;)



#### For within species chimps

group &lt;- rbind(D0C,D1C, D2C,  D3C)

corr_day_species &lt;- rbind(cor_D0C,  cor_D1C,  cor_D2C, cor_D3C)

aov_groups &lt;- cbind(corr_day_species, group)

dim(aov_groups)</code></pre>
<pre><code>## [1] 24  2</code></pre>
<pre class="r"><code>oneway.test(aov_groups$correlations ~ aov_groups$Sample)</code></pre>
<pre><code>## 
##  One-way analysis of means (not assuming equal variances)
## 
## data:  aov_groups$correlations and aov_groups$Sample
## F = 0.43012, num df = 3.000, denom df = 10.827, p-value = 0.7356</code></pre>
<pre class="r"><code>t.test(cor_D0C, cor_D1C) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D0C and cor_D1C
## t = 0.3782, df = 9.5065, p-value = 0.7136
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.02235482  0.03141787
## sample estimates:
## mean of x mean of y 
## 0.9570097 0.9524782</code></pre>
<pre class="r"><code>t.test(cor_D2C, cor_D1C)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D2C and cor_D1C
## t = 0.054331, df = 9.521, p-value = 0.9578
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.03299980  0.03463782
## sample estimates:
## mean of x mean of y 
## 0.9532972 0.9524782</code></pre>
<pre class="r"><code>t.test(cor_D3C, cor_D1C) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D3C and cor_D1C
## t = -0.8715, df = 8.5815, p-value = 0.4072
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.05429986  0.02425956
## sample estimates:
## mean of x mean of y 
## 0.9374580 0.9524782</code></pre>
<pre class="r"><code>#### For within species humans

group &lt;- rbind(D0H,D1H, D2H,  D3H)

corr_day_species &lt;- rbind(cor_D0H,  cor_D1H,  cor_D2H, cor_D3H)

aov_groups &lt;- cbind(corr_day_species, group)

dim(aov_groups)</code></pre>
<pre><code>## [1] 60  2</code></pre>
<pre class="r"><code>aov_bet &lt;- oneway.test(aov_groups$correlations ~ aov_groups$Sample)

t.test(cor_D0H, cor_D1H) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D0H and cor_D1H
## t = -4.5192, df = 17.413, p-value = 0.0002866
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.03112308 -0.01133634
## sample estimates:
## mean of x mean of y 
## 0.9583215 0.9795512</code></pre>
<pre class="r"><code>t.test(cor_D2H, cor_D1H)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D2H and cor_D1H
## t = -3.6334, df = 26.301, p-value = 0.001191
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.014519886 -0.004030923
## sample estimates:
## mean of x mean of y 
## 0.9702758 0.9795512</code></pre>
<pre class="r"><code>t.test(cor_D3H, cor_D1H) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D3H and cor_D1H
## t = -6.7783, df = 19.251, p-value = 1.669e-06
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.03425423 -0.01810187
## sample estimates:
## mean of x mean of y 
## 0.9533731 0.9795512</code></pre>
<pre class="r"><code># For between species 

#Make all the correlations

cor_D0H &lt;- as.data.frame(na.omit(day0cor_H))
colnames(cor_D0H) &lt;- c(&quot;correlations&quot;)
cor_D0C &lt;- as.data.frame(na.omit(day0cor_C))
colnames(cor_D0C) &lt;- c(&quot;correlations&quot;)

cor_D0 &lt;- rbind(cor_D0H, cor_D0C)

cor_D1H &lt;- as.data.frame(na.omit(day1cor_H)) 
colnames(cor_D1H) &lt;- c(&quot;correlations&quot;)
cor_D1C &lt;- as.data.frame(na.omit(day1cor_C))
colnames(cor_D1C) &lt;- c(&quot;correlations&quot;)
cor_D1 &lt;- rbind(cor_D1H, cor_D1C)

cor_D2H &lt;- as.data.frame(na.omit(day2cor_H))
colnames(cor_D2H) &lt;- c(&quot;correlations&quot;)
cor_D2C &lt;- as.data.frame(na.omit(day2cor_C))
colnames(cor_D2C) &lt;- c(&quot;correlations&quot;)
cor_D2 &lt;- rbind(cor_D2H, cor_D2C)
colnames(cor_D2) &lt;- c(&quot;correlations&quot;)

cor_D3H &lt;- as.data.frame(na.omit(day3cor_H))
colnames(cor_D3H) &lt;- c(&quot;correlations&quot;)
cor_D3C &lt;- as.data.frame(na.omit(day3cor_C))
colnames(cor_D3C) &lt;- c(&quot;correlations&quot;)
cor_D3 &lt;- rbind(cor_D3H, cor_D3C)


corr_day_species &lt;- rbind(cor_D0H, cor_D0C, cor_D1H, cor_D1C, cor_D2H, cor_D2C, cor_D3H, cor_D3C)

D0 &lt;- as.data.frame(rep(&quot;D0&quot;, each = 21))
colnames(D0) &lt;- c(&quot;Sample&quot;)

D1 &lt;- as.data.frame(rep(&quot;D1&quot;, each = 21))
colnames(D1) &lt;- c(&quot;Sample&quot;)

D2 &lt;- as.data.frame(rep(&quot;D2&quot;, each = 21))
colnames(D2) &lt;- c(&quot;Sample&quot;)

D3 &lt;- as.data.frame(rep(&quot;D3&quot;, each = 21))
colnames(D3) &lt;- c(&quot;Sample&quot;)


group &lt;- rbind(D0, D1, D2, D3)

aov_groups &lt;- cbind(corr_day_species, group)

# Perform the ANOVA

oneway.test(aov_groups$correlations ~ aov_groups$Sample)  </code></pre>
<pre><code>## 
##  One-way analysis of means (not assuming equal variances)
## 
## data:  aov_groups$correlations and aov_groups$Sample
## F = 5.0828, num df = 3.000, denom df = 44.268, p-value = 0.004134</code></pre>
<pre class="r"><code>t.test(cor_D0, cor_D1)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D0 and cor_D1
## t = -2.5851, df = 39.931, p-value = 0.0135
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.024713422 -0.003025293
## sample estimates:
## mean of x mean of y 
## 0.9579467 0.9718160</code></pre>
<pre class="r"><code>t.test(cor_D2, cor_D1) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D2 and cor_D1
## t = -1.1686, df = 40, p-value = 0.2495
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.017444706  0.004662136
## sample estimates:
## mean of x mean of y 
## 0.9654248 0.9718160</code></pre>
<pre class="r"><code>t.test(cor_D3, cor_D1) </code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D3 and cor_D1
## t = -3.6938, df = 38.054, p-value = 0.0006915
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.03558936 -0.01039079
## sample estimates:
## mean of x mean of y 
##  0.948826  0.971816</code></pre>
</div>
<div id="using-average-of-log2cpm-expression-to-combine-technical-replicates" class="section level2">
<h2>Using average of log2(CPM) expression to combine technical replicates</h2>
<pre class="r"><code># Make the endoderm-counts-filtered document


# Find the cpm for the biological replicates (no technical replicates)
#bio_rep &lt;- c(1:4,6, 7, 8, 10, 12, 14, 17:20, 22, 23, 24, 26, 28, 30, 33:36,38, 39, 40, 42, 44, 46, 49:52, 53, 54, 56, 58, 60, 62)
#bio_rep &lt;- c(1:4,6, 7, 9, 11, 13, 15, 17:20, 22, 23, 25, 27, 29, 31, 33:36,38, 39, 41, 43, 45, 47, 49:52, 54, 55, 57, 59, 61, 63)

# # Note: unlike in Sammy&#39;s original script, we are going to use log2(CPM) values instead of count data and we are going to take the means of the technical replicates


################# ATTEMPT #1 ##########################

# First, we are going to try this with the means of all available technical replicates (except D0_20157, which we removed way upstream of this)
  # Day 0 technical replicates
D0_28815 &lt;- as.data.frame(apply(cpm_in_cutoff[,5:6], 1, mean))
D0_3647 &lt;- as.data.frame(apply(cpm_in_cutoff[,8:9], 1, mean))
D0_3649 &lt;- as.data.frame(apply(cpm_in_cutoff[,10:11], 1, mean))
D0_40300 &lt;- as.data.frame(apply(cpm_in_cutoff[,12:13], 1, mean))
D0_4955 &lt;- as.data.frame(apply(cpm_in_cutoff[,14:15], 1, mean))
  # Day 1 technical replicates
D1_20157 &lt;- as.data.frame(apply(cpm_in_cutoff[,16:17], 1, mean))
D1_28815 &lt;- as.data.frame(apply(cpm_in_cutoff[,21:22], 1, mean))
D1_3647 &lt;- as.data.frame(apply(cpm_in_cutoff[,24:25], 1, mean))
D1_3649 &lt;- as.data.frame(apply(cpm_in_cutoff[,26:27], 1, mean))
D1_40300 &lt;- as.data.frame(apply(cpm_in_cutoff[,28:29], 1, mean))
D1_4955 &lt;- as.data.frame(apply(cpm_in_cutoff[,30:31], 1, mean))
  # Day 2 technical replicates
D2_20157 &lt;- as.data.frame(apply(cpm_in_cutoff[,32:33], 1, mean))
D2_28815 &lt;- as.data.frame(apply(cpm_in_cutoff[,37:38], 1, mean))
D2_3647 &lt;- as.data.frame(apply(cpm_in_cutoff[,40:41], 1, mean))
D2_3649 &lt;- as.data.frame(apply(cpm_in_cutoff[,42:43], 1, mean))
D2_40300 &lt;- as.data.frame(apply(cpm_in_cutoff[,44:45], 1, mean))
D2_4955 &lt;- as.data.frame(apply(cpm_in_cutoff[,46:47], 1, mean))
  # Day 3 technical replicates
D3_20157 &lt;- as.data.frame(apply(cpm_in_cutoff[,48:49], 1, mean))
D3_28815 &lt;- as.data.frame(apply(cpm_in_cutoff[,53:54], 1, mean))
D3_3647 &lt;- as.data.frame(apply(cpm_in_cutoff[,56:57], 1, mean))
D3_3649 &lt;- as.data.frame(apply(cpm_in_cutoff[,58:59], 1, mean))
D3_40300 &lt;- as.data.frame(apply(cpm_in_cutoff[,60:61], 1, mean))
D3_4955 &lt;- as.data.frame(apply(cpm_in_cutoff[,62:63], 1, mean))

# Create a new data frame with all of the combined technical replicates
mean_tech_reps &lt;- cbind(cpm_in_cutoff[,1:4], D0_28815, cpm_in_cutoff[,7], D0_3647, D0_3649, D0_40300, D0_4955, D1_20157, cpm_in_cutoff[,18:20], D1_28815, cpm_in_cutoff[,23], D1_3647, D1_3649, D1_40300, D1_4955, D2_20157, cpm_in_cutoff[,34:36], D2_28815, cpm_in_cutoff[,39], D2_3647, D2_3649, D2_40300, D2_4955, D3_20157, cpm_in_cutoff[,50:52], D3_28815, cpm_in_cutoff[,55], D3_3647, D3_3649, D3_40300, D3_4955)


bio_rep_cpm_in_cutoff &lt;- mean_tech_reps

colnames(bio_rep_cpm_in_cutoff) &lt;- c(&quot;D0_20157&quot;, &quot;D0_20961&quot;, &quot;D0_21792&quot;, &quot;D0_28162&quot;, &quot;D0_28815&quot;, &quot;D0_29089&quot;, &quot;D0_3647&quot;, &quot;D0_3649&quot;, &quot;D0_40300&quot;, &quot;D0_4955&quot;, &quot;D1_20157&quot;, &quot;D1_20961&quot;, &quot;D1_21792&quot;, &quot;D1_28162&quot;, &quot;D1_28815&quot;, &quot;D1_29089&quot;, &quot;D1_3647&quot;, &quot;D1_3649&quot;, &quot;D1_40300&quot;, &quot;D1_4955&quot;, &quot;D2_20157&quot;, &quot;D2_20961&quot;, &quot;D2_21792&quot;, &quot;D2_28162&quot;, &quot;D2_28815&quot;, &quot;D2_29089&quot;, &quot;D2_3647&quot;, &quot;D2_3649&quot;, &quot;D2_40300&quot;, &quot;D2_4955&quot;, &quot;D3_20157&quot;, &quot;D3_20961&quot;, &quot;D3_21792&quot;, &quot;D3_28162&quot;, &quot;D3_28815&quot;, &quot;D3_29089&quot;, &quot;D3_3647&quot;, &quot;D3_3649&quot;, &quot;D3_40300&quot;, &quot;D3_4955&quot;)


# Find the technical factors for the biological replicates (no technical replicates)

bio_rep_samplefactors &lt;- read.delim(&quot;~/Downloads/samplefactors-filtered.txt&quot;, stringsAsFactors=FALSE) 

# Find the correlations between the individuals

pair_cor &lt;- cor(bio_rep_cpm_in_cutoff, method = &quot;spearman&quot;)
pair_cor &lt;- as.data.frame(pair_cor)
pair_cor$sample1 &lt;- rownames(pair_cor)

pair_cor_long &lt;- gather(pair_cor, key = &quot;sample2&quot;, value = &quot;r&quot;, -sample1) %&gt;%
  mutate(sample2 = as.character(sample2)) %&gt;%
  filter(sample1 &lt; sample2) %&gt;% # ensures only one of the two entries is
  arrange(sample1, sample2)    # maintained
stopifnot(pair_cor_long$r &lt; 1,
          nrow(pair_cor_long) == (40*40 - 40) / 2)
head(pair_cor_long)</code></pre>
<pre><code>##    sample1  sample2         r
## 1 D0_20157 D0_20961 0.9566170
## 2 D0_20157 D0_21792 0.9624744
## 3 D0_20157 D0_28162 0.9595328
## 4 D0_20157 D0_28815 0.9782665
## 5 D0_20157 D0_29089 0.9542512
## 6 D0_20157  D0_3647 0.9312925</code></pre>
<pre class="r"><code>rownames(bio_rep_samplefactors) &lt;- colnames(bio_rep_cpm_in_cutoff)
pair_cor_long$day1 &lt;- &quot;&quot;
pair_cor_long$day2 &lt;- &quot;&quot;
pair_cor_long$species1 &lt;- &quot;&quot;
pair_cor_long$species2 &lt;- &quot;&quot;

for (i in 1:nrow(pair_cor_long)) {
  pair_cor_long$day1[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample1[i], &quot;Day&quot;]
  pair_cor_long$day2[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample2[i], &quot;Day&quot;]
  pair_cor_long$species1[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample1[i], &quot;Species&quot;]
  pair_cor_long$species2[i] &lt;- bio_rep_samplefactors[pair_cor_long$sample2[i], &quot;Species&quot;]
}
head(pair_cor_long)</code></pre>
<pre><code>##    sample1  sample2         r day1 day2 species1 species2
## 1 D0_20157 D0_20961 0.9566170    0    0        H        H
## 2 D0_20157 D0_21792 0.9624744    0    0        H        H
## 3 D0_20157 D0_28162 0.9595328    0    0        H        H
## 4 D0_20157 D0_28815 0.9782665    0    0        H        H
## 5 D0_20157 D0_29089 0.9542512    0    0        H        H
## 6 D0_20157  D0_3647 0.9312925    0    0        H        C</code></pre>
<pre class="r"><code>pair_cor_long$human &lt;- NA
pair_cor_long$chimp &lt;- NA
pair_cor_long$cross &lt;- NA
pair_cor_long$all &lt;- NA
for (i in 1:nrow(pair_cor_long)) {
  # Do not include comparisons across days
  if (pair_cor_long$day1[i] != pair_cor_long$day2[i]) {
    next
  }
  pair_cor_long$all[i] &lt;- pair_cor_long$r[i]
  if (pair_cor_long$species1[i] == &quot;H&quot; &amp;
      pair_cor_long$species2[i] == &quot;H&quot;) {
    pair_cor_long$human[i] &lt;- pair_cor_long$r[i]
  } else if (pair_cor_long$species1[i] == &quot;C&quot; &amp;
             pair_cor_long$species2[i] == &quot;C&quot;) {
    pair_cor_long$chimp[i] &lt;- pair_cor_long$r[i]
  } else {
    pair_cor_long$cross[i] &lt;- pair_cor_long$r[i]
  }
}
head(pair_cor_long)</code></pre>
<pre><code>##    sample1  sample2         r day1 day2 species1 species2     human chimp
## 1 D0_20157 D0_20961 0.9566170    0    0        H        H 0.9566170    NA
## 2 D0_20157 D0_21792 0.9624744    0    0        H        H 0.9624744    NA
## 3 D0_20157 D0_28162 0.9595328    0    0        H        H 0.9595328    NA
## 4 D0_20157 D0_28815 0.9782665    0    0        H        H 0.9782665    NA
## 5 D0_20157 D0_29089 0.9542512    0    0        H        H 0.9542512    NA
## 6 D0_20157  D0_3647 0.9312925    0    0        H        C        NA    NA
##       cross       all
## 1        NA 0.9566170
## 2        NA 0.9624744
## 3        NA 0.9595328
## 4        NA 0.9782665
## 5        NA 0.9542512
## 6 0.9312925 0.9312925</code></pre>
<pre class="r"><code>df_plot &lt;- gather(pair_cor_long, key = &quot;comparison&quot;, value = &quot;correlation&quot;,
                  human, chimp, cross, all, na.rm = TRUE)
stopifnot(sum(df_plot$comparison == &quot;all&quot;) ==
            sum(df_plot$comparison == &quot;human&quot;) +
            sum(df_plot$comparison == &quot;chimp&quot;) +
            sum(df_plot$comparison == &quot;cross&quot;),
          df_plot$day1 == df_plot$day2)
head(df_plot)</code></pre>
<pre><code>##     sample1  sample2         r day1 day2 species1 species2 comparison
## 1  D0_20157 D0_20961 0.9566170    0    0        H        H      human
## 2  D0_20157 D0_21792 0.9624744    0    0        H        H      human
## 3  D0_20157 D0_28162 0.9595328    0    0        H        H      human
## 4  D0_20157 D0_28815 0.9782665    0    0        H        H      human
## 5  D0_20157 D0_29089 0.9542512    0    0        H        H      human
## 40 D0_20961 D0_21792 0.9285112    0    0        H        H      human
##    correlation
## 1    0.9566170
## 2    0.9624744
## 3    0.9595328
## 4    0.9782665
## 5    0.9542512
## 40   0.9285112</code></pre>
<pre class="r"><code>df_plot$comparison &lt;- factor(df_plot$comparison,
                             levels = c(&quot;chimp&quot;, &quot;human&quot;, &quot;cross&quot;, &quot;all&quot;),
                             labels = c(&quot;Within chimps&quot;, &quot;Within humans&quot;,
                                        &quot;Between species&quot;, &quot;All comparisons&quot;))
table(df_plot$comparison)</code></pre>
<pre><code>## 
##   Within chimps   Within humans Between species All comparisons 
##              24              60              96             180</code></pre>
<pre class="r"><code>ggplot(df_plot, aes(x = day1, y = correlation)) +
  geom_boxplot() +
  facet_wrap(~comparison) +
  labs(x = &quot;Day&quot;, y = &quot;Spearman&#39;s r&quot;, title = &quot;Average of technical replicates when available (4 chimps, 6 humans&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<pre class="r"><code>  day1cor_cross &lt;- pair_cor_long[pair_cor_long$day1==1,]$cross
  day0cor_cross &lt;- pair_cor_long[pair_cor_long$day1==0,]$cross
  day2cor_cross &lt;- pair_cor_long[pair_cor_long$day1==2,]$cross
  day3cor_cross &lt;- pair_cor_long[pair_cor_long$day1==3,]$cross
    day1cor_H &lt;- pair_cor_long[pair_cor_long$day1==1,]$human
  day0cor_H &lt;- pair_cor_long[pair_cor_long$day1==0,]$human
  day2cor_H &lt;- pair_cor_long[pair_cor_long$day1==2,]$human
  day3cor_H &lt;- pair_cor_long[pair_cor_long$day1==3,]$human
     day1cor_C &lt;- pair_cor_long[pair_cor_long$day1==1,]$chimp
  day0cor_C &lt;- pair_cor_long[pair_cor_long$day1==0,]$chimp
  day2cor_C &lt;- pair_cor_long[pair_cor_long$day1==2,]$chimp
  day3cor_C &lt;- pair_cor_long[pair_cor_long$day1==3,]$chimp
  t.test(na.omit(day1cor_cross), na.omit(day3cor_cross))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  na.omit(day1cor_cross) and na.omit(day3cor_cross)
## t = 7.4993, df = 29.568, p-value = 2.553e-08
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.03039399 0.05316286
## sample estimates:
## mean of x mean of y 
## 0.9287558 0.8869774</code></pre>
<pre class="r"><code>  t.test(na.omit(day1cor_H), na.omit(day0cor_H))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  na.omit(day1cor_H) and na.omit(day0cor_H)
## t = 4.2139, df = 18.903, p-value = 0.0004748
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.01034362 0.03077296
## sample estimates:
## mean of x mean of y 
## 0.9801389 0.9595806</code></pre>
<pre class="r"><code>  t.test(na.omit(day1cor_C), na.omit(day0cor_C))</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  na.omit(day1cor_C) and na.omit(day0cor_C)
## t = 0.032537, df = 8.2333, p-value = 0.9748
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.007639769  0.007859541
## sample estimates:
## mean of x mean of y 
## 0.9814218 0.9813119</code></pre>
<pre class="r"><code>mean(na.omit(day1cor_cross))</code></pre>
<pre><code>## [1] 0.9287558</code></pre>
<pre class="r"><code>mean(na.omit(day0cor_cross))</code></pre>
<pre><code>## [1] 0.9183733</code></pre>
<pre class="r"><code># Real statistics with the correlations

# Within species
#Make all the correlations

cor_D0H &lt;- as.data.frame(na.omit(day0cor_H))
colnames(cor_D0H) &lt;- c(&quot;correlations&quot;)
cor_D0C &lt;- as.data.frame(na.omit(day0cor_C))
colnames(cor_D0C) &lt;- c(&quot;correlations&quot;)
cor_D1H &lt;- as.data.frame(na.omit(day1cor_H)) 
colnames(cor_D1H) &lt;- c(&quot;correlations&quot;)
cor_D1C &lt;- as.data.frame(na.omit(day1cor_C))
colnames(cor_D1C) &lt;- c(&quot;correlations&quot;)
cor_D2H &lt;- as.data.frame(na.omit(day2cor_H))
colnames(cor_D2H) &lt;- c(&quot;correlations&quot;)
cor_D2C &lt;- as.data.frame(na.omit(day2cor_C))
colnames(cor_D2C) &lt;- c(&quot;correlations&quot;)
cor_D3H &lt;- as.data.frame(na.omit(day3cor_H))
colnames(cor_D3H) &lt;- c(&quot;correlations&quot;)
cor_D3C &lt;- as.data.frame(na.omit(day3cor_C))
colnames(cor_D3C) &lt;- c(&quot;correlations&quot;)


corr_day_species &lt;- rbind(cor_D0H, cor_D0C, cor_D1H, cor_D1C, cor_D2H, cor_D2C, cor_D3H, cor_D3C)

D0H &lt;- as.data.frame(rep(&quot;D0H&quot;, each = 15))
colnames(D0H) &lt;- c(&quot;Sample&quot;)

D0C &lt;- as.data.frame(rep(&quot;D0C&quot;, each = 6))
colnames(D0C) &lt;- c(&quot;Sample&quot;)
D1H &lt;- as.data.frame(rep(&quot;D1H&quot;, each = 15))
colnames(D1H) &lt;- c(&quot;Sample&quot;)
D1C &lt;- as.data.frame(rep(&quot;D1C&quot;, each = 6))
colnames(D1C) &lt;- c(&quot;Sample&quot;)
D2H &lt;- as.data.frame(rep(&quot;D2H&quot;, each = 15))
colnames(D2H) &lt;- c(&quot;Sample&quot;)
D2C &lt;- as.data.frame(rep(&quot;D2C&quot;, each = 6))
colnames(D2C) &lt;- c(&quot;Sample&quot;)
D3H &lt;- as.data.frame(rep(&quot;D3H&quot;, each = 15))
colnames(D3H) &lt;- c(&quot;Sample&quot;)
D3C &lt;- as.data.frame(rep(&quot;D3C&quot;, each = 6))
colnames(D3C) &lt;- c(&quot;Sample&quot;)



#### For within species chimps

group &lt;- rbind(D0C,D1C, D2C,  D3C)

corr_day_species &lt;- rbind(cor_D0C,  cor_D1C,  cor_D2C, cor_D3C)

aov_groups &lt;- cbind(corr_day_species, group)

dim(aov_groups)</code></pre>
<pre><code>## [1] 24  2</code></pre>
<pre class="r"><code>oneway.test(aov_groups$correlations ~ aov_groups$Sample)</code></pre>
<pre><code>## 
##  One-way analysis of means (not assuming equal variances)
## 
## data:  aov_groups$correlations and aov_groups$Sample
## F = 1.5518, num df = 3.000, denom df = 10.163, p-value = 0.2606</code></pre>
<pre class="r"><code>#### For within species humans

group &lt;- rbind(D0H,D1H, D2H,  D3H)

corr_day_species &lt;- rbind(cor_D0H,  cor_D1H,  cor_D2H, cor_D3H)

aov_groups &lt;- cbind(corr_day_species, group)

dim(aov_groups)</code></pre>
<pre><code>## [1] 60  2</code></pre>
<pre class="r"><code>aov_bet &lt;- oneway.test(aov_groups$correlations ~ aov_groups$Sample)

t.test(cor_D0H, cor_D1H) # 0.0004748</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D0H and cor_D1H
## t = -4.2139, df = 18.903, p-value = 0.0004748
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.03077296 -0.01034362
## sample estimates:
## mean of x mean of y 
## 0.9595806 0.9801389</code></pre>
<pre class="r"><code>t.test(cor_D2H, cor_D1H) # 0.0009124</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D2H and cor_D1H
## t = -3.8713, df = 20.55, p-value = 0.0009124
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.025483704 -0.007657046
## sample estimates:
## mean of x mean of y 
## 0.9635685 0.9801389</code></pre>
<pre class="r"><code>t.test(cor_D3H, cor_D1H) # 4.172e-05</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D3H and cor_D1H
## t = -5.2163, df = 20.026, p-value = 4.172e-05
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.03245577 -0.01391431
## sample estimates:
## mean of x mean of y 
## 0.9569539 0.9801389</code></pre>
<pre class="r"><code># For between species 

#Make all the correlations

cor_D0H &lt;- as.data.frame(na.omit(day0cor_H))
colnames(cor_D0H) &lt;- c(&quot;correlations&quot;)
cor_D0C &lt;- as.data.frame(na.omit(day0cor_C))
colnames(cor_D0C) &lt;- c(&quot;correlations&quot;)

cor_D0 &lt;- rbind(cor_D0H, cor_D0C)

cor_D1H &lt;- as.data.frame(na.omit(day1cor_H)) 
colnames(cor_D1H) &lt;- c(&quot;correlations&quot;)
cor_D1C &lt;- as.data.frame(na.omit(day1cor_C))
colnames(cor_D1C) &lt;- c(&quot;correlations&quot;)
cor_D1 &lt;- rbind(cor_D1H, cor_D1C)

cor_D2H &lt;- as.data.frame(na.omit(day2cor_H))
colnames(cor_D2H) &lt;- c(&quot;correlations&quot;)
cor_D2C &lt;- as.data.frame(na.omit(day2cor_C))
colnames(cor_D2C) &lt;- c(&quot;correlations&quot;)
cor_D2 &lt;- rbind(cor_D2H, cor_D2C)
colnames(cor_D2) &lt;- c(&quot;correlations&quot;)

cor_D3H &lt;- as.data.frame(na.omit(day3cor_H))
colnames(cor_D3H) &lt;- c(&quot;correlations&quot;)
cor_D3C &lt;- as.data.frame(na.omit(day3cor_C))
colnames(cor_D3C) &lt;- c(&quot;correlations&quot;)
cor_D3 &lt;- rbind(cor_D3H, cor_D3C)


corr_day_species &lt;- rbind(cor_D0H, cor_D0C, cor_D1H, cor_D1C, cor_D2H, cor_D2C, cor_D3H, cor_D3C)

D0 &lt;- as.data.frame(rep(&quot;D0&quot;, each = 21))
colnames(D0) &lt;- c(&quot;Sample&quot;)

D1 &lt;- as.data.frame(rep(&quot;D1&quot;, each = 21))
colnames(D1) &lt;- c(&quot;Sample&quot;)

D2 &lt;- as.data.frame(rep(&quot;D2&quot;, each = 21))
colnames(D2) &lt;- c(&quot;Sample&quot;)

D3 &lt;- as.data.frame(rep(&quot;D3&quot;, each = 21))
colnames(D3) &lt;- c(&quot;Sample&quot;)


group &lt;- rbind(D0, D1, D2, D3)

aov_groups &lt;- cbind(corr_day_species, group)

# Perform the ANOVA

oneway.test(aov_groups$correlations ~ aov_groups$Sample)  # 1.632e-06</code></pre>
<pre><code>## 
##  One-way analysis of means (not assuming equal variances)
## 
## data:  aov_groups$correlations and aov_groups$Sample
## F = 14.373, num df = 3.000, denom df = 40.286, p-value = 1.632e-06</code></pre>
<pre class="r"><code>t.test(cor_D0, cor_D1) # 0.001695</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D0 and cor_D1
## t = -3.5127, df = 25.221, p-value = 0.001695
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.023340254 -0.006091521
## sample estimates:
## mean of x mean of y 
## 0.9657896 0.9805054</code></pre>
<pre class="r"><code>t.test(cor_D2, cor_D1) # 0.0006723</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D2 and cor_D1
## t = -3.8294, df = 27.681, p-value = 0.0006723
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.020671578 -0.006258641
## sample estimates:
## mean of x mean of y 
## 0.9670403 0.9805054</code></pre>
<pre class="r"><code>t.test(cor_D3, cor_D1) # 1.091e-05</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  cor_D3 and cor_D1
## t = -5.4033, df = 26.533, p-value = 1.091e-05
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.02821013 -0.01267260
## sample estimates:
## mean of x mean of y 
## 0.9600641 0.9805054</code></pre>
</div>
<div id="plot-the-correlations-genewise" class="section level2">
<h2>Plot the correlations genewise</h2>
<pre class="r"><code>##### Plot the correlations genewise

humans_day0_var &lt;- apply(as.data.frame(mean_tech_reps[,1:6]),1, var) 
chimps_day0_var &lt;- apply(as.data.frame(mean_tech_reps[,7:10]),1, var)
humans_day1_var &lt;- apply(as.data.frame(mean_tech_reps[,11:16]),1, var)
chimps_day1_var &lt;- apply(as.data.frame(mean_tech_reps[,17:20]),1, var)
humans_day2_var &lt;- apply(as.data.frame(mean_tech_reps[,21:26]),1, var)
chimps_day2_var &lt;- apply(as.data.frame(mean_tech_reps[,27:30]),1, var)
humans_day3_var &lt;- apply(as.data.frame(mean_tech_reps[,31:36]),1, var)
chimps_day3_var &lt;- apply(as.data.frame(mean_tech_reps[,37:40]),1, var)

HC_var &lt;- cbind(humans_day0_var, chimps_day0_var, humans_day1_var, chimps_day1_var, humans_day2_var, chimps_day2_var, humans_day3_var, chimps_day3_var)
boxplot(HC_var, ylim = c(0,0.5))</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<pre class="r"><code>humans_day0_mean &lt;- apply(as.data.frame(mean_tech_reps[,1:6]),1, mean)
chimps_day0_mean &lt;- apply(as.data.frame(mean_tech_reps[,7:10]),1, mean)
humans_day1_mean &lt;- apply(as.data.frame(mean_tech_reps[,11:16]),1, mean)
chimps_day1_mean &lt;- apply(as.data.frame(mean_tech_reps[,17:20]),1, mean)
humans_day2_mean &lt;- apply(as.data.frame(mean_tech_reps[,21:26]),1, mean)
chimps_day2_mean &lt;- apply(as.data.frame(mean_tech_reps[,27:30]),1, mean)
humans_day3_mean &lt;- apply(as.data.frame(mean_tech_reps[,31:36]),1, mean)
chimps_day3_mean &lt;- apply(as.data.frame(mean_tech_reps[,37:40]),1, mean)
 
HC_var_mean &lt;- cbind((humans_day0_var/humans_day0_mean), (chimps_day0_var/chimps_day0_mean), (humans_day1_var/humans_day1_mean), (chimps_day1_var/chimps_day1_mean), (humans_day2_var/humans_day2_mean), (chimps_day2_var/chimps_day2_mean), (humans_day3_var/humans_day3_mean), (chimps_day3_var/chimps_day3_mean))
boxplot(HC_var_mean, ylim = c(0,0.1))</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-25-2.png" width="672" /></p>
<pre class="r"><code>plot(log10(humans_day0_var), log10(humans_day1_var))
abline(0,1, col = &quot;red&quot;, v = 0, h = 0)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-25-3.png" width="672" /></p>
<pre class="r"><code># Make density plot 
#ggplot(variance, aes(x=var, fill=day)) + geom_density(alpha=0.5)+xlab(&quot;Variance&quot;) + theme(legend.position=c(.75,.75), legend.title=element_blank(), axis.title=element_text(size=14), panel.background=element_rect(fill=&#39;white&#39;)) + theme(text = element_text(size=18)) +xlim(c(-0.1,1))+ theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())
#t.test(var_d1b, var_d3b)</code></pre>
</div>
</div>
<div id="assessing-differential-gene-expression-using-limma" class="section level1">
<h1>Assessing differential gene expression using Limma</h1>
<p>Here, the model for expression has fixed effects for day, species, and a day-by-species interaction term.</p>
<pre class="r"><code>species &lt;- c(&quot;H&quot;, &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;, &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;)

day &lt;- factor(day)

# Make the design matrix with an interaction term between species and day
design &lt;- model.matrix(~species+day+species:day)

# We want a random effect term for individual. As a result, we want to run voom twice. See https://support.bioconductor.org/p/59700/

cpm.voom &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;none&quot;)

# We want a random effect term for individual
#corfit &lt;- duplicateCorrelation(cpm.voom, design,block=individual)
corfit.correlation = 0.1947084

cpm.voom.corfit &lt;- voom(dge_in_cutoff, design, plot = TRUE, normalize.method=&quot;none&quot;, block=individual, correlation = corfit.correlation )</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<pre class="r"><code># Plot the density to see shape of the distribution

plotDensities(cpm.voom.corfit, col=pal[as.numeric(day)], legend = F)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-2.png" width="672" /></p>
<pre class="r"><code># Run lmFit and eBayes in limma

fit &lt;- lmFit(cpm.voom.corfit , design, block=individual, correlation=corfit.correlation)
fit &lt;- eBayes(fit)

#cont.matrix &lt;- cbind(HvC = c(1,0,0,0,0,0,0),  levels=design)
#try &lt;- contrasts.fit(fit, cont.matrix)

# See DE genes
pv_list &lt;- fit$p.value
output_all &lt;- topTable(fit, number=Inf, sort.by=&quot;none&quot;)</code></pre>
<pre><code>## Removing intercept from test coefficients</code></pre>
<pre class="r"><code>head(output_all)</code></pre>
<pre><code>##                           genes    speciesH         day1        day2
## ENSG00000000003 ENSG00000000003 -0.18619499 -0.153714075 -0.41548685
## ENSG00000000419 ENSG00000000419  0.70105075  0.183830909 -0.02119966
## ENSG00000000457 ENSG00000000457 -0.94862229  0.003473239  0.20428506
## ENSG00000000460 ENSG00000000460  0.06439064  0.013782089 -0.45118234
## ENSG00000001036 ENSG00000001036  0.28859221 -0.265806917 -0.48065074
## ENSG00000001084 ENSG00000001084  0.27362005 -0.121040903  0.20635545
##                       day3 speciesH.day1 speciesH.day2 speciesH.day3
## ENSG00000000003 -0.6417390    0.12450608   -0.30714162   -0.07570601
## ENSG00000000419 -0.2894722    0.14957912    0.15126358    0.11368160
## ENSG00000000457  0.2877446   -0.19386265   -0.01871508   -0.24235822
## ENSG00000000460 -1.3306645    0.07421102    0.05744897    0.49278944
## ENSG00000001036 -0.6371074   -0.32833370    0.25727905    0.52333905
## ENSG00000001084  0.2793245    0.14119656    0.04878254   -0.22611736
##                  AveExpr         F      P.Value    adj.P.Val
## ENSG00000000003 7.300291 13.958350 1.457233e-10 3.597348e-10
## ENSG00000000419 6.219241 15.845103 1.468128e-11 4.127584e-11
## ENSG00000000457 4.383056 21.472263 3.747086e-14 1.545636e-13
## ENSG00000000460 5.201460 27.194667 2.370634e-16 1.418526e-15
## ENSG00000001036 6.238005 10.183301 2.575717e-08 4.917581e-08
## ENSG00000001084 6.824361  7.396568 2.189730e-06 3.391400e-06</code></pre>
<pre class="r"><code>dim(output_all)</code></pre>
<pre><code>## [1] 10304    12</code></pre>
<pre class="r"><code>mean(abs(output_all$speciesH.day2))</code></pre>
<pre><code>## [1] 0.3496331</code></pre>
<pre class="r"><code># Find the top hits by BH adjusted p-value
#top2 &lt;- list(speciesdiff = topTable(fit2, coeff = 2, adjust=&quot;BH&quot;, number = Inf, sort.by = &quot;none&quot;), day01diff = topTable(fit2, coeff = 3, adjust=&quot;BH&quot;,number = Inf, sort.by = &quot;none&quot;), day02sdiff = topTable(fit2, coeff = 4, adjust=&quot;BH&quot;,number = Inf, sort.by = &quot;none&quot;), day03diff = topTable(fit2, coeff = 5, adjust=&quot;BH&quot;,number = Inf, sort.by = &quot;none&quot;), Hday1inter = topTable(fit2, coeff = 6, adjust=&quot;BH&quot;,number = Inf, sort.by = &quot;none&quot;), Hday2inter = topTable(fit2, coeff = 7, adjust=&quot;BH&quot;,number = Inf, sort.by = &quot;none&quot;), Hday3diff = topTable(fit2, coeff = 8, adjust=&quot;BH&quot;,number = Inf, sort.by = &quot;none&quot;))


# Find genes DE between species (BH-adjusted p-value of 5%)
Bspecies &lt;- output_all$speciesH
Bspecies_chimpref &lt;- Bspecies
Bspecies_humanref &lt;- Bspecies

names(Bspecies) &lt;- rownames(output_all)
Bspecies &lt;- Bspecies[order(Bspecies)]
Pspecies &lt;- pv_list[,2]
adjPspecies &lt;- p.adjust(Pspecies, method = &quot;BH&quot;)
sigspecies &lt;- adjPspecies[adjPspecies &lt; 0.05]
sigspecies &lt;- sigspecies[order(sigspecies)]
sigspeciesnames &lt;- names(sigspecies)
length(sigspecies)</code></pre>
<pre><code>## [1] 4473</code></pre>
<pre class="r"><code># Make plots for genes with this significant coefficient
sig &lt;- cpm_in_cutoff[sigspeciesnames,]
for (i in 1:10) {
  play &lt;- sig[i,]
  playdf &lt;- data.frame(play,day,species)
  print(ggplot(data=playdf, aes(x=as.factor(day), y=play, fill=species)) + geom_boxplot(aes(fill=species)) + ggtitle(paste0(&quot;Expression for &quot;, rownames(sig)[i])) + ylab(&quot;log2(CPM) expression value&quot;))
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-3.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-4.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-5.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-6.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-7.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-8.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-9.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-10.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-11.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-12.png" width="672" /></p>
<pre class="r"><code># Differentially expressed genes between days 0 and 1

BD1 &lt;- output_all$day1
Chimp_ref_BD1 &lt;- BD1
Human_ref_BD1 &lt;- BD1
BD1_comp &lt;- cbind(Chimp_ref_BD1, Human_ref_BD1)
cor(BD1_comp)</code></pre>
<pre><code>##               Chimp_ref_BD1 Human_ref_BD1
## Chimp_ref_BD1             1             1
## Human_ref_BD1             1             1</code></pre>
<pre class="r"><code>names(BD1) &lt;- rownames(output_all)
BD1 &lt;- BD1[order(BD1)]
PBD1 &lt;- pv_list[,3]
adjPBD1 &lt;- p.adjust(PBD1, method = &quot;BH&quot;)
sigBD1 &lt;- adjPBD1[adjPBD1 &lt; 0.05]
sigBD1 &lt;- sigBD1[order(sigBD1)]
sigBD1names &lt;- names(sigBD1)
length(sigBD1)</code></pre>
<pre><code>## [1] 3291</code></pre>
<pre class="r"><code># Make plots for genes with this significant coefficient
sig3 &lt;- cpm_in_cutoff[sigBD1names,]
for (i in 1:10) {
  play &lt;- sig3[i,]
  playdf &lt;- data.frame(play,day,species)
  print(ggplot(data=playdf, aes(x=as.factor(day), y=play, fill=species)) + geom_boxplot(aes(fill=species)) + ggtitle(paste0(&quot;Expression for &quot;, rownames(sig3)[i])) + ylab(&quot;log2(CPM) expression value&quot;))
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-13.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-14.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-15.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-16.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-17.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-18.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-19.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-20.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-21.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-22.png" width="672" /></p>
<pre class="r"><code># Differentially expressed genes between days 0 and 2

BD2 &lt;- output_all$day2
names(BD2) &lt;- rownames(output_all)
BD2 &lt;- BD2[order(BD2)]
PBD2 &lt;- pv_list[,4]
adjPBD2 &lt;- p.adjust(PBD2, method = &quot;BH&quot;)
sigBD2 &lt;- adjPBD2[adjPBD2 &lt; 0.05]
sigBD2 &lt;- sigBD2[order(sigBD2)]
sigBD2names &lt;- names(sigBD2)
length(sigBD2)</code></pre>
<pre><code>## [1] 5584</code></pre>
<pre class="r"><code># Make plots for genes with this significant coefficient
sig3 &lt;- cpm_in_cutoff[sigBD2names,]
for (i in 1:10) {
  play &lt;- sig3[i,]
  playdf &lt;- data.frame(play,day,species)
  print(ggplot(data=playdf, aes(x=as.factor(day), y=play, fill=species)) + geom_boxplot(aes(fill=species)) + ggtitle(paste0(&quot;Expression for &quot;, rownames(sig3)[i])) + ylab(&quot;log2(CPM) expression value&quot;))
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-23.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-24.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-25.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-26.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-27.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-28.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-29.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-30.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-31.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-32.png" width="672" /></p>
<pre class="r"><code># Differentially expressed genes between days 0 and 3

BD3 &lt;- output_all$day3
names(BD3) &lt;- rownames(output_all)
BD3 &lt;- BD3[order(BD3)]
PBD3 &lt;- pv_list[,5]
adjPBD3 &lt;- p.adjust(PBD3, method = &quot;BH&quot;)
sigBD3 &lt;- adjPBD1[adjPBD3 &lt; 0.05]
sigBD3 &lt;- sigBD3[order(sigBD3)]
sigBD3names &lt;- names(sigBD3)
length(sigBD3)</code></pre>
<pre><code>## [1] 7091</code></pre>
<pre class="r"><code># Make plots for genes with this significant coefficient
sig3 &lt;- cpm_in_cutoff[sigBD3names,]
for (i in 1:10) {
  play &lt;- sig3[i,]
  playdf &lt;- data.frame(play,day,species)
  print(ggplot(data=playdf, aes(x=as.factor(day), y=play, fill=species)) + geom_boxplot(aes(fill=species)) + ggtitle(paste0(&quot;Expression for &quot;, rownames(sig3)[i])) + ylab(&quot;log2(CPM) expression value&quot;))
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-33.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-34.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-35.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-36.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-37.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-38.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-39.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-40.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-41.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-42.png" width="672" /></p>
<pre class="r"><code># Genes with a significant species by day interaction term, Human-by-day1

BspeciesD1 &lt;- output_all$speciesH.day1
names(BspeciesD1) &lt;- rownames(output_all)
BspeciesD1 &lt;- BspeciesD1[order(BspeciesD1)]
Pspecies1 &lt;- pv_list[,6]
adjPspecies1 &lt;- p.adjust(Pspecies1, method = &quot;BH&quot;)
sigspecies1 &lt;- adjPspecies1[adjPspecies1 &lt; 0.05]
sigspecies1 &lt;- sigspecies1[order(sigspecies1)]
sigspecies1names &lt;- names(sigspecies1)
length(sigspecies1)</code></pre>
<pre><code>## [1] 253</code></pre>
<pre class="r"><code># Genes with a significant species by day interaction term, Human-by-day1

BspeciesD1 &lt;- output_all$speciesH.day1
names(BspeciesD1) &lt;- rownames(output_all)
BspeciesD1 &lt;- BspeciesD1[order(BspeciesD1)]
Pspecies1 &lt;- pv_list[,6]
adjPspecies1 &lt;- p.adjust(Pspecies1, method = &quot;BH&quot;)
sigspecies1 &lt;- adjPspecies1[adjPspecies1 &lt; 0.05]
sigspecies1 &lt;- sigspecies1[order(sigspecies1)]
sigspecies1names &lt;- names(sigspecies1)
length(sigspecies1)</code></pre>
<pre><code>## [1] 253</code></pre>
<pre class="r"><code>ensembl &lt;- useMart(host = &quot;dec2013.archive.ensembl.org&quot;,
                   biomart = &quot;ENSEMBL_MART_ENSEMBL&quot;,
                   dataset = &quot;hsapiens_gene_ensembl&quot;)
gene_names &lt;- getBM(filters= &quot;ensembl_gene_id&quot;, attributes= c(&quot;ensembl_gene_id&quot;,&quot;hgnc_symbol&quot;),values=sigspecies1names,mart= ensembl)
sigspd1names &lt;- sigspecies1names[sigspecies1names %in% gene_names[,1]]
#sigspd1names_add &lt;- names(sigspd1names)
#all(sigspd1names == gene_names[,1])
#sigspd1names == gene_names[,1]
#sigspd1names &lt;- gene_names[,2]
write.table(sigspd1names, file=&quot;sigspd1names.txt&quot;)

# Make plots for genes with this significant coefficient
sig1 &lt;- cpm_in_cutoff[sigspd1names,]

for (i in 1:3) {
  play &lt;- sig1[i,]
  playdf &lt;- data.frame(play,day,species)
  print(ggplot(data=playdf, aes(x=as.factor(day), y=play, fill=species)) + geom_boxplot(aes(fill=species)) + ggtitle(paste0(&quot;Expression for &quot;, rownames(sig1)[i])) + ylab(&quot;log2(CPM) expression value&quot;))
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-43.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-44.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-45.png" width="672" /></p>
<pre class="r"><code># Genes with a significant species by day interaction term, Human-by-day2

BspeciesD2 &lt;- output_all$speciesH.day2
names(BspeciesD2) &lt;- rownames(output_all)
BspeciesD2 &lt;- BspeciesD2[order(BspeciesD2)]
Pspecies2 &lt;- pv_list[,7]
adjPspecies2 &lt;- p.adjust(Pspecies2, method = &quot;BH&quot;)
sigspecies2 &lt;- adjPspecies2[adjPspecies2 &lt; 0.05]
sigspecies2 &lt;- sigspecies2[order(sigspecies2)]
sigspecies2names &lt;- names(sigspecies2)
length(sigspecies2)</code></pre>
<pre><code>## [1] 1082</code></pre>
<pre class="r"><code># Make plots for genes with this significant coefficient
sig2 &lt;- cpm_in_cutoff[sigspecies2names,]
for (i in 1:3) {
  play &lt;- sig2[i,]
  playdf &lt;- data.frame(play,day,species)
  print(ggplot(data=playdf, aes(x=as.factor(day), y=play, fill=species)) + geom_boxplot(aes(fill=species)) + ggtitle(paste0(&quot;Expression for &quot;, rownames(sig2)[i])) + ylab(&quot;log2(CPM) expression value&quot;))
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-46.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-47.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-48.png" width="672" /></p>
<pre class="r"><code># Genes with a significant species by day interaction term, Human-by-day3

BspeciesD3 &lt;- output_all$speciesH.day3
names(BspeciesD3) &lt;- rownames(output_all)
BspeciesD3 &lt;- BspeciesD3[order(BspeciesD3)]
Pspecies3 &lt;- pv_list[,8]
adjPspecies3 &lt;- p.adjust(Pspecies3, method = &quot;BH&quot;)
sigspecies3 &lt;- adjPspecies3[adjPspecies3 &lt; 0.05]
sigspecies3 &lt;- sigspecies3[order(sigspecies3)]
sigspecies3names &lt;- names(sigspecies3)
length(sigspecies3)</code></pre>
<pre><code>## [1] 2325</code></pre>
<pre class="r"><code># Make plots for genes with this significant coefficient
sig3 &lt;- cpm_in_cutoff[sigspecies3names,]
for (i in 1:3) {
  play &lt;- sig3[i,]
  playdf &lt;- data.frame(play,day,species)
  print(ggplot(data=playdf, aes(x=as.factor(day), y=play, fill=species)) + geom_boxplot(aes(fill=species)) + ggtitle(paste0(&quot;Expression for &quot;, rownames(sig3)[i])) + ylab(&quot;log2(CPM) expression value&quot;))
}</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-49.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-50.png" width="672" /><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-26-51.png" width="672" /></p>
<div id="make-a-venn-diagrams" class="section level2">
<h2>Make a Venn Diagrams</h2>
<pre class="r"><code>library(&quot;R.utils&quot;)</code></pre>
<pre><code>## Warning: package &#39;R.utils&#39; was built under R version 3.2.5</code></pre>
<pre><code>## Loading required package: R.oo</code></pre>
<pre><code>## Warning: package &#39;R.oo&#39; was built under R version 3.2.5</code></pre>
<pre><code>## Loading required package: R.methodsS3</code></pre>
<pre><code>## Warning: package &#39;R.methodsS3&#39; was built under R version 3.2.3</code></pre>
<pre><code>## R.methodsS3 v1.7.1 (2016-02-15) successfully loaded. See ?R.methodsS3 for help.</code></pre>
<pre><code>## R.oo v1.21.0 (2016-10-30) successfully loaded. See ?R.oo for help.</code></pre>
<pre><code>## 
## Attaching package: &#39;R.oo&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:GenomicRanges&#39;:
## 
##     trim</code></pre>
<pre><code>## The following object is masked from &#39;package:IRanges&#39;:
## 
##     trim</code></pre>
<pre><code>## The following objects are masked from &#39;package:methods&#39;:
## 
##     getClasses, getMethods</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     attach, detach, gc, load, save</code></pre>
<pre><code>## R.utils v2.4.0 (2016-09-13) successfully loaded. See ?R.utils for help.</code></pre>
<pre><code>## 
## Attaching package: &#39;R.utils&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     extract</code></pre>
<pre><code>## The following object is masked from &#39;package:GEOquery&#39;:
## 
##     gunzip</code></pre>
<pre><code>## The following object is masked from &#39;package:utils&#39;:
## 
##     timestamp</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     cat, commandArgs, getOption, inherits, isOpen, parse, warnings</code></pre>
<pre class="r"><code>venn(list(d1=sigBD1names, d2=sigBD2names, d3=sigBD3names))</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<pre class="r"><code># Significant between days

mylist &lt;- list()
mylist[[&quot;Days 0 and 1&quot;]] &lt;- sigBD1names
mylist[[&quot;Days 0 and 2&quot;]] &lt;- sigBD2names
mylist[[&quot;Days 0 and 3&quot;]] &lt;- sigBD3names


# Make as png file
species_by_day &lt;- venn.diagram(mylist, filename=NULL,  main=&quot;Genes with Significant Day Coefficients&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=2500)

pdf(file = &quot;Figure 3.6A.pdf&quot;)
  grid.draw(species_by_day)
dev.off()</code></pre>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
<pre class="r"><code># Significant day-interaction term

venn(list(spd1=sigspecies1names, spd2=sigspecies2names, spd3=sigspecies3names))</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-27-2.png" width="672" /></p>
<pre class="r"><code>mylist &lt;- list()
mylist[[&quot;Human x Day 1&quot;]] &lt;- sigspecies1names
mylist[[&quot;Human x Day 2&quot;]] &lt;- sigspecies2names
mylist[[&quot;Human x Day 3&quot;]] &lt;- sigspecies3names



# Make as png file
venn.diagram(mylist, filename=&quot;Genes with Significant Interaction Terms.png&quot;, imagetype = &quot;png&quot;, main=&quot;Genes with Significant Interaction Terms&quot;, cex=1.5 , fill = pal[1:3], lty=1, height=2000, width=3000)</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code># Differences between genes with significant day terms and significant day by species interaction terms

mylist_day1 &lt;- list()
mylist_day1[[&quot;Human x Day 1&quot;]] &lt;- sigspecies1names
mylist_day1[[&quot;Day 1&quot;]] &lt;- sigBD1names

venn.diagram(mylist_day1, filename=&quot;Day 1 Comparison.png&quot;, imagetype = &quot;png&quot;, main=&quot;Day 1 Comparison&quot;, cex=1.5 , fill = pal[1:2], lty=1, height=2000, width=3000)</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>mylist_day2 &lt;- list()
mylist_day2[[&quot;Human x Day 2&quot;]] &lt;- sigspecies2names
mylist_day2[[&quot;Day 2&quot;]] &lt;- sigBD2names

venn.diagram(mylist_day2, filename=&quot;Day 2 Comparison.png&quot;, imagetype = &quot;png&quot;, main=&quot;Day 2 Comparison&quot;, cex=1.5 , fill = pal[1:2], lty=1, height=2000, width=3000)</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>mylist_day3 &lt;- list()
mylist_day3[[&quot;Human x Day 3&quot;]] &lt;- sigspecies3names
mylist_day3[[&quot;Day 3&quot;]] &lt;- sigBD3names

venn.diagram(mylist_day3, filename=&quot;Day 3 Comparison.png&quot;, imagetype = &quot;png&quot;, main=&quot;Day 3 Comparison&quot;, cex=1.5 , fill = pal[1:2], lty=1, height=2000, width=3000)</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>rl &lt;- lapply(list(&quot;Day 1 Comparison.png&quot;, &quot;Day 2 Comparison.png&quot;, &quot;Day 3 Comparison.png&quot;), png::readPNG)
gl &lt;- lapply(rl, grid::rasterGrob)
do.call(gridExtra::grid.arrange, gl)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-27-3.png" width="672" /></p>
<pre class="r"><code>par(mfrow = c(1,1)) 
length_days &lt;- c(length(sigBD1names), length(sigBD2names), length(sigBD3names))
length_interaction &lt;- c(length(sigspecies1names), length(sigspecies2names), length(sigspecies3names))
meets1 &lt;- 
length_both &lt;- rbind(length_days, length_interaction)
colnames(length_both)&lt;- c(&quot;Day 1&quot;, &quot;Day 2&quot;, &quot;Day 3&quot;)
barplot(length_both, col=c(&quot;darkblue&quot;, &quot;red&quot;), ylim = c(0,17000), ylab = (&quot;Number of DE genes&quot;), legend=rownames(length_both))</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-27-4.png" width="672" /></p>
</div>
</div>
<div id="differentially-expressed-genes-using-a-model-with-no-interaction-term" class="section level1">
<h1>Differentially expressed genes using a model with no interaction term</h1>
<p>Here, the model for expression has fixed effects for day and species.</p>
<pre class="r"><code># To make contrasts easier to formulate, we rename factors species and day in a single factor 

# Label the species-day pairs
species &lt;- c(&quot;H&quot;, &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;, &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,
             &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,
             &quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,&quot;H&quot;,  &quot;C&quot;, &quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;,&quot;C&quot;)
condition &lt;- factor(paste(species, day, sep=&quot;.&quot;))

# Make the design matrix

design &lt;- model.matrix(~ 0+condition)
colnames(design) &lt;- gsub(&quot;condition&quot;, &quot;&quot;, dput(colnames(design)))</code></pre>
<pre><code>## c(&quot;conditionC.0&quot;, &quot;conditionC.1&quot;, &quot;conditionC.2&quot;, &quot;conditionC.3&quot;, 
## &quot;conditionH.0&quot;, &quot;conditionH.1&quot;, &quot;conditionH.2&quot;, &quot;conditionH.3&quot;
## )</code></pre>
<pre class="r"><code># We want a random effect term for individual. As a result, we want to run voom twice. See https://support.bioconductor.org/p/59700/

cpm.voom &lt;- voom(dge_in_cutoff, design, normalize.method=&quot;none&quot;)

# We want a random effect term for individual
#corfit &lt;- duplicateCorrelation(cpm.voom, design,block=individual)
corfit.correlation = 0.1947084

cpm.voom.corfit &lt;- voom(dge_in_cutoff, design, plot = TRUE, normalize.method=&quot;none&quot;, block=individual, correlation = corfit.correlation )</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
<pre class="r"><code># Plot the density to see shape of the distribution

plotDensities(cpm.voom.corfit, col=pal[as.numeric(day+1)], legend = F)</code></pre>
<pre><code>## Warning in Ops.factor(day, 1): &#39;+&#39; not meaningful for factors</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-28-2.png" width="672" /></p>
<pre class="r"><code># Run lmFit and eBayes in limma

fit1 &lt;- lmFit(cpm.voom.corfit , design, block=individual, correlation=corfit.correlation)
fit1 &lt;- eBayes(fit1)</code></pre>
</div>
<div id="differences-between-species-at-each-time-points" class="section level1">
<h1>Differences between species at each time points</h1>
<pre class="r"><code># Make the contrast matrix

cont_matrix&lt;-makeContrasts(HvsC_D0=H.0-C.0, HvsC_D1 = H.1-C.1, HvsC_D2 = H.2-C.2, HvsC_D3 = H.3-C.3, levels=design)

# Test on contrast matrix
fit2 &lt;- contrasts.fit(fit1, cont_matrix)
fit2 &lt;- eBayes(fit2)
head(fit2$t)</code></pre>
<pre><code>##                  Contrasts
##                      HvsC_D0    HvsC_D1   HvsC_D2    HvsC_D3
##   ENSG00000000003 -1.2328068 -0.4198084 -3.320348 -1.7447032
##   ENSG00000000419  5.0898343  6.3988402  6.198700  5.7310579
##   ENSG00000000457 -6.7929398 -8.2214596 -7.096535 -8.6916991
##   ENSG00000000460  0.4823213  1.0627122  0.849881  3.4139568
##   ENSG00000001036  2.0008084 -0.2748422  3.689035  5.4068766
##   ENSG00000001084  2.6234977  4.0623969  3.179010  0.4668392</code></pre>
<pre class="r"><code>stats_list &lt;- list()
pv_list &lt;- fit2$p.value

output_all_0 &lt;- topTable(fit2, coef=&quot;HvsC_D0&quot;, number=Inf, sort.by=&quot;none&quot;)
output_all_1 &lt;- topTable(fit2, coef=&quot;HvsC_D1&quot;, number=Inf, sort.by=&quot;none&quot;)
output_all_2 &lt;- topTable(fit2, coef=&quot;HvsC_D2&quot;, number=Inf, sort.by=&quot;none&quot;)
output_all_3 &lt;- topTable(fit2, coef=&quot;HvsC_D3&quot;, number=Inf, sort.by=&quot;none&quot;)
head(output_all_0)</code></pre>
<pre><code>##                           genes       logFC  AveExpr          t
## ENSG00000000003 ENSG00000000003 -0.18619499 7.300291 -1.2328068
## ENSG00000000419 ENSG00000000419  0.70105075 6.219241  5.0898343
## ENSG00000000457 ENSG00000000457 -0.94862229 4.383056 -6.7929398
## ENSG00000000460 ENSG00000000460  0.06439064 5.201460  0.4823213
## ENSG00000001036 ENSG00000001036  0.28859221 6.238005  2.0008084
## ENSG00000001084 ENSG00000001084  0.27362005 6.824361  2.6234977
##                      P.Value    adj.P.Val         B
## ENSG00000000003 2.224814e-01 3.304192e-01 -6.598877
## ENSG00000000419 3.840530e-06 3.278610e-05  3.583851
## ENSG00000000457 5.762583e-09 1.107792e-07 10.154548
## ENSG00000000460 6.313418e-01 7.306094e-01 -7.128350
## ENSG00000001036 4.996718e-02 9.949021e-02 -5.362204
## ENSG00000001084 1.103195e-02 2.843963e-02 -4.036950</code></pre>
<pre class="r"><code># Make a heatmap
#df.norm &lt;- data.frame(output_all_0$B, output_all_1$B, output_all_2$B, output_all_3$B)
#for(i in 1:nrow(df.norm)){df.norm[i,]=df.norm[i,]/max(abs(df.norm[i,]))}
#colnames(df.norm)=c(&quot;d0&quot;,&quot;d1&quot;,&quot;d2&quot;,&quot;d3&quot;)
#heatmap(as.matrix(df.norm))

# Use hclust

#hclustfunc &lt;- function(x, method = &quot;complete&quot;, dmeth = &quot;euclidean&quot;) {    
#    hclust(dist(x, method = dmeth), method = method)
#}

#df.norm.hclust=hclustfunc(as.matrix(df.norm[1:500,]))

#dist.pear &lt;- as.dist(1-cor(t(df.norm[1:500,])))

#heatmap.2(as.matrix(df.norm[1:500,]), trace=&quot;none&quot;, dist.pear, df.norm.hclust)

#dev.off()

#effect_size_spday &lt;- data.frame(ef=c(abs(output_all_0$logFC), abs(output_all_1$logFC), abs(output_all_2$logFC), abs(output_all_3$logFC)), day = rep(c(&quot;D0&quot;, &quot;D1&quot;, &quot;D2&quot;, &quot;D3&quot;), times =c(length(output_all_0$logFC), length(output_all_1$logFC), length(output_all_2$logFC), length(output_all_3$logFC))))

#ggplot(effect_size_spday, aes(x=ef, fill=day)) + geom_density(alpha=0.5)+xlab(&quot;Effect size&quot;) + theme(legend.position=c(.75,.75), legend.title=element_blank(), axis.title=element_text(size=14), panel.background=element_rect(fill=&#39;white&#39;)) + theme(text = element_text(size=18)) +xlim(c(-0.1,2))
t.test(BspeciesD2,BspeciesD3)</code></pre>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  BspeciesD2 and BspeciesD3
## t = 0.10038, df = 19633, p-value = 0.92
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  -0.01657022  0.01835897
## sample estimates:
##   mean of x   mean of y 
## 0.002803466 0.001909093</code></pre>
<div id="find-non-significant-genes-between-species-for-each-day" class="section level2">
<h2>Find non-significant genes between species for each day</h2>
<pre class="r"><code># Set FDR level at 5% 

FDR_level &lt;- 0.05

# HvsC at Day 0
notsig0 &lt;- output_all_0[(output_all_0$adj.P.Val)&gt;FDR_level,]
dim(notsig0)</code></pre>
<pre><code>## [1] 5831    7</code></pre>
<pre class="r"><code># HvsC at Day 1

notsig1 &lt;- output_all_1[(output_all_1$adj.P.Val)&gt;FDR_level,]
dim(notsig1)</code></pre>
<pre><code>## [1] 5924    7</code></pre>
<pre class="r"><code># HvsC at Day 2

notsig2 &lt;- output_all_2[(output_all_2$adj.P.Val)&gt;FDR_level,]
dim(notsig2)</code></pre>
<pre><code>## [1] 5707    7</code></pre>
<pre class="r"><code># HvsC at Day 3

notsig3 &lt;- output_all_3[(output_all_3$adj.P.Val)&gt;FDR_level,]
dim(notsig3)</code></pre>
<pre><code>## [1] 5322    7</code></pre>
<pre class="r"><code># Not significant between species in any of the days

notsig_any &lt;- intersect(intersect(intersect(rownames(notsig0), rownames(notsig1)), rownames(notsig2)), rownames(notsig3))

notsig_any &lt;- as.data.frame(notsig_any)
dim(notsig_any)</code></pre>
<pre><code>## [1] 2880    1</code></pre>
</div>
<div id="find-genes-between-species-for-each-day" class="section level2">
<h2>Find genes between species for each day</h2>
<pre class="r"><code># Significant between species in all of the days

sig_D0 &lt;- output_all_0[(output_all_0$adj.P.Val)&lt;FDR_level,]
dim(sig_D0)</code></pre>
<pre><code>## [1] 4473    7</code></pre>
<pre class="r"><code>sig_D1 &lt;- output_all_1[(output_all_1$adj.P.Val)&lt;FDR_level,]
dim(sig_D1)</code></pre>
<pre><code>## [1] 4380    7</code></pre>
<pre class="r"><code>sig_D2 &lt;- output_all_2[(output_all_2$adj.P.Val)&lt;FDR_level,] 
dim(sig_D2)</code></pre>
<pre><code>## [1] 4597    7</code></pre>
<pre class="r"><code>sig_D3 &lt;- output_all_3[(output_all_3$adj.P.Val)&lt;FDR_level,]
dim(sig_D3)</code></pre>
<pre><code>## [1] 4982    7</code></pre>
<pre class="r"><code># Mean of the absolute value of the fold change
mean_effect_D0 &lt;- mean(abs(output_all_0$logFC))
mean_effect_D1 &lt;- mean(abs(output_all_1$logFC))
mean_effect_D2 &lt;- mean(abs(output_all_2$logFC))
mean_effect_D3 &lt;- mean(abs(output_all_3$logFC))

# Plot mean log FC for all genes

library(&quot;vioplot&quot;)</code></pre>
<pre><code>## Loading required package: sm</code></pre>
<pre><code>## Package &#39;sm&#39;, version 2.2-5.4: type help(sm) for summary information</code></pre>
<pre class="r"><code>mean_effects_0 &lt;- output_all_0$logFC
mean_effects_1 &lt;- output_all_1$logFC
mean_effects_2 &lt;- output_all_2$logFC
mean_effects_3 &lt;- output_all_3$logFC
vioplot(mean_effects_0, mean_effects_1, mean_effects_2, mean_effects_3, names=c(&quot;Day 0&quot;, &quot;Day 1&quot;, &quot;Day 2&quot;, &quot;Day 3&quot;), col = &quot;red&quot;)
title(&quot;Violin Plots of Fold Change for Each Gene at Each Day&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-31-1.png" width="672" /></p>
<pre class="r"><code># Plot log fold change for all significant genes

mean_effects_0 &lt;- sig_D0$logFC
mean_effects_1 &lt;- sig_D1$logFC
mean_effects_2 &lt;- sig_D2$logFC
mean_effects_3 &lt;- sig_D3$logFC
vioplot(mean_effects_0, mean_effects_1, mean_effects_2, mean_effects_3, names=c(&quot;Day 0&quot;, &quot;Day 1&quot;, &quot;Day 2&quot;, &quot;Day 3&quot;), col = &quot;gold&quot;)
title(&quot;Violin Plots of Fold Change for Differently Expressed Genes at Each Day&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-31-2.png" width="672" /></p>
<pre class="r"><code># Plot effect size for all significant genes

effect_size_spday &lt;- data.frame(ef=c(abs(sig_D0$B), abs(sig_D1$B), abs(sig_D2$B), abs(sig_D3$B)), day = rep(c(&quot;Day 0&quot;, &quot;Day 1&quot;, &quot;Day 2&quot;, &quot;Day 3&quot;), times =c(length(sig_D0$logFC), length(sig_D1$logFC), length(sig_D2$logFC), length(sig_D3$logFC))))

ggplot(effect_size_spday, aes(x=ef, fill=day)) + geom_density(alpha=0.5)+xlab(&quot;Effect size&quot;) + theme(legend.position=c(.75,.75), legend.title=element_blank(), axis.title=element_text(size=14), panel.background=element_rect(fill=&#39;white&#39;)) + theme(text = element_text(size=18)) +xlim(c(-1.5,15)) + labs(title = &quot;Effect Sizes for Significant Genes by Day&quot;)</code></pre>
<pre><code>## Warning: Removed 1209 rows containing non-finite values (stat_density).</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-31-3.png" width="672" /></p>
<pre class="r"><code>#t.test(abs(output_D1$logFC), abs(output_D3$logFC))

# Make a Venn diagram

par(mfrow=c(1,1))

mylist &lt;- list()
mylist[[&quot;DE Day 0&quot;]] &lt;- rownames(sig_D0)
mylist[[&quot;DE Day 1&quot;]] &lt;- rownames(sig_D1)
mylist[[&quot;DE Day 2&quot;]] &lt;- rownames(sig_D2)
mylist[[&quot;DE Day 3&quot;]] &lt;- rownames(sig_D3)

# Make as pdf 
Four_comp &lt;- venn.diagram(mylist, filename= NULL, main=&quot;DE genes between species per day&quot;, cex=1.5 , fill = pal[1:4], lty=1, height=2000, width=3000)
pdf(file = &quot;Figure 3.7C.pdf&quot;)
  grid.draw(Four_comp)
dev.off()</code></pre>
<pre><code>## quartz_off_screen 
##                 2</code></pre>
</div>
<div id="find-genes-that-are-significant-between-species-only-for-one-day" class="section level2">
<h2>Find genes that are significant between species only for one day</h2>
<pre class="r"><code># Get the rownames for the genes
sig_d0 &lt;- rownames(sig_D0)
sig_d1 &lt;- rownames(sig_D1)
sig_d2 &lt;- rownames(sig_D2)
sig_d3 &lt;- rownames(sig_D3)

# Only significant in day 0
sig_d0_only &lt;- sig_d0[!(sig_d0 %in% sig_d1)]
sig_d0_only &lt;- sig_d0_only[!(sig_d0_only %in% sig_d2)]
sig_d0_only &lt;- sig_d0_only[!(sig_d0_only %in% sig_d3)]
length(sig_d0_only)</code></pre>
<pre><code>## [1] 522</code></pre>
<pre class="r"><code># Only significant in day 1

sig_d1_only &lt;- sig_d1[!(sig_d1 %in% sig_d0)]
sig_d1_only &lt;- sig_d1_only[!(sig_d1_only %in% sig_d2)]
sig_d1_only &lt;- sig_d1_only[!(sig_d1_only %in% sig_d3)]
length(sig_d1_only)</code></pre>
<pre><code>## [1] 315</code></pre>
<pre class="r"><code># Only significant in day 2

sig_d2_only &lt;- sig_d2[!(sig_d2 %in% sig_d0)]
sig_d2_only &lt;- sig_d2_only[!(sig_d2_only %in% sig_d3)]
sig_d2_only &lt;- sig_d2_only[!(sig_d2_only %in% sig_d1)]
length(sig_d2_only)</code></pre>
<pre><code>## [1] 365</code></pre>
<pre class="r"><code># Only significant in day 3

sig_d3_only &lt;- sig_d3[!(sig_d3 %in% sig_d0)]
sig_d3_only &lt;- sig_d3_only[!(sig_d3_only %in% sig_d2)]
sig_d3_only &lt;- sig_d3_only[!(sig_d3_only %in% sig_d1)]
length(sig_d3_only)</code></pre>
<pre><code>## [1] 966</code></pre>
<pre class="r"><code># Significant in all 4 days

sig_all &lt;- sig_d2[(sig_d2 %in% sig_d0)]
sig_all &lt;- sig_all[(sig_all %in% sig_d3)]
sig_all &lt;- sig_all[(sig_all %in% sig_d1)]
length(sig_all)</code></pre>
<pre><code>## [1] 2193</code></pre>
<pre class="r"><code># Unclear what you are trying to do here
#dfs &lt;- data.frame(sig_D0$logFC, sig_D1$logFC, sig_D2$logFC, sig_D3$logFC)
#dfs &lt;- stack(dfs)
#head(dfs)
dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
</div>
<div id="plot-the-number-of-genes-de-between-species-at-each-day" class="section level2">
<h2>Plot the number of genes DE between species at each day</h2>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
<pre class="r"><code># Numbers for non-unique DE genes
collect_total_barplot &lt;- c(length(sig_d0)-length(sig_d0_only), length(sig_d1)-length(sig_d1_only), length(sig_d2)-length(sig_d2_only), length(sig_d3)-length(sig_d3_only), nrow(notsig_any), length(sig_all))

# Numbers for unique DE genes
collect_only &lt;- c(length(sig_d0_only), length(sig_d1_only), length(sig_d2_only), length(sig_d3_only), 0, 0)

# Combine the unique and non-unique DE genes
msig &lt;- rbind(collect_total_barplot, collect_only)
colnames(msig) &lt;- c(&quot;Day 0&quot;, &quot;Day 1&quot;, &quot;Day 2&quot;, &quot;Day 3&quot;, &quot;Never DE&quot;, &quot;Always DE&quot;)

# Make a stacked barplot

barplot(msig, ylab=&quot;Number of DE genes&quot;, ylim = c(0, 6000), main = &quot;Differentially Expressed Genes between Humans and Chimps Across Days&quot;)</code></pre>
</div>
<div id="find-significant-genes-arising-at-a-given-day" class="section level2">
<h2>Find significant genes arising at a given day</h2>
<pre class="r"><code>  # Arise at Day 0
sig_d0_arise &lt;- sig_d0
length(sig_d0_arise)</code></pre>
<pre><code>## [1] 4473</code></pre>
<pre class="r"><code>  # Arise at Day 1
sig_d1_arise &lt;- sig_d1[!(sig_d1 %in% sig_d0)]
length(sig_d1_arise)</code></pre>
<pre><code>## [1] 1013</code></pre>
<pre class="r"><code>  # Arise at Day 2
sig_d2_arise &lt;- sig_d2[!(sig_d2 %in% sig_d0)]
sig_d2_arise &lt;- sig_d2_arise[!(sig_d2_arise %in% sig_d1)]
length(sig_d2_arise)</code></pre>
<pre><code>## [1] 972</code></pre>
<pre class="r"><code>  # Arise at Day 3
sig_d3_arise &lt;- sig_d3_only
length(sig_d3_arise)</code></pre>
<pre><code>## [1] 966</code></pre>
<pre class="r"><code>collect_arise &lt;- c(length(sig_d0_arise), length(sig_d1_arise), length(sig_d2_arise), length(sig_d3_arise))

plot(collect_arise, xaxt=&quot;n&quot;, ylab=&quot;Number of DE genes arising&quot;, main=&quot; Significant Genes Arising At Each Day&quot;, ylim = c(0, 5000), pch = 19)
axis(side=1, at=1:4, labels=c(&quot;Day 0&quot;, &quot;Day 1&quot;, &quot;Day 2&quot;, &quot;Day 3&quot;))</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-34-1.png" width="672" /></p>
<div id="plot-distributions-of-signficant-de-genes-between-species-at-each-day-line-946" class="section level3">
<h3>Plot distributions of signficant DE genes between species at each day (line 946)</h3>
<pre class="r"><code># Here is the function to find the normalized cpm of a subset of genes
subset_func &lt;- function(subset_genes){
  
  inshared_lists = row.names(cpm_in_cutoff) %in% subset_genes
  inshared_lists_data &lt;- as.data.frame(inshared_lists)
  cpm_genes_in &lt;- cbind(cpm_in_cutoff, inshared_lists_data)
  cpm_genes_in_sig &lt;- subset(cpm_genes_in, inshared_lists_data == &quot;TRUE&quot;)

  return(cpm_genes_in_sig[,1:63])
}

# Find the normalized cpm of all of the significant genes

  # Significant at day 0 
cpm_sig0 &lt;- subset_func(sig_d0)
dim(cpm_sig0)</code></pre>
<pre><code>## [1] 4473   63</code></pre>
<pre class="r"><code>### TO DO: PLOT EFFECT SIZE OF THESE


  # Significant at day 1

cpm_sig1 &lt;- subset_func(sig_d1)
dim(cpm_sig1)</code></pre>
<pre><code>## [1] 4380   63</code></pre>
<pre class="r"><code>  # Significant at day 2

cpm_sig2 &lt;- subset_func(sig_d2)
dim(cpm_sig2)</code></pre>
<pre><code>## [1] 4597   63</code></pre>
<pre class="r"><code>  # Significant at day 3

cpm_sig3 &lt;- subset_func(sig_d3)
dim(cpm_sig3)</code></pre>
<pre><code>## [1] 4982   63</code></pre>
</div>
</div>
<div id="plot-distributions-of-non-signficant-de-genes-between-species-at-each-day" class="section level2">
<h2>Plot distributions of non-signficant DE genes between species at each day</h2>
<pre class="r"><code>###Plot nonsignificant genes###

cpm_notsig_any &lt;- subset_func(notsig_any)
dim(notsig_any)</code></pre>
<pre><code>## [1] 2880    1</code></pre>
</div>
</div>
<div id="expression-values-for-individual-genes" class="section level1">
<h1>Expression values for individual genes</h1>
<pre class="r"><code># Get day and species information
day &lt;- as.data.frame(day)
Species &lt;- as.data.frame(After_removal_sample_info$Species)

# Trial

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000000003&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;ENSG00000000003 Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
<pre class="r"><code># EOMES (ENSG00000163508)

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000163508&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;EOMES Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;) +  theme_bw() </code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-2.png" width="672" /></p>
<pre class="r"><code># HHEX (ENSG00000152804)

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000152804&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;HHEX Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-3.png" width="672" /></p>
<pre class="r"><code># CXCR4 (ENSG00000121966)

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000121966&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;CXCR4 Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;) + theme_bw()</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-4.png" width="672" /></p>
<pre class="r"><code># Potentially not needed: OTX2 (ENSG00000165588)

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000165588&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;OTX2 Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-5.png" width="672" /></p>
<pre class="r"><code># FOXH1 (ENSG00000160973)

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000160973&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;FOXH1 Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-6.png" width="672" /></p>
<pre class="r"><code># Brian suggested using BRACHYURY (ENSG00000164458) instead of FOXH1

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000164458&quot;, rownames(cpm_in_cutoff)), ])

make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;BRACHYURY Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-7.png" width="672" /></p>
<pre class="r"><code># WNT3 (ENSG00000108379)

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000108379&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;WNT3A Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-8.png" width="672" /></p>
<pre class="r"><code># ISOC1 (ENSG00000066583)

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000066583&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;ISOC1 Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-9.png" width="672" /></p>
<pre class="r"><code># TLE4 (ENSG00000106829)

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000106829&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;TLE4 Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-10.png" width="672" /></p>
<pre class="r"><code># APOE

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000130203&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;ApoE Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-11.png" width="672" /></p>
<pre class="r"><code># TMEM65 (ENSG00000164983)

exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000164983&quot;, rownames(cpm_in_cutoff)), ])


make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species)) + ggtitle(&quot;TMEM65 Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-12.png" width="672" /></p>
<pre class="r"><code># SOX 2
exp &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000181449&quot;, rownames(cpm_in_cutoff)), ])
make_exp_df &lt;- as.data.frame(cbind(exp, day, species))
colnames(make_exp_df) &lt;- c(&quot;CPM&quot;, &quot;Day&quot;, &quot;Species&quot;)
ggplot(make_exp_df, aes(x=as.factor(Day), y=CPM, fill=Species)) + geom_boxplot(aes(fill=Species))  + ggtitle(&quot;SOX2 Expression&quot;) + xlab(&quot;Day&quot;) + ylab(&quot;Log2(CPM) expression&quot;)</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-37-13.png" width="672" /></p>
<div id="make-heatmap-of-all-the-markers" class="section level4">
<h4>Make heatmap of all the markers</h4>
<pre class="r"><code># PRDM1 ENSG00000057657

exp_PRDM1 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000057657&quot;, rownames(cpm_in_cutoff)), ])

# HHEX (ENSG00000152804)

exp_HHEX &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000152804&quot;, rownames(cpm_in_cutoff)), ])

# CXCR4 (ENSG00000121966)

exp_CXCR4 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000121966&quot;, rownames(cpm_in_cutoff)), ])

# SOX17 (ENSG00000164736)

exp_SOX17 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000164736&quot;, rownames(cpm_in_cutoff)), ])

# GSC (ENSG00000133937)

exp_GSC &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000133937&quot;, rownames(cpm_in_cutoff)), ])

# OTX2 (ENSG00000165588)

exp_OTX2 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000165588&quot;, rownames(cpm_in_cutoff)), ])


# EOMES (ENSG00000163508)

exp_EOMES &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000163508&quot;, rownames(cpm_in_cutoff)), ])


# SOX2 (ENSG00000181449)

exp_SOX2 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000181449&quot;, rownames(cpm_in_cutoff)), ])

# PRDM14 (ENSG00000147596)

exp_PRDM14 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000147596&quot;, rownames(cpm_in_cutoff)), ])


# WNT3 (ENSG00000108379)

exp_WNT3 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000108379&quot;, rownames(cpm_in_cutoff)), ])

# FGF8 (ENSG00000107831)

exp_FGF8 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000107831&quot;, rownames(cpm_in_cutoff)), ])


# PDGFRA (ENSG00000134853)

exp_PDGFRA &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000134853&quot;, rownames(cpm_in_cutoff)), ])

# FOXA1 (ENSG00000129514)
  # This is a lowly expressed gene and therefore has been filtered

exp_FOXA1 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000129514&quot;, rownames(cpm_in_cutoff)), ])

# FOXA2 (ENSG00000125798)

exp_FOXA2 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000125798&quot;, rownames(cpm_in_cutoff)), ])

# BRACHYURY (ENSG00000164458)

exp_BRACHYURY &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000164458&quot;, rownames(cpm_in_cutoff)), ])

# FGF4 (ENSG00000075388)
  # This is a lowly expressed gene and therefore has been filtered

exp_FGF4 &lt;- as.data.frame(cpm_in_cutoff[grepl(&quot;ENSG00000075388&quot;, rownames(cpm_in_cutoff)), ])

# Combine expression values for the different markers

TF_heatmap &lt;- cbind(exp_PRDM1, exp_HHEX, exp_CXCR4, exp_SOX17, exp_GSC, exp_OTX2, exp_EOMES, exp_SOX2, exp_PRDM14, exp_WNT3, exp_FGF8, exp_PDGFRA, exp_FOXA2, exp_BRACHYURY)
colnames(TF_heatmap) &lt;- c(&quot;PRDM1&quot;, &quot;HHEX&quot;, &quot;CXCR4&quot;, &quot;SOX17&quot;, &quot;GSC&quot;, &quot;OTX2&quot;, &quot;EOMES&quot;, &quot;SOX2&quot;, &quot;PRDM14&quot;, &quot;WNT3&quot;, &quot;FGF8&quot;, &quot;PDGFRA&quot;, &quot;FOXA2&quot;, &quot;BRACHYURY&quot;)

dim(TF_heatmap)  </code></pre>
<pre><code>## [1] 63 14</code></pre>
<pre class="r"><code>View(TF_heatmap)

# Find the correlations and then plot the results
Species &lt;- After_removal_sample_info$Species
#Day &lt;- After_removal_sample_info$Day

# Get a library
library(&quot;colorspace&quot;)</code></pre>
<pre><code>## Warning: package &#39;colorspace&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code>library(&quot;magrittr&quot;)</code></pre>
<pre><code>## 
## Attaching package: &#39;magrittr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:R.utils&#39;:
## 
##     extract</code></pre>
<pre><code>## The following object is masked from &#39;package:R.oo&#39;:
## 
##     equals</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     extract</code></pre>
<pre class="r"><code>library(&quot;dendextend&quot;)</code></pre>
<pre><code>## Warning: package &#39;dendextend&#39; was built under R version 3.2.5</code></pre>
<pre><code>## Warning: replacing previous import by &#39;magrittr::%&gt;%&#39; when loading
## &#39;dendextend&#39;</code></pre>
<pre><code>## 
## ---------------------
## Welcome to dendextend version 1.3.0
## Type browseVignettes(package = &#39;dendextend&#39;) for the package vignette.
## The github page is: https://github.com/talgalili/dendextend/
## 
## Suggestions and bug-reports can be submitted at: https://github.com/talgalili/dendextend/issues
## Or contact: &lt;tal.galili@gmail.com&gt;
## 
##  To suppress this message use:  suppressPackageStartupMessages(library(dendextend))
## ---------------------</code></pre>
<pre><code>## 
## Attaching package: &#39;dendextend&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:VennDiagram&#39;:
## 
##     rotate</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     cutree</code></pre>
<pre class="r"><code>#dend_r &lt;- cor(t(TF_heatmap)) %&gt;% dist(method = &quot;euclidean&quot;) %&gt;% #hclust(method = &quot;average&quot;) 
#dend1 &lt;- as.dendrogram(dend_r)
#plot(dend1)


#dend_c &lt;- cor(TF_heatmap) %&gt;% dist(method = &quot;euclidean&quot;) %&gt;% hclust(method = &quot;average&quot;)
#dend2 &lt;- as.dendrogram(dend_c)
#plot(dend2)


gplots::heatmap.2(x=as.matrix(t(TF_heatmap)),
#                  , distfun = dist(x, method = &quot;euclidean&quot;), 
                  hclustfun = function(x) hclust(dist(x), method = &quot;average&quot;), tracecol=NA, col=colors, denscol=&quot;white&quot;, labCol= labels, ColSideColors=pal[as.integer(as.factor(Species))]) #RowSideColors=pal[as.integer(as.factor(Day))+9])</code></pre>
<p><img src="Endoderm_original_analysis_files/figure-html/unnamed-chunk-38-1.png" width="672" /></p>
</div>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
