---
title: "Use PCA to visualize single-cell data"
author: "Joyce Hsiao"
date: 2015-12-18
---

**Last updated:** `r Sys.Date()`

**Code version:** `r system("git log -1 --format='%H'", intern = TRUE)`

```{r chunk-options, include=FALSE}
source("chunk-options.R")
```

## Objectives

I imported a subset of the [singleCellSeq][] data and performed a principal component analysis to explore visually the similarity between batches in transcriptional expression profile. The data set includes 225 induced pluripotent stem cells (iPSCs) derived from a Yoruba cell line (NA19239) and a subsample of 1,000 genes). The cells were collected from the same cell culture and processed in three different batches. 

I used the PCA functions created by [John Blischak][JB] (*ashlar/codes*) to inspect the first two major sources of variation in expression profiles. I was interested in learning: Do cells collected in different batches share similar variation? A batch  


[singleCellSeq]: http://jdblischak.github.io/singleCellSeq/analysis
[JB]: http://jdblischak.github.io



## Prepare data

Import gene by sample (cell) matrix. Each column consists of molecule counts of a single cell. 

```{r}
molecules_filter_subsample <- read.table("../data/molecules-filter-subsample.txt",
                                         header = TRUE, 
                                         stringsAsFactors = FALSE)
dim(molecules_filter_subsample)
```

Import metadata information. This annotates for every cell the individual (cell line) labels, replicate labels, etc. 

```{r}
annotation_filter_subsample <- read.table("../data/annotation-filter-subsample.txt",
                                          header = TRUE,
                                          stringsAsFactors = FALSE)
head(annotation_filter_subsample)
dim(annotation_filter_subsample)
```


## Making PCA plots

Source PCA functions.

```{r}
source("../code/functions.R")
```

Run PCA.

```{r}
pca_results <- run_pca(as.matrix(molecules_filter_subsample))
```

Visualize PCA results.

```{r}
plot_pca(pca_results$PCs,
         explained = pca_results$explained,
         metadata = annotation_filter_subsample,
         color = "batch") +
  labs(title = "Single-cell transcript counts")
```



## Session information

```{r info}
sessionInfo()
```
