<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />

<meta name="date" content="2017-01-12" />

<title>Endoderm_TC_Technical_Factor_Analysis</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Endoderm_TimeCourse</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Endoderm_TC">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Endoderm_TC_Technical_Factor_Analysis</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>January 12, 2017</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#part-one-see-if-any-of-the-variables-for-rna-seq-correlate-with-the-expression-pcs-for-genes-63-samples">PART ONE: See if any of the variables for RNA-Seq correlate with the expression PCs for genes (63 samples)</a><ul>
<li><a href="#initialization">Initialization</a></li>
<li><a href="#obtain-gene-expression-pcs">Obtain gene expression PCs</a></li>
<li><a href="#plots-for-pcs-versus-technical-factors">Plots for PCs versus technical factors</a></li>
<li><a href="#testing-association-between-a-particular-variable-and-pcs-with-a-linear-model">Testing association between a particular variable and PCs with a linear model</a></li>
<li><a href="#test-for-potential-violations-of-the-assumptions-of-the-linear-model">Test for potential violations of the assumptions of the linear model</a></li>
<li><a href="#find-which-factors-are-statistically-significant">Find which factors are statistically significant</a></li>
</ul></li>
<li><a href="#part-two-for-the-variables-that-correlate-see-if-these-segregate-with-either-species-or-tissue">PART TWO: For the variable(s) that correlate, see if these segregate with either species or tissue</a><ul>
<li><a href="#testing-to-see-if-differences-across-variable-groups-are-statistically-significant-for-day.">Testing to see if differences across variable groups are statistically significant for day.</a></li>
<li><a href="#testing-to-see-if-differences-across-variable-groups-are-statistically-significant-for-species.">Testing to see if differences across variable groups are statistically significant for species.</a></li>
<li><a href="#find-factors-with-fdr-0.1">Find factors with FDR &lt; 0.1</a><ul>
<li><a href="#combine-and-test-day-versus-adapter-for-each-sequencing-pool">Combine and test day versus adapter for each sequencing pool</a></li>
</ul></li>
<li><a href="#part-three-which-variables-to-put-in-the-model">PART THREE: Which variables to put in the model?</a><ul>
<li><a href="#check-correlation-between-harvest-density-and-max.-purity">Check correlation between harvest density and max. purity</a></li>
<li><a href="#rin-score">RIN Score</a></li>
<li><a href="#harvest-time-and-harvest-densities-supplement">Harvest time and harvest densities (supplement)</a></li>
</ul></li>
</ul></li>
<li><a href="#part-one-see-if-any-of-the-variables-for-rna-seq-correlate-with-the-expression-pcs-for-genes-40-samples">PART ONE: See if any of the variables for RNA-Seq correlate with the expression PCs for genes (40 samples)</a><ul>
<li><a href="#initialization-1">Initialization</a></li>
<li><a href="#obtain-gene-expression-pcs-40-samples">Obtain gene expression PCs (40 samples)</a></li>
<li><a href="#testing-association-between-a-particular-variable-and-pcs-with-a-linear-model-40-samples">Testing association between a particular variable and PCs with a linear model (40 samples)</a></li>
</ul></li>
</ul>
</div>

<p>The goal of this is to establish which, if any, technical factors are correlated with our biological variables of interest.</p>
<div id="part-one-see-if-any-of-the-variables-for-rna-seq-correlate-with-the-expression-pcs-for-genes-63-samples" class="section level1">
<h1>PART ONE: See if any of the variables for RNA-Seq correlate with the expression PCs for genes (63 samples)</h1>
<div id="initialization" class="section level2">
<h2>Initialization</h2>
<pre class="r"><code># Load libraries

library(&quot;gdata&quot;)</code></pre>
<pre><code>## gdata: read.xls support for &#39;XLS&#39; (Excel 97-2004) files ENABLED.</code></pre>
<pre><code>## </code></pre>
<pre><code>## gdata: read.xls support for &#39;XLSX&#39; (Excel 2007+) files ENABLED.</code></pre>
<pre><code>## 
## Attaching package: &#39;gdata&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     nobs</code></pre>
<pre><code>## The following object is masked from &#39;package:utils&#39;:
## 
##     object.size</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)</code></pre>
<pre><code>## Warning: package &#39;ggplot2&#39; was built under R version 3.2.4</code></pre>
<pre class="r"><code>library(&quot;qvalue&quot;)</code></pre>
<pre><code>## Warning: package &#39;qvalue&#39; was built under R version 3.2.3</code></pre>
<pre class="r"><code>library(&quot;glmnet&quot;)</code></pre>
<pre><code>## Warning: package &#39;glmnet&#39; was built under R version 3.2.4</code></pre>
<pre><code>## Loading required package: Matrix</code></pre>
<pre><code>## Warning: package &#39;Matrix&#39; was built under R version 3.2.5</code></pre>
<pre><code>## Loading required package: foreach</code></pre>
<pre><code>## Loaded glmnet 2.0-5</code></pre>
<pre class="r"><code>source(&quot;~/Desktop/Endoderm_TC/ashlar-trial/analysis/chunk-options.R&quot;)</code></pre>
<pre><code>## Warning: package &#39;knitr&#39; was built under R version 3.2.5</code></pre>
<pre class="r"><code># Load cpm data 

cpm_in_cutoff &lt;- read.delim(&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/cpm_cyclicloess.txt&quot;)

# Load sample information

After_removal_sample_info &lt;- read.csv(&quot;~/Desktop/Endoderm_TC/After_removal_sample_info.csv&quot;)

Species &lt;- After_removal_sample_info$Species
species &lt;- After_removal_sample_info$Species
day &lt;- After_removal_sample_info$Day
individual &lt;- After_removal_sample_info$Individual
Sample_ID &lt;- After_removal_sample_info$Sample_ID
labels &lt;- paste(Sample_ID, day, sep=&quot; &quot;)

# Load technical factor information

RNA_seq_info_all &lt;- read.csv(&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/Endo_TC_Data_Share_Sorting.csv&quot;,  header = T)

dim(RNA_seq_info_all)</code></pre>
<pre><code>## [1] 999  43</code></pre>
<pre class="r"><code>RNA_seq_info &lt;- as.data.frame(cbind(RNA_seq_info_all[1:63, 4], RNA_seq_info_all[1:63, 3], RNA_seq_info_all[1:63, 5:27], RNA_seq_info_all[1:63, 30:35], RNA_seq_info_all[1:63, 37:43]))

# Remove day (biological variable of interest)
RNA_seq_info &lt;- RNA_seq_info[,-8]

# Remove library well (only 1/well)
RNA_seq_info &lt;- RNA_seq_info[,-19]

# Full data set
dim(RNA_seq_info)</code></pre>
<pre><code>## [1] 63 36</code></pre>
</div>
<div id="obtain-gene-expression-pcs" class="section level2">
<h2>Obtain gene expression PCs</h2>
<pre class="r"><code># PCs

pca_genes &lt;- prcomp(t(cpm_in_cutoff), scale = T, retx = TRUE, center = TRUE)

matrixpca &lt;- pca_genes$x
pc1 &lt;- matrixpca[,1]
pc2 &lt;- matrixpca[,2]
pc3 &lt;- matrixpca[,3]
pc4 &lt;- matrixpca[,4]
pc5 &lt;- matrixpca[,5]

pcs &lt;- data.frame(pc1, pc2, pc3, pc4, pc5)

summary &lt;- summary(pca_genes)</code></pre>
</div>
<div id="plots-for-pcs-versus-technical-factors" class="section level2">
<h2>Plots for PCs versus technical factors</h2>
<pre class="r"><code>#Create plots for each of the possible confounders versus PCs 1-5

pdf(&#39;~/Desktop/Endoderm_TC/ashlar-trial/data/VarVsGenePCs.pdf&#39;)

for (i in 2:length(RNA_seq_info)) { 
  par(mfrow=c(1,5))
  plot(RNA_seq_info[,i], pcs[,1],  ylab = &quot;PC1&quot;, xlab = &quot; &quot;)
  plot(RNA_seq_info[,i], pcs[,2],  ylab = &quot;PC2&quot;, xlab = &quot; &quot;)
  plot(RNA_seq_info[,i], pcs[,3],  ylab = &quot;PC3&quot;, xlab = &quot; &quot;)
  plot(RNA_seq_info[,i], pcs[,4],  ylab = &quot;PC4&quot;, xlab = &quot; &quot;)
  plot(RNA_seq_info[,i], pcs[,5],  ylab = &quot;PC5&quot;, xlab = &quot; &quot;)
  title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
  mtext(substitute(paste(&#39;PCs vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)

}
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div id="testing-association-between-a-particular-variable-and-pcs-with-a-linear-model" class="section level2">
<h2>Testing association between a particular variable and PCs with a linear model</h2>
<pre class="r"><code># TESTING BIOLOGICAL VARIABLES OF INTEREST

PC_pvalues_day = matrix(data = NA, nrow = 5, ncol = 1, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Day&quot;)))

for(i in 1:5){
# PC versus day

 checkPC1 &lt;- lm(pcs[,i] ~ as.factor(day))

#Get the summary statistics from it
  summary(checkPC1)

#Get the p-value of the F-statistic
  summary(checkPC1)$fstatistic

  fstat &lt;- as.data.frame(summary(checkPC1)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  PC_pvalues_day[i,1] &lt;- p_fstat
  
#Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC1)$r.squared

}
# PC  versus species

PC_pvalues_species = matrix(data = NA, nrow = 5, ncol = 1, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Species&quot;)))

for(i in 1:5){
# PC versus species

   checkPC1 &lt;- lm(pcs[,i] ~ as.factor(species))

#Get the summary statistics from it
  summary(checkPC1)

#Get the p-value of the F-statistic
  summary(checkPC1)$fstatistic

  fstat &lt;- as.data.frame(summary(checkPC1)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  PC_pvalues_species [i,1] &lt;- p_fstat 
  
#Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC1)$r.squared
}

# TESTING TECHNICAL VARIABLES OF INTEREST
  
#Make an empty matrix to put all of the data in
# Note: Do not include TC day, as it is a biological variable of interest

PC_pvalues = matrix(data = NA, nrow = 5, ncol = 35, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Cell line&quot;, &quot;SPSX&quot;, &quot;Batch&quot;, &quot;Passage at seed&quot;,  &quot;Start date&quot;, &quot;Density at seed&quot;, &quot;Harvest time&quot;, &quot;Harvest density&quot;, &quot;High conf purity&quot;, &quot;Max purity&quot;, &quot;RNA Extraction Date&quot;, &quot;RNA conc&quot;, &quot;RIN&quot;, &quot;260 280 RNA&quot;, &quot;260 230 RNA&quot;, &quot;DNA concentration&quot;, &quot;Library prep batch&quot;, &quot;Library concentration&quot;, &quot;uL sample&quot;, &quot;uL EB&quot;, &quot;Index sequence&quot;, &quot;Seq pool&quot;, &quot;Lane r1&quot;, &quot;Mseqs R1&quot;, &quot;Total lane reads 1&quot;, &quot;Lane prop r1&quot;, &quot;Dup r1&quot;, &quot;GC r1&quot;, &quot;Lane r2&quot;, &quot;Mseqs r2&quot;, &quot;Total lane reads r2&quot;, &quot;Lane prop r2&quot;, &quot;Dups r2&quot;, &quot;GC r2&quot;, &quot;Total reads&quot;)))

PC_r2 = matrix(data = NA, nrow = 5, ncol = 35, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Cell line&quot;, &quot;SPSX&quot;, &quot;Batch&quot;, &quot;Passage at seed&quot;,  &quot;Start date&quot;, &quot;Density at seed&quot;,   &quot;Harvest time&quot;, &quot;Harvest density&quot;, &quot;High conf purity&quot;, &quot;Max purity&quot;, &quot;RNA Extraction Date&quot;, &quot;RNA conc&quot;, &quot;RIN&quot;, &quot;260 280 RNA&quot;, &quot;260 230 RNA&quot;, &quot;DNA concentration&quot;, &quot;Library prep batch&quot;, &quot;Library concentration&quot;, &quot;uL sample&quot;, &quot;uL EB&quot;, &quot;Index sequence&quot;, &quot;Seq pool&quot;, &quot;Lane r1&quot;, &quot;Mseqs R1&quot;, &quot;Total lane reads 1&quot;, &quot;Lane prop r1&quot;, &quot;Dup r1&quot;, &quot;GC r1&quot;, &quot;Lane r2&quot;, &quot;Mseqs r2&quot;, &quot;Total lane reads r2&quot;, &quot;Lane prop r2&quot;, &quot;Dups r2&quot;, &quot;GC r2&quot;, &quot;Total reads&quot;)))



numerical_tech_factors &lt;- c(5,7, 9:11, 13:17, 19:21, 25:29, 31:36)
categorical_tech_factors &lt;- c(2:4,6, 8, 12, 18, 22:24, 30 )

#Check lm to see how well the variables containing numerical data are correlated with a PC

#For PCs 1-5
j=1
for (i in numerical_tech_factors){
  for (j in 1:5){
  
  checkPC1 &lt;- lm(pcs[,j] ~ RNA_seq_info[,i])

#Get the summary statistics from it
  summary(checkPC1)

#Get the p-value of the F-statistic
  summary(checkPC1)$fstatistic

  fstat &lt;- as.data.frame(summary(checkPC1)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  
#Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC1)$r.squared

#Put the summary statistics into the matrix w

  PC_pvalues[j, i-1] &lt;- p_fstat
  PC_r2[j, i-1] &lt;- r2_value
  
  }

}
#Check lm to see how well the variables containing ordinal data are correlated with a PC

#For PCs 1-5
j=1
for (i in categorical_tech_factors){
  for (j in 1:5){
  
  checkPC1 &lt;- lm(pcs[,j] ~ as.factor(RNA_seq_info[,i]))

#Get the summary statistics from it
  summary(checkPC1)

#Get the p-value of the F-statistic
  summary(checkPC1)$fstatistic

  fstat &lt;- as.data.frame(summary(checkPC1)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  
#Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC1)$r.squared

#Put the summary statistics into the matrix w

  PC_pvalues[j, i-1] &lt;- p_fstat
  PC_r2[j, i-1] &lt;- r2_value
  
  }

}</code></pre>
</div>
<div id="test-for-potential-violations-of-the-assumptions-of-the-linear-model" class="section level2">
<h2>Test for potential violations of the assumptions of the linear model</h2>
<p>Note: I learned in <a href="http://lauren-blake.github.io/Reg_Evo_Primates/analysis/Tech_factor_analysis1_gene_exp.html" class="uri">http://lauren-blake.github.io/Reg_Evo_Primates/analysis/Tech_factor_analysis1_gene_exp.html</a> that this doesn’t work when one or more cells in a column contains an “NA”.</p>
<pre class="r"><code>#Plot the residuals to look for violations of the assumptions of lm

#pdf(&#39;~/Desktop/Endoderm_TC/ashlar-trial/data/Residuals_vs_var_GenePCs.pdf&#39;)


#non_NA_numerical_tech_factor &lt;- c(5,8,  14, 20:22, 26:30, 32:37)
#non_NA_categorical_tech_factor &lt;- c(2:4,6, 9, 13, 18:19, 23:25, 31 )

#for (i in non_NA_numerical_tech_factor){
  
#  par(mfrow=c(1,5))
  
#  checkPC1 &lt;- lm(pcs[,1] ~ RNA_seq_info[,i])
#  plot(RNA_seq_info[,i], resid(checkPC1), ylab = &quot;Residuals (lm with PC1)&quot;, xlab = &quot; &quot;)
  
#  checkPC2 &lt;- lm(pcs[,2] ~ RNA_seq_info[,i])
#  plot(RNA_seq_info[,i], resid(checkPC2), ylab = &quot;Residuals (lm with PC2)&quot;, xlab = &quot; &quot;)
  
#  checkPC3 &lt;- lm(pcs[,3] ~ RNA_seq_info[,i])
#  plot(RNA_seq_info[,i], resid(checkPC3), ylab = &quot;Residuals (lm with PC3)&quot;, xlab = #&quot; &quot;)
  
#  checkPC4 &lt;- lm(pcs[,4] ~ RNA_seq_info[,i])
#  plot(RNA_seq_info[,i], resid(checkPC4), ylab = &quot;Residuals (lm with PC4)&quot;, xlab = &quot; &quot;)
  
#  checkPC5 &lt;- lm(pcs[,5] ~ RNA_seq_info[,i])
#  plot(RNA_seq_info[,i], resid(checkPC5), ylab = &quot;Residuals (lm with PC5)&quot;, xlab = &quot; &quot;)
  
#  title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
#  mtext(substitute(paste(&#39;Residuals vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
  
#}

#  for (i in non_NA_categorical_tech_factor){
#    par(mfrow=c(1,5))
   
#    checkPC1 &lt;- lm(pcs[,1] ~ as.factor(RNA_seq_info[,i]))
#    plot(RNA_seq_info[,i], resid(checkPC1), ylab = &quot;Residuals (lm with PC1)&quot;, xlab = &quot; &quot;)

#    checkPC2 &lt;- lm(pcs[,2] ~ as.factor(RNA_seq_info[,i]))
#    plot(RNA_seq_info[,i], resid(checkPC2),  ylab = &quot;Residuals (lm with PC2)&quot;, xlab = &quot; &quot;)
    
#    checkPC3 &lt;- lm(pcs[,3] ~ as.factor(RNA_seq_info[,i]))
#    plot(RNA_seq_info[,i], resid(checkPC3),  ylab = &quot;Residuals (lm with PC3)&quot;, xlab = &quot; &quot;)
    
#    checkPC4 &lt;- lm(pcs[,4] ~ as.factor(RNA_seq_info[,i]))
#    plot(RNA_seq_info[,i], resid(checkPC4),  ylab = &quot;Residuals (lm with PC4)&quot;, xlab = &quot; &quot;)
    
#    checkPC5 &lt;- lm(pcs[,5] ~ as.factor(RNA_seq_info[,i]))
#    plot(RNA_seq_info[,i], resid(checkPC5),  ylab = &quot;Residuals (lm with PC5)&quot;, xlab = &quot; &quot;)
    
#    title(xlab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])), outer = TRUE, line = -2)
#    mtext(substitute(paste(&#39;Residuals vs. &#39;, k), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)
    
#  }

#dev.off()</code></pre>
</div>
<div id="find-which-factors-are-statistically-significant" class="section level2">
<h2>Find which factors are statistically significant</h2>
<pre class="r"><code>#Distribution of p-values adjusted by FDR not including species or tissue

fdr_val = p.adjust(PC_pvalues, method = &quot;fdr&quot;, n = length(PC_pvalues))
fdr_val_order = fdr_val[order(fdr_val)]
hist(fdr_val_order, ylab = &quot;BH-adjusted p-values&quot;, main = &quot;Distribution of Benjamini and Hochberg adjusted p-values&quot;, breaks = 10)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Number of values significant at 10% FDR 

fdr_val &lt;- matrix(fdr_val, nrow = 5, ncol = 35)
matrix_fdr_val = matrix(data = fdr_val, nrow = 5, ncol = 35, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;),  c(&quot;Cell line&quot;, &quot;SPSX&quot;, &quot;Batch&quot;, &quot;Passage at seed&quot;,  &quot;Start date&quot;, &quot;Density at seed&quot;, &quot;Harvest time&quot;, &quot;Harvest density&quot;, &quot;High conf purity&quot;, &quot;Max purity&quot;, &quot;RNA Extraction Date&quot;, &quot;RNA conc&quot;, &quot;RIN&quot;, &quot;260 280 RNA&quot;, &quot;260 230 RNA&quot;, &quot;DNA concentration&quot;, &quot;Library prep batch&quot;, &quot;Library concentration&quot;, &quot;uL sample&quot;, &quot;uL EB&quot;, &quot;Index sequence&quot;, &quot;Seq pool&quot;, &quot;Lane r1&quot;, &quot;Mseqs R1&quot;, &quot;Total lane reads 1&quot;, &quot;Lane prop r1&quot;, &quot;Dup r1&quot;, &quot;GC r1&quot;, &quot;Lane r2&quot;, &quot;Mseqs r2&quot;, &quot;Total lane reads r2&quot;, &quot;Lane prop r2&quot;, &quot;Dups r2&quot;, &quot;GC r2&quot;, &quot;Total reads&quot;)))

# Number of values significant at 10% FDR 
sum(matrix_fdr_val &lt;= 0.1)</code></pre>
<pre><code>[1] 28</code></pre>
<pre class="r"><code>#Get the coordinates of which variables/PC combinations are significant at FDR 10%

TorF_matrix_fdr &lt;- matrix_fdr_val &lt;=0.1
coor_to_check &lt;- which(matrix_fdr_val &lt;= 0.1, arr.ind=T)
coor_to_check &lt;- as.data.frame(coor_to_check)

# Number of variables significant at 10% FDR  (note: off by 1 from column number in RNA_seq_info file; see notes in Part Two)

coor_to_check_col &lt;- coor_to_check$col
unique(coor_to_check_col)</code></pre>
<pre><code> [1]  1  2  3  4  5  7  8  9 10 11 12 13 14 17 21 22 23 25 29</code></pre>
<p>** Conclusions from Part I**</p>
<p>The following variables are associated with one of the PCs tested and will be investigated further in Part 2.</p>
<ul>
<li>Cell line</li>
<li>SPSX</li>
<li>Batch</li>
<li>Passage at seed</li>
<li>Start date</li>
<li>Harvest time</li>
<li>Harvest density</li>
<li>High conf purity</li>
<li>Max purity</li>
<li>RNA Extraction Date</li>
<li>RNA conc</li>
<li>RIN</li>
<li>260 280 RNA</li>
<li>Library prep batch</li>
<li>Index sequence</li>
<li>Seq pool</li>
<li>Lane r1</li>
<li>Total lane reads 1</li>
<li>Lane r2</li>
</ul>
</div>
</div>
<div id="part-two-for-the-variables-that-correlate-see-if-these-segregate-with-either-species-or-tissue" class="section level1">
<h1>PART TWO: For the variable(s) that correlate, see if these segregate with either species or tissue</h1>
<p>In coor_to_check_col, row is the PC # and col is the column # -1 that is associated with the PC. Want to take the coor_to_check_col column # and add one.</p>
<pre class="r"><code>var_numb = unique(coor_to_check_col) + 1
var.numb &lt;- as.data.frame(var_numb)

pdf(&#39;~/Desktop/Endoderm_TC/ashlar-trial/data/Var_GenePCsVsSpeciesTissue.pdf&#39;)

for (i in var.numb[1:nrow(var.numb),]) { 
par(mfrow=c(1,2))
plot(day, RNA_seq_info[,i], xlab = &quot;Species&quot;, ylab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])))

plot(Species, RNA_seq_info[,i], xlab = &quot;Tissue&quot;, ylab = substitute(paste(k), list(k=colnames(RNA_seq_info)[i])))

mtext(substitute(paste(k, &#39; vs. Species and Tissue&#39;), list(k=colnames(RNA_seq_info)[i])), side = 3, line = -2, outer = TRUE)

}
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<div id="testing-to-see-if-differences-across-variable-groups-are-statistically-significant-for-day." class="section level2">
<h2>Testing to see if differences across variable groups are statistically significant for day.</h2>
<pre class="r"><code># Technical factors to be tested 

numerical_prev_sign_tech_factors &lt;- c(5, 9:11, 13:15, 26)
categorical_prev_sign_tech_factors &lt;- c(2:4,6, 8, 12, 18, 22:24, 30 )

#Make a matrix to store the p-values

pvalues_day = matrix(data = NA, nrow = 1, ncol = 35, dimnames = list(c(&quot;p-value&quot;),  c(&quot;Cell line&quot;, &quot;SPSX&quot;, &quot;Batch&quot;, &quot;Passage at seed&quot;,  &quot;Start date&quot;, &quot;Density at seed&quot;, &quot;Harvest time&quot;, &quot;Harvest density&quot;, &quot;High conf purity&quot;, &quot;Max purity&quot;, &quot;RNA Extraction Date&quot;, &quot;RNA conc&quot;, &quot;RIN&quot;, &quot;260 280 RNA&quot;, &quot;260 230 RNA&quot;, &quot;DNA concentration&quot;, &quot;Library prep batch&quot;, &quot;Library concentration&quot;, &quot;uL sample&quot;, &quot;uL EB&quot;, &quot;Index sequence&quot;, &quot;Seq pool&quot;, &quot;Lane r1&quot;, &quot;Mseqs R1&quot;, &quot;Total lane reads 1&quot;, &quot;Lane prop r1&quot;, &quot;Dup r1&quot;, &quot;GC r1&quot;, &quot;Lane r2&quot;, &quot;Mseqs r2&quot;, &quot;Total lane reads r2&quot;, &quot;Lane prop r2&quot;, &quot;Dups r2&quot;, &quot;GC r2&quot;, &quot;Total reads&quot;)))

# Numerical

#Performing this test of significance for variables that are numerical data (Using an ANOVA)
j=1
for (i in numerical_prev_sign_tech_factors) {  
summary_anova = summary(aov(RNA_seq_info[,i]~ as.factor(day)))
data_summary_anova &lt;- as.data.frame(summary_anova[[1]]$`Pr(&gt;F)`)
p_val_anova &lt;- data_summary_anova[1,]
pvalues_day[, i-1] &lt;- p_val_anova  
j=j+1
}

# Ordinal

#Performing this test of significance for variables that are categorical data (Using Pearson&#39;s chi-squared test)
j=1
for (i in categorical_prev_sign_tech_factors) { 
  
  pval_chi &lt;- chisq.test(as.factor(RNA_seq_info[,i]), as.factor(day), simulate.p.value = TRUE)$p.value
  pvalues_day[, i-1] &lt;- pval_chi  
  j=j+1
  
}</code></pre>
</div>
<div id="testing-to-see-if-differences-across-variable-groups-are-statistically-significant-for-species." class="section level2">
<h2>Testing to see if differences across variable groups are statistically significant for species.</h2>
<pre class="r"><code># Technical factors to be tested 

numerical_prev_sign_tech_factors &lt;- c(5, 9:11, 13:15, 26)
categorical_prev_sign_tech_factors &lt;- c(2:4,6, 8, 12, 18, 22:24, 30 )

#Make a matrix to store the p-values

pvalues_species = matrix(data = NA, nrow = 1, ncol = 35, dimnames = list(c(&quot;p-value&quot;),  c(&quot;Cell line&quot;, &quot;SPSX&quot;, &quot;Batch&quot;, &quot;Passage at seed&quot;,  &quot;Start date&quot;, &quot;Density at seed&quot;, &quot;Harvest time&quot;, &quot;Harvest density&quot;, &quot;High conf purity&quot;, &quot;Max purity&quot;, &quot;RNA Extraction Date&quot;, &quot;RNA conc&quot;, &quot;RIN&quot;, &quot;260 280 RNA&quot;, &quot;260 230 RNA&quot;, &quot;DNA concentration&quot;, &quot;Library prep batch&quot;, &quot;Library concentration&quot;, &quot;uL sample&quot;, &quot;uL EB&quot;, &quot;Index sequence&quot;, &quot;Seq pool&quot;, &quot;Lane r1&quot;, &quot;Mseqs R1&quot;, &quot;Total lane reads 1&quot;, &quot;Lane prop r1&quot;, &quot;Dup r1&quot;, &quot;GC r1&quot;, &quot;Lane r2&quot;, &quot;Mseqs r2&quot;, &quot;Total lane reads r2&quot;, &quot;Lane prop r2&quot;, &quot;Dups r2&quot;, &quot;GC r2&quot;, &quot;Total reads&quot;)))

# Numerical

#Performing this test of significance for variables that are numerical data (Using an ANOVA. Note: in this case, the p-value of the ANOVA is the same as the p-value of the beta1 coefficient in lm)
j=1
for (i in numerical_prev_sign_tech_factors) {  
summary_anova = summary(aov(RNA_seq_info[,i]~ as.factor(Species)))
data_summary_anova &lt;- as.data.frame(summary_anova[[1]]$`Pr(&gt;F)`)
p_val_anova &lt;- data_summary_anova[1,]
pvalues_species[, i-1] &lt;- p_val_anova  
j=j+1
}


# Ordinal

#Performing this test of significance for variables that are categorical data (Using Pearson&#39;s chi-squared test)
j=1
for (i in categorical_prev_sign_tech_factors) { 
  
  pval_chi &lt;- chisq.test(as.factor(RNA_seq_info[,i]), as.factor(Species), simulate.p.value = TRUE)$p.value
  pvalues_species[, i-1] &lt;- pval_chi  
  j=j+1
  
}

# Combine tables

collapse_table_full &lt;- rbind(pvalues_day, pvalues_species)

# Get rid of the NAs

collapse_table &lt;- collapse_table_full[,(var_numb-1)]
collapse_table</code></pre>
<pre><code>           Cell line         SPSX Batch Passage at seed Start date
p-value 1.0000000000 1.0000000000     1      0.99977773          1
p-value 0.0004997501 0.0004997501     1      0.01227519          1
        Harvest time Harvest density High conf purity  Max purity
p-value 0.0004997501    0.0008736023       0.16424734 0.823375937
p-value 1.0000000000    0.0026053622       0.02830242 0.004607077
        RNA Extraction Date   RNA conc       RIN 260 280 RNA
p-value                   1 0.02126824 0.4553421  0.01963749
p-value                   1 0.09748413 0.7176148  0.44575390
        Library prep batch Index sequence Seq pool Lane r1
p-value                  1   0.0004997501        1       1
p-value                  1   0.9435282359        1       1
        Total lane reads 1 Lane r2
p-value          0.9985858       1
p-value          0.9201341       1</code></pre>
</div>
<div id="find-factors-with-fdr-0.1" class="section level2">
<h2>Find factors with FDR &lt; 0.1</h2>
<pre class="r"><code>#Calculate q-values (FDR = 10%)

fdr_val = p.adjust(collapse_table, method = &quot;fdr&quot;, n = length(collapse_table)*2)
fdr_val_order = fdr_val[order(fdr_val)]
hist(fdr_val_order, ylab = &quot;BH-adjusted p-values&quot;, main = &quot;Distribution of Benjamini and Hochberg adjusted p-values&quot;, breaks = 10)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>collapse_table_fdr_val = matrix(data = fdr_val, nrow = 2, ncol = nrow(var.numb), dimnames = list(c(&quot;Day&quot;, &quot;Species&quot;), colnames(collapse_table)))

collapse_table_fdr_val</code></pre>
<pre><code>          Cell line        SPSX Batch Passage at seed Start date
Day     1.000000000 1.000000000     1       1.0000000          1
Species 0.009495252 0.009495252     1       0.1166143          1
        Harvest time Harvest density High conf purity Max purity
Day      0.009495252      0.01327875        0.9602152 1.00000000
Species  1.000000000      0.03300125        0.1955440 0.05001969
        RNA Extraction Date  RNA conc RIN 260 280 RNA Library prep batch
Day                       1 0.1616386   1   0.1616386                  1
Species                   1 0.6173995   1   1.0000000                  1
        Index sequence Seq pool Lane r1 Total lane reads 1 Lane r2
Day        0.009495252        1       1                  1       1
Species    1.000000000        1       1                  1       1</code></pre>
<p>** Conclusions from Part 2 **</p>
<p>The following variables are confounded with day:</p>
<ul>
<li>Harvest time (BH adjusted p-value = 0.009495252)</li>
<li>Harvest density (BH adjusted p-value = 0.01327875)</li>
<li>Index sequence (BH adjusted p-value = 0.009495252)</li>
</ul>
<p>The following variables are confounded with species:</p>
<ul>
<li>Cell line (BH adjusted p-value = 0.009495252)</li>
<li>SPSX (BH adjusted p-value = 0.009495252)</li>
<li>Harvest density (BH adjusted p-value = 0.03300125)</li>
<li>Max. purity (BH adjusted p-value = 0.05001969)</li>
</ul>
<pre class="r"><code>plot(day, as.factor(RNA_seq_info[,2]), main = &quot;Cell line versus day&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(Species, as.factor(RNA_seq_info[,2]), main = &quot;Cell line versus species (sign.)&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(day, as.factor(RNA_seq_info[,3]), main = &quot;SPSX versus day&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(Species, as.factor(RNA_seq_info[,3]), main = &quot;SPSX versus species (sign.)&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(day, as.factor(RNA_seq_info[,8]), main =  &quot;Harvest time versus day (sign.)&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(Species, RNA_seq_info[,8], main = &quot;Harvest time versus species&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(as.factor(day), RNA_seq_info[,9], main = &quot;Harvest density versus day (sign.)&quot;, ylab = &quot;Harvest density&quot;, xlab = &quot;Day&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(Species, RNA_seq_info[,9], main = &quot;Harvest density versus species (sign.)&quot;, ylab = &quot;Harvest density&quot;, xlab = &quot;Species&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(as.factor(day), RNA_seq_info[,10], main = &quot;High confidence purity versus day&quot;, ylab = &quot;High confidence purity&quot;, xlab = &quot;Day&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(Species, RNA_seq_info[,10], main = &quot;High confidence purity versus species&quot;, ylab = &quot;High confidence purity&quot;, xlab = &quot;Species&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(as.factor(day), RNA_seq_info[,11], main = &quot;Max. purity versus day&quot;,  ylab = &quot;Max. purity&quot;, xlab = &quot;Day&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(Species, RNA_seq_info[,11], main = &quot;Max. purity versus species (sign.)&quot;, ylab = &quot;Max. purity&quot;, xlab = &quot;Species&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(as.factor(day), as.factor(RNA_seq_info[,22]), main = &quot;Index sequence versus day (sign.)&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(Species, RNA_seq_info[,22], main = &quot;Index sequence versus species&quot;)</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-11-14.png" width="672" style="display: block; margin: auto;" /></p>
<div id="combine-and-test-day-versus-adapter-for-each-sequencing-pool" class="section level3">
<h3>Combine and test day versus adapter for each sequencing pool</h3>
<pre class="r"><code># Make table of day, adaptor, and sequencing pool

day_index_pool &lt;- cbind(day, RNA_seq_info[,22], RNA_seq_info[,23]) 

# Test dependency with the adaptors by day 

  # Sequencing pool 1
adaptors_pool1 &lt;- day_index_pool[ which(day_index_pool[,3]==1),]

chisq.test(as.factor(adaptors_pool1[,2]), as.factor(adaptors_pool1[,1]), simulate.p.value = TRUE)$p.value</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>#fisher.test(adaptors_pool1[,1:2], simulate.p.value=TRUE)$p.value
#fisher.test(adaptors_pool2[,2:1], simulate.p.value=TRUE)$p.value
#fisher.test(adaptors_pool3[,2:1], simulate.p.value=TRUE)$p.value
#fisher.test(adaptors_pool4[,2:1], simulate.p.value=TRUE)$p.value

  # p-value = 1

  # Sequencing pool 2
adaptors_pool2 &lt;- day_index_pool[ which(day_index_pool[,3]==2),]

chisq.test(as.factor(adaptors_pool2[,2]), as.factor(adaptors_pool2[,1]), simulate.p.value = TRUE)$p.value</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>  # p-value = 1

  # Sequencing pool 3
adaptors_pool3 &lt;- day_index_pool[ which(day_index_pool[,3]==3),]

chisq.test(as.factor(adaptors_pool3[,2]), as.factor(adaptors_pool3[,1]), simulate.p.value = TRUE)$p.value</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>  # p-value = 1

  # Sequencing pool 4
adaptors_pool4 &lt;- day_index_pool[ which(day_index_pool[,3]==4),]

chisq.test(as.factor(adaptors_pool4[,2]), as.factor(adaptors_pool4[,1]), simulate.p.value = TRUE)$p.value</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>  # p-value = 1

fdr_val_4= p.adjust(c(1,1,1,1), method = &quot;fdr&quot;)</code></pre>
</div>
</div>
<div id="part-three-which-variables-to-put-in-the-model" class="section level2">
<h2>PART THREE: Which variables to put in the model?</h2>
<div id="check-correlation-between-harvest-density-and-max.-purity" class="section level3">
<h3>Check correlation between harvest density and max. purity</h3>
<p>Check for degree of similarity between the values of the different technical variables, harvest density and max. purity. For ones that show high correlation (multi-colinearity), we will only want to incorporate one of these variables into the model.</p>
<pre class="r"><code># Correlation bet. harvest density and max. purity.
  corr_tech &lt;- summary(lm(RNA_seq_info[,9] ~ RNA_seq_info[,11]))


#Get the p-value of the F-statistic
  corr_tech$fstatistic</code></pre>
<pre><code>     value      numdf      dendf 
 0.5549879  1.0000000 20.0000000 </code></pre>
<pre class="r"><code>  fstat &lt;- as.data.frame(corr_tech$fstatistic)
  1-pf(fstat[1,], fstat[2,], fstat[3,])</code></pre>
<pre><code>[1] 0.4649518</code></pre>
<p>These two factors are not highly correlated (p-value = 0.465).</p>
</div>
<div id="rin-score" class="section level3">
<h3>RIN Score</h3>
<pre class="r"><code># Make a table with species, day, and RIN score
species &lt;- After_removal_sample_info$Species
day &lt;- After_removal_sample_info$Day

iPSC_prop &lt;- as.data.frame(cbind(day, species, RNA_seq_info[,14]), stringsAsFactors = FALSE)

# Remove the 1 sample with a missing RIN score

remove_NA &lt;- c(26)

iPSC_prop &lt;- iPSC_prop[-remove_NA, ]

colnames(iPSC_prop) &lt;- c(&quot;Day&quot;, &quot;Species&quot;, &quot;RIN_Score&quot;) 

iPSC_prop$Species[iPSC_prop$Species == &quot;2&quot;] &lt;- &quot;Human&quot;
iPSC_prop$Species[iPSC_prop$Species == &quot;1&quot;] &lt;- &quot;Chimp&quot;

iPSC_prop$Day[iPSC_prop$Day == &quot;0&quot;] &lt;- &quot;Day 0&quot;
iPSC_prop$Day[iPSC_prop$Day == &quot;1&quot;] &lt;- &quot;Day 1&quot;
iPSC_prop$Day[iPSC_prop$Day == &quot;2&quot;] &lt;- &quot;Day 2&quot;
iPSC_prop$Day[iPSC_prop$Day == &quot;3&quot;] &lt;- &quot;Day 3&quot;

ggplot(data = iPSC_prop, aes(y = RIN_Score, x = as.factor(Species)))  + facet_wrap(~ Day, nrow = 1) + geom_boxplot() + geom_point(size = 5,  position=position_jitter(width=0.2, height=0.1), show.legend = FALSE) + labs(x = &quot;Species&quot;, y = &quot;RIN score&quot;) + theme_bw()</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data = iPSC_prop, aes(y = RIN_Score, x = as.factor(Species)))  + facet_wrap(~ Day, nrow = 1) + geom_boxplot()  + geom_dotplot(binaxis=&#39;y&#39;, stackdir=&#39;center&#39;, binwidth = 0.05) + labs(x = &quot;Species&quot;, y = &quot;RIN score&quot;) + theme_bw()</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="harvest-time-and-harvest-densities-supplement" class="section level3">
<h3>Harvest time and harvest densities (supplement)</h3>
<pre class="r"><code>iPSC_prop &lt;- as.data.frame(cbind(day, species, RNA_seq_info[,9],  as.character(RNA_seq_info[,8])), stringsAsFactors = FALSE)

  
colnames(iPSC_prop) &lt;- c(&quot;Day&quot;, &quot;Species&quot;, &quot;Harvest_density&quot;, &quot;Harvest_time&quot;) 

iPSC_prop$Species[iPSC_prop$Species == &quot;2&quot;] &lt;- &quot;Human&quot;
iPSC_prop$Species[iPSC_prop$Species == &quot;1&quot;] &lt;- &quot;Chimp&quot;

iPSC_prop$Day[iPSC_prop$Day == &quot;0&quot;] &lt;- &quot;Day 0&quot;
iPSC_prop$Day[iPSC_prop$Day == &quot;1&quot;] &lt;- &quot;Day 1&quot;
iPSC_prop$Day[iPSC_prop$Day == &quot;2&quot;] &lt;- &quot;Day 2&quot;
iPSC_prop$Day[iPSC_prop$Day == &quot;3&quot;] &lt;- &quot;Day 3&quot;

# Time is 7

remove_NA &lt;- c(7,23,39,55)

iPSC_prop &lt;- iPSC_prop[-remove_NA, ]

iPSC_prop$Harvest_density &lt;- as.numeric(iPSC_prop$Harvest_density)

ggplot(data = iPSC_prop, aes(y = Harvest_density, x = as.factor(Day)))  + facet_wrap(~ Species, nrow = 1) + geom_boxplot() + geom_point(aes(color = as.factor(Species)), size = 5,  position=position_jitter(width=0.2, height=0.1), show.legend = FALSE)</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data = iPSC_prop, aes(y = Harvest_density, x = as.factor(Species)))  + facet_wrap(~ Day, nrow = 1) + geom_boxplot() </code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite values (stat_boxplot).</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Harvest density (supplement)

ggplot(data = iPSC_prop, aes(y = Harvest_density, x = Species))  + facet_wrap(~ Day, nrow = 1) + geom_boxplot() + geom_dotplot(binaxis=&#39;y&#39;, stackdir=&#39;center&#39;) + labs(y = &quot;Harvest density&quot;, title = &quot;Harvest densities for each sample&quot;) + theme_bw()</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite values (stat_boxplot).</code></pre>
<pre><code>`stat_bindot()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite values (stat_bindot).</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(data = iPSC_prop, aes(y = Harvest_density, x = Species))  + facet_wrap(~ Day, nrow = 1) + geom_boxplot() + geom_point(size = 5,  position=position_jitter(width=0.2, height=0.1), show.legend = FALSE) + labs(y = &quot;Harvest density&quot;, title = &quot;Harvest densities for each sample&quot;) + theme_bw()</code></pre>
<pre><code>Warning: Removed 2 rows containing non-finite values (stat_boxplot).

Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-15-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Harvest density for each day species

median(iPSC_prop[which(iPSC_prop$Day == &quot;Day 0&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 3])</code></pre>
<pre><code>[1] NA</code></pre>
<pre class="r"><code>median(iPSC_prop[which(iPSC_prop$Day == &quot;Day 0&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 3])</code></pre>
<pre><code>[1] NA</code></pre>
<pre class="r"><code>median(iPSC_prop[which(iPSC_prop$Day == &quot;Day 1&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 3])</code></pre>
<pre><code>[1] NA</code></pre>
<pre class="r"><code>median(iPSC_prop[which(iPSC_prop$Day == &quot;Day 1&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 3])</code></pre>
<pre><code>[1] NA</code></pre>
<pre class="r"><code>median(iPSC_prop[which(iPSC_prop$Day == &quot;Day 2&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 3])</code></pre>
<pre><code>[1] NA</code></pre>
<pre class="r"><code>median(iPSC_prop[which(iPSC_prop$Day == &quot;Day 2&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 3])</code></pre>
<pre><code>[1] NA</code></pre>
<pre class="r"><code>median(iPSC_prop[which(iPSC_prop$Day == &quot;Day 3&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 3])</code></pre>
<pre><code>[1] NA</code></pre>
<pre class="r"><code>median(iPSC_prop[which(iPSC_prop$Day == &quot;Day 3&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 3])</code></pre>
<pre><code>[1] NA</code></pre>
<pre class="r"><code># Harvest time

species &lt;- After_removal_sample_info$Species
day &lt;- After_removal_sample_info$Day

iPSC_prop &lt;- as.data.frame(cbind(day, species, RNA_seq_info[,9],  as.character(RNA_seq_info[,8])), stringsAsFactors = FALSE)

colnames(iPSC_prop) &lt;- c(&quot;Day&quot;, &quot;Species&quot;, &quot;Harvest_density&quot;, &quot;Harvest_time&quot;) 

iPSC_prop$Species[iPSC_prop$Species == &quot;2&quot;] &lt;- &quot;Human&quot;
iPSC_prop$Species[iPSC_prop$Species == &quot;1&quot;] &lt;- &quot;Chimp&quot;

iPSC_prop$Day[iPSC_prop$Day == &quot;0&quot;] &lt;- &quot;Day 0&quot;
iPSC_prop$Day[iPSC_prop$Day == &quot;1&quot;] &lt;- &quot;Day 1&quot;
iPSC_prop$Day[iPSC_prop$Day == &quot;2&quot;] &lt;- &quot;Day 2&quot;
iPSC_prop$Day[iPSC_prop$Day == &quot;3&quot;] &lt;- &quot;Day 3&quot;

table(iPSC_prop$Day, iPSC_prop$Harvest_time)</code></pre>
<pre><code>       
        1:00 PM 1:05 PM 1:06 PM 1:08 PM 1:17 PM 1:20 PM 1:30 PM 11:56
  Day 0       0       0       0       0       0       0       7     0
  Day 1       0       4       0       0       0       3       1     0
  Day 2       4       0       3       0       0       1       0     0
  Day 3       0       0       1       4       3       0       0     7
       
        12:10 12:18 12:30
  Day 0     0     8     0
  Day 1     3     0     5
  Day 2     8     0     0
  Day 3     1     0     0</code></pre>
<pre class="r"><code>ggplot(data = iPSC_prop, aes(y = as.factor(Harvest_time), x = Species))  + facet_wrap(~ Day, nrow = 1)  + geom_point(aes(color = Species), size = 5,  position=position_jitter(width=0.2, height=0.1), show.legend = FALSE) + theme_bw()</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-15-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>iPSC_prop[which(iPSC_prop$Day == &quot;Day 0&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 4]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>d0c &lt;- table(factor(iPSC_prop[which(iPSC_prop$Day == &quot;Day 0&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 4], levels = c(&quot;11:56&quot;, &quot;12:10&quot;, &quot;12:18&quot;, &quot;12:30&quot;, &quot;1:00 PM&quot;, &quot;1:05 PM&quot;, &quot;1:06 PM&quot;, &quot;1:08 PM&quot;, &quot;1:17 PM&quot;, &quot;1:20 PM&quot;, &quot;1:30 PM&quot;)))

iPSC_prop[which(iPSC_prop$Day == &quot;Day 0&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 4]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>d0h &lt;- table(factor(iPSC_prop[which(iPSC_prop$Day == &quot;Day 0&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 4], levels = c(&quot;11:56&quot;, &quot;12:10&quot;, &quot;12:18&quot;, &quot;12:30&quot;, &quot;1:00 PM&quot;, &quot;1:05 PM&quot;, &quot;1:06 PM&quot;, &quot;1:08 PM&quot;, &quot;1:17 PM&quot;, &quot;1:20 PM&quot;, &quot;1:30 PM&quot;)))

iPSC_prop[which(iPSC_prop$Day == &quot;Day 1&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 4]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>d1c &lt;- table(factor(iPSC_prop[which(iPSC_prop$Day == &quot;Day 1&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 4], levels = c(&quot;11:56&quot;, &quot;12:10&quot;, &quot;12:18&quot;, &quot;12:30&quot;, &quot;1:00 PM&quot;, &quot;1:05 PM&quot;, &quot;1:06 PM&quot;, &quot;1:08 PM&quot;, &quot;1:17 PM&quot;, &quot;1:20 PM&quot;, &quot;1:30 PM&quot;)))

iPSC_prop[which(iPSC_prop$Day == &quot;Day 1&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 4]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>d1h &lt;- table(factor(iPSC_prop[which(iPSC_prop$Day == &quot;Day 1&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 4], levels = c(&quot;11:56&quot;, &quot;12:10&quot;, &quot;12:18&quot;, &quot;12:30&quot;, &quot;1:00 PM&quot;, &quot;1:05 PM&quot;, &quot;1:06 PM&quot;, &quot;1:08 PM&quot;, &quot;1:17 PM&quot;, &quot;1:20 PM&quot;, &quot;1:30 PM&quot;)))

iPSC_prop[which(iPSC_prop$Day == &quot;Day 2&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 4]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>d2c &lt;- table(factor(iPSC_prop[which(iPSC_prop$Day == &quot;Day 2&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 4], levels = c(&quot;11:56&quot;, &quot;12:10&quot;, &quot;12:18&quot;, &quot;12:30&quot;, &quot;1:00 PM&quot;, &quot;1:05 PM&quot;, &quot;1:06 PM&quot;, &quot;1:08 PM&quot;, &quot;1:17 PM&quot;, &quot;1:20 PM&quot;, &quot;1:30 PM&quot;)))

iPSC_prop[which(iPSC_prop$Day == &quot;Day 2&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 4]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>d2h &lt;- table(factor(iPSC_prop[which(iPSC_prop$Day == &quot;Day 2&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 4], levels = c(&quot;11:56&quot;, &quot;12:10&quot;, &quot;12:18&quot;, &quot;12:30&quot;, &quot;1:00 PM&quot;, &quot;1:05 PM&quot;, &quot;1:06 PM&quot;, &quot;1:08 PM&quot;, &quot;1:17 PM&quot;, &quot;1:20 PM&quot;, &quot;1:30 PM&quot;)))

iPSC_prop[which(iPSC_prop$Day == &quot;Day 3&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 4]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>d3c &lt;- table(factor(iPSC_prop[which(iPSC_prop$Day == &quot;Day 3&quot; &amp; iPSC_prop$Species == &quot;chimp&quot;) , 4], levels = c(&quot;11:56&quot;, &quot;12:10&quot;, &quot;12:18&quot;, &quot;12:30&quot;, &quot;1:00 PM&quot;, &quot;1:05 PM&quot;, &quot;1:06 PM&quot;, &quot;1:08 PM&quot;, &quot;1:17 PM&quot;, &quot;1:20 PM&quot;, &quot;1:30 PM&quot;)))

iPSC_prop[which(iPSC_prop$Day == &quot;Day 3&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 4]</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code>d3h &lt;- table(factor(iPSC_prop[which(iPSC_prop$Day == &quot;Day 3&quot; &amp; iPSC_prop$Species == &quot;human&quot;) , 4], levels = c(&quot;11:56&quot;, &quot;12:10&quot;, &quot;12:18&quot;, &quot;12:30&quot;, &quot;1:00 PM&quot;, &quot;1:05 PM&quot;, &quot;1:06 PM&quot;, &quot;1:08 PM&quot;, &quot;1:17 PM&quot;, &quot;1:20 PM&quot;, &quot;1:30 PM&quot;)))

#Check that the frequencies match the input (n = 59)

frequency_time &lt;- as.data.frame(cbind(d0c, d0h, d1c, d1h, d2c, d2h, d3c, d3h))

frequency_time</code></pre>
<pre><code>        d0c d0h d1c d1h d2c d2h d3c d3h
11:56     0   0   0   0   0   0   0   0
12:10     0   0   0   0   0   0   0   0
12:18     0   0   0   0   0   0   0   0
12:30     0   0   0   0   0   0   0   0
1:00 PM   0   0   0   0   0   0   0   0
1:05 PM   0   0   0   0   0   0   0   0
1:06 PM   0   0   0   0   0   0   0   0
1:08 PM   0   0   0   0   0   0   0   0
1:17 PM   0   0   0   0   0   0   0   0
1:20 PM   0   0   0   0   0   0   0   0
1:30 PM   0   0   0   0   0   0   0   0</code></pre>
<pre class="r"><code>plot(frequency_time[,1])</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-15-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rowSums(frequency_time)</code></pre>
<pre><code>  11:56   12:10   12:18   12:30 1:00 PM 1:05 PM 1:06 PM 1:08 PM 1:17 PM 
      0       0       0       0       0       0       0       0       0 
1:20 PM 1:30 PM 
      0       0 </code></pre>
<pre class="r"><code># Order correctly

iPSC_prop$Harvest_time[iPSC_prop$Harvest_time == &quot;11:56&quot;] &lt;- &quot;11:56 AM&quot;
iPSC_prop$Harvest_time[iPSC_prop$Harvest_time == &quot;12:10&quot;] &lt;- &quot;12:10 PM&quot;
iPSC_prop$Harvest_time[iPSC_prop$Harvest_time == &quot;12:18&quot;] &lt;- &quot;12:18 PM&quot;
iPSC_prop$Harvest_time[iPSC_prop$Harvest_time == &quot;12:30&quot;] &lt;- &quot;12:30 PM&quot;

lv &lt;- c(&quot;1:30 PM&quot;, &quot;1:20 PM&quot;, &quot;1:17 PM&quot;, &quot;1:08 PM&quot;, &quot;1:06 PM&quot;, &quot;1:05 PM&quot;, &quot;1:00 PM&quot;, &quot;12:30 PM&quot;, &quot;12:18 PM&quot;, &quot;12:10 PM&quot;, &quot;11:56 AM&quot;)
  
  
x &lt;- factor(iPSC_prop$Harvest_time,levels = lv)

# Make plot

ggplot(data = iPSC_prop, aes(y = as.factor(x), x = as.factor(Species)))  + facet_wrap(~ Day, nrow = 1)  + geom_dotplot(binaxis=&#39;y&#39;, stackdir=&#39;center&#39;, binwidth = 0.2) + labs(x = &quot;Species&quot;, y = &quot;Harvest time&quot;) + theme_bw()</code></pre>
<p><img src="figure/Technical_factor_analysis_cyclic_loess.Rmd/unnamed-chunk-15-7.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="part-one-see-if-any-of-the-variables-for-rna-seq-correlate-with-the-expression-pcs-for-genes-40-samples" class="section level1">
<h1>PART ONE: See if any of the variables for RNA-Seq correlate with the expression PCs for genes (40 samples)</h1>
<div id="initialization-1" class="section level2">
<h2>Initialization</h2>
<pre class="r"><code># Load cpm data 

cpm_in_cutoff_40 &lt;- read.delim(&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/cpm_cyclicloess_40.txt&quot;)

# Load sample information

bio_rep_samplefactors &lt;- read.delim(&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/samplefactors-filtered.txt&quot;, stringsAsFactors=FALSE)
day &lt;- bio_rep_samplefactors$Day
species &lt;- bio_rep_samplefactors$Species</code></pre>
</div>
<div id="obtain-gene-expression-pcs-40-samples" class="section level2">
<h2>Obtain gene expression PCs (40 samples)</h2>
<pre class="r"><code># PCs

pca_genes &lt;- prcomp(t(cpm_in_cutoff_40), scale = T, retx = TRUE, center = TRUE)

matrixpca &lt;- pca_genes$x
pc1 &lt;- matrixpca[,1]
pc2 &lt;- matrixpca[,2]
pc3 &lt;- matrixpca[,3]
pc4 &lt;- matrixpca[,4]
pc5 &lt;- matrixpca[,5]

pcs &lt;- data.frame(pc1, pc2, pc3, pc4, pc5)

summary &lt;- summary(pca_genes)</code></pre>
</div>
<div id="testing-association-between-a-particular-variable-and-pcs-with-a-linear-model-40-samples" class="section level2">
<h2>Testing association between a particular variable and PCs with a linear model (40 samples)</h2>
<pre class="r"><code># TESTING BIOLOGICAL VARIABLES OF INTEREST

PC_pvalues_day = matrix(data = NA, nrow = 5, ncol = 1, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Day&quot;)))

for(i in 1:5){
# PC versus day

 checkPC1 &lt;- lm(pcs[,i] ~ as.factor(day))

#Get the summary statistics from it
  summary(checkPC1)

#Get the p-value of the F-statistic
  summary(checkPC1)$fstatistic

  fstat &lt;- as.data.frame(summary(checkPC1)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  PC_pvalues_day[i,1] &lt;- p_fstat
  
#Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC1)$r.squared

}
# PC  versus species

PC_pvalues_species = matrix(data = NA, nrow = 5, ncol = 1, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Species&quot;)))

for(i in 1:5){
# PC versus species

   checkPC1 &lt;- lm(pcs[,i] ~ as.factor(species))

#Get the summary statistics from it
  summary(checkPC1)

#Get the p-value of the F-statistic
  summary(checkPC1)$fstatistic

  fstat &lt;- as.data.frame(summary(checkPC1)$fstatistic)
  p_fstat &lt;- 1-pf(fstat[1,], fstat[2,], fstat[3,])
  PC_pvalues_species [i,1] &lt;- p_fstat 
  
#Fraction of the variance explained by the model
  r2_value &lt;- summary(checkPC1)$r.squared
}

 # Combine tables

collapse_table_full &lt;- rbind(PC_pvalues_day, PC_pvalues_species)

#Calculate q-values (FDR = 10%)

fdr_val = p.adjust(collapse_table_full, method = &quot;fdr&quot;, n = length(collapse_table_full)*2)

collapse_table_fdr_val = matrix(data = fdr_val, nrow = 5, ncol = 2, dimnames = list(c(&quot;PC1&quot;, &quot;PC2&quot;, &quot;PC3&quot;, &quot;PC4&quot;, &quot;PC5&quot;), c(&quot;Day&quot;, &quot;Species&quot;)))

collapse_table_fdr_val</code></pre>
<pre><code>             Day   Species
PC1 0.000000e+00 0.4791868
PC2 1.000000e+00 0.0000000
PC3 1.494613e-08 0.8253925
PC4 1.670156e-01 0.9296140
PC5 5.072326e-01 1.0000000</code></pre>
</div>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
