<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Lauren Blake" />


<title>Explore_temporal_changes_in_variation</title>

<script src="libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<link href="libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}


/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}



/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}

</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>



<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Endoderm_TimeCourse</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Home</a></li>
        <li><a href="about.html">About</a></li>
        <li><a href="license.html">License</a></li>
        <li><a href="https://github.com/Lauren-Blake/Endoderm_TC">GitHub</a></li>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Explore_temporal_changes_in_variation</h1>
<h4 class="author"><em>Lauren Blake</em></h4>
<h4 class="date"><em>January 9, 2017</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#global-variance-trends">Global variance trends</a><ul>
<li><a href="#check-data">Check data</a></li>
<li><a href="#take-the-mean-of-technical-replicates-for-each-individual-and-visualize-the-data">Take the mean of technical replicates for each individual and visualize the data</a></li>
<li><a href="#examine-the-means-and-the-variances-for-each-day-species-pairs">Examine the means and the variances for each day-species pairs</a></li>
</ul></li>
<li><a href="#evaluating-global-changes-in-variation">Evaluating global changes in variation</a><ul>
<li><a href="#no-global-reduction-in-means-between-day-0-and-day-1-in-both-species">No global reduction in means between day 0 and day 1 in both species</a></li>
<li><a href="#the-effect-size-for-the-global-reduction-in-variance-between-day-0-and-1-in-chimpanzees-is-small-but-larger-in-humans.">The effect size for the global reduction in variance between day 0 and 1 in chimpanzees is small but larger in humans.</a></li>
</ul></li>
<li><a href="#characterizing-variance-trends-between-days-tested-at-the-gene-level-using-f-tests">Characterizing variance trends between days tested at the gene level using F tests</a><ul>
<li><a href="#general-functions-for-reductionincrease-in-variance">General functions for reduction/increase in variance</a></li>
<li><a href="#number-of-significant-genes-and-storeys-pi_0-values-for-a-reduction-in-variation-between-days-main-paper">Number of significant genes and Storey’s pi_0 values for a reduction in variation between days (main paper)</a><ul>
<li><a href="#chimpanzees">Chimpanzees</a></li>
</ul></li>
<li><a href="#number-of-significant-genes-and-storeys-pi_0-values-for-an-increase-in-variation-between-days-main-paper">Number of significant genes and Storey’s pi_0 values for an increase in variation between days (main paper)</a><ul>
<li><a href="#chimpanzees-1">Chimpanzees</a></li>
</ul></li>
<li><a href="#the-pairwise-changes-in-variance-were-robust-with-respect-to-p-value-cutoff">The pairwise changes in variance were robust with respect to p-value cutoff</a><ul>
<li><a href="#changes-in-variance-in-chimp-samples-p-value-cutoff-0.05">Changes in variance in chimp samples (p-value cutoff 0.05)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.05">Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.05)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-increase-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.05">Estimates of pi0 and pi1 for increase of variation in chimps conditioned on significance in humans (p = 0.05)</a></li>
<li><a href="#changes-of-variation-in-humans">Changes of variation in humans</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-reduction-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.05">Estimates of pi0 and pi1 for reduction of variation in humans conditioned on significance in chimps (p = 0.05)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-increase-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.05">Estimates of pi0 and pi1 for increase of variation in humans conditioned on significance in chimps (p = 0.05)</a></li>
</ul></li>
<li><a href="#permutation-tests-this-code-should-not-be-run-on-a-laptop-computer.-use-a-hpc-instead.-for-sample-results-see-supplemental-data-s15-of-the-endoderm-paper">Permutation tests (This code should not be run on a laptop computer. Use a HPC instead. For sample results, see Supplemental Data S15 of the Endoderm paper)</a><ul>
<li><a href="#permutation-test-to-find-the-null-hypothesis-of-sharing-reduction-in-chimps-significant-in-humans-for-day-0-to-1-with-p-value-cutoff-of-0.05">Permutation test to find the null hypothesis of sharing (reduction in chimps | significant in humans for day 0 to 1 with p-value cutoff of 0.05)</a></li>
<li><a href="#permutation-test-to-find-the-null-hypothesis-of-sharing-reduction-in-humans-significant-in-chimps-for-day-0-to-1-with-p-value-cutoff-of-0.05">Permutation test to find the null hypothesis of sharing (reduction in humans | significant in chimps for day 0 to 1 with p-value cutoff of 0.05)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.01">Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.01)</a></li>
<li><a href="#changes-in-variance-in-chimp-samples-p-value-cutoff-0.01">Changes in variance in chimp samples (p-value cutoff 0.01)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.01-1">Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.01)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-increase-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.01">Estimates of pi0 and pi1 for increase of variation in chimps conditioned on significance in humans (p = 0.01)</a></li>
<li><a href="#changes-of-variation-in-humans-p-0.01">Changes of variation in humans (p = 0.01)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-reduction-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.01">Estimates of pi0 and pi1 for reduction of variation in humans conditioned on significance in chimps (p = 0.01)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-increase-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.01">Estimates of pi0 and pi1 for increase of variation in humans conditioned on significance in chimps (p = 0.01)</a></li>
</ul></li>
<li><a href="#estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.1">Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.1)</a><ul>
<li><a href="#changes-in-variance-in-chimp-samples-p-value-cutoff-0.1">Changes in variance in chimp samples (p-value cutoff 0.1)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.1-1">Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.1)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-increase-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.10">Estimates of pi0 and pi1 for increase of variation in chimps conditioned on significance in humans (p = 0.10)</a></li>
<li><a href="#changes-of-variation-in-humans-p-0.10">Changes of variation in humans (p = 0.10)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-reduction-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.10">Estimates of pi0 and pi1 for reduction of variation in humans conditioned on significance in chimps (p = 0.10)</a></li>
<li><a href="#estimates-of-pi0-and-pi1-for-increase-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.10">Estimates of pi0 and pi1 for increase of variation in humans conditioned on significance in chimps (p = 0.10)</a></li>
<li><a href="#sharing-estimates-chimps-conditioned-on-humans-for-x-number-of-significant-genes-p-0.05-cutoff">Sharing estimates (chimps conditioned on humans) for x number of significant genes (p = 0.05 cutoff)</a></li>
<li><a href="#sharing-estimates-humans-conditioned-on-chimps-for-x-number-of-significant-genes">Sharing estimates (humans conditioned on chimps) for x number of significant genes</a></li>
<li><a href="#note-code-to-make-other-figures">Note: code to make other figures</a></li>
<li><a href="#gene-names-showing-reduction-increase-in-variance-from-days-0-to-1-days-1-to-2">Gene names showing reduction (increase) in variance from days 0 to 1 (days 1 to 2)</a></li>
</ul></li>
</ul></li>
</ul>
</div>

<p>The goal of this script is to assess the extent of regulatory variation in our cell types.</p>
<div id="global-variance-trends" class="section level1">
<h1>Global variance trends</h1>
<div id="check-data" class="section level2">
<h2>Check data</h2>
<pre class="r"><code># Load libraries

library(&quot;ggplot2&quot;)
library(&quot;qvalue&quot;)
source(&quot;~/Desktop/Endoderm_TC/ashlar-trial/analysis/chunk-options.R&quot;)
library(&quot;RColorBrewer&quot;)

# Load colors
pal &lt;- c(brewer.pal(9, &quot;Set1&quot;), brewer.pal(8, &quot;Set2&quot;), brewer.pal(12, &quot;Set3&quot;))

# Functions for plots

bjpm&lt;-
theme(
  panel.border = element_rect(colour = &quot;black&quot;, fill = NA, size = 2),
  plot.title = element_text(size = 16, face = &quot;bold&quot;),
  axis.text.y =  element_text(size = 14,face = &quot;bold&quot;,color = &quot;black&quot;),
  axis.text.x =  element_text(size = 14,face = &quot;bold&quot;,color = &quot;black&quot;),
  axis.title.y = element_text(size = 14,face = &quot;bold&quot;),
  axis.title.x=element_blank(),
  legend.text = element_text(size = 14,face = &quot;bold&quot;),
  legend.title = element_text(size = 14,face = &quot;bold&quot;),
  strip.text.x = element_text(size = 14,face = &quot;bold&quot;),
  strip.text.y = element_text(size = 14,face = &quot;bold&quot;),
  strip.background = element_rect(colour = &quot;black&quot;, size = 2))

bjp&lt;-
theme(
  axis.text.y =  element_text(size = 9,face = &quot;bold&quot;,color = &quot;black&quot;),
  axis.title.y = element_text(size = 10,face = &quot;bold&quot;,color = &quot;black&quot;),
  axis.text.x =  element_text(size = 9,face = &quot;bold&quot;,color = &quot;black&quot;),
  axis.title.x = element_text(size = 10,face = &quot;bold&quot;,color = &quot;black&quot;), 
  plot.title = element_text(size = 13, face = &quot;bold&quot;))


# Load cyclic loess normalized data

cyclicloess_norm &lt;- read.delim(&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/cpm_cyclicloess.txt&quot;)</code></pre>
</div>
<div id="take-the-mean-of-technical-replicates-for-each-individual-and-visualize-the-data" class="section level2">
<h2>Take the mean of technical replicates for each individual and visualize the data</h2>
<pre class="r"><code># Take the mean of the technical replicates when available

  # Day 0 technical replicates
D0_28815 &lt;- as.data.frame(apply(cyclicloess_norm[,5:6], 1, mean))
D0_3647 &lt;- as.data.frame(apply(cyclicloess_norm[,8:9], 1, mean))
D0_3649 &lt;- as.data.frame(apply(cyclicloess_norm[,10:11], 1, mean))
D0_40300 &lt;- as.data.frame(apply(cyclicloess_norm[,12:13], 1, mean))
D0_4955 &lt;- as.data.frame(apply(cyclicloess_norm[,14:15], 1, mean))
  # Day 1 technical replicates
D1_20157 &lt;- as.data.frame(apply(cyclicloess_norm[,16:17], 1, mean))
D1_28815 &lt;- as.data.frame(apply(cyclicloess_norm[,21:22], 1, mean))
D1_3647 &lt;- as.data.frame(apply(cyclicloess_norm[,24:25], 1, mean))
D1_3649 &lt;- as.data.frame(apply(cyclicloess_norm[,26:27], 1, mean))
D1_40300 &lt;- as.data.frame(apply(cyclicloess_norm[,28:29], 1, mean))
D1_4955 &lt;- as.data.frame(apply(cyclicloess_norm[,30:31], 1, mean))
  # Day 2 technical replicates
D2_20157 &lt;- as.data.frame(apply(cyclicloess_norm[,32:33], 1, mean))
D2_28815 &lt;- as.data.frame(apply(cyclicloess_norm[,37:38], 1, mean))
D2_3647 &lt;- as.data.frame(apply(cyclicloess_norm[,40:41], 1, mean))
D2_3649 &lt;- as.data.frame(apply(cyclicloess_norm[,42:43], 1, mean))
D2_40300 &lt;- as.data.frame(apply(cyclicloess_norm[,44:45], 1, mean))
D2_4955 &lt;- as.data.frame(apply(cyclicloess_norm[,46:47], 1, mean))
  # Day 3 technical replicates
D3_20157 &lt;- as.data.frame(apply(cyclicloess_norm[,48:49], 1, mean))
D3_28815 &lt;- as.data.frame(apply(cyclicloess_norm[,53:54], 1, mean))
D3_3647 &lt;- as.data.frame(apply(cyclicloess_norm[,56:57], 1, mean))
D3_3649 &lt;- as.data.frame(apply(cyclicloess_norm[,58:59], 1, mean))
D3_40300 &lt;- as.data.frame(apply(cyclicloess_norm[,60:61], 1, mean))
D3_4955 &lt;- as.data.frame(apply(cyclicloess_norm[,62:63], 1, mean))

# Create a new data frame with all of the combined technical replicates
mean_tech_reps &lt;- cbind(cyclicloess_norm[,1:4], D0_28815, cyclicloess_norm[,7], D0_3647, D0_3649, D0_40300, D0_4955, D1_20157, cyclicloess_norm[,18:20], D1_28815, cyclicloess_norm[,23], D1_3647, D1_3649, D1_40300, D1_4955, D2_20157, cyclicloess_norm[,34:36], D2_28815, cyclicloess_norm[,39], D2_3647, D2_3649, D2_40300, D2_4955, D3_20157, cyclicloess_norm[,50:52], D3_28815, cyclicloess_norm[,55], D3_3647, D3_3649, D3_40300, D3_4955)


colnames(mean_tech_reps) &lt;- c(&quot;D0_20157&quot;, &quot;D0_20961&quot;, &quot;D0_21792&quot;, &quot;D0_28162&quot;, &quot;D0_28815&quot;, &quot;D0_29089&quot;, &quot;D0_3647&quot;, &quot;D0_3649&quot;, &quot;D0_40300&quot;, &quot;D0_4955&quot;, &quot;D1_20157&quot;, &quot;D1_20961&quot;, &quot;D1_21792&quot;, &quot;D1_28162&quot;, &quot;D1_28815&quot;, &quot;D1_29089&quot;, &quot;D1_3647&quot;, &quot;D1_3649&quot;, &quot;D1_40300&quot;, &quot;D1_4955&quot;, &quot;D2_20157&quot;, &quot;D2_20961&quot;, &quot;D2_21792&quot;, &quot;D2_28162&quot;, &quot;D2_28815&quot;, &quot;D2_29089&quot;, &quot;D2_3647&quot;, &quot;D2_3649&quot;, &quot;D2_40300&quot;, &quot;D2_4955&quot;, &quot;D3_20157&quot;, &quot;D3_20961&quot;, &quot;D3_21792&quot;, &quot;D3_28162&quot;, &quot;D3_28815&quot;, &quot;D3_29089&quot;, &quot;D3_3647&quot;, &quot;D3_3649&quot;, &quot;D3_40300&quot;, &quot;D3_4955&quot;)

dim(mean_tech_reps)</code></pre>
<pre><code>[1] 10304    40</code></pre>
<pre class="r"><code>#write.table(mean_tech_reps, file=&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/cpm_cyclicloess_40.txt&quot;,sep=&quot;\t&quot;, col.names = T, row.names = T) 

# Make a column for which are averaged or not

averaged_status &lt;- c(1,1,1,1,2,1,2,2,2,2,2,1,1,1,2,1,2,2,2,2,2,1,1,1,2,1,2,2,2,2,2,1,1,1,2,1,2,2,2,2)

# Find the technical factors for the biological replicates (no technical replicates)

bio_rep_samplefactors &lt;- read.delim(&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/samplefactors-filtered.txt&quot;, stringsAsFactors=FALSE)
day &lt;- bio_rep_samplefactors$Day
species &lt;- bio_rep_samplefactors$Species

# Make PCA plots with the factors colored by day

pca_genes &lt;- prcomp(t(mean_tech_reps), scale = T, retx = TRUE, center = TRUE)

matrixpca &lt;- pca_genes$x
pc1 &lt;- matrixpca[,1]
pc2 &lt;- matrixpca[,2]
pc3 &lt;- matrixpca[,3]
pc4 &lt;- matrixpca[,4]
pc5 &lt;- matrixpca[,5]

pcs &lt;- data.frame(pc1, pc2, pc3, pc4, pc5)

summary &lt;- summary(pca_genes)

#dev.off()

ggplot(data=pcs, aes(x=pc1, y=pc2, color=as.factor(day+1), shape=as.factor(averaged_status), size=2)) + geom_point() + xlab(paste(&quot;PC1 (&quot;,(summary$importance[2,1]*100), &quot;% of variance)&quot;)) + ylab(paste(&quot;PC2 (&quot;,(summary$importance[2,2]*100), &quot;% of variance)&quot;)) + theme_minimal() + guides(color = guide_legend(order=1), size = FALSE, shape = guide_legend(order=3)) + scale_color_discrete(name  =&quot;Day&quot;, labels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;)) +  scale_shape_manual(values = c(5, 7)) + scale_shape_discrete(name  =&quot;Replicate status&quot;, labels = c(&quot;Single&quot; ,&quot;Averaged&quot;)) + labs(title = &quot;PCs 1 and 2 from global normalized expression&quot;)</code></pre>
<pre><code>Scale for &#39;shape&#39; is already present. Adding another scale for &#39;shape&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggplot(data=pcs, aes(x=pc1, y=pc2, color=as.factor(day+1), shape=as.factor(averaged_status), size=2)) + geom_point() + xlab(paste(&quot;PC1 (&quot;,(summary$importance[2,1]*100), &quot;% of variance)&quot;)) + ylab(paste(&quot;PC2 (&quot;,(summary$importance[2,2]*100), &quot;% of variance)&quot;)) + scale_shape_manual(values = c(5, 8)) + scale_color_discrete(name  =&quot;Day&quot;, labels = c(&quot;0&quot;, &quot;1&quot;, &quot;2&quot;, &quot;3&quot;))  + labs(title = &quot;PCs 1 and 2 from global normalized gene expression (Averaged bet. technical replicates)&quot;)

#ggplotly()</code></pre>
</div>
<div id="examine-the-means-and-the-variances-for-each-day-species-pairs" class="section level2">
<h2>Examine the means and the variances for each day-species pairs</h2>
<pre class="r"><code># Calculate the variance for each species-time pair

humans_day0_var &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,1:6]),1, var) )
colnames(humans_day0_var) &lt;- c(&quot;Variance&quot;) 
chimps_day0_var &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,7:10]),1, var))
colnames(chimps_day0_var) &lt;- c(&quot;Variance&quot;) 
humans_day1_var &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,11:16]),1, var))
colnames(humans_day1_var) &lt;- c(&quot;Variance&quot;) 
chimps_day1_var &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,17:20]),1, var))
colnames(chimps_day1_var) &lt;- c(&quot;Variance&quot;)
humans_day2_var &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,21:26]),1, var))
colnames(humans_day2_var) &lt;- c(&quot;Variance&quot;) 
chimps_day2_var &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,27:30]),1, var))
colnames(chimps_day2_var) &lt;- c(&quot;Variance&quot;)
humans_day3_var &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,31:36]),1, var))
colnames(humans_day3_var) &lt;- c(&quot;Variance&quot;) 
chimps_day3_var &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,37:40]),1, var))
colnames(chimps_day3_var) &lt;- c(&quot;Variance&quot;)

# Calculate the mean for each species-time pair
humans_day0_mean &lt;-  as.data.frame(apply(as.data.frame(mean_tech_reps[,1:6]),1, mean))
colnames(humans_day0_mean) &lt;- c(&quot;Mean&quot;)
chimps_day0_mean &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,7:10]),1, mean))
colnames(chimps_day0_mean) &lt;- c(&quot;Mean&quot;)
humans_day1_mean &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,11:16]),1, mean))
colnames(humans_day1_mean) &lt;- c(&quot;Mean&quot;)
chimps_day1_mean &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,17:20]),1, mean))
colnames(chimps_day1_mean) &lt;- c(&quot;Mean&quot;)
humans_day2_mean &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,21:26]),1, mean))
colnames(humans_day2_mean) &lt;- c(&quot;Mean&quot;)
chimps_day2_mean &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,27:30]),1, mean))
colnames(chimps_day2_mean) &lt;- c(&quot;Mean&quot;)
humans_day3_mean &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,31:36]),1, mean))
colnames(humans_day3_mean) &lt;- c(&quot;Mean&quot;)
chimps_day3_mean &lt;- as.data.frame(apply(as.data.frame(mean_tech_reps[,37:40]),1, mean))
colnames(chimps_day3_mean) &lt;- c(&quot;Mean&quot;)

# Make arrays with all the means

labels1 &lt;- array(&quot;Chimp Day 0&quot;, dim = c(10304, 1)) 
labels2 &lt;- array(&quot;Chimp Day 1&quot;, dim = c(10304, 1)) 
labels3 &lt;- array(&quot;Chimp Day 2&quot;, dim = c(10304, 1)) 
labels4 &lt;- array(&quot;Chimp Day 3&quot;, dim = c(10304, 1)) 
labels5 &lt;- array(&quot;Human Day 0&quot;, dim = c(10304, 1)) 
labels6 &lt;- array(&quot;Human Day 1&quot;, dim = c(10304, 1)) 
labels7 &lt;- array(&quot;Human Day 2&quot;, dim = c(10304, 1)) 
labels8 &lt;- array(&quot;Human Day 3&quot;, dim = c(10304, 1)) 

# Make species-day labels
labels9 &lt;- rbind(labels1, labels2, labels3, labels4, labels5, labels6, labels7, labels8)
labels &lt;- as.numeric(as.factor(labels9))
# Make labels so same days from different species are the same color
labels10 &lt;- rbind(labels1, labels2, labels3, labels4, labels1, labels2, labels3, labels4)
labels10 &lt;- as.numeric(as.factor(labels10))

# Make species labels

labels11 &lt;- array(&quot;Chimpanzee&quot;, dim = c(41216, 1))
labels12 &lt;- array(&quot;Human&quot;, dim = c(41216, 1))
labels13 &lt;- rbind(labels11, labels12)

#1 Overall trend across genes

# Boxplot of means gives general trend

# Combine means
HC_mean &lt;- rbind(as.data.frame(chimps_day0_mean), as.data.frame(chimps_day1_mean), as.data.frame(chimps_day2_mean), as.data.frame(chimps_day3_mean), as.data.frame(humans_day0_mean), as.data.frame(humans_day1_mean), as.data.frame(humans_day2_mean), as.data.frame(humans_day3_mean))

HC_mean_labels &lt;- cbind(HC_mean, labels, labels10)

m &lt;- ggplot(HC_mean_labels, aes(x = factor(labels), y = HC_mean))
m &lt;- m + geom_violin(aes(fill = factor(labels10)), show.legend = FALSE) + geom_boxplot(aes(fill = factor(labels10)), show.legend = FALSE, outlier.shape = NA,width=0.2) + theme_bw() +  ggtitle(&quot;Mean for each gene by species and day&quot;) + xlab(&quot;Species-Day Pair&quot;) + ylab(&quot;Mean for each gene&quot;) 
m &lt;- m + scale_x_discrete(labels=c(&quot;1&quot; = &quot;C Day 0&quot;, &quot;2&quot; = &quot;C Day 1&quot;, &quot;3&quot; = &quot;C Day 2&quot;, &quot;4&quot; = &quot;C Day 3&quot;, &quot;5&quot; = &quot;H Day 0&quot;, &quot;6&quot; = &quot;H Day 1&quot;, &quot;7&quot; = &quot;H Day 2&quot;, &quot;8&quot; = &quot;H Day 3&quot;)) + bjp
m</code></pre>
<pre><code>Don&#39;t know how to automatically pick scale for object of type data.frame. Defaulting to continuous.</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Take log2 of each data frame

log_chimps_day0_var &lt;- log2(chimps_day0_var)
log_chimps_day1_var &lt;- log2(chimps_day1_var)
log_chimps_day2_var &lt;- log2(chimps_day2_var)
log_chimps_day3_var &lt;- log2(chimps_day3_var)
log_humans_day0_var &lt;- log(humans_day0_var)
log_humans_day1_var &lt;- log2(humans_day1_var)
log_humans_day2_var &lt;- log2(humans_day2_var)
log_humans_day3_var &lt;- log2(humans_day3_var)


# Boxplot of variances gives general trend
HC_var &lt;- rbind(as.data.frame(log_chimps_day0_var), as.data.frame(log_chimps_day1_var), as.data.frame(log_chimps_day2_var), as.data.frame(log_chimps_day3_var), as.data.frame(log_humans_day0_var), as.data.frame(log_humans_day1_var), as.data.frame(log_humans_day2_var), as.data.frame(log_humans_day3_var))


# Make a boxplot of log2(variance of gene expression levels)

HC_var_labels &lt;- cbind(HC_var, labels, labels10)

dim(HC_var_labels)</code></pre>
<pre><code>[1] 82432     3</code></pre>
<pre class="r"><code>p &lt;- ggplot(HC_var_labels, aes(x = factor(labels), y = HC_var))
p &lt;- p + geom_violin(aes(fill = factor(labels10)), show.legend = FALSE) + geom_boxplot(aes(fill = factor(labels10)), show.legend = FALSE, outlier.shape = NA,width=0.2) + theme_bw() + xlab(&quot;Species-Day Pair&quot;) + ylab(&quot;log2 variance of gene expression&quot;) +  ggtitle(&quot;Log2 variance for each gene by species and day&quot;)
p &lt;- p + scale_x_discrete(labels=c(&quot;1&quot; = &quot;C Day 0&quot;, &quot;2&quot; = &quot;C Day 1&quot;, &quot;3&quot; = &quot;C Day 2&quot;, &quot;4&quot; = &quot;C Day 3&quot;, &quot;5&quot; = &quot;H Day 0&quot;, &quot;6&quot; = &quot;H Day 1&quot;, &quot;7&quot; = &quot;H Day 2&quot;, &quot;8&quot; = &quot;H Day 3&quot;))
p + bjp</code></pre>
<pre><code>Don&#39;t know how to automatically pick scale for object of type data.frame. Defaulting to continuous.</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Make boxplots that are not violin plots
HC_var_labels &lt;- cbind(HC_var, labels, labels10, labels13)
colnames(HC_var_labels) &lt;- c(&quot;Variance&quot;, &quot;SpeciesDay&quot;, &quot;Day&quot;, &quot;Species&quot;)

p &lt;- ggplot(HC_var_labels, aes(x = factor(Species), y = Variance)) + geom_boxplot(aes(fill = as.factor(Day)),width=0.4) + xlab(&quot;Species&quot;) + ylab(expression(bold(&#39;log&#39;[2]*&#39; variance in gene expression&#39;))) + scale_fill_discrete(name= &quot;Day&quot;, labels=c(&quot;Day 0&quot;,&quot;Day 1&quot;,&quot;Day 2&quot;, &quot;Day 3&quot;))
p + bjp + theme(legend.text=element_text(size=7), legend.title=element_text(size=7, face = &quot;bold&quot; ))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="evaluating-global-changes-in-variation" class="section level1">
<h1>Evaluating global changes in variation</h1>
<div id="no-global-reduction-in-means-between-day-0-and-day-1-in-both-species" class="section level2">
<h2>No global reduction in means between day 0 and day 1 in both species</h2>
<p>Check to see if differences between means (of all gene expression values) within humans and chimps between days</p>
<pre class="r"><code>########## Within chimps ##########

# Test at least one of the means is different
group &lt;- rbind(labels1, labels2, labels3, labels4)

mean_day_species &lt;- rbind(chimps_day0_mean,chimps_day1_mean, chimps_day2_mean, chimps_day3_mean)

aov_groups &lt;- cbind(mean_day_species, group)

dim(aov_groups)</code></pre>
<pre><code>[1] 41216     2</code></pre>
<pre class="r"><code>aov_bet &lt;- oneway.test(aov_groups$Mean ~ aov_groups$group)
aov_bet</code></pre>
<pre><code>
    One-way analysis of means (not assuming equal variances)

data:  aov_groups$Mean and aov_groups$group
F = 3.8146, num df = 3, denom df = 22892, p-value = 0.009564</code></pre>
<pre class="r"><code>#F = 4.4683, num df = 3, denom df = 22893, p-value = 0.003845F = 3.8146, num df = 3, denom df = 22892, p-value = 0.009564

# Test groups individually

# Day 0 and Day 1
t.test(chimps_day0_mean, chimps_day1_mean) # 0.1716</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  chimps_day0_mean and chimps_day1_mean
t = -1.3672, df = 20583, p-value = 0.1716
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.08994855  0.01602921
sample estimates:
mean of x mean of y 
 5.191453  5.228413 </code></pre>
<pre class="r"><code># Day 1 and Day 2
t.test(chimps_day1_mean, chimps_day2_mean) # 0.2033</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  chimps_day1_mean and chimps_day2_mean
t = -1.2721, df = 20606, p-value = 0.2033
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.08578381  0.01825800
sample estimates:
mean of x mean of y 
 5.228413  5.262176 </code></pre>
<pre class="r"><code># Day 2 and Day 3
t.test(chimps_day2_mean, chimps_day3_mean) # 0.6032</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  chimps_day2_mean and chimps_day3_mean
t = -0.51978, df = 20565, p-value = 0.6032
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.06724364  0.03905490
sample estimates:
mean of x mean of y 
 5.262176  5.276270 </code></pre>
<pre class="r"><code>########## Within humans ##########

# Test at least one of the means is different
group &lt;- rbind(labels5, labels6, labels7, labels8)

mean_day_species &lt;- rbind(humans_day0_mean,humans_day1_mean, humans_day2_mean, humans_day3_mean)

aov_groups &lt;- cbind(mean_day_species, group)

dim(aov_groups)</code></pre>
<pre><code>[1] 41216     2</code></pre>
<pre class="r"><code>aov_bet &lt;- oneway.test(aov_groups$Mean ~ aov_groups$group)
aov_bet</code></pre>
<pre><code>
    One-way analysis of means (not assuming equal variances)

data:  aov_groups$Mean and aov_groups$group
F = 6.3119, num df = 3, denom df = 22893, p-value = 0.0002829</code></pre>
<pre class="r"><code># F = 5.5523, num df = 3, denom df = 22893, p-value = 0.0008336

# Test groups individually

# Day 0 and Day 1
t.test(humans_day0_mean, humans_day1_mean) # 0.2702</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  humans_day0_mean and humans_day1_mean
t = -1.1027, df = 20596, p-value = 0.2702
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.08260299  0.02312380
sample estimates:
mean of x mean of y 
 5.211648  5.241388 </code></pre>
<pre class="r"><code># Day 1 and Day 2
t.test(humans_day1_mean, humans_day2_mean) # 0.02824</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  humans_day1_mean and humans_day2_mean
t = -2.1941, df = 20598, p-value = 0.02824
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.109737567 -0.006181887
sample estimates:
mean of x mean of y 
 5.241388  5.299347 </code></pre>
<pre class="r"><code># Day 2 and Day 3
t.test(humans_day2_mean, humans_day3_mean) # 0.628</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  humans_day2_mean and humans_day3_mean
t = -0.48453, df = 20602, p-value = 0.628
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.06437644  0.03885704
sample estimates:
mean of x mean of y 
 5.299347  5.312107 </code></pre>
<p>Conclusion: For both species, there is at least 1 mean that is different; however, when testing between the humans and chimps day 0 and 1, there is not enough statistical evidence to suggest a difference in the means of the variances.</p>
</div>
<div id="the-effect-size-for-the-global-reduction-in-variance-between-day-0-and-1-in-chimpanzees-is-small-but-larger-in-humans." class="section level2">
<h2>The effect size for the global reduction in variance between day 0 and 1 in chimpanzees is small but larger in humans.</h2>
<p>Check to see if differences variances of all of the genes between humans and chimps</p>
<pre class="r"><code># Untransformed variances have a range of 0 to infinity. Therefore, we will log transform the variance values so that there is a range of negative infinity to positive infinity. Then we can use a t-test. 


########## Within chimps ##########

t.test(log2((unlist(chimps_day0_var))), log2((unlist(chimps_day1_var))), alternative = c(&quot;greater&quot;))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  log2((unlist(chimps_day0_var))) and log2((unlist(chimps_day1_var)))
t = 10.235, df = 20439, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 0.2388908       Inf
sample estimates:
mean of x mean of y 
-4.751923 -5.036558 </code></pre>
<pre class="r"><code>2^(5.036558 - 4.751923)</code></pre>
<pre><code>[1] 1.218102</code></pre>
<pre class="r"><code># t = 10.235, df = 20439, p-value &lt; 2.2e-16

# Is Day 1 to 2 or 2 to 3 greater?

# Day 1 and Day 2
t.test(log2((unlist(chimps_day1_var))), log2((unlist(chimps_day2_var))), alternative = c(&quot;greater&quot;)) # p-value = 1</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  log2((unlist(chimps_day1_var))) and log2((unlist(chimps_day2_var)))
t = -15.864, df = 20550, p-value = 1
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 -0.4955952        Inf
sample estimates:
mean of x mean of y 
-5.036558 -4.587522 </code></pre>
<pre class="r"><code># Day 2 and Day 3
t.test(log2((unlist(chimps_day2_var))), log2((unlist(chimps_day3_var))), alternative = c(&quot;greater&quot;)) # p-value = 1</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  log2((unlist(chimps_day2_var))) and log2((unlist(chimps_day3_var)))
t = -12.813, df = 20595, p-value = 1
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 -0.4030692        Inf
sample estimates:
mean of x mean of y 
-4.587522 -4.230311 </code></pre>
<pre class="r"><code>########## Within humans ##########


# Day 0 and Day 1
t.test(log2((unlist(humans_day0_var))), log2((unlist(humans_day1_var))), alternative = c(&quot;greater&quot;))</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  log2((unlist(humans_day0_var))) and log2((unlist(humans_day1_var)))
t = 49.053, df = 20465, p-value &lt; 2.2e-16
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 1.189854      Inf
sample estimates:
mean of x mean of y 
-3.636568 -4.867707 </code></pre>
<pre class="r"><code>2^(4.867707-3.636568)</code></pre>
<pre><code>[1] 2.347523</code></pre>
<pre class="r"><code># Day 1 and Day 2
t.test(log2((unlist(humans_day1_var))), log2((unlist(humans_day2_var))), alternative = c(&quot;greater&quot;)) # p-value = 1</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  log2((unlist(humans_day1_var))) and log2((unlist(humans_day2_var)))
t = -45.735, df = 19997, p-value = 1
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 -1.141847       Inf
sample estimates:
mean of x mean of y 
-4.867707 -3.765502 </code></pre>
<pre class="r"><code># Day 2 and Day 3
t.test(log2((unlist(humans_day2_var))), log2((unlist(humans_day3_var))), alternative = c(&quot;greater&quot;)) # p-value = 0.9986</code></pre>
<pre><code>
    Welch Two Sample t-test

data:  log2((unlist(humans_day2_var))) and log2((unlist(humans_day3_var)))
t = -2.9879, df = 20393, p-value = 0.9986
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 -0.107056       Inf
sample estimates:
mean of x mean of y 
-3.765502 -3.696457 </code></pre>
<p>Conclusion: For both species, there is at least 1 mean (of the variances) that is different; however, when testing between the chimps day 0 and 1, there is not enough statistical evidence to suggest a difference in the means of the variances in the chimpanzees. Even if you make the argument that biologically we expect a reduction in variance between day 0 an day 1 (due to cannalization, for example) and therefore try a one-sided t-test, the p-value is still not statistically significant.</p>
</div>
</div>
<div id="characterizing-variance-trends-between-days-tested-at-the-gene-level-using-f-tests" class="section level1">
<h1>Characterizing variance trends between days tested at the gene level using F tests</h1>
<p>We are interested in the genes that experience either an increase or reduction in variance between days in both species. We are going to use a parametric test.</p>
<div id="general-functions-for-reductionincrease-in-variance" class="section level2">
<h2>General functions for reduction/increase in variance</h2>
<pre class="r"><code>## Original code for the qqplots provided by Bryce van de Geijn.
#Output:                                                                                                                                  
#adds a set of qq points to an existing graph                                                                                             

addqqplot=function(pvals, always.plot,density, col_designated){
  len = length(pvals)
  res=qqplot(-log10((1:len)/(1+len)),pvals,plot.it=F)
  return(res)
}

#newqqplot creates a new plot with a set of qq points                                                                                     
#Output:                                                                                                                                  
#creates a new qq plot                                                                                                                    

newqqplot=function(pvals, always.plot,density){
  len = length(pvals)
  res=qqplot(-log10((1:len)/(1+len)),pvals,plot.it=F)
}</code></pre>
<pre class="r"><code># 1A) Reduction in variation. Enrichment in chimps given that the gene was significant in humans

# We need 4 values for each species: day t-1 start and end values and day t start and end values.

find_qqplot_red_chimps &lt;- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t, p_val_cutoff){

  # Make an array to store the Chimp p-values

  chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
  }

  hist(chimp_var_pval)
  
  # Test
#  chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

  # for(i in 1:10304){
  #   x &lt;- t(mean_tech_reps[i,7:10])
  #   y &lt;- t(mean_tech_reps[i,17:20])
  #    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
  #   chimp_var_pval[i,1] &lt;- htest$p.value
# }
# hist(chimp_var_pval)
  #   q_chimp_var_adj_pval &lt;- qvalue(chimp_var_pval)
  # length(q_chimp_var_adj_pval[which(q_chimp_var_adj_pval &lt; 0.05) , ])

############## For humans

  human_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
  }

  hist(human_var_pval)

  #   human_var_pval &lt;- array(NA, dim = c(10304, 1))
  # for(i in 1:10304){
  #   x &lt;- t(mean_tech_reps[i,1:6])
  #   y &lt;- t(mean_tech_reps[i,11:16])
  #   htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
  #   human_var_pval[i,1] &lt;- htest$p.value
# }
 
# hist(human_var_pval)
#   q_human_var_adj_pval &lt;- qvalue(human_var_pval)$qvalues
#  length(q_human_var_adj_pval[which(q_human_var_adj_pval &lt; 0.05) , ])
# 3498
  
############ Find p-values of the F_statistics

# Make one data frame for corrected pvalues
  
#  chimp_var_adj_pval &lt;- qvalue(chimp_var_pval)
#  human_var_adj_pval &lt;- qvalue(human_var_pval)$qvalues
#  human_var_pval_adj_pval &lt;- as.data.frame(cbind(human_var_pval, human_var_adj_pval))
#  human_var_pval_adj_pval_order &lt;- human_var_pval_adj_pval[order(human_var_pval_adj_pval[,2]),]
  
# Set p-value  (for FDR)
#  p_val &lt;- human_var_pval_adj_pval_order[max(which(human_var_pval_adj_pval_order[,2] &lt; 0.05)), 1] 

  p_val &lt;- p_val_cutoff
  var_pval &lt;- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) &lt;- rownames(mean_tech_reps)
  colnames(var_pval) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;)
  summary(var_pval)
  
  
  
# Make one data frame for uncorrected pvalues

  var_pval &lt;- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) &lt;- rownames(mean_tech_reps)
  colnames(var_pval) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;)
  summary(var_pval)

  dim(var_pval)

# Set p-value

#  p_val &lt;- 0.05

# P-val &lt; 0.05 for chimps only
  num_var_pval_chimps &lt;- var_pval[ which(var_pval[,1] &lt; p_val), ]

                    
# P-val &lt; 0.05 for humans only
  num_var_pval_humans &lt;- var_pval[ which(var_pval[,2] &lt; p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val &lt;- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
  num_var_pval_chimps_neg_log &lt;- -log10(var_pval[,1])

  res_chimp &lt;- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the chimps given that it was significant in the humans

  subset_var &lt;- as.data.frame(var_pval[ which(var_pval[,2] &lt; p_val), 1:2])
  num_var_pval_chimp_given_human_no_df &lt;- subset_var[,1]
  num_var_pval_chimp_given_human &lt;- as.data.frame(subset_var[,1])
 
  num_var_pval_shared_neg_log &lt;- -log10(num_var_pval_chimp_given_human)

  res_shared &lt;- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(chimp_var_pval, num_var_pval_chimp_given_human_no_df, sig_p_val, res_chimp$x, res_chimp$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}

# 2A) Increase in variation. Enrichment in chimps given that the gene was significant in humans

# We need 4 values for each species: day t-1 start and end values and day t start and end values.

find_qqplot_inc_chimps &lt;- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t, p_val_cutoff){

  # Make an array to store the Chimp p-values

  chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval &lt;- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) &lt;- rownames(mean_tech_reps)
  colnames(var_pval) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;)
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val &lt;- p_val_cutoff

# P-val &lt; 0.05 for chimps only
  num_var_pval_chimps &lt;- var_pval[ which(var_pval[,1] &lt; p_val), ]

                    
# P-val &lt; 0.05 for humans only
  num_var_pval_humans &lt;- var_pval[ which(var_pval[,2] &lt; p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val &lt;- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the chimps
  num_var_pval_chimps_neg_log &lt;- -log10(var_pval[,1])

  res_chimp &lt;- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the chimps given that it was significant in the humans

  subset_var &lt;- as.data.frame(var_pval[ which(var_pval[,2] &lt; p_val), 1:2])
  num_var_pval_chimp_given_human_no_df &lt;- subset_var[,1]
  num_var_pval_chimp_given_human &lt;- as.data.frame(subset_var[,1])
 
  num_var_pval_shared_neg_log &lt;- -log10(num_var_pval_chimp_given_human)

  res_shared &lt;- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(chimp_var_pval, num_var_pval_chimp_given_human_no_df, sig_p_val, res_chimp$x, res_chimp$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}

# 1B) Reduction in variation. Enrichment in humans given that the gene was significant in chimps

# We need 4 values for each species: day t-1 start and end values and day t start and end values.

find_qqplot_red_humans &lt;- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t, p_val_cutoff){

  # Make an array to store the Chimp p-values

  chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval &lt;- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) &lt;- rownames(mean_tech_reps)
  colnames(var_pval) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;)
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val &lt;- p_val_cutoff

# P-val &lt; 0.05 for chimps only
  num_var_pval_chimps &lt;- var_pval[ which(var_pval[,1] &lt; p_val), ]

                    
# P-val &lt; 0.05 for humans only
  num_var_pval_humans &lt;- var_pval[ which(var_pval[,2] &lt; p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val &lt;- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the humans
  num_var_pval_chimps_neg_log &lt;- -log10(var_pval[,2])

  res_chimp &lt;- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the humans given that it was significant in the chimps

  subset_var &lt;- as.data.frame(var_pval[ which(var_pval[,1] &lt; p_val), 1:2])
  num_var_pval_human_given_chimp_no_df &lt;- subset_var[,2]
  num_var_pval_human_given_chimp &lt;- as.data.frame(subset_var[,2])
 
  num_var_pval_shared_neg_log &lt;- -log10(num_var_pval_human_given_chimp)

  res_shared &lt;- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(human_var_pval, num_var_pval_human_given_chimp_no_df, sig_p_val, res_chimp$x, res_chimp$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}

# 2B) Increase in variation. Enrichment in humans given that the gene was significant in chimps

# We need 4 values for each species: day t-1 start and end values and day t start and end values.

find_qqplot_inc_humans &lt;- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t, p_val_cutoff){

  # Make an array to store the Chimp p-values

  chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval &lt;- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) &lt;- rownames(mean_tech_reps)
  colnames(var_pval) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;)
  summary(var_pval)

  dim(var_pval)

# Set p-value

  p_val &lt;- p_val_cutoff

# P-val &lt; 0.05 for chimps only
  num_var_pval_chimps &lt;- var_pval[ which(var_pval[,1] &lt; p_val), ]

                    
# P-val &lt; 0.05 for humans only
  num_var_pval_humans &lt;- var_pval[ which(var_pval[,2] &lt; p_val), ]
  
# A data frame with the # of genes with significant p-values  
  sig_p_val &lt;- as.data.frame(rbind(dim(num_var_pval_chimps), dim(num_var_pval_humans)))[,1]
  
# Run for the humans
  num_var_pval_chimps_neg_log &lt;- -log10(var_pval[,2])

  res_chimp &lt;- newqqplot(num_var_pval_chimps_neg_log, -1, 100)

# Run for the shared
  # Find the p-values of the humans given that it was significant in the chimps

  subset_var &lt;- as.data.frame(var_pval[ which(var_pval[,1] &lt; p_val), 1:2])
  num_var_pval_human_given_chimp_no_df &lt;- subset_var[,2]
  num_var_pval_human_given_chimp &lt;- as.data.frame(subset_var[,2])
 
  num_var_pval_shared_neg_log &lt;- -log10(num_var_pval_human_given_chimp)

  res_shared &lt;- addqqplot(num_var_pval_shared_neg_log[,1], -1, 100, pal[3])

  # Information for plotting

  list_values = list(human_var_pval, num_var_pval_human_given_chimp_no_df, sig_p_val, res_chimp$x, res_chimp$y, res_shared$x, res_shared$y, max(res_chimp$x), max(res_chimp$y),  max(res_shared$x),  max(res_shared$y))
  
  
  return(list_values)

}

# Make multiple ggplots on the same page

# Multiplot function (from http://www.cookbook-r.com/Graphs/Multiple_graphs_on_one_page_%28ggplot2%29/)

multiplot &lt;- function(..., plotlist=NULL, file, cols=1, layout=NULL) {
  library(grid)

  # Make a list from the ... arguments and plotlist
  plots &lt;- c(list(...), plotlist)

  numPlots = length(plots)

  # If layout is NULL, then use &#39;cols&#39; to determine layout
  if (is.null(layout)) {
    # Make the panel
    # ncol: Number of columns of plots
    # nrow: Number of rows needed, calculated from # of cols
    layout &lt;- matrix(seq(1, cols * ceiling(numPlots/cols)),
                    ncol = cols, nrow = ceiling(numPlots/cols))
  }

 if (numPlots==1) {
    print(plots[[1]])

  } else {
    # Set up the page
    grid.newpage()
    pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))

    # Make each plot, in the correct location
    for (i in 1:numPlots) {
      # Get the i,j matrix positions of the regions that contain this subplot
      matchidx &lt;- as.data.frame(which(layout == i, arr.ind = TRUE))

      print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,
                                      layout.pos.col = matchidx$col))
    }
  }
}</code></pre>
</div>
<div id="number-of-significant-genes-and-storeys-pi_0-values-for-a-reduction-in-variation-between-days-main-paper" class="section level2">
<h2>Number of significant genes and Storey’s pi_0 values for a reduction in variation between days (main paper)</h2>
<div id="chimpanzees" class="section level3">
<h3>Chimpanzees</h3>
<div id="testing-chimps-reduction-in-variance-from-days-0-to-1" class="section level4">
<h4>Testing chimps reduction in variance from days 0 to 1</h4>
<pre class="r"><code>chimp_var_pval &lt;- array(NA, dim = c(10304, 1))
for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,7:10])
     y &lt;- t(mean_tech_reps[i,17:20])
      htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
     chimp_var_pval[i,1] &lt;- htest$p.value
 }

chimp_var_pval_red01 &lt;- as.data.frame(chimp_var_pval)
length(which(chimp_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 664</code></pre>
<pre class="r"><code># Make a histogram of the unadjusted p-values
hist(chimp_var_pval)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(chimp_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.3703  0.8591  0.8835  0.8793  0.9167  0.9999 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

which(fdr_adj &lt; 0.05)</code></pre>
<pre><code>integer(0)</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(chimp_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Chimp samples Day 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.7440476</code></pre>
<pre class="r"><code>boot_pi0 &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.7259161</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

# Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.2559524</code></pre>
<pre class="r"><code>1-qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.2740839</code></pre>
<pre class="r"><code>length(which(chimp_var_pval &lt; 0.05))*(1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 169.9524</code></pre>
<pre class="r"><code># Make a histogram of the p-value distribution

scaleFUN &lt;- function(x) sprintf(&quot;%.1f&quot;, x)

p1 &lt;- ggplot(chimp_var_pval_red01, aes(chimp_var_pval_red01[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#E77642&quot;) + 
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;0.74&quot;)&#39;, parse=TRUE, x=0.8, y=(1.25), size = 9, colour = &quot;#E77642&quot;)
             
            

p1</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p1s &lt;- ggplot(chimp_var_pval_red01, aes(chimp_var_pval_red01[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#E77642&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;0.73&quot;)&#39;, parse=TRUE, x=0.80, y=(1.25), size = 9, colour = &quot;#E77642&quot;) +bjp
             
p1s</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval &lt;- chimp_var_pval_red01[,1]
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = 10304)
line &lt;- rep(c(&quot;0.73&quot;), times = 10304)
figS12A &lt;- cbind(pval, resp, line)</code></pre>
</div>
<div id="testing-chimps-reduction-in-variance-from-days-1-to-2" class="section level4">
<h4>Testing chimps reduction in variance from days 1 to 2</h4>
<pre class="r"><code>chimp_var_pval &lt;- array(NA, dim = c(10304, 1))
for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,17:20])
     y &lt;- t(mean_tech_reps[i,27:30])
      htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
     chimp_var_pval[i,1] &lt;- htest$p.value
 }

chimp_var_pval_red12 &lt;- as.data.frame(chimp_var_pval)
length(which(chimp_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 272</code></pre>
<pre class="r"><code># Make a histogram of the unadjusted p-values
hist(chimp_var_pval)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(chimp_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.2234  0.9998  0.9998  0.9997  0.9998  0.9999 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

which(fdr_adj &lt; 0.05)</code></pre>
<pre><code>integer(0)</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(chimp_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Chimp samples Day 1 to 2&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>1-qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>length(which(chimp_var_pval &lt; 0.05))*(1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Make a histogram of the p-value distribution
# Since there is no apparent enrichment of small p-values, we won&#39;t apply Storey&#39;s pi0 

p2 &lt;- ggplot(chimp_var_pval_red12, aes(chimp_var_pval_red12[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)  + scale_y_continuous(labels=scaleFUN) + theme(plot.title = element_text(face = &quot;bold&quot;))
             
p2</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-9-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p2s &lt;- ggplot(chimp_var_pval_red12, aes(chimp_var_pval_red12[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)  + scale_y_continuous(labels=scaleFUN) + theme(plot.title = element_text(face = &quot;bold&quot;))  +bjp + ggtitle(&quot;Days 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;))
             
p2s</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-9-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval &lt;- chimp_var_pval_red12[,1]
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = 10304)
line &lt;- rep(c(&quot;NA&quot;), times = 10304)
figS12A_fig &lt;- cbind(pval, resp, line)

figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
</div>
<div id="testing-chimps-reduction-in-variance-from-days-2-to-3" class="section level4">
<h4>Testing chimps reduction in variance from days 2 to 3</h4>
<pre class="r"><code>chimp_var_pval &lt;- array(NA, dim = c(10304, 1))
for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,27:30])
     y &lt;- t(mean_tech_reps[i,37:40])
      htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
     chimp_var_pval[i,1] &lt;- htest$p.value
 }

chimp_var_pval_red23 &lt;- as.data.frame(chimp_var_pval)
length(which(chimp_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 452</code></pre>
<pre class="r"><code># Make a histogram of the unadjusted p-values
hist(chimp_var_pval, breaks = 500, main = c(&quot;Chimpanzees days 2 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(chimp_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.3806  0.8308  0.8308  0.8455  0.8344  0.9999 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

which(fdr_adj &lt; 0.05)</code></pre>
<pre><code>integer(0)</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(chimp_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Chimp samples Day 2 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.4600155</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.3652978</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.5399845</code></pre>
<pre class="r"><code>1-qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.6347022</code></pre>
<pre class="r"><code>length(which(chimp_var_pval &lt; 0.05))*(1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 244.073</code></pre>
<pre class="r"><code>p3 &lt;- ggplot(chimp_var_pval_red23, aes(chimp_var_pval_red23[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 2 to 3&quot;) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)  + scale_y_continuous(labels=scaleFUN) + theme(plot.title = element_text(face = &quot;bold&quot;))</code></pre>
<pre><code>Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
will replace the existing scale.</code></pre>
<pre class="r"><code>p3</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-10-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p3s &lt;- ggplot(chimp_var_pval_red23, aes(chimp_var_pval_red23[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 2 to 3&quot;) + theme_bw() +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)   + scale_y_continuous(labels=scaleFUN) + theme(plot.title = element_text(face = &quot;bold&quot;)) + bjp
             
p3s</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-10-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval &lt;- chimp_var_pval_red23[,1]
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = 10304)
line &lt;- rep(c(&quot;NA&quot;), times = 10304)
figS12A_fig &lt;- cbind(pval, resp, line)

figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
</div>
<div id="testing-chimps-reduction-in-variance-from-days-1-to-3" class="section level4">
<h4>Testing chimps reduction in variance from days 1 to 3</h4>
<pre class="r"><code>chimp_var_pval &lt;- array(NA, dim = c(10304, 1))
for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,17:20])
     y &lt;- t(mean_tech_reps[i,37:40])
      htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
     chimp_var_pval[i,1] &lt;- htest$p.value
 }

chimp_var_pval_red13 &lt;- as.data.frame(chimp_var_pval)

# Make a histogram of the unadjusted p-values
hist(chimp_var_pval, main = c(&quot;Chimpanzees days 1 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(chimp_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.9999  0.9999  0.9999  0.9999  0.9999  0.9999 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

which(fdr_adj &lt; 0.05)</code></pre>
<pre><code>integer(0)</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(chimp_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Chimp samples Day 1 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>1-qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>length(which(chimp_var_pval &lt; 0.05))*(1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>p3_13 &lt;- ggplot(chimp_var_pval_red13, aes(chimp_var_pval_red13[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 1 to 3&quot;) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + ggtitle(&quot;Days 1 to 2&quot;)  + scale_y_continuous(labels=scaleFUN) + theme(plot.title = element_text(face = &quot;bold&quot;))        </code></pre>
<pre><code>Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
will replace the existing scale.</code></pre>
<pre class="r"><code>p3_13</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-11-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p3_13s &lt;- ggplot(chimp_var_pval_red13, aes(chimp_var_pval_red13[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 1 to 3&quot;) + theme_bw() + 
  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + ggtitle(&quot;Days 1 to 3&quot;)  + scale_y_continuous(labels=scaleFUN) + theme(plot.title = element_text(face = &quot;bold&quot;)) + bjp
             
p3_13s</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-11-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval &lt;- chimp_var_pval_red13[,1]
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = 10304)
line &lt;- rep(c(&quot;NA&quot;), times = 10304)
figS12A_fig &lt;- cbind(pval, resp, line)

figS12A &lt;- rbind(figS12A, figS12A_fig)
colnames(figS12A) &lt;- c(&quot;pval&quot;, &quot;days&quot;, &quot;line_value&quot;)
write.table(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS12A.txt&quot;, quote = FALSE, row.names = FALSE)</code></pre>
</div>
<div id="testing-humans-reduction-in-variance-from-days-0-to-1" class="section level4">
<h4>Testing humans reduction in variance from days 0 to 1</h4>
<pre class="r"><code>human_var_pval &lt;- array(NA, dim = c(10304, 1))
   for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,1:6])
     y &lt;- t(mean_tech_reps[i,11:16])
     htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
     human_var_pval[i,1] &lt;- htest$p.value
   }
human_var_pval_red01 &lt;- as.data.frame(human_var_pval)

# Make a histogram of the unadjusted p-values
hist(human_var_pval, main = c(&quot;Humans days 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 2309</code></pre>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(human_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.02558 0.23606 0.38498 0.42280 0.58436 0.99874 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

length(which(fdr_adj &lt; 0.05))</code></pre>
<pre><code>[1] 22</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(human_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Human samples Day 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.2316253</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=human_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.174912</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.7683747</code></pre>
<pre class="r"><code>1-qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.825088</code></pre>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))*(1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 1774.177</code></pre>
<pre class="r"><code>p4 &lt;- ggplot(human_var_pval_red01, aes(human_var_pval_red01[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + ggtitle(&quot;Days 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;0.23&quot;)&#39;, parse=TRUE, x=0.80, y=(2), size = 6, colour = &quot;#00A4F4&quot;)
             
p4</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p4s &lt;- ggplot(human_var_pval_red01, aes(human_var_pval_red01[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + ggtitle(&quot;Days 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;0.17&quot;)&#39;, parse=TRUE, x=0.80, y=(2), size = 6, colour = &quot;#00A4F4&quot;) + bjp
             
p4s</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-12-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval &lt;- human_var_pval_red01[,1]
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = 10304)
line &lt;- rep(c(&quot;0.23&quot;), times = 10304)
figS12A &lt;- cbind(pval, resp, line)</code></pre>
</div>
<div id="testing-humans-reduction-in-variance-from-days-1-to-2" class="section level4">
<h4>Testing humans reduction in variance from days 1 to 2</h4>
<pre class="r"><code>human_var_pval &lt;- array(NA, dim = c(10304, 1))
   for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,11:16])
     y &lt;- t(mean_tech_reps[i,21:26])
     htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
     human_var_pval[i,1] &lt;- htest$p.value
   }

human_var_pval_red12 &lt;- as.data.frame(human_var_pval)

# Make a histogram of the unadjusted p-values
hist(human_var_pval, main = c(&quot;Humans days 1 to 2&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 114</code></pre>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(human_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
      1       1       1       1       1       1 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

length(which(fdr_adj &lt; 0.05))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(human_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Human samples Day 1 to 2&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-13-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=human_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>1-qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))*(1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>p5 &lt;- ggplot(human_var_pval_red12, aes(human_var_pval_red12[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + ggtitle(&quot;Days 1 to 2&quot;) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)  + scale_y_continuous(labels=scaleFUN) + theme(plot.title = element_text(face = &quot;bold&quot;))
             
p5</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-13-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p5s &lt;- ggplot(human_var_pval_red12, aes(human_var_pval_red12[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + ggtitle(&quot;Days 1 to 2&quot;) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)  + scale_y_continuous(labels=scaleFUN) + theme(plot.title = element_text(face = &quot;bold&quot;)) + bjp
             
p5s</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-13-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval &lt;- human_var_pval_red12[,1]
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = 10304)
line &lt;- rep(c(&quot;NA&quot;), times = 10304)
figS12A_fig &lt;- cbind(pval, resp, line)

figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
</div>
<div id="testing-humans-reduction-in-variance-from-days-2-to-3" class="section level4">
<h4>Testing humans reduction in variance from days 2 to 3</h4>
<pre class="r"><code>human_var_pval &lt;- array(NA, dim = c(10304, 1))
   for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,21:26])
     y &lt;- t(mean_tech_reps[i,31:36])
     htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
     human_var_pval[i,1] &lt;- htest$p.value
   }

human_var_pval_red23 &lt;- as.data.frame(human_var_pval)

# Make a histogram of the unadjusted p-values
hist(human_var_pval, main = c(&quot;Humans days 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 450</code></pre>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(human_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.2764  0.9855  0.9855  0.9850  0.9855  0.9997 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

length(which(fdr_adj &lt; 0.05))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(human_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Human samples Day 2 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.951734</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=human_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.8664877</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.04826605</code></pre>
<pre class="r"><code>1-qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.1335123</code></pre>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))*(1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 21.71972</code></pre>
<pre class="r"><code>p6 &lt;- ggplot(human_var_pval_red23, aes(human_var_pval_red23[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + ggtitle(&quot;Days 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + scale_y_continuous(labels=scaleFUN) +
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) 
             
p6</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p6s &lt;- ggplot(human_var_pval_red23, aes(human_var_pval_red23[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() +  ggtitle(&quot;Days 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +  scale_y_continuous(labels=scaleFUN) + bjp

             
p6s</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-14-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval &lt;- human_var_pval_red23[,1]
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = 10304)
line &lt;- rep(c(&quot;NA&quot;), times = 10304)
figS12A_fig &lt;- cbind(pval, resp, line)

figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
</div>
<div id="testing-humans-reduction-in-variance-from-days-1-to-3" class="section level4">
<h4>Testing humans reduction in variance from days 1 to 3</h4>
<pre class="r"><code>human_var_pval &lt;- array(NA, dim = c(10304, 1))
   for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,11:16])
     y &lt;- t(mean_tech_reps[i,31:36])
     htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
     human_var_pval[i,1] &lt;- htest$p.value
   }

human_var_pval_red13 &lt;- as.data.frame(human_var_pval)

# Make a histogram of the unadjusted p-values
hist(human_var_pval, main = c(&quot;Humans days 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(human_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
      1       1       1       1       1       1 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

length(which(fdr_adj &lt; 0.05))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(human_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Human samples Day 2 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=human_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>1-qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))*(1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>p6_13 &lt;- ggplot(human_var_pval_red13, aes(human_var_pval_red13[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + ggtitle(&quot;Days 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  scale_y_continuous(labels=scaleFUN) +
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)  
             
p6_13</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-15-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p6_13s &lt;- ggplot(human_var_pval_red13, aes(human_var_pval_red13[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + ggtitle(&quot;Days 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)  + scale_y_continuous(labels=scaleFUN)  + bjp

             
p6_13s</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-15-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval &lt;- human_var_pval_red13[,1]
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = 10304)
line &lt;- rep(c(&quot;NA&quot;), times = 10304)
figS12A_fig &lt;- cbind(pval, resp, line)

figS12A &lt;- rbind(figS12A, figS12A_fig)

write.table(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS12B.txt&quot;, quote = FALSE, row.names = FALSE)</code></pre>
</div>
<div id="plot-reduction-in-variation-pi_0-determined-by-a-bootstrap-method" class="section level4">
<h4>Plot reduction in variation (pi_0 determined by a bootstrap method)</h4>
<pre class="r"><code># Chimps on the top; humans on the bottom
multiplot(p1,p4,p2,p5,p3,p6,p3_13, p6_13, cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Chimps on the left, humans on the right
#multiplot(p1,p2,p3,p3_13,p4,p5,p6, p6_13, cols=2)</code></pre>
</div>
<div id="plot-reduction-in-variation-pi_0-determined-by-a-smoother-method" class="section level4">
<h4>Plot reduction in variation (pi_0 determined by a smoother method)</h4>
<pre class="r"><code># Chimps on the top; humans on the bottom
multiplot(p1s,p4s,p2s,p5s,p3s,p6s,p3_13s, p6_13s, cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Chimps on the left, humans on the right
#multiplot(p1s,p2s,p3s,p3_13s,p4s,p5s,p6s, p6_13s, cols=2)

# Chimps across
multiplot(p1s,p2s,p3s,p3_13s, cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-17-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Humans across
multiplot(p4s,p5s,p6s,p6_13s, cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-17-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="number-of-significant-genes-and-storeys-pi_0-values-for-an-increase-in-variation-between-days-main-paper" class="section level2">
<h2>Number of significant genes and Storey’s pi_0 values for an increase in variation between days (main paper)</h2>
<div id="chimpanzees-1" class="section level3">
<h3>Chimpanzees</h3>
<div id="testing-chimps-increase-in-variance-from-days-0-to-1" class="section level4">
<h4>Testing chimps increase in variance from days 0 to 1</h4>
<pre class="r"><code>chimp_var_pval &lt;- array(NA, dim = c(10304, 1))
for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,7:10])
     y &lt;- t(mean_tech_reps[i,17:20])
      htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
     chimp_var_pval[i,1] &lt;- htest$p.value
 }

chimp_var_pval_inc01 &lt;- as.data.frame(chimp_var_pval)

# Make a histogram of the unadjusted p-values
hist(chimp_var_pval)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(chimp_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.8624  0.9999  0.9999  0.9994  0.9999  0.9999 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

which(fdr_adj &lt; 0.05)</code></pre>
<pre><code>integer(0)</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(chimp_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Chimp samples Day 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-18-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>boot_pi0 &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

# Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>1-qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>length(which(chimp_var_pval &lt; 0.05))*(1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Make a histogram of the p-value distribution

p1_inc &lt;- ggplot(chimp_var_pval_inc01, aes(chimp_var_pval_inc01[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#E77642&quot;) + scale_y_continuous(labels=scaleFUN) + 
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.16, y=(1.25), size = 6, colour = &quot;#E77642&quot;)
             
            

p1_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-18-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p1s_inc &lt;- ggplot(chimp_var_pval_inc01, aes(chimp_var_pval_inc01[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#E77642&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.16, y=(1.25), size = 6, colour = &quot;#E77642&quot;)
             
p1s_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-18-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="testing-chimps-increase-in-variance-from-days-1-to-2" class="section level4">
<h4>Testing chimps increase in variance from days 1 to 2</h4>
<pre class="r"><code>chimp_var_pval &lt;- array(NA, dim = c(10304, 1))
for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,17:20])
     y &lt;- t(mean_tech_reps[i,27:30])
      htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
     chimp_var_pval[i,1] &lt;- htest$p.value
 }

chimp_var_pval_inc12 &lt;- as.data.frame(chimp_var_pval)

# Make a histogram of the unadjusted p-values
hist(chimp_var_pval)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(chimp_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.5974  0.7833  0.7907  0.8195  0.8336  1.0000 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

which(fdr_adj &lt; 0.05)</code></pre>
<pre><code>integer(0)</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(chimp_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Chimp samples Day 1 to 2&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.5725932</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.5221613</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.4274068</code></pre>
<pre class="r"><code>1-qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.4778387</code></pre>
<pre class="r"><code>length(which(chimp_var_pval &lt; 0.05))*(1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 238.493</code></pre>
<pre class="r"><code># Make a histogram of the p-value distribution

p2_inc &lt;- ggplot(chimp_var_pval_inc12, aes(chimp_var_pval_inc12[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#E77642&quot;) + ggtitle(&quot;Days 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;0.57&quot;)&#39;, parse=TRUE, x=0.80, y=(1.25), size = 6, colour = &quot;#E77642&quot;)
             
p2_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-19-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p2s_inc &lt;- ggplot(chimp_var_pval_inc12, aes(chimp_var_pval_inc12[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + ggtitle(&quot;Days 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#E77642&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;0.52&quot;)&#39;, parse=TRUE, x=0.80, y=(1.25), size = 6, colour = &quot;#E77642&quot;)
             
p2s_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-19-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="testing-chimps-increase-in-variance-from-days-2-to-3" class="section level4">
<h4>Testing chimps increase in variance from days 2 to 3</h4>
<pre class="r"><code>chimp_var_pval &lt;- array(NA, dim = c(10304, 1))
for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,27:30])
     y &lt;- t(mean_tech_reps[i,37:40])
      htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
     chimp_var_pval[i,1] &lt;- htest$p.value
 }

chimp_var_pval_inc23 &lt;- as.data.frame(chimp_var_pval)

# Make a histogram of the unadjusted p-values
hist(chimp_var_pval, breaks = 500, main = c(&quot;Chimpanzees days 2 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(chimp_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.3806  0.8308  0.8308  0.8455  0.8344  0.9999 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

which(fdr_adj &lt; 0.05)</code></pre>
<pre><code>integer(0)</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(chimp_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Chimp samples Day 2 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-20-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.4600155</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.3652978</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.5399845</code></pre>
<pre class="r"><code>1-qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.6347022</code></pre>
<pre class="r"><code>length(which(chimp_var_pval &lt; 0.05))*(1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 244.073</code></pre>
<pre class="r"><code>p3_inc &lt;- ggplot(chimp_var_pval_inc23, aes(chimp_var_pval_inc23[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#E77642&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;0.46&quot;)&#39;, parse=TRUE, x=0.8, y=(1.25), size = 6, colour = &quot;#E77642&quot;)
             
p3_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-20-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p3s_inc &lt;- ggplot(chimp_var_pval_inc23, aes(chimp_var_pval_inc23[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#E77642&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;0.37&quot;)&#39;, parse=TRUE, x=0.80, y=(1.25), size = 6, colour = &quot;#E77642&quot;)
             
p3s_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-20-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="testing-chimps-increase-in-variance-from-days-1-to-3" class="section level4">
<h4>Testing chimps increase in variance from days 1 to 3</h4>
<pre class="r"><code>chimp_var_pval &lt;- array(NA, dim = c(10304, 1))
for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,17:20])
     y &lt;- t(mean_tech_reps[i,37:40])
      htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
     chimp_var_pval[i,1] &lt;- htest$p.value
 }

chimp_var_pval_inc13 &lt;- as.data.frame(chimp_var_pval)

# Make a histogram of the unadjusted p-values
hist(chimp_var_pval, main = c(&quot;Chimpanzees days 1 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(chimp_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.2511  0.5429  0.6683  0.6713  0.7869  0.9996 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

which(fdr_adj &lt; 0.05)</code></pre>
<pre><code>integer(0)</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(chimp_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Chimp samples Day 1 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-21-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.550757</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=chimp_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.5071291</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.449243</code></pre>
<pre class="r"><code>1-qvalue(p=chimp_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.4928709</code></pre>
<pre class="r"><code>length(which(chimp_var_pval &lt; 0.05))*(1-qvalue(p=chimp_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 487.8779</code></pre>
<pre class="r"><code>p3_13_inc &lt;- ggplot(chimp_var_pval_inc13, aes(chimp_var_pval_inc13[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#E77642&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;0.55&quot;)&#39;, parse=TRUE, x=0.80, y=(1.25), size = 6, colour = &quot;#E77642&quot;)
             
p3_13_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-21-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p3_13s_inc &lt;- ggplot(chimp_var_pval_inc13, aes(chimp_var_pval_inc13[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) + ggtitle(&quot;Days 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#E77642&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, chimp&quot;])==&quot;0.51&quot;)&#39;, parse=TRUE, x=0.80, y=(1.25), size = 6, colour = &quot;#E77642&quot;)
             
p3_13s_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-21-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="testing-humans-increase-in-variance-from-days-0-to-1" class="section level4">
<h4>Testing humans increase in variance from days 0 to 1</h4>
<pre class="r"><code>human_var_pval &lt;- array(NA, dim = c(10304, 1))
   for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,1:6])
     y &lt;- t(mean_tech_reps[i,11:16])
     htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
     human_var_pval[i,1] &lt;- htest$p.value
   }
human_var_pval_inc01 &lt;- as.data.frame(human_var_pval)

# Make a histogram of the unadjusted p-values
hist(human_var_pval, main = c(&quot;Humans days 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(human_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
      1       1       1       1       1       1 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

length(which(fdr_adj &lt; 0.05))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(human_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Human samples Day 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=human_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>1-qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))*(1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>p4_inc &lt;- ggplot(human_var_pval_inc01, aes(human_var_pval_inc01[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.16, y=(2), size = 6, colour = &quot;#00A4F4&quot;)
             
p4_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-22-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p4s_inc &lt;- ggplot(human_var_pval_inc01, aes(human_var_pval_inc01[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.16, y=(2), size = 6, colour = &quot;#00A4F4&quot;)
             
p4s_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-22-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="testing-humans-increase-in-variance-from-days-1-to-2" class="section level4">
<h4>Testing humans increase in variance from days 1 to 2</h4>
<pre class="r"><code>human_var_pval &lt;- array(NA, dim = c(10304, 1))
   for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,11:16])
     y &lt;- t(mean_tech_reps[i,21:26])
     htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
     human_var_pval[i,1] &lt;- htest$p.value
   }

human_var_pval_inc12 &lt;- as.data.frame(human_var_pval)

# Make a histogram of the unadjusted p-values
hist(human_var_pval, main = c(&quot;Humans days 1 to 2&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(human_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.06298 0.29350 0.43367 0.46480 0.61330 0.99838 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

length(which(fdr_adj &lt; 0.05))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(human_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Human samples Day 1 to 2&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-23-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.2685041</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=human_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.2141993</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.7314959</code></pre>
<pre class="r"><code>1-qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.7858007</code></pre>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))*(1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 1439.584</code></pre>
<pre class="r"><code>p5_inc &lt;- ggplot(human_var_pval_inc12, aes(human_var_pval_inc12[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;0.27&quot;)&#39;, parse=TRUE, x=0.80, y=(2), size = 6, colour = &quot;#00A4F4&quot;)
             
p5_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-23-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p5s_inc &lt;- ggplot(human_var_pval_inc12, aes(human_var_pval_inc12[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;0.21&quot;)&#39;, parse=TRUE, x=0.80, y=(2), size = 6, colour = &quot;#00A4F4&quot;)
             
p5s_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-23-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="testing-humans-increase-in-variance-from-days-2-to-3" class="section level4">
<h4>Testing humans increase in variance from days 2 to 3</h4>
<pre class="r"><code>human_var_pval &lt;- array(NA, dim = c(10304, 1))
   for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,21:26])
     y &lt;- t(mean_tech_reps[i,31:36])
     htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
     human_var_pval[i,1] &lt;- htest$p.value
   }

human_var_pval_inc23 &lt;- as.data.frame(human_var_pval)

# Make a histogram of the unadjusted p-values
hist(human_var_pval, main = c(&quot;Humans days 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(human_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.9201  0.9201  0.9207  0.9360  0.9438  1.0000 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

length(which(fdr_adj &lt; 0.05))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(human_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Human samples Day 2 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-24-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.8614777</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=human_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.8477893</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.1385223</code></pre>
<pre class="r"><code>1-qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.1522107</code></pre>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))*(1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 53.05402</code></pre>
<pre class="r"><code>p6_inc &lt;- ggplot(human_var_pval_inc23, aes(human_var_pval_inc23[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;0.86&quot;)&#39;, parse=TRUE, x=0.80, y=(1.25), size = 6, colour = &quot;#00A4F4&quot;)
             
p6_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-24-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p6s_inc &lt;- ggplot(human_var_pval_inc23, aes(human_var_pval_inc23[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;0.85&quot;)&#39;, parse=TRUE, x=0.80, y=(1.25), size = 6, colour = &quot;#00A4F4&quot;) 
             
p6s_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-24-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="testing-humans-increase-in-variance-from-days-1-to-3" class="section level4">
<h4>Testing humans increase in variance from days 1 to 3</h4>
<pre class="r"><code>human_var_pval &lt;- array(NA, dim = c(10304, 1))
   for(i in 1:10304){
     x &lt;- t(mean_tech_reps[i,11:16])
     y &lt;- t(mean_tech_reps[i,31:36])
     htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
     human_var_pval[i,1] &lt;- htest$p.value
   }

human_var_pval_inc13 &lt;- as.data.frame(human_var_pval)

# Make a histogram of the unadjusted p-values
hist(human_var_pval, main = c(&quot;Humans days 0 to 1&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Go from unadjusted to B.H. adjusted p-values
fdr_adj &lt;- p.adjust(human_var_pval, method = c(&quot;fdr&quot;) )

summary(fdr_adj)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.03533 0.26036 0.40614 0.44127 0.59814 0.99985 </code></pre>
<pre class="r"><code># How many B.H. adjusted p-values &lt; 0.05?

length(which(fdr_adj &lt; 0.05))</code></pre>
<pre><code>[1] 23</code></pre>
<pre class="r"><code># Plot the unadjusted versus adjusted p-values

plot(human_var_pval, fdr_adj, xlim = c(0,1), ylim = c(0,1), xlab = c(&quot;p-value of the F-test&quot;), ylab = c(&quot;BH-adj p-value&quot;), main = c(&quot;F test results from Human samples Day 2 to 3&quot;))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-25-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Obtain Storey&#39;s pi_0

boot_pi0 &lt;- qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.2413302</code></pre>
<pre class="r"><code>boot_pi0s &lt;- qvalue(p=human_var_pval, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.203399</code></pre>
<pre class="r"><code># Obtain pi_1 based on Storey&#39;s pi_0

1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.7586698</code></pre>
<pre class="r"><code>1-qvalue(p=human_var_pval,  pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0</code></pre>
<pre><code>[1] 0.796601</code></pre>
<pre class="r"><code>length(which(human_var_pval &lt; 0.05))*(1-qvalue(p=human_var_pval, pi0.method=&quot;bootstrap&quot;, fdr.level=NULL, robust=TRUE)$pi0)</code></pre>
<pre><code>[1] 1625.071</code></pre>
<pre class="r"><code>p6_13_inc &lt;- ggplot(human_var_pval_inc13, aes(human_var_pval_inc13[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + geom_hline(yintercept = boot_pi0, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;0.24&quot;)&#39;, parse=TRUE, x=0.80, y=(2), size = 6, colour = &quot;#00A4F4&quot;)
             
p6_13_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-25-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p6_13s_inc &lt;- ggplot(human_var_pval_inc13, aes(human_var_pval_inc13[,1])) + geom_histogram(aes(y = ..density..), fill=&quot;white&quot;, colour=&quot;black&quot;, breaks=seq(0, 1, by = 0.02)) + theme_bw() + 
  scale_x_continuous(limits = c(0,1)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)  + geom_hline(yintercept = boot_pi0s, size=2.5, colour = &quot;#00A4F4&quot;) + scale_y_continuous(labels=scaleFUN) +
  annotate(&quot;text&quot;, label=&#39;bold(bolditalic(hat(pi)[0][&quot;, human&quot;])==&quot;0.20&quot;)&#39;, parse=TRUE, x=0.80, y=(2), size = 6, colour = &quot;#00A4F4&quot;)
             
p6_13s_inc</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-25-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-reduction-in-variation-pi_0-determined-by-a-bootstrap-method-1" class="section level4">
<h4>Plot reduction in variation (pi_0 determined by a bootstrap method)</h4>
<pre class="r"><code># Chimps on the top; humans on the bottom
multiplot(p1_inc,p4_inc,p2_inc,p5_inc,p3_inc,p6_inc,p3_13_inc, p6_13_inc, cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Chimps on the left, humans on the right
#multiplot(p1,p2,p3,p3_13,p4,p5,p6, p6_13, cols=2)</code></pre>
</div>
<div id="plot-reduction-in-variation-pi_0-determined-by-a-smoother-method-1" class="section level4">
<h4>Plot reduction in variation (pi_0 determined by a smoother method)</h4>
<pre class="r"><code># Chimps on the top; humans on the bottom
multiplot(p1s_inc,p4s_inc,p2s_inc,p5s_inc,p3s_inc,p6s_inc,p3_13s_inc, p6_13s_inc, cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Chimps on the left, humans on the right
#multiplot(p1s,p2s,p3s,p3_13s,p4s,p5s,p6s, p6_13s, cols=2)</code></pre>
</div>
</div>
</div>
<div id="the-pairwise-changes-in-variance-were-robust-with-respect-to-p-value-cutoff" class="section level2">
<h2>The pairwise changes in variance were robust with respect to p-value cutoff</h2>
<div id="changes-in-variance-in-chimp-samples-p-value-cutoff-0.05" class="section level3">
<h3>Changes in variance in chimp samples (p-value cutoff 0.05)</h3>
<pre class="r"><code># Reduction in variance from day 0 to day 1 chimps

qqplot_red_day01 &lt;- find_qqplot_red_chimps(7,10,17,20,1,6,11,16, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 2 chimps

qqplot_red_day12 &lt;- find_qqplot_red_chimps(17,20,27,30,11,16,21,26, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 2 to day 3 chimps

qqplot_red_day23 &lt;- find_qqplot_red_chimps(27,30,37,40,21,26,31,36, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 0 to day 2 chimps

qqplot_red_day02 &lt;- find_qqplot_red_chimps(7,10,27,30,1,6,21,26, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 3 chimps

qqplot_red_day13 &lt;- find_qqplot_red_chimps(17,20,37,40,11,16,31,36, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 1

qqplot_inc_day01 &lt;- find_qqplot_inc_chimps(7,10,17,20,1,6,11,16, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 2

qqplot_inc_day12 &lt;- find_qqplot_inc_chimps(17,20,27,30,11,16,21,26, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 2 to day 3

qqplot_inc_day23 &lt;- find_qqplot_inc_chimps(27,30,37,40,21,26,31,36, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 2

qqplot_inc_day02 &lt;- find_qqplot_inc_chimps(7,10,27,30,1,6,21,26, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 3

qqplot_inc_day13 &lt;- find_qqplot_inc_chimps(17,20,37,40,11,16,31,36, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction of variance qqplots- chimps

par(mfrow=c(2,4),oma = c(2, 2, 2, 2))

# Reduction of variance to day 0 to 1 chimps
plot(qqplot_red_day01[[4]],qqplot_red_day01[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;, main = &quot;Day 0 to 1&quot;)
abline(0,1, col = &quot;black&quot;)

# Add to the plot
points(qqplot_red_day01[[6]],qqplot_red_day01[[7]], pch = 16, col = &quot;red&quot;)

# Reduction of variance to day 1 to 2
plot(qqplot_red_day12[[4]],qqplot_red_day12[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;, main = &quot;Day 1 to 2&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_red_day12[[6]],qqplot_red_day12[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 2 to 3
plot(qqplot_red_day23[[4]],qqplot_red_day23[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;, main = &quot;Day 2 to 3&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_red_day23[[6]],qqplot_red_day23[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 0 to 2
plot(qqplot_red_day02[[4]],qqplot_red_day02[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;, main = &quot;Day 0 to 2&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_red_day02[[6]],qqplot_red_day02[[7]], pch = 16, col = pal[3])

# Add bottom title

mtext(&quot;Reduction in Variation&quot;, font = 2, side = 3, line = -1.5, outer = TRUE)

### Increase in variance

# Inc variance to day 0 to 1
plot(qqplot_inc_day01[[4]],qqplot_inc_day01[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;,  main = &quot;Day 0 to 1&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_inc_day01[[6]],qqplot_inc_day01[[7]], pch = 16, col = pal[3])

# Increase of variance to day 1 to 2
plot(qqplot_inc_day12[[4]],qqplot_inc_day12[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;,  main = &quot;Day 1 to 2&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_inc_day12[[6]],qqplot_inc_day12[[7]], pch = 16, col = pal[3])


# Increase of variance to day 2 to 3
plot(qqplot_inc_day23[[4]],qqplot_inc_day23[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;,  main = &quot;Day 2 to 3&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_inc_day23[[6]],qqplot_inc_day23[[7]], pch = 16, col = pal[3])

# Increase of variance to day 0 to 2
plot(qqplot_inc_day02[[4]],qqplot_inc_day02[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;,  main = &quot;Day 0 to 2&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_inc_day02[[6]],qqplot_inc_day02[[7]], pch = 16, col = pal[3])


mtext(&quot;Increase in Variation&quot;, font = 2, side = 3, line = -29, outer = TRUE)

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
legend(&quot;bottomright&quot;, c(&quot;Chimpanzee&quot;, &quot;Chimpanzee|significant in humans&quot;), xpd = TRUE, horiz = TRUE, inset = c(0,  0), bty = &quot;n&quot;, pch = c(16, 16), col = pal[2:3], cex = 1.5)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-28-21.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#legend(&quot;top&quot;, c(&quot;Chimpanzee&quot;, &quot;Chimpanzee|significant in humans&quot;), xpd = TRUE, horiz = TRUE, inset = c(0,  0), bty = &quot;n&quot;, pch = c(16, 16), col = pal[2:3], cex = 0.7)</code></pre>
</div>
<div id="estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.05" class="section level3">
<h3>Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.05)</h3>
<pre class="r"><code>library(&quot;qvalue&quot;)
################ REDUCTION IN VARIATION FOR CHIMPS (p = 0.05) ################

# Day 0 to 1


# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day01[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.7259161</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.2740839</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.5348868</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.4651132</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Chimpanzee|Sig. in Humans&quot;, dim = c(as.data.frame(qqplot_red_day01[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p1_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(-0.2,1.2), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +  scale_y_continuous(labels=scaleFUN) +     
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.73&quot;)&#39;, parse=TRUE, x=0.4, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.53&quot;)&#39;, parse=TRUE, x=0.4, y=(pi0est_chimps_given_humans+0.07), size = 6)

p1_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pal &lt;- c( &quot;#E77642&quot;, &quot;gray&quot;) 

ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) + geom_histogram(binwidth = 0.07)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(chimps[,1:2], aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) + geom_histogram(aes(y=..density..), binwidth = 0.07) + geom_density() </code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code> ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) + geom_density() +   scale_x_continuous(limits = c(-0.2,1.2), breaks = seq(0,1,0.25))</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +  geom_histogram(alpha = 1, aes(y = ..density..), position = &#39;identity&#39;, bins = 15, show.legend = FALSE) + theme_bw() + facet_wrap(~Species) + labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + bjp +
  geom_hline(data=data.frame(Species = &quot;Chimpanzee&quot;), aes(yintercept = 0.27,  show.legend = NA)) + geom_hline(data=data.frame(Species = &quot;Chimpanzee|Sig. in Humans&quot;), aes(yintercept = 0.47, show.legend = NA))</code></pre>
<pre><code>Warning: Ignoring unknown aesthetics: show.legend

Warning: Ignoring unknown aesthetics: show.legend</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +  geom_histogram(alpha = 0.6, aes(y = ..density..), position = &#39;identity&#39;, bins = 15, show.legend = FALSE, color = &quot;black&quot;) + theme_bw()</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(all_pval, aes(pval)) + 
  geom_histogram(data=subset(all_pval, Species == &#39;Chimpanzee&#39;), alpha = 1, aes(y = ..density..), position = &#39;identity&#39;, bins = 15, show.legend = FALSE, fill = &quot;#E77642&quot;, color = &quot;black&quot;) +
  geom_histogram(data=subset(all_pval, Species == &#39;Chimpanzee|Sig. in Humans&#39;), alpha = 0.8, aes(y = ..density..), position = &#39;identity&#39;, bins = 15, show.legend = FALSE, fill = &quot;gray&quot;)  + labs(x = &quot;Unadjusted p-values&quot;, y = &quot;Density&quot;) +  theme_bw() + bjpm +  scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 1.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 1.5) +     
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.73&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.07), size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.53&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6) + labs(x = &quot;Unadjusted p-values&quot;)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(all_pval, aes(pval)) + 
  geom_histogram(data=subset(all_pval, Species == &#39;Chimpanzee&#39;), alpha = 1, aes(y = ..density..), position = &#39;identity&#39;, bins = 15, show.legend = FALSE, fill = &quot;#E77642&quot;, color = &quot;black&quot;) +
  geom_histogram(data=subset(all_pval, Species == &#39;Chimpanzee|Sig. in Humans&#39;), alpha = 0.8, aes(y = ..density..), position = &#39;identity&#39;, bins = 15, show.legend = FALSE, fill = &quot;gray&quot;) + theme_bw() + bjpm +  scale_y_continuous(labels=scaleFUN) + labs(x = &quot;Unadjusted p-values&quot;, y = &quot;Density&quot;) +
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.73&quot;)&#39;, parse=TRUE, x=0.83, y=1.3, size = 7, colour = &quot;#E77642&quot;) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.53&quot;)&#39;, color = &quot;grey32&quot;, parse=TRUE, x=0.83, y=1.2, size = 7)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(all_pval, aes(pval, fill = as.factor(Species))) + scale_fill_manual(values=pal[2:3]) + geom_density(data=subset(all_pval, Species == &#39;Chimpanzee&#39;), alpha = 1, fill = &quot;#E77642&quot;) + geom_density(data=subset(all_pval, Species == &#39;Chimpanzee|Sig. in Humans&#39;), alpha = 0.5, fill = &quot;cyan4&quot;) + theme_bw() +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +  scale_y_continuous(labels=scaleFUN) +     
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.73&quot;)&#39;, parse=TRUE, x=0.4, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.53&quot;)&#39;, parse=TRUE, x=0.4, y=(pi0est_chimps_given_humans+0.07), size = 6)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>neg_p &lt;- as.data.frame(as.numeric(-(all_pval$pval)))

chimps_neg &lt;- cbind(labels_c, neg_p[1:10304,])
colnames(chimps_neg) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans_neg &lt;- cbind(labels_h, neg_p[10305:12613,])
colnames(humans_neg) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)


all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])


right_side &lt;- as.data.frame(2-all_pval$pval)

chimps_rs &lt;- cbind(labels_c, right_side[1:10304,])
colnames(chimps_rs) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans_rs &lt;- cbind(labels_h, right_side[10305:12613,])
colnames(humans_rs) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)


all_pval_ref &lt;- as.data.frame(rbind(chimps[,1:2], humans[,1:2], chimps_neg, humans_neg, chimps_rs, humans_rs   ))
all_pval_ref$pval &lt;- as.numeric(all_pval_ref$pval)

ggplot(all_pval_ref, aes(pval, fill = as.factor(Species))) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) + theme_bw() +
  scale_x_continuous(limits = c(-0.5,1.5), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +  scale_y_continuous(labels=scaleFUN) +     
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.73&quot;)&#39;, parse=TRUE, x=0.4, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.53&quot;)&#39;, parse=TRUE, x=0.4, y=(pi0est_chimps_given_humans+0.07), size = 6)</code></pre>
<pre><code>Warning: Removed 12613 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Day 1 to 2


# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day12[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.2469456</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.7530544</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day12[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p2_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.5, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.25&quot;)&#39;, parse=TRUE, x=0.5, y=(pi0est_chimps_given_humans+0.07), size = 6)

p2_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Day 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))
line &lt;- rep(c(&quot;NA&quot;), times = nrow(pval_c)+nrow(pval_h))
figS12A &lt;- cbind(pval_ch, resp, line)

# Day 2 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day23[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.903342</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.09665802</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day23[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p3_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.9&quot;)&#39;, parse=TRUE, x=0.8, y=(pi0est_chimps-0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.8, y=(pi0est_chimps_given_humans+0.07), size = 6)

p3_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Day 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))
line &lt;- rep(c(&quot;NA&quot;), times = nrow(pval_c)+nrow(pval_h))
figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)


# Day 1 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day13[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day13[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p4_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;black&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.2), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p4_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-29-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Day 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))
line &lt;- rep(c(&quot;NA&quot;), times = nrow(pval_c)+nrow(pval_h))
figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS13A.csv&quot;, quote = FALSE, row.names = FALSE)</code></pre>
</div>
<div id="estimates-of-pi0-and-pi1-for-increase-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.05" class="section level3">
<h3>Estimates of pi0 and pi1 for increase of variation in chimps conditioned on significance in humans (p = 0.05)</h3>
<pre class="r"><code>############### Increase in Variation ############

# Day 0 to 1

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day01[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p1_inc_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.3), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.15), size = 6)

p1_inc_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Day 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))
line &lt;- rep(c(&quot;NA&quot;), times = nrow(pval_c)+nrow(pval_h))
figS12A &lt;- cbind(pval_ch, resp, line)


# Day 1 to 2

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.5221613</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.4778387</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.3388645</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.6611355</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day12[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p2_inc_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.52&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.34&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p2_inc_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-30-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Day 2 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.3652978</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.6347022</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.2443566</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.7556434</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day23[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p3_inc_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) + scale_y_continuous(labels=scaleFUN) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.37&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.24&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.06), size = 6)

p3_inc_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-30-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Day 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.37&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.24&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)


write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS13C.csv&quot;, quote = FALSE, row.names = FALSE)

# Day 1 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.5071291</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.4928709</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.3889643</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.6110357</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day13[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p4_inc_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) + scale_y_continuous(labels=scaleFUN) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.51&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.08), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.39&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans-0.08), size = 6)

p4_inc_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-30-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.51&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.39&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
</div>
<div id="changes-of-variation-in-humans" class="section level3">
<h3>Changes of variation in humans</h3>
<pre class="r"><code># Reduction in variance from day 0 to day 1 humans

qqplot_red_day01 &lt;- find_qqplot_red_humans(7,10,17,20,1,6,11,16, 0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 2 humans

qqplot_red_day12 &lt;- find_qqplot_red_humans(17,20,27,30,11,16,21,26,0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 2 to day 3 humans

qqplot_red_day23 &lt;- find_qqplot_red_humans(27,30,37,40,21,26,31,36,0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 3 humans

qqplot_red_day13 &lt;- find_qqplot_red_humans(17,20,37,40,11,16,31,36,0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 0 to day 2 humans

qqplot_red_day02 &lt;- find_qqplot_red_humans(7,10,27,30,1,6,21,26,0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 1 humans

qqplot_inc_day01 &lt;- find_qqplot_inc_humans(7,10,17,20,1,6,11,16,0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 2 humans

qqplot_inc_day12 &lt;- find_qqplot_inc_humans(17,20,27,30,11,16,21,26,0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 2 to day 3 humans

qqplot_inc_day23 &lt;- find_qqplot_inc_humans(27,30,37,40,21,26,31,36,0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 2

qqplot_inc_day02 &lt;- find_qqplot_inc_humans(7,10,27,30,1,6,21,26,0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 3 humans

qqplot_inc_day13 &lt;- find_qqplot_inc_humans(17,20,37,40,11,16,31,36,0.05)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-31-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction of variance qqplots- humans

dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>par(mfrow=c(2,4),oma = c(2, 2, 2, 2))

# Reduction of variance to day 0 to 1 humans
plot(qqplot_red_day01[[4]],qqplot_red_day01[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;, main = &quot;Day 0 to 1&quot;)
abline(0,1, col = &quot;black&quot;)

# Add to the plot
points(qqplot_red_day01[[6]],qqplot_red_day01[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 1 to 2
plot(qqplot_red_day12[[4]],qqplot_red_day12[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;, main = &quot;Day 1 to 2&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_red_day12[[6]],qqplot_red_day12[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 2 to 3
plot(qqplot_red_day23[[4]],qqplot_red_day23[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;, main = &quot;Day 2 to 3&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_red_day23[[6]],qqplot_red_day23[[7]], pch = 16, col = pal[3])

# Reduction of variance to day 0 to 2
plot(qqplot_red_day02[[4]],qqplot_red_day02[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;, main = &quot;Day 0 to 2&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_red_day02[[6]],qqplot_red_day02[[7]], pch = 16, col = pal[3])

# Add bottom title

mtext(&quot;Reduction in Variation&quot;, font = 2, side = 3, line = -1.5, outer = TRUE)

### Increase in variance

# Inc variance to day 0 to 1
plot(qqplot_inc_day01[[4]],qqplot_inc_day01[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;,  main = &quot;Day 0 to 1&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_inc_day01[[6]],qqplot_inc_day01[[7]], pch = 16, col = pal[3])

# Increase of variance to day 1 to 2
plot(qqplot_inc_day12[[4]],qqplot_inc_day12[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;,  main = &quot;Day 1 to 2&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_inc_day12[[6]],qqplot_inc_day12[[7]], pch = 16, col = pal[3])


# Increase of variance to day 2 to 3
plot(qqplot_inc_day23[[4]],qqplot_inc_day23[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;,  main = &quot;Day 2 to 3&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_inc_day23[[6]],qqplot_inc_day23[[7]], pch = 16, col = pal[3])

# Increase of variance to day 0 to 2
plot(qqplot_inc_day02[[4]],qqplot_inc_day02[[5]], pch = 16, xlim = c(0,6), ylim = c(0,6), col = pal[2], xlab = &quot;-log10(p) Expected&quot;, ylab = &quot;-log10(p) Obeserved&quot;,  main = &quot;Day 0 to 2&quot;)
abline(0,1, col = &quot;black&quot;)


# Add to the plot
points(qqplot_inc_day02[[6]],qqplot_inc_day02[[7]], pch = 16, col = pal[3])


mtext(&quot;Increase in Variation&quot;, font = 2, side = 3, line = -29, outer = TRUE)

par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
legend(&quot;bottomright&quot;, c(&quot;Human&quot;, &quot;Human|significant in chimpanzee&quot;), xpd = TRUE, horiz = TRUE, inset = c(0,  0), bty = &quot;n&quot;, pch = c(16, 16), col = pal[2:3], cex = 1.5)

#legend(&quot;top&quot;, c(&quot;Human&quot;, &quot;Human|significant in chimpanzee&quot;), xpd = TRUE, horiz = TRUE, inset = c(0,  0), bty = &quot;n&quot;, pch = c(16, 16), col = pal[2:3], cex = 0.7)</code></pre>
</div>
<div id="estimates-of-pi0-and-pi1-for-reduction-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.05" class="section level3">
<h3>Estimates of pi0 and pi1 for reduction of variation in humans conditioned on significance in chimps (p = 0.05)</h3>
<pre class="r"><code>####################### REDUCTION OF VARIATION IN HUMANS (p = 0.05)


# Day 0 to 1

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day01[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.174912</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.825088</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.1556767</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.8443233</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day01[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p5_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.17&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.32), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.16&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p5_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Day 1 to 2

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day12[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day12[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p6_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p6_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-32-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_h &lt;- humans[,1:2]
pval_c &lt;- chimps[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Day 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)


# Day 2 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day23[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.8664877</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.1335123</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.8071014</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.1928986</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day23[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p7_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.87&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.81&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.3), size = 6)

p7_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-32-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_h &lt;- humans[,1:2]
pval_c &lt;- chimps[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Day 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS13B.csv&quot;, quote = FALSE, row.names = FALSE)

# Day 1 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day13[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day13[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p8_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.30), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.12), size = 6)

p8_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-32-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_h &lt;- humans[,1:2]
pval_c &lt;- chimps[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Day 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

multiplot(p1_05,p5_05,p2_05,p6_05,p3_05,p7_05,p4_05,p8_05,cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-32-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="estimates-of-pi0-and-pi1-for-increase-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.05" class="section level3">
<h3>Estimates of pi0 and pi1 for increase of variation in humans conditioned on significance in chimps (p = 0.05)</h3>
<pre class="r"><code>####################### INCREASE IN VARIATION IN HUMANS (p = 0.05)


# Day 0 to 1

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day01[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p5_inc_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +  scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.32), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p5_inc_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_h &lt;- humans[,1:2]
pval_c &lt;- chimps[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Day 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)


# Day 1 to 2

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.2141993</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.7858007</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.09012664</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.9098734</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day12[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p6_inc_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +  scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.21&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.32), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.09&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.27), size = 6)

p6_inc_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-33-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Day 2 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.8477893</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.1522107</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.9526267</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.04737329</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day23[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p7_inc_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +  scale_y_continuous(labels=scaleFUN) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + 
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.85&quot;)&#39;, parse=TRUE, x=0.25, y=(pi0est_humans+0.05), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.95&quot;)&#39;, parse=TRUE, x=0.25, y=(pi0est_humans_given_chimps+0.07), size = 6)

p7_inc_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-33-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_h &lt;- humans[,1:2]
pval_c &lt;- chimps[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Day 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.95&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.85&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

#write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS13D.csv&quot;, quote = FALSE, row.names = FALSE)


# Day 1 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.203399</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.796601</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.177843</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.822157</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day13[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p8_inc_05 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +  scale_y_continuous(labels=scaleFUN) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + 
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.20&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.18&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.16), size = 6)

p8_inc_05</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-33-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_h &lt;- humans[,1:2]
pval_c &lt;- chimps[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Day 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.20&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.18&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

multiplot(p1_inc_05,p5_inc_05,p2_inc_05,p6_inc_05,p3_inc_05,p7_inc_05,p4_inc_05,p8_inc_05,cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-33-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="permutation-tests-this-code-should-not-be-run-on-a-laptop-computer.-use-a-hpc-instead.-for-sample-results-see-supplemental-data-s15-of-the-endoderm-paper" class="section level2">
<h2>Permutation tests (This code should not be run on a laptop computer. Use a HPC instead. For sample results, see Supplemental Data S15 of the Endoderm paper)</h2>
<div id="permutation-test-to-find-the-null-hypothesis-of-sharing-reduction-in-chimps-significant-in-humans-for-day-0-to-1-with-p-value-cutoff-of-0.05" class="section level3">
<h3>Permutation test to find the null hypothesis of sharing (reduction in chimps | significant in humans for day 0 to 1 with p-value cutoff of 0.05)</h3>
<pre class="r"><code>### library

#library(&quot;picante&quot;)

### Make a matrix of the p-values ###

  # Chimp p-values
#chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

#for(i in 1:10304){
#     x &lt;- t(mean_tech_reps[i,7:10])
#     y &lt;- t(mean_tech_reps[i,17:20])
#      htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
#     chimp_var_pval[i,1] &lt;- htest$p.value

#}

  # Human p-values

#human_var_pval &lt;- array(NA, dim = c(10304, 1))

#for(i in 1:10304){
#     x &lt;- t(mean_tech_reps[i,1:6])
#     y &lt;- t(mean_tech_reps[i,11:16])
#     htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
#     human_var_pval[i,1] &lt;- htest$p.value
# }
 

### Matrix for the permuation values

#shared_p0_given_permutation &lt;- array(NA, dim = c(100000, 1))
  
### Randomize chimp p-values ###  

#for(i in 1:100000){
#out_p_val &lt;- randomizeMatrix(chimp_var_pval, null.model = &quot;frequency&quot;,iterations = 1000)

#var_pval &lt;- as.data.frame(cbind(out_p_val, human_var_pval))
#rownames(var_pval) &lt;- rownames(mean_tech_reps)
#colnames(var_pval) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;)
#summary(var_pval)
  
# Set p-value

#p_val &lt;- 0.05

# P-val &lt; 0.05 for chimps only
#  num_var_pval_chimps &lt;- var_pval[ which(var_pval[,1] &lt; p_val), ]

                    
# P-val &lt; 0.05 for humans only
#  num_var_pval_humans &lt;- var_pval[ which(var_pval[,2] &lt; p_val), ]
  
# Run for the shared
  # Find the p-values of the chimps given that it was significant in the humans

#  subset_var &lt;- as.data.frame(var_pval[ which(var_pval[,2] &lt; p_val), 1:2])
#  num_var_pval_chimp_given_human_no_df &lt;- subset_var[,1]
#  num_var_pval_chimp_given_human &lt;- as.data.frame(subset_var[,1])
 

### Find the Pi_0 estimate for the chimps given significant in the humans

# Obtain pvalues
#chimp_p_values_gen &lt;- as.data.frame(num_var_pval_chimps)
#shared_p_values_gen &lt;- as.data.frame(num_var_pval_chimp_given_human)



# Pi_0 estimate for the chimps given significant in the humans
#pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#shared_p0_given_permutation[i,1] &lt;- pi0est_chimps_given_humans
#print(i)
#}

#exp_p1 &lt;- 1-mean(shared_p0_given_permutation)
#exp_p1
#sd_p1 &lt;- sd(1-shared_p0_given_permutation)

#1-pnorm(0.47, exp_p1, sd_p1)</code></pre>
</div>
<div id="permutation-test-to-find-the-null-hypothesis-of-sharing-reduction-in-humans-significant-in-chimps-for-day-0-to-1-with-p-value-cutoff-of-0.05" class="section level3">
<h3>Permutation test to find the null hypothesis of sharing (reduction in humans | significant in chimps for day 0 to 1 with p-value cutoff of 0.05)</h3>
<pre class="r"><code>### Now do the same thing but with humans given significant in chimps


### Make a matrix of the p-values ###

  # Chimp p-values
#chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

#for(i in 1:10304){
#     x &lt;- t(mean_tech_reps[i,7:10])
#     y &lt;- t(mean_tech_reps[i,17:20])
#      htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
#     chimp_var_pval[i,1] &lt;- htest$p.value

# }

  # Human p-values

# human_var_pval &lt;- array(NA, dim = c(10304, 1))

#for(i in 1:10304){
#     x &lt;- t(mean_tech_reps[i,1:6])
#     y &lt;- t(mean_tech_reps[i,11:16])
#     htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
#     human_var_pval[i,1] &lt;- htest$p.value
# }

### Matrix for the permuation values

# shared_p0_given_permutation_human &lt;- array(NA, dim = c(100000, 1))
  
### Randomize chimp p-values ###  

# for(i in 1:100000){
# out_p_val &lt;- randomizeMatrix(human_var_pval,null.model = &quot;frequency&quot;,iterations = 1000)

# var_pval &lt;- as.data.frame(cbind(out_p_val, chimp_var_pval))
# rownames(var_pval) &lt;- rownames(mean_tech_reps)
# colnames(var_pval) &lt;- c(&quot;Human&quot;, &quot;Chimpanzee&quot;)
# summary(var_pval)
  
# Set p-value

# p_val &lt;- 0.05

# P-val &lt; 0.05 for chimps only
#  num_var_pval_humans &lt;- var_pval[ which(var_pval[,1] &lt; p_val), ]

                    
# P-val &lt; 0.05 for humans only
#  num_var_pval_chimps &lt;- var_pval[ which(var_pval[,2] &lt; p_val), ]
  
# Run for the shared
  # Find the p-values of the humans given that it was significant in the chimps

#  subset_var &lt;- as.data.frame(var_pval[ which(var_pval[,2] &lt; p_val), 1:2])
#  num_var_pval_human_given_chimp_no_df &lt;- subset_var[,1]
#  num_var_pval_human_given_chimp &lt;- as.data.frame(subset_var[,1])
 

### Find the Pi_0 estimate for the humans given significant in the chimps

# Obtain pvalues
# shared_p_values_gen &lt;- as.data.frame(num_var_pval_human_given_chimp)



# Pi_0 estimate for the humans given significant in the chimps

# pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
# shared_p0_given_permutation_human[i,1] &lt;- pi0est_humans_given_chimps

# print(i)
# }


# comp_mean &lt;- 1-mean(shared_p0_given_permutation_human)
# comp_mean

# sd_human &lt;- 1-shared_p0_given_permutation_human

# 1-pnorm(0.84, comp_mean, sd(sd_human))</code></pre>
</div>
<div id="estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.01" class="section level3">
<h3>Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.01)</h3>
</div>
<div id="changes-in-variance-in-chimp-samples-p-value-cutoff-0.01" class="section level3">
<h3>Changes in variance in chimp samples (p-value cutoff 0.01)</h3>
<pre class="r"><code># Reduction in variance from day 0 to day 1 chimps

qqplot_red_day01 &lt;- find_qqplot_red_chimps(7,10,17,20,1,6,11,16, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 2 chimps

qqplot_red_day12 &lt;- find_qqplot_red_chimps(17,20,27,30,11,16,21,26, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 2 to day 3 chimps

qqplot_red_day23 &lt;- find_qqplot_red_chimps(27,30,37,40,21,26,31,36, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 0 to day 2 chimps

qqplot_red_day02 &lt;- find_qqplot_red_chimps(7,10,27,30,1,6,21,26, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 3 chimps

qqplot_red_day13 &lt;- find_qqplot_red_chimps(17,20,37,40,11,16,31,36, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 1

qqplot_inc_day01 &lt;- find_qqplot_inc_chimps(7,10,17,20,1,6,11,16, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 2

qqplot_inc_day12 &lt;- find_qqplot_inc_chimps(17,20,27,30,11,16,21,26, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 2 to day 3

qqplot_inc_day23 &lt;- find_qqplot_inc_chimps(27,30,37,40,21,26,31,36, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 2

qqplot_inc_day02 &lt;- find_qqplot_inc_chimps(7,10,27,30,1,6,21,26, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 3

qqplot_inc_day13 &lt;- find_qqplot_inc_chimps(17,20,37,40,11,16,31,36, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-36-20.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.01-1" class="section level3">
<h3>Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.01)</h3>
<pre class="r"><code>library(&quot;qvalue&quot;)
################ REDUCTION IN VARIATION FOR CHIMPS (p = 0.01) ################

# Day 0 to 1


# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day01[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.7259161</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.2740839</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.4760713</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.5239287</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day01[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p1_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)  +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.73&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.48&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p1_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.73&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.48&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)

# Day 1 to 2


# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day12[2])

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day12[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p1_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Days 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;)

p1_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-37-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)



# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

# pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0

# Note: There&#39;s an issue with estimating pi_0 from only 15 points. 


# Day 2 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day23[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.903342</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.09665802</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day23[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p3_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.9&quot;)&#39;, parse=TRUE, x=0.8, y=(pi0est_chimps-0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.8, y=(pi0est_chimps_given_humans+0.07), size = 6)

p3_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-37-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

# Day 1 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day13[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day13[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p4_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;black&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.2), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p4_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-37-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14E_with_days12.csv&quot;, quote = FALSE, row.names = FALSE)</code></pre>
</div>
<div id="estimates-of-pi0-and-pi1-for-increase-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.01" class="section level3">
<h3>Estimates of pi0 and pi1 for increase of variation in chimps conditioned on significance in humans (p = 0.01)</h3>
<pre class="r"><code>############### Increase in Variation ############

# Day 0 to 1

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[2])

# Pi_0 estimate for the chimps

#pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_chimps

# Pi_1 estimate for the chimps

#pi1est_chimps &lt;- 1 - pi0est_chimps
#pi1est_chimps

# Pi_0 estimate for the chimps given significant in the humans

#pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_chimps_given_humans

# Pi_1 estimate for the humans

#pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
#pi1est_chimps_given_humans

# Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(nrow(shared_p_values_gen[1]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p1_inc_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.80&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p1_inc_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)</code></pre>
<pre class="r"><code># Day 1 to 2

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.5221613</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.4778387</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.448485</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.551515</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day12[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p2_inc_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.52&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.45&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans-0.07), size = 6)

p2_inc_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.52&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.45&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
<pre class="r"><code># Day 2 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.3652978</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.6347022</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.3625405</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.6374595</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day23[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p3_inc_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.37&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.15), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.36&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p3_inc_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.37&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.36&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
<pre class="r"><code># Day 1 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.5071291</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.4928709</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.2368051</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.7631949</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day13[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p4_inc_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.51&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.08), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.24&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.08), size = 6)

p4_inc_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-41-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.51&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.24&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

#write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14I.csv&quot;, quote = FALSE, row.names = FALSE)</code></pre>
</div>
<div id="changes-of-variation-in-humans-p-0.01" class="section level3">
<h3>Changes of variation in humans (p = 0.01)</h3>
<pre class="r"><code># Reduction in variance from day 0 to day 1 humans

qqplot_red_day01 &lt;- find_qqplot_red_humans(7,10,17,20,1,6,11,16, 0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 2 humans

qqplot_red_day12 &lt;- find_qqplot_red_humans(17,20,27,30,11,16,21,26,0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 2 to day 3 humans

qqplot_red_day23 &lt;- find_qqplot_red_humans(27,30,37,40,21,26,31,36,0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 3 humans

qqplot_red_day13 &lt;- find_qqplot_red_humans(17,20,37,40,11,16,31,36,0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 0 to day 2 humans

qqplot_red_day02 &lt;- find_qqplot_red_humans(7,10,27,30,1,6,21,26,0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 1 humans

qqplot_inc_day01 &lt;- find_qqplot_inc_humans(7,10,17,20,1,6,11,16,0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 2 humans

qqplot_inc_day12 &lt;- find_qqplot_inc_humans(17,20,27,30,11,16,21,26,0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 2 to day 3 humans

qqplot_inc_day23 &lt;- find_qqplot_inc_humans(27,30,37,40,21,26,31,36,0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 2

qqplot_inc_day02 &lt;- find_qqplot_inc_humans(7,10,27,30,1,6,21,26,0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day ` to day 3 humans

qqplot_inc_day13 &lt;- find_qqplot_inc_humans(17,20,37,40,11,16,31,36,0.01)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-42-20.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="estimates-of-pi0-and-pi1-for-reduction-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.01" class="section level3">
<h3>Estimates of pi0 and pi1 for reduction of variation in humans conditioned on significance in chimps (p = 0.01)</h3>
<pre class="r"><code>####################### REDUCTION OF VARIATION IN HUMANS (p = 0.01)


# Day 0 to 1

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day01[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.174912</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.825088</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.1469146</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.8530854</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day01[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p5_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.17&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.15&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p5_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-43-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.17&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.15&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)</code></pre>
<pre class="r"><code># Day 1 to 2

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day12[2])


# Pi_0 estimate for the humans

#pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans

# Pi_1 estimate for the humans

#pi1est_humans &lt;- 1 - pi0est_humans
#pi1est_humans

# Pi_0 estimate for the chimps given significant in the humans

#pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, #fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans_given_chimps

# Pi_1 estimate for the humans

#pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
#pi1est_humans_given_chimps

# Note: There&#39;s an issue with estimating pi_0 from only 15 points. 


# Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day12[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p6_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p6_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
<pre class="r"><code># Day 2 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day23[2])


# Pi_0 estimate for the humans

#pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans

# Pi_1 estimate for the humans

#pi1est_humans &lt;- 1 - pi0est_humans
#pi1est_humans

# Pi_0 estimate for the chimps given significant in the humans

#pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans_given_chimps

# Pi_1 estimate for the humans

#pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
#pi1est_humans_given_chimps

# Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day23[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p7_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.87&quot;)&#39;, parse=TRUE, x=0.6, y=(pi0est_humans+0.07), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.36&quot;)&#39;, parse=TRUE, x=0.6, y=(pi0est_humans_given_chimps+0.07), size = 6)

p7_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
<pre class="r"><code># Day 1 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day13[2])


# Pi_0 estimate for the humans

#pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans

# Pi_1 estimate for the humans

#pi1est_humans &lt;- 1 - pi0est_humans
#pi1est_humans

# Pi_0 estimate for the chimps given significant in the humans

#pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans_given_chimps

# Pi_1 estimate for the humans

#pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
#pi1est_humans_given_chimps

# Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day13[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p8_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) + 
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p8_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

#write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14G.csv&quot;, quote = FALSE, row.names = FALSE)

multiplot(p1_01,p5_01,p3_01,p6_01, p4_01,p7_01,p8_01,cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-46-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#multiplot(p1_01,p3_01,p4_01,p5_01,p6_01,p7_01,p8_01,cols=2)</code></pre>
</div>
<div id="estimates-of-pi0-and-pi1-for-increase-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.01" class="section level3">
<h3>Estimates of pi0 and pi1 for increase of variation in humans conditioned on significance in chimps (p = 0.01)</h3>
<pre class="r"><code>####################### INCREASE IN VARIATION IN HUMANS (p = 0.01)


# Day 0 to 1

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[2])


# Pi_0 estimate for the humans

#pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans

# Pi_1 estimate for the humans

#pi1est_humans &lt;- 1 - pi0est_humans
#pi1est_humans

# Pi_0 estimate for the chimps given significant in the humans

#pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans_given_chimps

# Pi_1 estimate for the humans

#pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
#pi1est_humans_given_chimps

# Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day01[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p5_inc_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) + scale_y_continuous(labels=scaleFUN) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p5_inc_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)</code></pre>
<pre class="r"><code># Day 1 to 2

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[2])


# Pi_0 estimate for the humans

#pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans

# Pi_1 estimate for the humans

#pi1est_humans &lt;- 1 - pi0est_humans
#pi1est_humans

# Pi_0 estimate for the chimps given significant in the humans

#pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans_given_chimps

# Pi_1 estimate for the humans

#pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
#pi1est_humans_given_chimps

# Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day12[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p6_inc_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) + scale_y_continuous(labels=scaleFUN) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.21&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.32), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.07&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.27), size = 6)

p6_inc_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.21&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.07&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
<pre class="r"><code># Day 2 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[2])


# Pi_0 estimate for the humans

#pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans

# Pi_1 estimate for the humans

#pi1est_humans &lt;- 1 - pi0est_humans
#pi1est_humans

# Pi_0 estimate for the chimps given significant in the humans

#pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
#pi0est_humans_given_chimps

# Pi_1 estimate for the humans

#pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
#pi1est_humans_given_chimps

# Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day23[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p7_inc_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) + scale_y_continuous(labels=scaleFUN) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + 
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.85&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.05), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.07), size = 6)

p7_inc_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;1&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.85&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)</code></pre>
<pre class="r"><code># Day 1 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.203399</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.796601</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.1274361</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.8725639</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day13[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p8_inc_01 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) + scale_y_continuous(labels=scaleFUN) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + 
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.20&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.35), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.13&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.2), size = 6)

p8_inc_01</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-50-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.20&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.13&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

#write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14K.csv&quot;, quote = FALSE, row.names = FALSE)

multiplot(p1_inc_01,p5_inc_01,p2_inc_01,p6_inc_01,p3_inc_01,p7_inc_01,p4_inc_01,p8_inc_01,cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-50-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.1" class="section level2">
<h2>Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.1)</h2>
<div id="changes-in-variance-in-chimp-samples-p-value-cutoff-0.1" class="section level3">
<h3>Changes in variance in chimp samples (p-value cutoff 0.1)</h3>
<pre class="r"><code># Reduction in variance from day 0 to day 1 chimps

qqplot_red_day01 &lt;- find_qqplot_red_chimps(7,10,17,20,1,6,11,16, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 2 chimps

qqplot_red_day12 &lt;- find_qqplot_red_chimps(17,20,27,30,11,16,21,26, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 2 to day 3 chimps

qqplot_red_day23 &lt;- find_qqplot_red_chimps(27,30,37,40,21,26,31,36, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 0 to day 2 chimps

qqplot_red_day02 &lt;- find_qqplot_red_chimps(7,10,27,30,1,6,21,26, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 3 chimps

qqplot_red_day13 &lt;- find_qqplot_red_chimps(17,20,37,40,11,16,31,36, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 1

qqplot_inc_day01 &lt;- find_qqplot_inc_chimps(7,10,17,20,1,6,11,16, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 2

qqplot_inc_day12 &lt;- find_qqplot_inc_chimps(17,20,27,30,11,16,21,26, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 2 to day 3

qqplot_inc_day23 &lt;- find_qqplot_inc_chimps(27,30,37,40,21,26,31,36, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 2

qqplot_inc_day02 &lt;- find_qqplot_inc_chimps(7,10,27,30,1,6,21,26, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 3

qqplot_inc_day13 &lt;- find_qqplot_inc_chimps(17,20,37,40,11,16,31,36, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-51-20.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="estimates-of-pi0-and-pi1-for-reduction-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.1-1" class="section level3">
<h3>Estimates of pi0 and pi1 for reduction of variation in chimps conditioned on significance in humans (p = 0.1)</h3>
<pre class="r"><code>library(&quot;qvalue&quot;)
################ REDUCTION IN VARIATION FOR CHIMPS (p = 0.1) ################

# Day 0 to 1


# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day01[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.7259161</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.2740839</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.5197874</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.4802126</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day01[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p1_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.73&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.52&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p1_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-52-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.73&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.52&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)
   

# Day 1 to 2


# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day12[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.7243204</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.2756796</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day12[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p2_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) + scale_y_continuous(labels=scaleFUN) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.5, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.72&quot;)&#39;, parse=TRUE, x=0.5, y=(pi0est_chimps_given_humans+0.07), size = 6)

p2_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-52-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

# Day 2 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day23[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.903342</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.09665802</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day23[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p3_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.9&quot;)&#39;, parse=TRUE, x=0.8, y=(pi0est_chimps-0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.8, y=(pi0est_chimps_given_humans+0.07), size = 6)

p3_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-52-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

# Day 1 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_red_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day13[2])


# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_red_day13[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p4_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;black&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.17), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p4_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-52-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

#write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14F.csv&quot;, quote = FALSE, row.names = FALSE)</code></pre>
</div>
<div id="estimates-of-pi0-and-pi1-for-increase-of-variation-in-chimps-conditioned-on-significance-in-humans-p-0.10" class="section level3">
<h3>Estimates of pi0 and pi1 for increase of variation in chimps conditioned on significance in humans (p = 0.10)</h3>
<pre class="r"><code>############### Increase in Variation ############

# Day 0 to 1

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.9219031</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.07809693</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(nrow(shared_p_values_gen[1]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p1_inc_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.92&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans-0.07), size = 6)

p1_inc_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-53-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)

# Day 1 to 2

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.5221613</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.4778387</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.3286427</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.6713573</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day12[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p2_inc_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.52&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.07), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.33&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p2_inc_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-53-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.52&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.33&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

# Day 2 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.3652978</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.6347022</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.3533745</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.6466255</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day23[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p3_inc_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.37&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.13), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.35&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans+0.07), size = 6)

p3_inc_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-53-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.37&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.35&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)


# Day 1 to 3

# Obtain pvalues
chimp_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[2])

# Pi_0 estimate for the chimps

pi0est_chimps &lt;- qvalue(p=chimp_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps</code></pre>
<pre><code>[1] 0.5071291</code></pre>
<pre class="r"><code># Pi_1 estimate for the chimps

pi1est_chimps &lt;- 1 - pi0est_chimps
pi1est_chimps</code></pre>
<pre><code>[1] 0.4928709</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_chimps_given_humans &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_chimps_given_humans</code></pre>
<pre><code>[1] 0.4191553</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_chimps_given_humans &lt;- 1 - pi0est_chimps_given_humans
pi1est_chimps_given_humans</code></pre>
<pre><code>[1] 0.5808447</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_c &lt;- array(&quot;Chimpanzee&quot;, dim = c(10304,1))
labels_h &lt;- array(&quot;Shared (Chimpanzee|Significant in Humans)&quot;, dim = c(as.data.frame(qqplot_inc_day13[[3]][2]),1))

chimps &lt;- cbind(labels_c, chimp_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, shared_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(chimps[,1:2], humans[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;#E77642&quot;, &quot;white&quot;) 

p4_inc_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[2:3]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) + scale_y_continuous(labels=scaleFUN) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_chimps_given_humans, size = 2.5) + 
  geom_hline(yintercept = pi0est_chimps, size = 2.5, color = &quot;#E77642&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, chimps&quot;])==&quot;0.51&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps+0.08), colour = &quot;#E77642&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.42&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_chimps_given_humans-0.08), size = 6)

p4_inc_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-53-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_c, pval_h)
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.51&quot;), times = nrow(pval_c)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.42&quot;), times = nrow(pval_h)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

#write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14J.csv&quot;, quote = FALSE, row.names = FALSE)</code></pre>
</div>
<div id="changes-of-variation-in-humans-p-0.10" class="section level3">
<h3>Changes of variation in humans (p = 0.10)</h3>
<pre class="r"><code># Reduction in variance from day 0 to day 1 humans

qqplot_red_day01 &lt;- find_qqplot_red_humans(7,10,17,20,1,6,11,16, 0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 2 humans

qqplot_red_day12 &lt;- find_qqplot_red_humans(17,20,27,30,11,16,21,26,0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 2 to day 3 humans

qqplot_red_day23 &lt;- find_qqplot_red_humans(27,30,37,40,21,26,31,36,0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 1 to day 3 humans

qqplot_red_day13 &lt;- find_qqplot_red_humans(17,20,37,40,11,16,31,36,0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Reduction in variance from day 0 to day 2 humans

qqplot_red_day02 &lt;- find_qqplot_red_humans(7,10,27,30,1,6,21,26,0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 1 humans

qqplot_inc_day01 &lt;- find_qqplot_inc_humans(7,10,17,20,1,6,11,16,0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 1 to day 2 humans

qqplot_inc_day12 &lt;- find_qqplot_inc_humans(17,20,27,30,11,16,21,26,0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 2 to day 3 humans

qqplot_inc_day23 &lt;- find_qqplot_inc_humans(27,30,37,40,21,26,31,36,0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day 0 to day 2

qqplot_inc_day02 &lt;- find_qqplot_inc_humans(7,10,27,30,1,6,21,26,0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Inc. in variance from day ` to day 3 humans

qqplot_inc_day13 &lt;- find_qqplot_inc_humans(17,20,37,40,11,16,31,36,0.1)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-54-20.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="estimates-of-pi0-and-pi1-for-reduction-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.10" class="section level3">
<h3>Estimates of pi0 and pi1 for reduction of variation in humans conditioned on significance in chimps (p = 0.10)</h3>
<pre class="r"><code>####################### REDUCTION OF VARIATION IN HUMANS (p = 0.10)


# Day 0 to 1

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day01[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.174912</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.825088</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.1538704</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.8461296</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day01[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p5_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.17&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.15&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p5_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-55-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.17&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.15&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)


# Day 1 to 2

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day12[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day12[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p6_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p6_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-55-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

# Day 2 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day23[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.8664877</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.1335123</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.7630281</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.2369719</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day23[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p7_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.87&quot;)&#39;, parse=TRUE, x=0.7, y=(pi0est_humans+0.06), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.76&quot;)&#39;, parse=TRUE, x=0.7, y=(pi0est_humans_given_chimps+0.06), size = 6)

p7_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-55-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

# Day 1 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_red_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_red_day13[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_red_day13[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p8_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p8_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-55-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

#write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14H.csv&quot;, quote = FALSE, row.names = FALSE)

multiplot(p1_10,p5_10,p2_10,p6_10,p3_10,p7_10,p4_10,p8_10,cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-55-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="estimates-of-pi0-and-pi1-for-increase-of-variation-in-humans-conditioned-on-significance-in-chimps-p-0.10" class="section level3">
<h3>Estimates of pi0 and pi1 for increase of variation in humans conditioned on significance in chimps (p = 0.10)</h3>
<pre class="r"><code>####################### INCREASE IN VARIATION IN HUMANS (p = 0.1)


# Day 0 to 1

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day01[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day01[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p5_inc_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 0 to 1&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.15), size = 6)

p5_inc_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-56-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 0 to 1&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;NA&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A &lt;- cbind(pval_ch, resp, line)


# Day 1 to 2

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day12[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.2141993</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.7858007</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.09899168</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.9010083</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day12[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p6_inc_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 1 to 2&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + 
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.21&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.28), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.10&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.25), size = 6)

p6_inc_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-56-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 1 to 2&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.21&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.10&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

# Day 2 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day23[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.8477893</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.1522107</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 1</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day23[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p7_inc_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) +
  theme_bw() +  labs(title = &quot;Day 2 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + 
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) +  labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) + scale_y_continuous(labels=scaleFUN) +
  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.85&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.07), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;1&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.07), size = 6)

p7_inc_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-56-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 2 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;1&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.85&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

# Day 1 to 3

# Obtain pvalues
human_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[1])
shared_p_values_gen &lt;- as.data.frame(qqplot_inc_day13[2])


# Pi_0 estimate for the humans

pi0est_humans &lt;- qvalue(p= human_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans</code></pre>
<pre><code>[1] 0.203399</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans &lt;- 1 - pi0est_humans
pi1est_humans</code></pre>
<pre><code>[1] 0.796601</code></pre>
<pre class="r"><code># Pi_0 estimate for the chimps given significant in the humans

pi0est_humans_given_chimps &lt;- qvalue(p=shared_p_values_gen, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
pi0est_humans_given_chimps</code></pre>
<pre><code>[1] 0.1556774</code></pre>
<pre class="r"><code># Pi_1 estimate for the humans

pi1est_humans_given_chimps &lt;- 1 - pi0est_humans_given_chimps
pi1est_humans_given_chimps</code></pre>
<pre><code>[1] 0.8443226</code></pre>
<pre class="r"><code># Let&#39;s overlap the two histograms

labels_h &lt;- array(&quot;Human&quot;, dim = c(10304,1))
labels_c &lt;- array(&quot;Shared (Human|Significant in Chimps)&quot;, dim = c(as.data.frame(qqplot_inc_day13[[3]][1]),1))

chimps &lt;- cbind(labels_c, shared_p_values_gen)
colnames(chimps) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)
humans &lt;- cbind(labels_h, human_p_values_gen)
colnames(humans) &lt;- c(&quot;Species&quot;, &quot;pval&quot;)

# Combine the two

all_pval &lt;- rbind(humans[,1:2], chimps[,1:2])

pal &lt;- c(&quot;#00A4F4&quot;, &quot;white&quot;) 

p8_inc_10 &lt;- ggplot(all_pval, aes(pval, fill = Species)) + scale_fill_manual(values=pal[1:2]) +
  geom_density(show.legend = F, alpha = 1) +
  scale_x_continuous(limits = c(0,1), breaks = seq(0,1,0.25)) + scale_y_continuous(labels=scaleFUN) +
  theme_bw() +  labs(title = &quot;Day 1 to 3&quot;) + theme(plot.title = element_text(face = &quot;bold&quot;)) + 
  geom_hline(yintercept = pi0est_humans_given_chimps, size = 2.5) + labs(x = &quot;Unadjusted p-values&quot;) + labs(y = &quot;Density&quot;) +  geom_hline(yintercept = pi0est_humans, size = 2.5, color = &quot;#00A4F4&quot;) +       
  annotate(&quot;text&quot;, label= &#39;bold(bolditalic(hat(pi)[0][&quot;, humans&quot;])==&quot;0.20&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans+0.3), colour = &quot;#00A4F4&quot;, size = 6) +  annotate(&quot;text&quot;, label = &#39;bold(bolditalic(hat(pi)[0][&quot;, shared&quot;])==&quot;0.16&quot;)&#39;, parse=TRUE, x=0.2, y=(pi0est_humans_given_chimps+0.17), size = 6)

p8_inc_10</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-56-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pval_c &lt;- chimps[,1:2]
pval_h &lt;- humans[,1:2]
pval_ch &lt;- rbind(pval_h, pval_c)
resp &lt;- rep(c(&quot;Days 1 to 3&quot;), times = nrow(pval_c)+nrow(pval_h))

line1 &lt;- as.data.frame(rep(c(&quot;0.20&quot;), times = nrow(pval_h)))
colnames(line1) &lt;- c(&quot;line&quot;)
line2 &lt;- as.data.frame(rep(c(&quot;0.16&quot;), times = nrow(pval_c)))
colnames(line2) &lt;- c(&quot;line&quot;)
line &lt;- rbind(line1, line2)

figS12A_fig &lt;- cbind(pval_ch, resp, line)
figS12A &lt;- rbind(figS12A, figS12A_fig)

#write.csv(figS12A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14L.csv&quot;, quote = FALSE, row.names = FALSE)

multiplot(p1_inc_10,p5_inc_10,p2_inc_10,p6_inc_10,p3_inc_10,p7_inc_10,p4_inc_10,p8_inc_10,cols=4)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-56-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sharing-estimates-chimps-conditioned-on-humans-for-x-number-of-significant-genes-p-0.05-cutoff" class="section level3">
<h3>Sharing estimates (chimps conditioned on humans) for x number of significant genes (p = 0.05 cutoff)</h3>
<pre class="r"><code>find_qqplot_red_top_all_sharing_chimps &lt;- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
  }

 # hist(chimp_var_pval)

############## For humans

  human_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
  }

  #hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval &lt;- as.data.frame(cbind(human_var_pval, chimp_var_pval))
  rownames(var_pval) &lt;- rownames(mean_tech_reps)
  colnames(var_pval) &lt;- c(&quot;Human&quot;, &quot;Chimpanzee&quot;)
  summary(var_pval)

  dim(var_pval)

# Convert to data frame
  top_var_pval &lt;- var_pval[order(var_pval[,1]), ]
  
# Instead of going by p &lt; 0.05, take the top x number of p-values
  for_pi1_var_pval &lt;- array(NA, dim = c(10205, 2))
  
  for(i in 100:10304){
    num_var_pval &lt;- top_var_pval[1:i, 1:2]
  
# Get humans
    num_var_pval_chimps_i &lt;- top_var_pval[1:i, 1]
  
# Get sharing
    num_var_pval_shared_i &lt;- top_var_pval[1:i, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
  pi0est_humans_given_chimps_i &lt;- qvalue(p=num_var_pval_shared_i, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0
  
 



# Pi_1 estimate for the humans
  for_pi1_var_pval[i-99,1] &lt;- i
  for_pi1_var_pval[i-99,2] &lt;- 1 - pi0est_humans_given_chimps_i
  #print(i)
  }
  
  return(for_pi1_var_pval)
}

find_qqplot_inc_top_all_sharing_chimps &lt;- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
  }

  #hist(chimp_var_pval)

############## For humans

  human_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
  }

  #hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval &lt;- as.data.frame(cbind( human_var_pval, chimp_var_pval))
  rownames(var_pval) &lt;- rownames(mean_tech_reps)
  colnames(var_pval) &lt;- c( &quot;Human&quot;, &quot;Chimpanzee&quot;)
  summary(var_pval)

  dim(var_pval)

# Convert to data frame
  top_var_pval &lt;- var_pval[order(var_pval[,1]), ]
  
# Instead of going by p &lt; 0.05, take the top x number of p-values
  for_pi1_var_pval &lt;- array(NA, dim = c(10205, 2))
  
  for(i in 100:10304){
    num_var_pval &lt;- top_var_pval[1:i, 1:2]
  
# Get humans
    num_var_pval_chimps_i &lt;- top_var_pval[1:i, 1]
  
# Get sharing
    num_var_pval_shared_i &lt;- top_var_pval[1:i, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
  pi0est_humans_given_chimps_i &lt;- qvalue(p=num_var_pval_shared_i, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0


# Pi_1 estimate for the humans
  for_pi1_var_pval[i-99,1] &lt;- i
  for_pi1_var_pval[i-99,2] &lt;- 1 - pi0est_humans_given_chimps_i
  #print(i)
  }
  return(for_pi1_var_pval)
}

# Reduction in variance from day 0 to day 1

#qqplot_red_top_all_day01 &lt;- find_qqplot_red_top_all_sharing_chimps(7,10,17,20,1,6,11,16)

# Reduction in variance from day 1 to day 2

#qqplot_red_top_all_day12 &lt;- find_qqplot_red_top_all_sharing_chimps(17,20,27,30,11,16,21,26)

# Reduction in variance from day 2 to day 3

#qqplot_red_top_all_day23 &lt;- find_qqplot_red_top_all_sharing_chimps(27,30,37,40,21,26,31,36)

# Reduction in variance from day 0 to day 2

#qqplot_red_top_all_day02 &lt;- find_qqplot_red_top_all_sharing_chimps(7,10,27,30,1,6,21,26)

# Reduction in variance from day 1 to day 3

#qqplot_red_top_all_day13 &lt;- find_qqplot_red_top_all_sharing_chimps(17,20,37,40,11,16,31,36)

# Inc. in variance from day 0 to day 1

#qqplot_inc_top_all_day01 &lt;- find_qqplot_inc_top_all_sharing_chimps(7,10,17,20,1,6,11,16)

# Inc. in variance from day 1 to day 2

#qqplot_inc_top_all_day12 &lt;- find_qqplot_inc_top_all_sharing_chimps(17,20,27,30,11,16,21,26)

# Inc. in variance from day 2 to day 3

#qqplot_inc_top_all_day23 &lt;- find_qqplot_inc_top_all_sharing_chimps(27,30,37,40,21,26,31,36)

# Inc. in variance from day 0 to day 2

#qqplot_inc_top_all_day02 &lt;- find_qqplot_inc_top_all_sharing_chimps(7,10,27,30,1,6,21,26)

# Inc. in variance from day 1 to day 3

#qqplot_inc_top_all_day13 &lt;- find_qqplot_inc_top_all_sharing_chimps(17,20,37,40,11,16,31,36)

# Make everything a data frame
#qqplot_red_top_all_day01 &lt;- as.data.frame(qqplot_red_top_all_day01)
#qqplot_red_top_all_day12 &lt;- as.data.frame(qqplot_red_top_all_day12)
#qqplot_red_top_all_day23 &lt;- as.data.frame(qqplot_red_top_all_day23)
#qqplot_red_top_all_day02 &lt;- as.data.frame(qqplot_red_top_all_day02)
#qqplot_red_top_all_day13 &lt;- as.data.frame(qqplot_red_top_all_day13)
#qqplot_inc_top_all_day01 &lt;- as.data.frame(qqplot_inc_top_all_day01)
#qqplot_inc_top_all_day12 &lt;- as.data.frame(qqplot_inc_top_all_day12)
#qqplot_inc_top_all_day23 &lt;- as.data.frame(qqplot_inc_top_all_day23)
#qqplot_inc_top_all_day02 &lt;- as.data.frame(qqplot_inc_top_all_day02)
#qqplot_inc_top_all_day13 &lt;- as.data.frame(qqplot_inc_top_all_day13)

# Plots of pi_0

#qqplot_red_top_all_day01_pi0 &lt;- abs(1- qqplot_red_top_all_day01)
#qqplot_red_top_all_day12_pi0 &lt;- abs(1- qqplot_red_top_all_day12)
#qqplot_red_top_all_day23_pi0 &lt;- abs(1- qqplot_red_top_all_day23)
#qqplot_red_top_all_day02_pi0 &lt;- abs(1- qqplot_red_top_all_day02)
#qqplot_red_top_all_day13_pi0 &lt;- abs(1- qqplot_red_top_all_day13)

#pi_0_across_genes &lt;- rbind(qqplot_red_top_all_day01_pi0, qqplot_red_top_all_day12_pi0, qqplot_red_top_all_day23_pi0, qqplot_red_top_all_day13_pi0)

#resp01 &lt;- t(t(rep(c(&quot;Day 0 to 1&quot;), times = 10205)))
#resp12 &lt;- t(t(rep(c(&quot;Day 1 to 2&quot;), times = 10205)))
#resp23 &lt;- t(t(rep(c(&quot;Day 2 to 3&quot;), times = 10205)))
#resp13 &lt;- t(t(rep(c(&quot;Day 1 to 3&quot;), times = 10205)))

#labels &lt;- rbind(resp01, resp12, resp23, resp13)

#figS14A &lt;- cbind(pi_0_across_genes, labels)
#colnames(figS14A) &lt;- c(&quot;N Most Significant Genes&quot;, &quot;pi_0&quot;, &quot;Day label&quot;)

#write.table(figS14A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14A.txt&quot;, quote = FALSE, row.names = FALSE)

#qqplot_inc_top_all_day01_pi0 &lt;- abs(1- qqplot_inc_top_all_day01)
#qqplot_inc_top_all_day12_pi0 &lt;- abs(1- qqplot_inc_top_all_day12)
#qqplot_inc_top_all_day23_pi0 &lt;- abs(1- qqplot_inc_top_all_day23)
#qqplot_inc_top_all_day02_pi0 &lt;- abs(1- qqplot_inc_top_all_day02)
#qqplot_inc_top_all_day13_pi0 &lt;- abs(1- qqplot_inc_top_all_day13)


#pi_0_across_genes &lt;- rbind(qqplot_inc_top_all_day01_pi0, qqplot_inc_top_all_day12_pi0, qqplot_inc_top_all_day23_pi0, qqplot_inc_top_all_day13_pi0)

#resp01 &lt;- t(t(rep(c(&quot;Day 0 to 1&quot;), times = 10205)))
#resp12 &lt;- t(t(rep(c(&quot;Day 1 to 2&quot;), times = 10205)))
#resp23 &lt;- t(t(rep(c(&quot;Day 2 to 3&quot;), times = 10205)))
#resp13 &lt;- t(t(rep(c(&quot;Day 1 to 3&quot;), times = 10205)))

#labels &lt;- rbind(resp01, resp12, resp23, resp13)

#figS14A &lt;- cbind(pi_0_across_genes, labels)
#colnames(figS14A) &lt;- c(&quot;N Most Significant Genes&quot;, &quot;pi_0&quot;, &quot;Day label&quot;)

#write.table(figS14A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14C.txt&quot;, quote = FALSE, row.names = FALSE)


# Plot reduction in variation
#p_top_genes_red_var_chimps &lt;- ggplot(qqplot_red_top_all_day01_pi0, aes(x = qqplot_red_top_all_day01_pi0[,1], y = qqplot_red_top_all_day01_pi0[,2])) + geom_line(colour=&quot;#00B451&quot;) + theme_bw() + geom_line(aes(x = qqplot_red_top_all_day12_pi0[,1], y = qqplot_red_top_all_day12_pi0[,2]-0.001), colour =  &quot;#00B1DB&quot;) + geom_line(aes(x = qqplot_red_top_all_day23_pi0[,1], y = qqplot_red_top_all_day23_pi0[,2]), colour =  &quot;#C77CFF&quot;) + geom_line(aes(x = qqplot_red_top_all_day13_pi0[,1], y = qqplot_red_top_all_day13_pi0[,2]+0.001), colour = &quot;#000000&quot;) + ylim(0,1.01) + ggtitle(expression( italic(hat(pi)[0]) ~ &quot;For Reduction in Variance (Chimpanzees|Sign. in Humans)&quot; )) + xlab(&quot;N Most Significant Genes in Humans&quot;) + ylab(expression( italic(hat(pi)[0]) ~ &quot;(Chimps|Sign. in Humans)&quot;)) 

#p_top_genes_red_var_chimps

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0, 0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#00B451&quot;, &quot;#00B1DB&quot;,  &quot;#C77CFF&quot;,&quot;#000000&quot; ), cex = 1)



# Plot of Increase in Variation

#p_top_genes_inc_var_chimps &lt;- ggplot(qqplot_inc_top_all_day01_pi0, aes(x = qqplot_inc_top_all_day01_pi0[,1], y = qqplot_inc_top_all_day01_pi0[,2]+0.001)) + geom_line(colour=&quot;#00B451&quot;) + theme_bw() + geom_line(aes(x = qqplot_inc_top_all_day12_pi0[,1], y = qqplot_inc_top_all_day12_pi0[,2]), colour = &quot;#00B1DB&quot;) + geom_line(aes(x = qqplot_inc_top_all_day23_pi0[,1], y = qqplot_inc_top_all_day23_pi0[,2]-0.001), colour =  &quot;#C77CFF&quot;) + geom_line(aes(x = qqplot_inc_top_all_day13_pi0[,1], y = qqplot_inc_top_all_day13_pi0[,2]), colour = &quot;#000000&quot;) + ylim(0,1.01) + ggtitle(expression( italic(hat(pi)[0]) ~ &quot;For Increase in Variance (Chimpanzees|Sign. in Humans)&quot; )) + xlab(&quot;N Most Significant Genes in Humans&quot;) + ylab(expression( italic(hat(pi)[0]) ~ &quot;(Chimps|Sign. in Humans)&quot;)) 

#p_top_genes_inc_var_chimps

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0, 0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#00B451&quot;, &quot;#00B1DB&quot;,  &quot;#C77CFF&quot;,&quot;#000000&quot; ), cex = 1)

# Plots of pi_1

#dev.off()
# Plot reduction in variation
#ggplot(qqplot_red_top_all_day01, aes(x = qqplot_red_top_all_day01[,1], y = qqplot_red_top_all_day01[,2])) + geom_line(colour=&quot;#E69F00&quot;) + theme_bw() + geom_line(aes(x = qqplot_red_top_all_day12[,1], y = qqplot_red_top_all_day12[,2]), colour = &quot;#000000&quot;) + geom_line(aes(x = qqplot_red_top_all_day23[,1], y = qqplot_red_top_all_day23[,2]), colour =  &quot;#009E73&quot;) + geom_line(aes(x = qqplot_red_top_all_day13[,1], y = qqplot_red_top_all_day13[,2]), colour =  &quot;#0072B2&quot;) + ylim(0,1) + ggtitle(&quot;Proportion of Shared Genes with a Reduction in Variance between Days&quot;) + xlab(&quot;N Most Significant Genes in Humans&quot;) + ylab(expression( italic(hat(pi)[1]) ~ &quot;(Chimpanzees|Significant in Humans)&quot;)) 

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0, 0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#E69F00&quot;, &quot;#000000&quot;,  &quot;#009E73&quot;,&quot;#0072B2&quot; ), cex = 1)


# Plot inc in variation
#ggplot(qqplot_inc_top_all_day01, aes(x = qqplot_inc_top_all_day01[,1], y = qqplot_inc_top_all_day01[,2])) + geom_line(colour=&quot;#E69F00&quot;) + theme_bw() + geom_line(aes(x = qqplot_inc_top_all_day12[,1], y = qqplot_inc_top_all_day12[,2]), colour = &quot;#000000&quot;) + geom_line(aes(x = qqplot_inc_top_all_day23[,1], y = qqplot_inc_top_all_day23[,2]), colour =  &quot;#009E73&quot;) + geom_line(aes(x = qqplot_inc_top_all_day13[,1], y = qqplot_inc_top_all_day13[,2]), colour =  &quot;#0072B2&quot;) + ylim(0,1) + ggtitle(&quot;Proportion of Shared Genes with an Increase in Variance between Days&quot;) + xlab(&quot;N Most Significant Genes in Humans&quot;) + ylab(expression( italic(hat(pi)[1]) ~ &quot;(Chimpanzees|Significant in Humans)&quot;)) 
 

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0, 0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#E69F00&quot;, &quot;#000000&quot;,  &quot;#009E73&quot;,&quot;#0072B2&quot; ), cex = 1)

# ggplot(qqplot_inc_top_all_day01, aes(x = qqplot_inc_top_all_day01[,1], y = qqplot_inc_top_all_day01[,2])) + geom_line(colour=&quot;#E69F00&quot;) + theme_bw() + geom_line(aes(x = qqplot_inc_top_all_day12[,1], y = qqplot_inc_top_all_day12[,2]), colour = &quot;#000000&quot;) + geom_line(aes(x = qqplot_inc_top_all_day23[,1], y = qqplot_inc_top_all_day23[,2]), colour =  &quot;#009E73&quot;) + geom_line(aes(x = qqplot_inc_top_all_day13[,1], y = qqplot_inc_top_all_day13[,2]), colour =  &quot;#0072B2&quot;) + ylim(0,1) + ggtitle(&quot;Proportion of Shared Genes with an Increase in Variance between Days&quot;) + xlab(&quot;N Most Significant Genes in Humans&quot;) + ylab(expression( italic(hat(pi)[1]) ~ &quot;(Chimpanzees|Significant in Humans)&quot;)) </code></pre>
</div>
<div id="sharing-estimates-humans-conditioned-on-chimps-for-x-number-of-significant-genes" class="section level3">
<h3>Sharing estimates (humans conditioned on chimps) for x number of significant genes</h3>
<pre class="r"><code>find_qqplot_red_top_all_humans &lt;- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval &lt;- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) &lt;- rownames(mean_tech_reps)
  colnames(var_pval) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;)
  summary(var_pval)

  dim(var_pval)

# Convert to data frame
  top_var_pval &lt;- var_pval[order(var_pval[,1]), ]
  
# Instead of going by p &lt; 0.05, take the top x number of p-values
  for_pi1_var_pval &lt;- array(NA, dim = c(10205, 2))
  
  for(i in 100:10304){
    num_var_pval &lt;- top_var_pval[1:i, 1:2]
  
# Get chimps
    num_var_pval_chimps_i &lt;- top_var_pval[1:i, 1]
  
# Get sharing
    num_var_pval_shared_i &lt;- top_var_pval[1:i, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
  pi0est_humans_given_chimps_i &lt;- qvalue(p=num_var_pval_shared_i, pi0.method=&quot;smoother&quot;, fdr.level=NULL, robust=TRUE)$pi0


# Pi_1 estimate for the humans
  for_pi1_var_pval[i-99,1] &lt;- i
  for_pi1_var_pval[i-99,2] &lt;- 1 - pi0est_humans_given_chimps_i
  #print(i)
  }
  return(for_pi1_var_pval)
}

find_qqplot_inc_top_all_humans &lt;- function(chimp_begin_column_tm1, chimp_end_column_tm1, chimp_begin_column_t, chimp_end_column_t, human_begin_column_tm1, human_end_column_tm1, human_begin_column_t, human_end_column_t){

  # Make an array to store the Chimp p-values

  chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,chimp_begin_column_tm1:chimp_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,chimp_begin_column_t:chimp_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
  }

  hist(chimp_var_pval)

############## For humans

  human_var_pval &lt;- array(NA, dim = c(10304, 1))

  for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,human_begin_column_tm1:human_end_column_tm1])
    y &lt;- t(mean_tech_reps[i,human_begin_column_t:human_end_column_t])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
  }

  hist(human_var_pval)

############ Find p-values of the F_statistics

# Make one data frame

  var_pval &lt;- as.data.frame(cbind(chimp_var_pval, human_var_pval))
  rownames(var_pval) &lt;- rownames(mean_tech_reps)
  colnames(var_pval) &lt;- c(&quot;Chimpanzee&quot;, &quot;Human&quot;)
  summary(var_pval)

  dim(var_pval)

# Convert to data frame
  top_var_pval &lt;- var_pval[order(var_pval[,1]), ]
  
# Instead of going by p &lt; 0.05, take the top x number of p-values
  for_pi1_var_pval &lt;- array(NA, dim = c(10205, 2))
  
  for(i in 100:10304){
    num_var_pval &lt;- top_var_pval[1:i, 1:2]
  
# Get chimps
    num_var_pval_chimps_i &lt;- top_var_pval[1:i, 1]
  
# Get sharing
    num_var_pval_shared_i &lt;- top_var_pval[1:i, 2]
  

# Pi_0 estimate for the humans given significant in the chimps
  pi0est_humans_given_chimps_i &lt;- pi0est(num_var_pval_shared_i, lambda = seq(0.05, 0.95, 0.05))$pi0


# Pi_1 estimate for the humans
  for_pi1_var_pval[i-99,1] &lt;- i
  for_pi1_var_pval[i-99,2] &lt;- 1 - pi0est_humans_given_chimps_i
#  print(i)
  }
  return(for_pi1_var_pval)
}

# Reduction in variance from day 0 to day 1

#qqplot_red_top_all_day01_humans &lt;- find_qqplot_red_top_all_humans(7,10,17,20,1,6,11,16)

# Reduction in variance from day 1 to day 2

#qqplot_red_top_all_day12_humans &lt;- find_qqplot_red_top_all_humans(17,20,27,30,11,16,21,26)

# Reduction in variance from day 2 to day 3

#qqplot_red_top_all_day23_humans  &lt;- find_qqplot_red_top_all_humans(27,30,37,40,21,26,31,36)

# Reduction in variance from day 0 to day 2

#qqplot_red_top_all_day02_humans  &lt;- find_qqplot_red_top_all_humans(7,10,27,30,1,6,21,26)

# Reduction in variance from day 1 to day 3

#qqplot_red_top_all_day13_humans  &lt;- find_qqplot_red_top_all_humans(17,20,37,40,11,16,31,36)

# Inc. in variance from day 0 to day 1

#qqplot_inc_top_all_day01_humans  &lt;- find_qqplot_inc_top_all_humans(7,10,17,20,1,6,11,16)

# Inc. in variance from day 1 to day 2

#qqplot_inc_top_all_day12_humans  &lt;- find_qqplot_inc_top_all_humans(17,20,27,30,11,16,21,26)

# Inc. in variance from day 2 to day 3

#qqplot_inc_top_all_day23_humans  &lt;- find_qqplot_inc_top_all_humans(27,30,37,40,21,26,31,36)

# Inc. in variance from day 0 to day 2

#qqplot_inc_top_all_day02_humans  &lt;- find_qqplot_inc_top_all_humans(7,10,27,30,1,6,21,26)

# Inc. in variance from day 1 to day 3

#qqplot_inc_top_all_day13_humans  &lt;- find_qqplot_inc_top_all_humans(17,20,37,40,11,16,31,36)


# Make everything a data frame
#qqplot_red_top_all_day01_humans  &lt;- as.data.frame(qqplot_red_top_all_day01_humans)
#qqplot_red_top_all_day12_humans  &lt;- as.data.frame(qqplot_red_top_all_day12_humans)
#qqplot_red_top_all_day23_humans  &lt;- as.data.frame(qqplot_red_top_all_day23_humans)
#qqplot_red_top_all_day02_humans  &lt;- as.data.frame(qqplot_red_top_all_day02_humans)
#qqplot_red_top_all_day13_humans  &lt;- as.data.frame(qqplot_red_top_all_day13_humans)
#qqplot_inc_top_all_day01_humans  &lt;- as.data.frame(qqplot_inc_top_all_day01_humans)
#qqplot_inc_top_all_day12_humans  &lt;- as.data.frame(qqplot_inc_top_all_day12_humans)
#qqplot_inc_top_all_day23_humans  &lt;- as.data.frame(qqplot_inc_top_all_day23_humans)
#qqplot_inc_top_all_day02_humans  &lt;- as.data.frame(qqplot_inc_top_all_day02_humans)
#qqplot_inc_top_all_day13_humans  &lt;- as.data.frame(qqplot_inc_top_all_day13_humans)



# Plots of pi_0

#qqplot_red_top_all_day01_pi0_humans  &lt;- abs(1- qqplot_red_top_all_day01_humans)
#qqplot_red_top_all_day12_pi0_humans  &lt;- abs(1- qqplot_red_top_all_day12_humans)
#qqplot_red_top_all_day23_pi0_humans  &lt;- abs(1- qqplot_red_top_all_day23_humans)
#qqplot_red_top_all_day02_pi0_humans  &lt;- abs(1- qqplot_red_top_all_day02_humans)
#qqplot_red_top_all_day13_pi0_humans  &lt;- abs(1- qqplot_red_top_all_day13_humans)
#qqplot_inc_top_all_day01_pi0_humans  &lt;- abs(1- qqplot_inc_top_all_day01_humans)
#qqplot_inc_top_all_day12_pi0_humans  &lt;- abs(1- qqplot_inc_top_all_day12_humans)
#qqplot_inc_top_all_day23_pi0_humans  &lt;- abs(1- qqplot_inc_top_all_day23_humans)
#qqplot_inc_top_all_day02_pi0_humans  &lt;- abs(1- qqplot_inc_top_all_day02_humans)
#qqplot_inc_top_all_day13_pi0_humans  &lt;- abs(1- qqplot_inc_top_all_day13_humans)


#pi_0_across_genes &lt;- rbind(qqplot_red_top_all_day01_pi0_humans, qqplot_red_top_all_day12_pi0_humans, qqplot_red_top_all_day23_pi0_humans, qqplot_red_top_all_day13_pi0_humans)

#resp01 &lt;- t(t(rep(c(&quot;Day 0 to 1&quot;), times = 10205)))
#resp12 &lt;- t(t(rep(c(&quot;Day 1 to 2&quot;), times = 10205)))
#resp23 &lt;- t(t(rep(c(&quot;Day 2 to 3&quot;), times = 10205)))
#resp13 &lt;- t(t(rep(c(&quot;Day 1 to 3&quot;), times = 10205)))

#labels &lt;- rbind(resp01, resp12, resp23, resp13)

#figS14A &lt;- cbind(pi_0_across_genes, labels)
#colnames(figS14A) &lt;- c(&quot;N Most Significant Genes&quot;, &quot;pi_0&quot;, &quot;Day label&quot;)

#write.table(figS14A, &quot;/Users/laurenblake/Dropbox/Endoderm TC/Draft versions/Resubmit1_Genome Biology/Supplemental/Data_figS14A.txt&quot;, quote = FALSE, row.names = FALSE)

# Plots of pi_0

#p_top_genes_red_var_humans &lt;- ggplot(qqplot_red_top_all_day01_pi0_humans, aes(x = qqplot_red_top_all_day01_pi0_humans[,1], y = qqplot_red_top_all_day01_pi0_humans[,2])) + geom_line(colour=&quot;#00B451&quot;) + theme_bw() + geom_line(aes(x = qqplot_red_top_all_day12_pi0_humans[,1], y = qqplot_red_top_all_day12_pi0_humans[,2]+0.001), colour = &quot;#00B1DB&quot;) + geom_line(aes(x = qqplot_red_top_all_day23_pi0_humans[,1], y = qqplot_red_top_all_day23_pi0_humans[,2]), colour =  &quot;#C77CFF&quot;) + geom_line(aes(x = qqplot_red_top_all_day13_pi0_humans[,1], y = qqplot_red_top_all_day13_pi0_humans[,2]-0.001), colour =  &quot;#000000&quot;) + ylim(0,1.01) + ggtitle(expression( italic(hat(pi)[0]) ~ &quot;For Reduction in Variance Between Days (Humans|Sign. Chimps)&quot; )) + xlab(&quot;N Most Significant Genes in Chimpanzees&quot;) + ylab(expression( italic(hat(pi)[0]) ~ &quot;(Humans|Sign. in Chimps)&quot;)) 
#p_top_genes_red_var_humans

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0, 0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#00B451&quot;, &quot;#00B1DB&quot;,  &quot;#C77CFF&quot;,&quot;#000000&quot; ), cex = 1)

#p_top_genes_inc_var_humans &lt;- ggplot(qqplot_inc_top_all_day01_pi0_humans, aes(x = qqplot_inc_top_all_day01_pi0_humans[,1], y = qqplot_inc_top_all_day01_pi0_humans[,2]+0.001)) + geom_line(colour=&quot;#00B451&quot;) + theme_bw() + geom_line(aes(x = qqplot_inc_top_all_day12_pi0_humans[,1], y = qqplot_inc_top_all_day12_pi0_humans[,2]), colour = &quot;#00B1DB&quot;) + geom_line(aes(x = qqplot_inc_top_all_day23_pi0_humans[,1], y = qqplot_inc_top_all_day23_pi0_humans[,2]-0.001), colour =  &quot;#C77CFF&quot;) + geom_line(aes(x = qqplot_inc_top_all_day13_pi0_humans[,1], y = qqplot_inc_top_all_day13_pi0_humans[,2]), colour =  &quot;#000000&quot; ) + ylim(0,1.01) + ggtitle(expression( italic(hat(pi)[0]) ~ &quot;For Increase in Variance Between Days (Humans|Significant in Chimpanzees)&quot; )) + xlab(&quot;N Most Significant Genes in Chimpanzees&quot;) + ylab(expression( italic(hat(pi)[0]) ~ &quot;(Humans|Significant in Chimpanzees)&quot;)) 
#p_top_genes_inc_var_humans

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0, 0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#00B451&quot;, &quot;#00B1DB&quot;,  &quot;#C77CFF&quot;,&quot;#000000&quot; ), cex = 1)


# Export plots

#multiplot(p_top_genes_red_var_chimps,p_top_genes_red_var_humans,cols=1)

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0,    0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#00B451&quot;, &quot;#00B1DB&quot;,  &quot;#C77CFF&quot;,&quot;#000000&quot; ), cex = 1)

#multiplot(p_top_genes_inc_var_chimps,p_top_genes_inc_var_humans,cols=1)

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0,    0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#00B451&quot;, &quot;#00B1DB&quot;,  &quot;#C77CFF&quot;,&quot;#000000&quot; ), cex = 1)

# Plots of pi_1

# Plot reduction in variation
#ggplot(qqplot_red_top_all_day01, aes(x = qqplot_red_top_all_day01[,1], y = qqplot_red_top_all_day01[,2])) + geom_line(colour=&quot;#E69F00&quot;) + theme_bw() + geom_line(aes(x = qqplot_red_top_all_day12[,1], y = qqplot_red_top_all_day12[,2]), colour = &quot;#000000&quot;) + geom_line(aes(x = qqplot_red_top_all_day23[,1], y = qqplot_red_top_all_day23[,2]), colour =  &quot;#009E73&quot;) + geom_line(aes(x = qqplot_red_top_all_day13[,1], y = qqplot_red_top_all_day13[,2]), colour =  &quot;#0072B2&quot;) + ggtitle(expression( italic(hat(pi)[1]) ~ &quot;For Reduction in Variance Between Days (Humans|Significant in Chimpanzees)&quot; )) + xlab(&quot;N Most Significant Genes in Chimpanzees&quot;) + ylab(expression( italic(hat(pi)[1]) ~ &quot;(Humans|Significant in Chimpanzees)&quot;)) 

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0, 0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#E69F00&quot;, &quot;#000000&quot;,  &quot;#009E73&quot;,&quot;#0072B2&quot; ), cex = 1)


# Plot inc in variation
#ggplot(qqplot_inc_top_all_day01, aes(x = qqplot_inc_top_all_day01[,1], y = qqplot_inc_top_all_day01[,2])) + geom_line(colour=&quot;#E69F00&quot;) + theme_bw() + geom_line(aes(x = qqplot_inc_top_all_day12[,1], y = qqplot_inc_top_all_day12[,2]), colour = &quot;#000000&quot;) + geom_line(aes(x = qqplot_inc_top_all_day23[,1], y = qqplot_inc_top_all_day23[,2]), colour =  &quot;#009E73&quot;) + geom_line(aes(x = qqplot_inc_top_all_day13[,1], y = qqplot_inc_top_all_day13[,2]), colour =  &quot;#0072B2&quot;) + ggtitle(expression( italic(hat(pi)[1]) ~ &quot;For Increase in Variance Between Days (Humans|Significant in Chimpanzees)&quot; )) + xlab(&quot;N Most Significant Genes in Chimpanzees&quot;) + ylab(expression( italic(hat(pi)[1]) ~ &quot;(Humans|Significant in Chimpanzees)&quot;)) 

#par(fig = c(0, 1, 0, 1), oma = c(0, 0, 0, 0), mar = c(0, 0, 0, 0), new = TRUE)
#plot(0, 0, type = &quot;n&quot;, bty = &quot;n&quot;, xaxt = &quot;n&quot;, yaxt = &quot;n&quot;)
#legend(&quot;right&quot;, c(&quot;Day 0 to 1&quot;, &quot;Day 1 to 2&quot;, &quot;Day 2 to 3&quot;, &quot;Day 1 to 3&quot;), xpd = TRUE, horiz = FALSE, inset = c(0, 0), bty = &quot;n&quot;, pch = c(15, 15, 15, 15), col = c(&quot;#E69F00&quot;, &quot;#000000&quot;,  &quot;#009E73&quot;,&quot;#0072B2&quot; ), cex = 1)</code></pre>
</div>
<div id="note-code-to-make-other-figures" class="section level3">
<h3>Note: code to make other figures</h3>
<pre class="r"><code># 6A) 
multiplot(p1s,p3s,p2s,p3_13s, cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 6B)
multiplot(p1s,p3s,p2s,p3_13s, cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 6C-F)

multiplot(p1_05, p2_inc_05, p5_05,  p6_inc_05, cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF 12A) 
multiplot(p1,p2,p3,p3_13, cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF12 B)
multiplot(p4,p5,p6,p6_13, cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF13 A)
multiplot(p2_05, p3_05,p4_05,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF13 B)
multiplot(p6_05, p7_05,p8_05,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF13 C)
multiplot(p1_inc_05, p3_inc_05,p4_inc_05,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF13 D)
multiplot(p5_inc_05, p7_inc_05,p8_inc_05,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF14 E)
multiplot(p1_01, p3_01,p4_01,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF14 F)
multiplot(p5_01, p6_01, p7_01,p8_01,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF14 G)
multiplot(p1_inc_01, p2_inc_01, p3_inc_01,p4_inc_01,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF14 H)
multiplot(p5_inc_01, p6_inc_01, p7_inc_01,p8_inc_01,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF14 I)
multiplot(p1_10, p2_10, p3_10,p4_10,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF14 J)
multiplot(p5_10, p6_10, p7_10,p8_10,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF14 K)
multiplot(p1_inc_10, p2_inc_10, p3_inc_10,p4_inc_10,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># SF14 L)
multiplot(p5_inc_10, p6_inc_10, p7_inc_10,p8_inc_10,cols=2)</code></pre>
<p><img src="figure/Temporal_changes_variation_cyclicloess.Rmd/unnamed-chunk-59-17.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-names-showing-reduction-increase-in-variance-from-days-0-to-1-days-1-to-2" class="section level3">
<h3>Gene names showing reduction (increase) in variance from days 0 to 1 (days 1 to 2)</h3>
<pre class="r"><code># Reduction Day 0 to 1 chimps

chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,7:10])
    y &lt;- t(mean_tech_reps[i,17:20])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
}
rownames(chimp_var_pval) &lt;- rownames(mean_tech_reps)
length(which(chimp_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 664</code></pre>
<pre class="r"><code>chimp_red_01 &lt;- as.data.frame(chimp_var_pval[which(chimp_var_pval &lt; 0.05), ])
dim(chimp_red_01)</code></pre>
<pre><code>[1] 664   1</code></pre>
<pre class="r"><code>#write.table(chimp_red_01,file=&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/red_chimps_day0_1.txt&quot;,sep=&quot;\t&quot;, col.names = T, row.names = T)


# Reduction Day 0 to 1 humans
human_var_pval &lt;- array(NA, dim = c(10304, 1))

for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,1:6])
    y &lt;- t(mean_tech_reps[i,11:16])
    htest &lt;- var.test(x, y, alternative = c(&quot;greater&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
}
rownames(human_var_pval) &lt;- rownames(mean_tech_reps)
length(which(human_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 2309</code></pre>
<pre class="r"><code>human_red_01 &lt;- as.data.frame(human_var_pval[which(human_var_pval &lt; 0.05), ])
dim(human_red_01)</code></pre>
<pre><code>[1] 2309    1</code></pre>
<pre class="r"><code>#write.table(human_red_01,file=&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/red_human_day0_1.txt&quot;,sep=&quot;\t&quot;, col.names = T, row.names = T)


# Increase Day 1 to 2 chimps
chimp_var_pval &lt;- array(NA, dim = c(10304, 1))

for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,17:20])
    y &lt;- t(mean_tech_reps[i,27:30])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    chimp_var_pval[i,1] &lt;- htest$p.value
}
rownames(chimp_var_pval) &lt;- rownames(mean_tech_reps)
length(which(chimp_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 558</code></pre>
<pre class="r"><code>chimp_inc_12 &lt;- as.data.frame(chimp_var_pval[which(chimp_var_pval &lt; 0.05), ])
dim(chimp_inc_12)</code></pre>
<pre><code>[1] 558   1</code></pre>
<pre class="r"><code>write.table(chimp_inc_12,file=&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/inc_chimps_day1_2.txt&quot;,sep=&quot;\t&quot;, col.names = T, row.names = T)


# Increase Day 1 to 2 humans
human_var_pval &lt;- array(NA, dim = c(10304, 1))

for(i in 1:10304){
    x &lt;- t(mean_tech_reps[i,11:16])
    y &lt;- t(mean_tech_reps[i,21:26])
    htest &lt;- var.test(x, y, alternative = c(&quot;less&quot;))
    human_var_pval[i,1] &lt;- htest$p.value
}
rownames(human_var_pval) &lt;- rownames(mean_tech_reps)
length(which(human_var_pval &lt; 0.05))</code></pre>
<pre><code>[1] 1968</code></pre>
<pre class="r"><code>human_inc_12 &lt;- as.data.frame(human_var_pval[which(human_var_pval &lt; 0.05), ])
dim(human_inc_12)</code></pre>
<pre><code>[1] 1968    1</code></pre>
<pre class="r"><code>#write.table(human_inc_12,file=&quot;~/Desktop/Endoderm_TC/ashlar-trial/data/inc_human_day1_2.txt&quot;,sep=&quot;\t&quot;, col.names = T, row.names = T)</code></pre>
</div>
</div>
</div>


<!-- some extra javascript for older browsers -->
<script type="text/javascript" src="libs/polyfill.js"></script>

<script>

// manage active state of menu based on current page
$(document).ready(function () {

    // active menu
    href = window.location.pathname
    href = href.substr(href.lastIndexOf('/') + 1)
    $('a[href="' + href + '"]').parent().addClass('active');

    // manage active menu header
    if (href.startsWith('authoring_'))
      $('a[href="' + 'authoring' + '"]').parent().addClass('active');
    else if (href.endsWith('_format.html'))
      $('a[href="' + 'formats' + '"]').parent().addClass('active');
    else if (href.startsWith('developer_'))
      $('a[href="' + 'developer' + '"]').parent().addClass('active');

});

</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
